
# Introduction to R programming {#introR}

This notebook collects the scripts used for teaching in BIOF1001 for 
Introduction to R (1 hour teaching). You can get this Rmd file on Moodle or
[here](https://github.com/StatBiomed/BMDatSci-book/blob/main/notebooks/chapter1-R/01-introR.Rmd)
(right-click and "save link as" to download).

R is a programming language, particularly popular for its power in statistical
computing, elegant graphics, and also genomic data analysis. It is a free and
open-source software, with active support and development from the community.
Additionally, R is relatively easy to get started for scientific computing.

> __Note__ To learn and practice R programming, you need to install `R` and `RStudio` on 
> your computer. You can follow the instructions for installation in the 
> [Appendix A](https://statbiomed.github.io/BMDS-book/install.html) chapter.


## Data types
In R language setting (similar to some other programming languages), there are 
a few commonly used data types that are predefined in the built-in environment.
You can use the `class()` and `typeof()` functions to check the class and 
data type of any variable. In total, R has five data types:

1. Numeric
2. Integers
3. Complex
4. Logical
5. Characters



### nemeric (or double)
The `numeric` is for numeric values, as the most common and the **default** 
data type. The `numeric` datatype saves values in double precision (double 
number of bytes in memory), so the type is also `double`.

```{r}
x <- c(1.0, 2.0, 5.0, 7.0)
x
class(x)
typeof(x)
```

### integer
The integer is another data type used for the set of all integers. You can use 
the capital ‘L’ notation as a suffix to specify a particular value as the 
integer data type. Also, you can convert a value into an integer type using the `as.integer()` function.

```{r}
y <- c(1L, 2L, 5L, 7L)
y
class(y)
typeof(y)
```

```{r}
# Assign a integer value to y
y = 5
 
# is y an integer?
print(is.integer(y))
```


### logical
In R, the `logical` data type takes either a value of true or false. A `logical` 
value is often generated when comparing variables.

```{r}
z <- c(TRUE, TRUE, TRUE, FALSE)
z
typeof(z)
```


### character
In R, the `character` is a data type where you have all the alphabets and 
special characters. It stores `character` values or `strings`. Strings in R can 
contain alphabets, numbers, and symbols. The character type is usually denoted 
by wrapping the value inside single or double inverted commas.

```{r}
w <- c("aa", "bb", "5", "7")
w
typeof(w)
```

### Memeory usage
Each data type is explicitly defined, especially the size of the memory.

When initializing a certain data type, there are a few small bytes used to store
basic information. Let's look at the empty values.

```{r}
object.size(numeric())
object.size(integer())
object.size(logical())
object.size(character())
```

To illustrate, let's use 1000 elements below to show the memory size usage 
for each data type. 
As we will see, integer and logical use 4 bytes per element, which is only half 
of the memory usage by double (numeric) and character of 8 bytes.

```{r}
object.size(rep(1, 1000))
object.size(rep(1L, 1000))
object.size(rep(TRUE, 1000))
object.size(rep("aa", 1000))
```


## Data structures
Data structure is one of the most important features in programming. It involves 
how the data is organised and can be accessed and modified. Using an appropriate
data structure may largely improve computing efficiency.

### Vector
Vector is a basic data structure in R. It contains elements in the **same** data
type (no matter double, integer, character or others). 
You can check the data type by using `typeof()` function and the length of the 
vector by `length()` function.

Since a vector has elements of the same type, this function will try and coerce elements to the same type, if they are different. Coercion is from lower 
to higher types, i.e., from logical to integer to double to a character.

See more introduction [here](https://www.datamentor.io/r-programming/vector/).

```{r}
x <- c(1, 2, 5, 7)
x
typeof(x)

x <- rep(3, 5)
x
typeof(x)

x <- 1:12 # integer
x
typeof(x)
```

### Matrix
Matrix is a two-dimensional data structure. It is in principle built based on 
vector but has more convenient built-in functions for computation.
It has rows and columns, both of which can also have names. To check the dimensions,
you can use the `dim()` function.

See more introduction [here](https://www.datamentor.io/r-programming/matrix/).

```{r}
A <- matrix(1:12, nrow=3)
A

B <- matrix(1:12, nrow=3, byrow=TRUE)
B

colnames(A) <- c("C1","C2","C3","C4")
rownames(A) <- c("R1","R2","R3")
A
```

#### Index Vector and Matrix
To index vector, you can use `logical` or `integer`, or the `element name` if it
has.

Note, when using an `integer` for indexing, the index *starts from 1* in R, unlike 
most programming languages where the index starts from 0.

We can also use negative integers to return all elements except those 
specified. But we cannot mix positive and negative integers while indexing and 
real numbers if used are truncated to integers.

**N.B.**, when using logical as the index, be careful if the index length is 
different from the vector.


```{r}
x <- 1:12

x[3]

x[2:5]

x[c(2, 5, 6)]                   # index with integer

x[c(TRUE, FALSE, FALSE, TRUE)]  # index with logical value
```

Now, let's index a matrix. It is very similar to vector, but it has both rows and columns.
```{r}
A <- matrix(1:12, nrow=3)
colnames(A) <- c("C1","C2","C3","C4")
rownames(A) <- c("R1","R2","R3")
A

A[1, 2]

A[1, "C2"]

A[1, c(2, 3)]

A[1:2, c(2, 3)]
```

Single row or column matrix will become a vector, unless **using `drop=FALSE`**
```{r}
A[1, 2:4]
dim(A[1, 2:4])

A[1, 2:4, drop=FALSE]
dim(A[1, 2:4, drop=FALSE])
```

#### Modify values
```{r}
A[1, 2:4] <- c(-3, -5, 20)
A
```

### List
Different from `vector` that has all elements in the same data type, the `list`
data structure can have components of mixed data types.
More broadly, a `list` can contain a list of any data structure: value, vector, 
matrix, etc.

We can use `str()` function to view the structure of a list (or any object).
```{r}
x <- list(2.5, TRUE, 1:3)
x

str(x)
```

We can also have a name for each element:
```{r}
x <- list("a" = 2.5, "b" = TRUE, "c" = 1:3)
x
str(x)
```

#### Indexing list
Different from vector and matrix, for a `list`, you need to use **double-layer square brackets**, 
either by numeric index or name. Alternatively, you can also use `$` symbol with the name.
```{r}
x[[3]]

x[["c"]]

x$c
```

### Data Frame
Data frame is widely used for rectangular data, where each column has the same 
data type (vector) but different columns can have different data types (like Excel)

As you guess, the data frame is a special type of `list`: A list of vectors with the same length.

```{r}
df <- data.frame("SN" = 1:2, "Age" = c(21,15), 
                 "Name" = c("John","Dora"))
df
```

```{r}
df$Age[2]
```

```{r}
# View(df)
```

### Factor vs vector
For vector, if it only contains pre-defined values (which can have specified 
orders), you may consider using factor.

* Factor is a data structure used for fields that takes only predefined, finite 
number of values (categorical data)
* The order of predefined values can be specified, instead of alphabetic by default

```{r}
x = c("single", "married", "married", "single") # vector
x

class(x)
typeof(x)
```

```{r}
y = factor(c("single", "married", "married", "single"))
y

class(y)
typeof(y)
```


#### Change the order of factor levels
```{r}
z <- factor(c("single", "married", "married", "single") , levels=c("single", "married"))
z
```

#### Smaller memory as categorical data type
```{r}
x = c("single", "married", "married", "single") # vector
y = factor(c("single", "married", "married", "single"))

object.size(rep(x, 1000))
object.size(rep(y, 1000))
```


## Read and write files (tables)
Besides operating in the R environment, we also want to read data from files or
write results into files.

For tables, R has convenient built-in function `read.table()` and `write.table()`.

### Read file
We can use `read.table()` function to read tables, e.g., in comma separated 
values (csv) or tab separated values (tsv) formats. 
See full manuals: `help(read.table)` or `?read.table` or its 
[online manual](https://www.rdocumentation.org/packages/utils/versions/3.6.2/topics/read.table)
```{r}
help("read.table")

?read.table
```

Here, let's read an example file. Data is available on Moodle and on the 
[github repository](https://github.com/StatBiomed/BMDatSci-book/blob/main/notebooks/chapter1-R)

```{r}
df = read.table("./SRP029880.colData.tsv", sep="\t")
df
```


### Write file
The above file is loaded as a data frame. Here, let's add an extra column to 
indicate if the sample is frozen or not, then save it into a file.
```{r}
df$frozen <- c(1, 1, 0, 0, 0, 1, 1, 0, 0, 0)
write.table(df, "./SRP029880.colData.add_frozen.tsv", sep="\t", quote=FALSE)
```



## Functions and Packages
As experienced above, we have used function multiple times, e.g., 
`read.table` and `typeof`. 

As one more example `mean()` is a function here and it is from the `base` package
```{r}
x <- 4:10
mean(x)

base::mean(x)
```

Generally speaking, A function is a set of statements organized together to 
perform a specific task. Many lines of codes are packed into one function & it’s reusable. 

A function can be written in the same R file and loaded, or it can be 
distributed as part of a package. For using such functions, we need to install
the corresponding package and load it.


### Install packages
It depends on where the package is stored. Please refers to the documentation 
of the specific package you want to install and use.

1. CRAN  (the Comprehensive R Archive Network): main platform
    - For example: `install.packages("ggplot2")`

2. Bioconductor: primarily for biology related packages
    - For example: `BiocManager::install("DESeq2")`

As an example, we can install the powerful plotting package `ggplot2` from CRAN.
```{r}
#install.packages("ggplot2")
```


### Apply function repeatly
We may often want to use a certain function for multiple times, e.g., calculate
the sample mean for many genes. There are multiple ways to achieve it, e.g., via a
`for loop`. Here, we will introduce `apply` and its variants for this purpose.

See more introductions [here](https://ademos.people.uic.edu/Chapter4.html).

#### apply, lapply, sapply and vapply

In short, the `apply()` function and its variants apply a certain function to
each element of a vector, a matrix, or a list.

##### apply for matrix

The `apply(X, MARGIN, FUN)` function works for `matrix` (or `array`) for rows or columns. For example, calculating the median of each column:

```{r}
my.matrx <- matrix(1:15, nrow = 5, ncol = 3)
my.matrx

apply(my.matrx, 2, median)
```

##### lapply, sapply, and vapply for list or vector

The above `apply` function requires `MARGIN`, hence won't work for vector or 
list. There are a few variants to support lists or vectors for different purposes.

From the [manual](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/lapply), we can find out the arguments for these three functions:

* `lapply(X, FUN, …)`
* `sapply(X, FUN, …, simplify = TRUE, USE.NAMES = TRUE)`
* `vapply(X, FUN, FUN.VALUE, …, USE.NAMES = TRUE)`

Let's look at examples. The `lapply` works for vector and list and returns a `list`:
```{r}
A<-c(1:9)
B<-c(1:12)
C<-c(1:15)
my.lst<-list(A,B,C)

lapply(my.lst, median)
```

If you want the return as a vector, you can use `sapply()` for `simplified` output:
```{r}
sapply(my.lst, median)
```

If you want to check the data type of the return, you can further use the `vapply` function. An error will be raised if the data type does not match.

**Note**, the `FUN.VALUE` argument takes the value as a template, so the use of 
`numeric(1)` or any numeric value, e.g., `3` is the same.

```{r}
vapply(my.lst, median, numeric(1))

vapply(my.lst, median, 3)

# try this
# vapply(my.lst, median, character(1))
```



## Plotting
### datasets
Let's use a built-in dataset for illustration: `iris` (4 flower features in 3 plants)
```{r}
head(iris)
```

```{r}
summary(iris)
```

There are two common ways of plotting:

1. the built-in plotting functions
2. the ggplot format

### Basic plotting
#### Histogram
```{r fig1-hist}
hist(iris$Sepal.Length)
```

#### Scatter plot
```{r fig1-scatter}
plot(x=iris$Sepal.Length, y=iris$Sepal.Width)
```

#### boxplot
```{r fig1-box}
x1 <- iris$Sepal.Length[iris$Species == "setosa"]
x2 <- iris$Sepal.Length[iris$Species == "versicolor"]
x3 <- iris$Sepal.Length[iris$Species == "virginica"]

boxplot(x1, x2, x3)
```

### ggplot2
See more instructions: http://www.sthda.com/english/wiki/ggplot2-essentials

#### Install and load package
```{r}
if (!requireNamespace("ggplot2", quietly = TRUE))
    install.packages("ggplot2")

library(ggplot2)
```

#### Histogram
```{r fig1-gghist}
ggplot(iris, aes(x=Sepal.Length)) + 
  geom_histogram(bins = 8)
```

#### Scatter plot
```{r fig1-ggpoint}
ggplot(iris, aes(x=Sepal.Length, y=Sepal.Width)) + 
    geom_point()
```

#### Box plot
```{r fig1-ggbox}
ggplot(iris, aes(x=Species, y=Sepal.Length)) + 
    geom_boxplot()
```

## Scientific and statistical computating
### Orders of operators
See lecture slides.

If you are not sure about a certain ordering, use brackets!

```{r}
5 * 2 > 4
```

```{r}
5 * (2 > 4)
```

### Functions for statistics
More theory and practice to come in next session

### Correlation
```{r}
cor(iris$Sepal.Length, iris$Petal.Length)
```

```{r}
cor.test(iris$Sepal.Length, iris$Petal.Length)
```

### Hypothesis testing (t test)
```{r}
x1 <- iris$Sepal.Length[iris$Species == "setosa"]
x2 <- iris$Sepal.Length[iris$Species == "versicolor"]
x3 <- iris$Sepal.Length[iris$Species == "virginica"]

t.test(x2, x3)
```

### Regression
```{r}
fit <- lm(Sepal.Length ~ Sepal.Width + Petal.Length + Petal.Width, data=iris)
summary(fit) # show results
```

This means the fitted regression is: 

`Sepal.Length ~ 1.856 + 0.65*Sepal.Width + 0.709*Petal.Length - 0.556*Petal.Width`

We can check how good the regression is by plotting it out
```{r}
y_pred <- fit$coefficients[1] + 
  fit$coefficients[2] * iris$Sepal.Width +
  fit$coefficients[3] * iris$Petal.Length +
  fit$coefficients[4] * iris$Petal.Width

cor(iris$Sepal.Length, y_pred)
```


```{r fig1-reg}
plot(iris$Sepal.Length, y_pred)
```


## Resource links

This notebook adapts contents from these resources:

- https://www.datamentor.io/r-programming/
- https://www.geeksforgeeks.org/r-data-types/
- https://data-flair.training/blogs/r-data-types/
- http://www.sthda.com/english/wiki/ggplot2-essentials


