<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Genetic sequence | Biomedical Data Science - introduction with case studies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.44 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Genetic sequence | Biomedical Data Science - introduction with case studies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Genetic sequence | Biomedical Data Science - introduction with case studies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="BIOF1001 teaching team" />


<meta name="date" content="2025-09-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cancer-case2.html"/>
<link rel="next" href="genomics-regression.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">Biomedical Data Science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#introduction-for-readers"><i class="fa fa-check"></i>Introduction for readers</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-you-will-learn-from-this-coursebook"><i class="fa fa-check"></i>What you will learn from this course/book</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-recommend-you-do-while-reading-this-book"><i class="fa fa-check"></i>What we recommend you do while reading this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#other-reference-books"><i class="fa fa-check"></i>Other reference books</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="part"><span><b>I Data Science Foundations</b></span></li>
<li class="chapter" data-level="1" data-path="introR.html"><a href="introR.html"><i class="fa fa-check"></i><b>1</b> Introduction to R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introR.html"><a href="introR.html#introR"><i class="fa fa-check"></i><b>1.1</b> Intro to R programming (Session 1)</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introR.html"><a href="introR.html#r-as-a-basic-calculator"><i class="fa fa-check"></i><b>1.1.1</b> R as a basic calculator</a></li>
<li class="chapter" data-level="1.1.2" data-path="introR.html"><a href="introR.html#variables"><i class="fa fa-check"></i><b>1.1.2</b> Variables</a></li>
<li class="chapter" data-level="1.1.3" data-path="introR.html"><a href="introR.html#data-types"><i class="fa fa-check"></i><b>1.1.3</b> Data types</a></li>
<li class="chapter" data-level="1.1.4" data-path="introR.html"><a href="introR.html#functions"><i class="fa fa-check"></i><b>1.1.4</b> Functions</a></li>
<li class="chapter" data-level="1.1.5" data-path="introR.html"><a href="introR.html#r-scripts"><i class="fa fa-check"></i><b>1.1.5</b> R scripts</a></li>
<li class="chapter" data-level="1.1.6" data-path="introR.html"><a href="introR.html#resource-links"><i class="fa fa-check"></i><b>1.1.6</b> Resource links</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introR.html"><a href="introR.html#introR"><i class="fa fa-check"></i><b>1.2</b> Intro to R programming (Session 2)</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introR.html"><a href="introR.html#data-structures"><i class="fa fa-check"></i><b>1.2.1</b> Data structures</a></li>
<li class="chapter" data-level="1.2.2" data-path="introR.html"><a href="introR.html#vector"><i class="fa fa-check"></i><b>1.2.2</b> Vector</a></li>
<li class="chapter" data-level="1.2.3" data-path="introR.html"><a href="introR.html#matrix"><i class="fa fa-check"></i><b>1.2.3</b> Matrix</a></li>
<li class="chapter" data-level="1.2.4" data-path="introR.html"><a href="introR.html#list"><i class="fa fa-check"></i><b>1.2.4</b> List</a></li>
<li class="chapter" data-level="1.2.5" data-path="introR.html"><a href="introR.html#data-frame"><i class="fa fa-check"></i><b>1.2.5</b> Data Frame</a></li>
<li class="chapter" data-level="1.2.6" data-path="introR.html"><a href="introR.html#readingwriting-data-fromto-r"><i class="fa fa-check"></i><b>1.2.6</b> Reading/writing data from/to R</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="introR.html"><a href="introR.html#introR"><i class="fa fa-check"></i><b>1.3</b> Intro to R programming (Session 4)</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introR.html"><a href="introR.html#operators"><i class="fa fa-check"></i><b>1.3.1</b> Operators</a></li>
<li class="chapter" data-level="1.3.2" data-path="introR.html"><a href="introR.html#ifelse"><i class="fa fa-check"></i><b>1.3.2</b> If/else</a></li>
<li class="chapter" data-level="1.3.3" data-path="introR.html"><a href="introR.html#loop"><i class="fa fa-check"></i><b>1.3.3</b> Loop</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introHypoTest.html"><a href="introHypoTest.html"><i class="fa fa-check"></i><b>2</b> Introduction to Hypothesis testing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-testing-and-p-value"><i class="fa fa-check"></i><b>2.1</b> Hypothesis testing and <em>p</em> value</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-1-probability-of-rolling-a-six"><i class="fa fa-check"></i><b>2.1.1</b> Example 1: probability of rolling a six?</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#permutation-test"><i class="fa fa-check"></i><b>2.2</b> Permutation test</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-2-difference-in-birth-weight"><i class="fa fa-check"></i><b>2.2.1</b> Example 2: difference in birth weight</a></li>
<li class="chapter" data-level="2.2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-approximated-by-resampling"><i class="fa fa-check"></i><b>2.2.2</b> Null distribution approximated by resampling</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introHypoTest.html"><a href="introHypoTest.html#t-test"><i class="fa fa-check"></i><b>2.3</b> <em>t</em> test</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introHypoTest.html"><a href="introHypoTest.html#derivation-of-t-distribution"><i class="fa fa-check"></i><b>2.3.1</b> Derivation of t distribution</a></li>
<li class="chapter" data-level="2.3.2" data-path="introHypoTest.html"><a href="introHypoTest.html#direct-use-of-t.test"><i class="fa fa-check"></i><b>2.3.2</b> Direct use of <code>t.test()</code></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="introHypoTest.html"><a href="introHypoTest.html#regression-based-test"><i class="fa fa-check"></i><b>2.4</b> regression-based test</a></li>
<li class="chapter" data-level="2.5" data-path="introHypoTest.html"><a href="introHypoTest.html#multiple-testing"><i class="fa fa-check"></i><b>2.5</b> Multiple testing</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-test-statistic"><i class="fa fa-check"></i><b>2.5.1</b> Null distribution (of test statistic)</a></li>
<li class="chapter" data-level="2.5.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-p-value"><i class="fa fa-check"></i><b>2.5.2</b> Null distribution of p value</a></li>
<li class="chapter" data-level="2.5.3" data-path="introHypoTest.html"><a href="introHypoTest.html#minimal-p-values-in-10-tests"><i class="fa fa-check"></i><b>2.5.3</b> Minimal p values in 10 tests</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="introHypoTest.html"><a href="introHypoTest.html#explore-power-and-sample-size-optional-power"><i class="fa fa-check"></i><b>2.6</b> Explore power and sample size (optional) { power }</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introLinearReg.html"><a href="introLinearReg.html"><i class="fa fa-check"></i><b>3</b> Introduction to Linear Regression</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#linear-regression-using-simulated-data"><i class="fa fa-check"></i><b>3.1</b> Linear Regression Using Simulated Data</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="introLinearReg.html"><a href="introLinearReg.html#simulating-data"><i class="fa fa-check"></i><b>3.1.1</b> Simulating data:</a></li>
<li class="chapter" data-level="3.1.2" data-path="introLinearReg.html"><a href="introLinearReg.html#model-efficacy"><i class="fa fa-check"></i><b>3.1.2</b> Model efficacy</a></li>
<li class="chapter" data-level="3.1.3" data-path="introLinearReg.html"><a href="introLinearReg.html#r-squared"><i class="fa fa-check"></i><b>3.1.3</b> <em>R-Squared</em></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#least-squares-using-simulated-data"><i class="fa fa-check"></i><b>3.2</b> Least Squares Using Simulated Data</a></li>
<li class="chapter" data-level="3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#diagnostic-check-of-a-fitted-regression-model"><i class="fa fa-check"></i><b>3.3</b> Diagnostic check of a fitted regression model</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#residual-standard-errors-rse"><i class="fa fa-check"></i><b>3.3.1</b> Residual Standard Errors (RSE)</a></li>
<li class="chapter" data-level="3.3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#p-values"><i class="fa fa-check"></i><b>3.3.2</b> p-values</a></li>
<li class="chapter" data-level="3.3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#f-statistics"><i class="fa fa-check"></i><b>3.3.3</b> F-statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introLinearReg.html"><a href="introLinearReg.html#simple-linear-regression-with-lm-function"><i class="fa fa-check"></i><b>3.4</b> Simple Linear Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="3.5" data-path="introLinearReg.html"><a href="introLinearReg.html#multiple-regression-with-lm-function"><i class="fa fa-check"></i><b>3.5</b> Multiple Regression with <code>lm</code> function</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introClassifier.html"><a href="introClassifier.html"><i class="fa fa-check"></i><b>4</b> Introduction to Classification</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introClassifier.html"><a href="introClassifier.html#visualise-logistic-and-logit-functions"><i class="fa fa-check"></i><b>4.1</b> Visualise logistic and logit functions</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="introClassifier.html"><a href="introClassifier.html#logistic-function"><i class="fa fa-check"></i><b>4.1.1</b> Logistic function</a></li>
<li class="chapter" data-level="4.1.2" data-path="introClassifier.html"><a href="introClassifier.html#logit-function"><i class="fa fa-check"></i><b>4.1.2</b> Logit function</a></li>
<li class="chapter" data-level="4.1.3" data-path="introClassifier.html"><a href="introClassifier.html#visualise-the-distribution"><i class="fa fa-check"></i><b>4.1.3</b> Visualise the distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="introClassifier.html"><a href="introClassifier.html#logistic-regression-on-diabetes"><i class="fa fa-check"></i><b>4.2</b> Logistic regression on Diabetes</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="introClassifier.html"><a href="introClassifier.html#load-pima-indians-diabetes-database"><i class="fa fa-check"></i><b>4.2.1</b> Load Pima Indians Diabetes Database</a></li>
<li class="chapter" data-level="4.2.2" data-path="introClassifier.html"><a href="introClassifier.html#fit-logistic-regression"><i class="fa fa-check"></i><b>4.2.2</b> Fit logistic regression</a></li>
<li class="chapter" data-level="4.2.3" data-path="introClassifier.html"><a href="introClassifier.html#assess-on-test-data"><i class="fa fa-check"></i><b>4.2.3</b> Assess on test data</a></li>
<li class="chapter" data-level="4.2.4" data-path="introClassifier.html"><a href="introClassifier.html#model-selection-and-diagnosis"><i class="fa fa-check"></i><b>4.2.4</b> Model selection and diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="introClassifier.html"><a href="introClassifier.html#cross-validation"><i class="fa fa-check"></i><b>4.3</b> Cross-validation</a></li>
<li class="chapter" data-level="4.4" data-path="introClassifier.html"><a href="introClassifier.html#more-assessment-metrics"><i class="fa fa-check"></i><b>4.4</b> More assessment metrics</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="introClassifier.html"><a href="introClassifier.html#two-types-of-error"><i class="fa fa-check"></i><b>4.4.1</b> Two types of error</a></li>
<li class="chapter" data-level="4.4.2" data-path="introClassifier.html"><a href="introClassifier.html#roc-curve"><i class="fa fa-check"></i><b>4.4.2</b> ROC curve</a></li>
<li class="chapter" data-level="4.4.3" data-path="introClassifier.html"><a href="introClassifier.html#homework"><i class="fa fa-check"></i><b>4.4.3</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Biomedical Data Modules</b></span></li>
<li class="chapter" data-level="5" data-path="image-digital.html"><a href="image-digital.html"><i class="fa fa-check"></i><b>5</b> Medical Image and Digital Health</a></li>
<li class="chapter" data-level="6" data-path="cancer.html"><a href="cancer.html"><i class="fa fa-check"></i><b>6</b> Cancer genomics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="cancer.html"><a href="cancer.html#cancer-case1"><i class="fa fa-check"></i><b>6.1</b> Case study 1: analysis of cBioportal mutation data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="cancer.html"><a href="cancer.html#exploratory-analysis"><i class="fa fa-check"></i><b>6.1.1</b> Exploratory analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="cancer.html"><a href="cancer.html#statistical-analysis"><i class="fa fa-check"></i><b>6.1.2</b> Statistical analysis</a></li>
<li class="chapter" data-level="6.1.3" data-path="cancer.html"><a href="cancer.html#literature-search"><i class="fa fa-check"></i><b>6.1.3</b> Literature search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="epidemiology.html"><a href="epidemiology.html"><i class="fa fa-check"></i><b>7</b> Epidemiology</a></li>
<li class="chapter" data-level="8" data-path="pop-genetics.html"><a href="pop-genetics.html"><i class="fa fa-check"></i><b>8</b> Population Genetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="pop-genetics.html"><a href="pop-genetics.html#case-study-1-heritability-and-human-traits"><i class="fa fa-check"></i><b>8.1</b> Case study 1: Heritability and human traits</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="pop-genetics.html"><a href="pop-genetics.html#part-1"><i class="fa fa-check"></i><b>8.1.1</b> Part 1</a></li>
<li class="chapter" data-level="8.1.2" data-path="pop-genetics.html"><a href="pop-genetics.html#part-2"><i class="fa fa-check"></i><b>8.1.2</b> Part 2</a></li>
<li class="chapter" data-level="8.1.3" data-path="pop-genetics.html"><a href="pop-genetics.html#references"><i class="fa fa-check"></i><b>8.1.3</b> References</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Modules in previous years</b></span></li>
<li class="chapter" data-level="9" data-path="cancer-case2.html"><a href="cancer-case2.html"><i class="fa fa-check"></i><b>9</b> Cancer Epidemiology</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="cancer-case2.html"><a href="cancer-case2.html#scenario"><i class="fa fa-check"></i><b>9.0.1</b> Scenario</a></li>
<li class="chapter" data-level="9.0.2" data-path="cancer-case2.html"><a href="cancer-case2.html#hong-kong-population"><i class="fa fa-check"></i><b>9.0.2</b> Hong Kong population</a></li>
<li class="chapter" data-level="9.0.3" data-path="cancer-case2.html"><a href="cancer-case2.html#cancer-registry-data"><i class="fa fa-check"></i><b>9.0.3</b> Cancer registry data</a></li>
<li class="chapter" data-level="9.0.4" data-path="cancer-case2.html"><a href="cancer-case2.html#existing-cancer-funding-and-publication-data"><i class="fa fa-check"></i><b>9.0.4</b> Existing cancer funding and publication data</a></li>
<li class="chapter" data-level="9.0.5" data-path="cancer-case2.html"><a href="cancer-case2.html#open-discussion"><i class="fa fa-check"></i><b>9.0.5</b> Open discussion</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="genetic-seq.html"><a href="genetic-seq.html"><i class="fa fa-check"></i><b>10</b> Genetic sequence</a>
<ul>
<li class="chapter" data-level="10.1" data-path="genetic-seq.html"><a href="genetic-seq.html#case-study-1-genetic-sequence-analysis"><i class="fa fa-check"></i><b>10.1</b> Case study 1: Genetic sequence analysis</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="genetic-seq.html"><a href="genetic-seq.html#sequence-motif-and-k-mer"><i class="fa fa-check"></i><b>10.1.1</b> Sequence motif and k-mer</a></li>
<li class="chapter" data-level="10.1.2" data-path="genetic-seq.html"><a href="genetic-seq.html#functional-mapping"><i class="fa fa-check"></i><b>10.1.2</b> Functional mapping</a></li>
<li class="chapter" data-level="10.1.3" data-path="genetic-seq.html"><a href="genetic-seq.html#references-1"><i class="fa fa-check"></i><b>10.1.3</b> References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="genomics-regression.html"><a href="genomics-regression.html"><i class="fa fa-check"></i><b>11</b> Genomics and prediction</a>
<ul>
<li class="chapter" data-level="11.1" data-path="genomics-regression.html"><a href="genomics-regression.html#genomics-case1"><i class="fa fa-check"></i><b>11.1</b> Case study 1: splicing fedility prediction</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="genomics-regression.html"><a href="genomics-regression.html#questions-for-discussion-1"><i class="fa fa-check"></i><b>11.1.1</b> Questions for Discussion</a></li>
<li class="chapter" data-level="11.1.2" data-path="genomics-regression.html"><a href="genomics-regression.html#hands-on-with-regression-model"><i class="fa fa-check"></i><b>11.1.2</b> Hands-on with regression model</a></li>
<li class="chapter" data-level="11.1.3" data-path="genomics-regression.html"><a href="genomics-regression.html#open-question"><i class="fa fa-check"></i><b>11.1.3</b> Open question</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i>Appendix A: Install R &amp; RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.1-install-r-4.3.1"><i class="fa fa-check"></i>A.1 Install R (&gt;=4.3.1)</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-windows"><i class="fa fa-check"></i>R on Windows</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-macos"><i class="fa fa-check"></i>R on macOS</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-linux-ubuntu"><i class="fa fa-check"></i>R on Linux (Ubuntu)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.2-install-rstudio"><i class="fa fa-check"></i>A.2 Install RStudio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.3-use-r-inside-rstudio"><i class="fa fa-check"></i>A.3 Use R inside RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-studio"><i class="fa fa-check"></i>R studio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#some-general-knowledge"><i class="fa fa-check"></i>Some general knowledge</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#install-packages"><i class="fa fa-check"></i>Install packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a4.-cloud-computing"><i class="fa fa-check"></i>A4. Cloud computing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-2.html"><a href="references-2.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Data Science - introduction with case studies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genetic-seq" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Genetic sequence<a href="genetic-seq.html#genetic-seq" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="case-study-1-genetic-sequence-analysis" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Case study 1: Genetic sequence analysis<a href="genetic-seq.html#case-study-1-genetic-sequence-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="sequence-motif-and-k-mer" class="section level3 hasAnchor" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> Sequence motif and k-mer<a href="genetic-seq.html#sequence-motif-and-k-mer" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Scenario:</strong> You are a biomedical data science trainee, and you want to use
some real case studies to understand how to perform sequence analysis. From these
practices, you hope to understand how the sickle cell disease happens and how
the choice of gene editing site makes a potential therapy.</p>
<p>From the lecture, you already understood that:</p>
<ol style="list-style-type: decimal">
<li>the sickle cell disease happens due to a single nucleotide mutation on HBB (
coding the hemoglobin bata chain for adult use), causing a missense mutation on
amino acid 7 from E to V.
You can view more from ClinVar
<a href="https://www.ncbi.nlm.nih.gov/clinvar/RCV000016574/">RCV000016574</a> and
OMIM <a href="https://www.omim.org/entry/603903">603903</a>.</li>
<li>the therapy was to re-activate another gene HBG (gamma chain) that was used
in the fetus. The HBG gene was repressed by a transcription factor BCL11A (as a
protein) that serves as a repressor to HBG.</li>
</ol>
<p><strong>Q1.</strong> How can we understand where the protein of the BCL11A gene prefers to
bind on DNA?</p>
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Using the binding motif database JASPAR: BCL11A’s motif is
<a href="https://jaspar.elixir.no/matrix/MA2324.1/">MA2324.1</a></li>
<li>It collected the sequences of many binding sites already, check the
<code>FASTA file</code> or the
<a href="https://jaspar.elixir.no/sites/MA2324.1">HTML file</a>.</li>
<li>It’s Motif can be viewed as this file
<img src="https://jaspar.elixir.no/static/logos/all/svg/MA2324.1.svg" alt="MA2324.1 Motif logo" /></li>
</ul>
</details>
<p><br /></p>
<p><strong>Q2.</strong> What is the meaning of position frequency matrix (PFM) and how to
convert it into a position probability matrix (PPM, often used as position
weight matrix PWM)?</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="genetic-seq.html#cb181-1" tabindex="-1"></a>MA2324_PFM <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(</span>
<span id="cb181-2"><a href="genetic-seq.html#cb181-2" tabindex="-1"></a>  <span class="dv">343</span>,    <span class="dv">180</span>,    <span class="dv">125</span>,   <span class="dv">5724</span>,    <span class="dv">155</span>,    <span class="dv">143</span>,   <span class="dv">4495</span>,</span>
<span id="cb181-3"><a href="genetic-seq.html#cb181-3" tabindex="-1"></a>  <span class="dv">5024</span>,   <span class="dv">120</span>,    <span class="dv">281</span>,     <span class="dv">92</span>,   <span class="dv">5855</span>,   <span class="dv">5964</span>,    <span class="dv">641</span>,</span>
<span id="cb181-4"><a href="genetic-seq.html#cb181-4" tabindex="-1"></a>  <span class="dv">591</span>,    <span class="dv">165</span>,   <span class="dv">5747</span>,    <span class="dv">100</span>,    <span class="dv">141</span>,     <span class="dv">55</span>,    <span class="dv">385</span>,</span>
<span id="cb181-5"><a href="genetic-seq.html#cb181-5" tabindex="-1"></a>  <span class="dv">307</span>,   <span class="dv">5800</span>,    <span class="dv">112</span>,    <span class="dv">349</span>,    <span class="dv">114</span>,    <span class="dv">103</span>,    <span class="dv">744</span>),</span>
<span id="cb181-6"><a href="genetic-seq.html#cb181-6" tabindex="-1"></a>  <span class="at">nrow =</span> <span class="dv">4</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span></span>
<span id="cb181-7"><a href="genetic-seq.html#cb181-7" tabindex="-1"></a>)</span>
<span id="cb181-8"><a href="genetic-seq.html#cb181-8" tabindex="-1"></a><span class="fu">rownames</span>(MA2324_PFM) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;T&quot;</span>)</span>
<span id="cb181-9"><a href="genetic-seq.html#cb181-9" tabindex="-1"></a>MA2324_PFM</span>
<span id="cb181-10"><a href="genetic-seq.html#cb181-10" tabindex="-1"></a><span class="co">#&gt;   [,1] [,2] [,3] [,4] [,5] [,6] [,7]</span></span>
<span id="cb181-11"><a href="genetic-seq.html#cb181-11" tabindex="-1"></a><span class="co">#&gt; A  343  180  125 5724  155  143 4495</span></span>
<span id="cb181-12"><a href="genetic-seq.html#cb181-12" tabindex="-1"></a><span class="co">#&gt; C 5024  120  281   92 5855 5964  641</span></span>
<span id="cb181-13"><a href="genetic-seq.html#cb181-13" tabindex="-1"></a><span class="co">#&gt; G  591  165 5747  100  141   55  385</span></span>
<span id="cb181-14"><a href="genetic-seq.html#cb181-14" tabindex="-1"></a><span class="co">#&gt; T  307 5800  112  349  114  103  744</span></span>
<span id="cb181-15"><a href="genetic-seq.html#cb181-15" tabindex="-1"></a></span>
<span id="cb181-16"><a href="genetic-seq.html#cb181-16" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: define position probability matrix</span></span></code></pre></div>
<details>
<summary>
<strong>HowTo:</strong>
</summary>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="genetic-seq.html#cb182-1" tabindex="-1"></a>MA2324_PWM <span class="ot">=</span> MA2324_PFM <span class="sc">/</span> <span class="fu">sum</span>(MA2324_PFM[, <span class="dv">1</span>])</span>
<span id="cb182-2"><a href="genetic-seq.html#cb182-2" tabindex="-1"></a>MA2324_PWM</span>
<span id="cb182-3"><a href="genetic-seq.html#cb182-3" tabindex="-1"></a><span class="co">#&gt;         [,1]       [,2]       [,3]       [,4]       [,5]        [,6]       [,7]</span></span>
<span id="cb182-4"><a href="genetic-seq.html#cb182-4" tabindex="-1"></a><span class="co">#&gt; A 0.05474860 0.02873105 0.01995211 0.91364725 0.02474062 0.022825219 0.71747805</span></span>
<span id="cb182-5"><a href="genetic-seq.html#cb182-5" tabindex="-1"></a><span class="co">#&gt; C 0.80191540 0.01915403 0.04485235 0.01468476 0.93455706 0.951955307 0.10231445</span></span>
<span id="cb182-6"><a href="genetic-seq.html#cb182-6" tabindex="-1"></a><span class="co">#&gt; G 0.09433360 0.02633679 0.91731844 0.01596169 0.02250599 0.008778931 0.06145251</span></span>
<span id="cb182-7"><a href="genetic-seq.html#cb182-7" tabindex="-1"></a><span class="co">#&gt; T 0.04900239 0.92577813 0.01787709 0.05570630 0.01819633 0.016440543 0.11875499</span></span></code></pre></div>
</details>
<p><br /></p>
<p><strong>Q3.</strong> Let’s make the Motif logo ourselves by using
<a href="https://weblogo.threeplusone.com/create.cgi">weblogo.threeplusone.com</a></p>
<p>You may use the first 500 lines of the sequences that we compiled:
<a href="https://github.com/StatBiomed/BMDS-book/blob/main/notebooks/module4-genetics/MA2324.1.sites-h500.txt">MA2324.1.sites-h500.txt</a></p>
<details>
<summary>
<strong>HowTo:</strong>
</summary>
<ul>
<li>Copy and paste to <a href="https://weblogo.threeplusone.com/create.cgi">weblogo.threeplusone.com</a></li>
<li>Check which position has the highest information content in <em>bit</em>.</li>
</ul>
</details>
<p><br /></p>
<p><strong>Q4.</strong> Given the motif, how good is a certain sequence in terms of consistency?
Take these two sequences as examples:</p>
<ul>
<li>sequence 1: <code>CGGACCA</code> (&gt;hg38_chr1:1000830-1000836(+))</li>
<li>sequence 2: <code>CTGACCG</code> (hg38_chr1:940785-940791(-))</li>
</ul>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="genetic-seq.html#cb183-1" tabindex="-1"></a><span class="co"># Use this One hot encoding function</span></span>
<span id="cb183-2"><a href="genetic-seq.html#cb183-2" tabindex="-1"></a>Onehot_Encode <span class="ot">&lt;-</span> <span class="cf">function</span>(sequence) {</span>
<span id="cb183-3"><a href="genetic-seq.html#cb183-3" tabindex="-1"></a>  idx <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">strsplit</span>(sequence, <span class="st">&#39;&#39;</span>)[[<span class="dv">1</span>]], <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;T&quot;</span>))</span>
<span id="cb183-4"><a href="genetic-seq.html#cb183-4" tabindex="-1"></a>  seq_mat <span class="ot">=</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="dv">4</span>, <span class="fu">length</span>(idx))</span>
<span id="cb183-5"><a href="genetic-seq.html#cb183-5" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(idx)) seq_mat[idx[i], i] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb183-6"><a href="genetic-seq.html#cb183-6" tabindex="-1"></a>  seq_mat</span>
<span id="cb183-7"><a href="genetic-seq.html#cb183-7" tabindex="-1"></a>}</span>
<span id="cb183-8"><a href="genetic-seq.html#cb183-8" tabindex="-1"></a></span>
<span id="cb183-9"><a href="genetic-seq.html#cb183-9" tabindex="-1"></a>seq_vec1 <span class="ot">=</span> <span class="fu">Onehot_Encode</span>(<span class="st">&quot;CGGACCA&quot;</span>)</span>
<span id="cb183-10"><a href="genetic-seq.html#cb183-10" tabindex="-1"></a>seq_vec2 <span class="ot">=</span> <span class="fu">Onehot_Encode</span>(<span class="st">&quot;CTGACCG&quot;</span>)</span>
<span id="cb183-11"><a href="genetic-seq.html#cb183-11" tabindex="-1"></a>seq_vec2</span>
<span id="cb183-12"><a href="genetic-seq.html#cb183-12" tabindex="-1"></a><span class="co">#&gt;      [,1] [,2] [,3] [,4] [,5] [,6] [,7]</span></span>
<span id="cb183-13"><a href="genetic-seq.html#cb183-13" tabindex="-1"></a><span class="co">#&gt; [1,]    0    0    0    1    0    0    0</span></span>
<span id="cb183-14"><a href="genetic-seq.html#cb183-14" tabindex="-1"></a><span class="co">#&gt; [2,]    1    0    0    0    1    1    0</span></span>
<span id="cb183-15"><a href="genetic-seq.html#cb183-15" tabindex="-1"></a><span class="co">#&gt; [3,]    0    0    1    0    0    0    1</span></span>
<span id="cb183-16"><a href="genetic-seq.html#cb183-16" tabindex="-1"></a><span class="co">#&gt; [4,]    0    1    0    0    0    0    0</span></span></code></pre></div>
<p>To calculate the consistency score of a certain sequence <span class="math inline">\(S\)</span> to a given motif
(<span class="math inline">\(M\)</span> for the motif position weight matrix), you may recall what we
introduced in the lecture (same as the reference papers [1, 2]):</p>
<p><span class="math inline">\(P(S|M) = \prod_{i=1}^k{M[s_i, i]}\)</span></p>
<p>Alternatively, you can also transform the sequence via our one hot encoding.
Now, please try and see if you can calculate the motif score for the above two
sequences (possibly in log10-scale).</p>
<details>
<summary>
<strong>Example solution scripts:</strong>
</summary>
<ul>
<li><p>Meaning of motif score function: the log-likelihood of seeing a binding site
sequence given the binding motif.</p></li>
<li><p>Manual calculation step by step:</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="genetic-seq.html#cb184-1" tabindex="-1"></a><span class="fu">colSums</span>(seq_vec1 <span class="sc">*</span> MA2324_PWM)</span>
<span id="cb184-2"><a href="genetic-seq.html#cb184-2" tabindex="-1"></a><span class="co">#&gt; [1] 0.80191540 0.02633679 0.91731844 0.91364725 0.93455706 0.95195531 0.71747805</span></span>
<span id="cb184-3"><a href="genetic-seq.html#cb184-3" tabindex="-1"></a></span>
<span id="cb184-4"><a href="genetic-seq.html#cb184-4" tabindex="-1"></a><span class="fu">colSums</span>(seq_vec2 <span class="sc">*</span> MA2324_PWM)</span>
<span id="cb184-5"><a href="genetic-seq.html#cb184-5" tabindex="-1"></a><span class="co">#&gt; [1] 0.80191540 0.92577813 0.91731844 0.91364725 0.93455706 0.95195531 0.06145251</span></span>
<span id="cb184-6"><a href="genetic-seq.html#cb184-6" tabindex="-1"></a></span>
<span id="cb184-7"><a href="genetic-seq.html#cb184-7" tabindex="-1"></a>seq_score1 <span class="ot">=</span> <span class="fu">sum</span>(<span class="fu">log10</span>(<span class="fu">colSums</span>(seq_vec1 <span class="sc">*</span> MA2324_PWM)))</span>
<span id="cb184-8"><a href="genetic-seq.html#cb184-8" tabindex="-1"></a>seq_score2 <span class="ot">=</span> <span class="fu">sum</span>(<span class="fu">log10</span>(<span class="fu">colSums</span>(seq_vec2 <span class="sc">*</span> MA2324_PWM)))</span>
<span id="cb184-9"><a href="genetic-seq.html#cb184-9" tabindex="-1"></a></span>
<span id="cb184-10"><a href="genetic-seq.html#cb184-10" tabindex="-1"></a><span class="fu">c</span>(seq_score1, seq_score2)</span>
<span id="cb184-11"><a href="genetic-seq.html#cb184-11" tabindex="-1"></a><span class="co">#&gt; [1] -1.946979 -1.468304</span></span></code></pre></div></li>
</ul>
</details>
<p><br /></p>
<p><strong>Q5.</strong> Given a segment of a long sequence, how to find the potential motif
sites?</p>
<ul>
<li>Scan the whole sequence</li>
<li>Find the best matched position</li>
</ul>
<p>Here, we will take the gene <a href="https://www.ncbi.nlm.nih.gov/gene/3048">HBG2</a>
as an example, with the initial sequence segment as follows:</p>
<pre><code>ref|NC_000011.10|:c5255019-5254782 Homo sapiens chromosome 11, GRCh38.p14 Primary Assembly
ATAAAAAAAATTAAGCAGCAGTATCCTCTTGGGGGCCCCTTCCCCACACTATCTCAATGCAAATATCTGT
CTGAAACGGTCCCTGGCTAAACTCCACCCATGGGTTGGCCAGCCTTGCCTTGACCAATAGCCTTGACAAG
GCAAACTTGACCAATAGTCTTAGAGTATCCAGTGAGGCCAGGGGCCGGCGGCTGGCTAGGGATGAAGAAT
AAAAGGAAGCACCCTTCAGCAGTTCCAC</code></pre>
<p>We may load the sequence into R by our <code>Onehot_Encode</code> function:</p>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="genetic-seq.html#cb186-1" tabindex="-1"></a>HBG2_segment <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb186-2"><a href="genetic-seq.html#cb186-2" tabindex="-1"></a>  <span class="st">&quot;ATAAAAAAAATTAAGCAGCAGTATCCTCTTGGGGGCCCCTTCCCCACACTATCTCAATGCAAATATCTGT&quot;</span>,</span>
<span id="cb186-3"><a href="genetic-seq.html#cb186-3" tabindex="-1"></a>  <span class="st">&quot;CTGAAACGGTCCCTGGCTAAACTCCACCCATGGGTTGGCCAGCCTTGCCTTGACCAATAGCCTTGACAAG&quot;</span>,</span>
<span id="cb186-4"><a href="genetic-seq.html#cb186-4" tabindex="-1"></a>  <span class="st">&quot;GCAAACTTGACCAATAGTCTTAGAGTATCCAGTGAGGCCAGGGGCCGGCGGCTGGCTAGGGATGAAGAAT&quot;</span>,</span>
<span id="cb186-5"><a href="genetic-seq.html#cb186-5" tabindex="-1"></a>  <span class="st">&quot;AAAAGGAAGCACCCTTCAGCAGTTCCAC&quot;</span></span>
<span id="cb186-6"><a href="genetic-seq.html#cb186-6" tabindex="-1"></a>)</span>
<span id="cb186-7"><a href="genetic-seq.html#cb186-7" tabindex="-1"></a>HBG2_segment</span>
<span id="cb186-8"><a href="genetic-seq.html#cb186-8" tabindex="-1"></a><span class="co">#&gt; [1] &quot;ATAAAAAAAATTAAGCAGCAGTATCCTCTTGGGGGCCCCTTCCCCACACTATCTCAATGCAAATATCTGTCTGAAACGGTCCCTGGCTAAACTCCACCCATGGGTTGGCCAGCCTTGCCTTGACCAATAGCCTTGACAAGGCAAACTTGACCAATAGTCTTAGAGTATCCAGTGAGGCCAGGGGCCGGCGGCTGGCTAGGGATGAAGAATAAAAGGAAGCACCCTTCAGCAGTTCCAC&quot;</span></span>
<span id="cb186-9"><a href="genetic-seq.html#cb186-9" tabindex="-1"></a></span>
<span id="cb186-10"><a href="genetic-seq.html#cb186-10" tabindex="-1"></a>HBG2_seg_vec <span class="ot">=</span> <span class="fu">Onehot_Encode</span>(HBG2_segment)</span>
<span id="cb186-11"><a href="genetic-seq.html#cb186-11" tabindex="-1"></a><span class="co"># HBG2_seg_vec</span></span></code></pre></div>
<details>
<summary>
<strong>Example solution scripts:</strong>
</summary>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="genetic-seq.html#cb187-1" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: try it yourself</span></span></code></pre></div>
</details>
<p><br /></p>
<p><strong>Q6.</strong> With the same sequence above, count the frequency of each 3-mer:</p>
<ul>
<li>List the all 3-mer</li>
<li>Count each of the 3-mer in the sequence</li>
</ul>
<details>
<summary>
<strong>Example solution scripts:</strong>
</summary>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="genetic-seq.html#cb188-1" tabindex="-1"></a><span class="co"># </span><span class="al">TODO</span><span class="co">: try it yourself</span></span>
<span id="cb188-2"><a href="genetic-seq.html#cb188-2" tabindex="-1"></a></span>
<span id="cb188-3"><a href="genetic-seq.html#cb188-3" tabindex="-1"></a><span class="co"># Option 1: keep using one hot encoding matrix</span></span>
<span id="cb188-4"><a href="genetic-seq.html#cb188-4" tabindex="-1"></a></span>
<span id="cb188-5"><a href="genetic-seq.html#cb188-5" tabindex="-1"></a><span class="co"># Option 2: substr(x, start, stop) </span></span>
<span id="cb188-6"><a href="genetic-seq.html#cb188-6" tabindex="-1"></a><span class="co"># you may make a vector with k-mer as name and use the above substr as index</span></span>
<span id="cb188-7"><a href="genetic-seq.html#cb188-7" tabindex="-1"></a><span class="co"># this is related to dictionary data structure</span></span>
<span id="cb188-8"><a href="genetic-seq.html#cb188-8" tabindex="-1"></a></span>
<span id="cb188-9"><a href="genetic-seq.html#cb188-9" tabindex="-1"></a>k <span class="ot">=</span> <span class="dv">3</span></span>
<span id="cb188-10"><a href="genetic-seq.html#cb188-10" tabindex="-1"></a>bases <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;T&quot;</span>)</span>
<span id="cb188-11"><a href="genetic-seq.html#cb188-11" tabindex="-1"></a>kmer_mat <span class="ot">=</span> <span class="fu">unique</span>(<span class="fu">t</span>(<span class="fu">combn</span>(<span class="fu">rep</span>(bases, k), <span class="at">m =</span> k)))</span>
<span id="cb188-12"><a href="genetic-seq.html#cb188-12" tabindex="-1"></a>kmer <span class="ot">=</span> <span class="fu">apply</span>(kmer_mat, <span class="dv">1</span>, paste, <span class="at">collapse=</span><span class="st">&#39;&#39;</span>)</span>
<span id="cb188-13"><a href="genetic-seq.html#cb188-13" tabindex="-1"></a></span>
<span id="cb188-14"><a href="genetic-seq.html#cb188-14" tabindex="-1"></a>kmer_count <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="fu">length</span>(kmer))</span>
<span id="cb188-15"><a href="genetic-seq.html#cb188-15" tabindex="-1"></a><span class="fu">names</span>(kmer_count) <span class="ot">=</span> kmer</span>
<span id="cb188-16"><a href="genetic-seq.html#cb188-16" tabindex="-1"></a></span>
<span id="cb188-17"><a href="genetic-seq.html#cb188-17" tabindex="-1"></a><span class="co"># Keep trying</span></span></code></pre></div>
</details>
<p><br /></p>
</div>
<div id="functional-mapping" class="section level3 hasAnchor" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> Functional mapping<a href="genetic-seq.html#functional-mapping" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this part, we will see how the sequence features may help predict molecular
phenotype, namely splicing efficiency. You may read more in the original paper
<a href="https://doi.org/10.1093/bioinformatics/btac321">Hou &amp; Huang 2021</a> [4], if you
are interested.</p>
<p>Let’s look at the data
<a href="https://github.com/StatBiomed/BMDS-book/blob/main/notebooks/module4-genetics/plicing_efficiency_pred.csv">plicing_efficiency_pred.csv</a></p>
<ul>
<li>columns 1 &amp; 2: gene ID and gene name (1850 genes)</li>
<li>column 3: splicing efficiency (log scale)</li>
<li>columns 4 to 99: 96 octamer features (the normalized frequency in intron
sequence)</li>
</ul>
<p><strong>Q7.</strong> Now, we can try using the multiple linear regression model to check how
predictive the octamer frequencies are for splicing efficiency.</p>
<ul>
<li><p>Load the data (scripts below)</p></li>
<li><p>Linear regression model</p></li>
<li><p>Using 10-fold cross-validation to evaluate the model</p></li>
<li><p>You may propose other alternative methods for assessing the
association between octamer and splicing efficiency.</p>
<pre><code>library(caret)

df_splicing &lt;- read.csv(&quot;plicing_efficiency_pred.csv&quot;)

# Define training control
# We also want to have savePredictions=TRUE &amp; classProbs=TRUE
set.seed(0) 
my_trControl &lt;- trainControl(method = &quot;cv&quot;, number = 10, 
                             savePredictions = TRUE)

# TODO: try it yourself to fill the rest</code></pre>
<p><br /></p></li>
</ul>
</div>
<div id="references-1" class="section level3 hasAnchor" number="10.1.3">
<h3><span class="header-section-number">10.1.3</span> References<a href="genetic-seq.html#references-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Crooks, G. E., Hon, G., Chandonia, J. M., &amp; Brenner, S. E. (2004).
<a href="https://doi.org/10.1101/gr.849004">WebLogo: a sequence logo generator</a>.
Genome research, 14(6), 1188-1190.</li>
<li>Schneider, T. D., &amp; Stephens, R. M. (1990).
<a href="https://doi.org/10.1093/nar/18.20.6097">Sequence logos: a new way to display consensus sequences</a>.
Nucleic acids research, 18(20), 6097-6100.</li>
<li>Canver, M. C., Smith, E. C., Sher, F., Pinello, L., Sanjana, N. E., Shalem,
O., … &amp; Bauer, D. E. (2015).
<a href="https://www.nature.com/articles/nature15521">BCL11A enhancer dissection by Cas9-mediated in situ saturating mutagenesis</a>.
Nature, 527(7577), 192-197.</li>
<li>Hou, R., &amp; Huang, Y. (2022).
<a href="https://doi.org/10.1093/bioinformatics/btac321">Genomic sequences and RNA-binding proteins predict RNA splicing efficiency
in various single-cell contexts</a>.
Bioinformatics, 38(12), 3231-3237.</li>
<li>Splicing efficiency prediction dataset at
<a href="https://github.com/StatBiomed/scRNA-efficiency-prediction" class="uri">https://github.com/StatBiomed/scRNA-efficiency-prediction</a>.
<details>
<summary>
<strong>scripts to subset the data :</strong>
</summary>
<pre><code> df_Y = read.csv(paste0(
   &quot;https://github.com/StatBiomed/scRNA-efficiency-prediction/&quot;, 
   &quot;raw/refs/heads/main/data/estimated/sck562_all_stochastical_gamma.csv&quot;
 ), row.names = 1)

 df_X0 = read.csv(paste0(
   &quot;https://github.com/StatBiomed/scRNA-efficiency-prediction/&quot;, 
   &quot;raw/refs/heads/main/data/features/humanproteincode_octamer_gene_level.csv&quot;
 ))

 df_use = merge(df_Y, df_X0, by=&quot;gene_id&quot;, incomparables = NA)
 df_use$velocity_gamma = -log(df_use$velocity_gamma + 0.01)
 colnames(df_use)[3] = &quot;splicing_efficiency&quot;
 # df_use = df_use[-1600, ]

 df_use[, 4:ncol(df_use)] = df_use[, 4:ncol(df_use)] + 0.00001
 df_use[, 3:ncol(df_use)] = round(df_use[, 3:ncol(df_use)], digits=6)

 options(digits = 3)
 write.csv(df_use, &quot;plicing_efficiency_pred.csv&quot;, 
           quote = FALSE, row.names = FALSE)</code></pre>
</details>
<br /></li>
</ol>
<div id="potential-resources" class="section level4 hasAnchor" number="10.1.3.1">
<h4><span class="header-section-number">10.1.3.1</span> Potential resources<a href="genetic-seq.html#potential-resources" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<details>
<summary>
<strong>Scan motif</strong>
</summary>
<pre><code># For Q5
motif_score_all = rep(NA, ncol(HBG2_seg_vec)-ncol(MA2324_PWM))
for (i in 1:(ncol(HBG2_seg_vec) - ncol(MA2324_PWM))) {
  motif_score_all[i] = sum(log10(colSums(
    HBG2_seg_vec[, i:(i + ncol(MA2324_PWM) - 1)] * MA2324_PWM
    )))
}
motif_score_all[1:5]

sort(motif_score_all, decreasing = TRUE)[1:7]

# For Q4
# Potential motif score function
# motif_score &lt;- function(sequence, motif_PWM) {
#   seq_vec = Onehot_Encode(sequence)
#   sum(log(colSums(seq_vec * motif_PWM)))
# }</code></pre>
</details>
<p><br /></p>
<details>
<summary>
<strong>K-mer counting</strong>
</summary>
<pre><code># For Q6
kmer_count = rep(0, length(kmer))
names(kmer_count) = kmer
for (i in 1:(nchar(HBG2_segment) - k + 1)) {
  a_kmer = substr(HBG2_segment, i, i+2)
  kmer_count[a_kmer] = kmer_count[a_kmer] + 1
}
kmer_count</code></pre>
</details>
<p><br /></p>
<details>
<summary>
<strong>Cross-validation for linear regression</strong>
</summary>
<pre><code># For Q7
# Train the model
cv_model &lt;- train(splicing_efficiency ~ ., 
                  data = df_splicing[, 3:ncol(df_splicing)], 
                  method = &quot;lm&quot;, trControl = my_trControl)

# Check model performance
print(cv_model)
cor(cv_model$pred$obs, cv_model$pred$pred)

ggplot(cv_model$pred, aes(x=obs, y=pred)) + 
  geom_point()</code></pre>
</details>
<p><br /></p>
<!-- Since it is only one amino acid change, you are curious how the protein  -->
<!-- structure changes. You copy the protein sequence from UniProt [P68871](https://www.uniprot.org/uniprotkb/P68871/entry#sequences)  -->
<!-- and put it into [AlphaFold server](https://alphafoldserver.com). Similarly, you -->
<!-- can introduce the disease mutation and predict the structure again. -->
<!-- Note, the predicted structure is also available: -->
<!-- [AF-P68871-F1-v4](https://alphafold.ebi.ac.uk/entry/P68871) -->
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cancer-case2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genomics-regression.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/statbiomed/BMDS-book/tree/main/02-data_topics.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["BMDatSci.pdf", "BMDatSci.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
