<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Genomics and prediction | Biomedical Data Science - introduction with case studies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Genomics and prediction | Biomedical Data Science - introduction with case studies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Genomics and prediction | Biomedical Data Science - introduction with case studies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="BIOF1001 teaching team" />


<meta name="date" content="2025-09-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genetic-seq.html"/>
<link rel="next" href="install.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">Biomedical Data Science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#introduction-for-readers"><i class="fa fa-check"></i>Introduction for readers</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-you-will-learn-from-this-coursebook"><i class="fa fa-check"></i>What you will learn from this course/book</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-recommend-you-do-while-reading-this-book"><i class="fa fa-check"></i>What we recommend you do while reading this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#other-reference-books"><i class="fa fa-check"></i>Other reference books</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="part"><span><b>I Data Science Foundations</b></span></li>
<li class="chapter" data-level="1" data-path="introR.html"><a href="introR.html"><i class="fa fa-check"></i><b>1</b> Introduction to R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introR.html"><a href="introR.html#introR-w1"><i class="fa fa-check"></i><b>1.1</b> Intro to R programming (Session 1)</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introR.html"><a href="introR.html#r-as-a-basic-calculator"><i class="fa fa-check"></i><b>1.1.1</b> R as a basic calculator</a></li>
<li class="chapter" data-level="1.1.2" data-path="introR.html"><a href="introR.html#variables"><i class="fa fa-check"></i><b>1.1.2</b> Variables</a></li>
<li class="chapter" data-level="1.1.3" data-path="introR.html"><a href="introR.html#data-types"><i class="fa fa-check"></i><b>1.1.3</b> Data types</a></li>
<li class="chapter" data-level="1.1.4" data-path="introR.html"><a href="introR.html#functions"><i class="fa fa-check"></i><b>1.1.4</b> Functions</a></li>
<li class="chapter" data-level="1.1.5" data-path="introR.html"><a href="introR.html#r-scripts"><i class="fa fa-check"></i><b>1.1.5</b> R scripts</a></li>
<li class="chapter" data-level="1.1.6" data-path="introR.html"><a href="introR.html#resource-links"><i class="fa fa-check"></i><b>1.1.6</b> Resource links</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introR.html"><a href="introR.html#introR-w2"><i class="fa fa-check"></i><b>1.2</b> Intro to R programming (Session 2)</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introR.html"><a href="introR.html#data-structures"><i class="fa fa-check"></i><b>1.2.1</b> Data structures</a></li>
<li class="chapter" data-level="1.2.2" data-path="introR.html"><a href="introR.html#vector"><i class="fa fa-check"></i><b>1.2.2</b> Vector</a></li>
<li class="chapter" data-level="1.2.3" data-path="introR.html"><a href="introR.html#matrix"><i class="fa fa-check"></i><b>1.2.3</b> Matrix</a></li>
<li class="chapter" data-level="1.2.4" data-path="introR.html"><a href="introR.html#list"><i class="fa fa-check"></i><b>1.2.4</b> List</a></li>
<li class="chapter" data-level="1.2.5" data-path="introR.html"><a href="introR.html#data-frame"><i class="fa fa-check"></i><b>1.2.5</b> Data Frame</a></li>
<li class="chapter" data-level="1.2.6" data-path="introR.html"><a href="introR.html#readingwriting-data-fromto-r"><i class="fa fa-check"></i><b>1.2.6</b> Reading/writing data from/to R</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="introR.html"><a href="introR.html#introR-w3"><i class="fa fa-check"></i><b>1.3</b> Intro to R programming (Session 4)</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introR.html"><a href="introR.html#operators"><i class="fa fa-check"></i><b>1.3.1</b> Operators</a></li>
<li class="chapter" data-level="1.3.2" data-path="introR.html"><a href="introR.html#ifelse"><i class="fa fa-check"></i><b>1.3.2</b> If/else</a></li>
<li class="chapter" data-level="1.3.3" data-path="introR.html"><a href="introR.html#loop"><i class="fa fa-check"></i><b>1.3.3</b> Loop</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introHypoTest.html"><a href="introHypoTest.html"><i class="fa fa-check"></i><b>2</b> Introduction to Hypothesis test</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-test-and-p-value"><i class="fa fa-check"></i><b>2.1</b> Hypothesis test and <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="introHypoTest.html"><a href="introHypoTest.html#extreme-events-and-random-chance"><i class="fa fa-check"></i><b>2.1.1</b> Extreme events and random chance</a></li>
<li class="chapter" data-level="2.1.2" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-test-for-statistical-decision"><i class="fa fa-check"></i><b>2.1.2</b> Hypothesis test for statistical decision</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#basic-hypothesis-test-methods"><i class="fa fa-check"></i><b>2.2</b> Basic hypothesis test methods</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#permutation-test"><i class="fa fa-check"></i><b>2.2.1</b> Permutation test</a></li>
<li class="chapter" data-level="2.2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#t-test-and-regression-based-test"><i class="fa fa-check"></i><b>2.2.2</b> <em>t</em>-test and regression-based test</a></li>
<li class="chapter" data-level="2.2.3" data-path="introHypoTest.html"><a href="introHypoTest.html#fishers-exact-test"><i class="fa fa-check"></i><b>2.2.3</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introHypoTest.html"><a href="introHypoTest.html#multiple-testing-and-errors"><i class="fa fa-check"></i><b>2.3</b> Multiple testing and errors</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-test-statistic"><i class="fa fa-check"></i><b>2.3.1</b> Null distribution (of test statistic)</a></li>
<li class="chapter" data-level="2.3.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-p-value"><i class="fa fa-check"></i><b>2.3.2</b> Null distribution of p value</a></li>
<li class="chapter" data-level="2.3.3" data-path="introHypoTest.html"><a href="introHypoTest.html#minimal-p-values-in-10-tests"><i class="fa fa-check"></i><b>2.3.3</b> Minimal p values in 10 tests</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="introHypoTest.html"><a href="introHypoTest.html#power-analysis"><i class="fa fa-check"></i><b>2.4</b> Explore power and sample size (optional)</a></li>
<li class="chapter" data-level="2.5" data-path="introHypoTest.html"><a href="introHypoTest.html#summary"><i class="fa fa-check"></i><b>2.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introLinearReg.html"><a href="introLinearReg.html"><i class="fa fa-check"></i><b>3</b> Introduction to Linear Regression</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#linear-regression-using-simulated-data"><i class="fa fa-check"></i><b>3.1</b> Linear Regression Using Simulated Data</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="introLinearReg.html"><a href="introLinearReg.html#simulating-data"><i class="fa fa-check"></i><b>3.1.1</b> Simulating data:</a></li>
<li class="chapter" data-level="3.1.2" data-path="introLinearReg.html"><a href="introLinearReg.html#model-efficacy"><i class="fa fa-check"></i><b>3.1.2</b> Model efficacy</a></li>
<li class="chapter" data-level="3.1.3" data-path="introLinearReg.html"><a href="introLinearReg.html#r-squared"><i class="fa fa-check"></i><b>3.1.3</b> <em>R-Squared</em></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#least-squares-using-simulated-data"><i class="fa fa-check"></i><b>3.2</b> Least Squares Using Simulated Data</a></li>
<li class="chapter" data-level="3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#diagnostic-check-of-a-fitted-regression-model"><i class="fa fa-check"></i><b>3.3</b> Diagnostic check of a fitted regression model</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#residual-standard-errors-rse"><i class="fa fa-check"></i><b>3.3.1</b> Residual Standard Errors (RSE)</a></li>
<li class="chapter" data-level="3.3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#p-values"><i class="fa fa-check"></i><b>3.3.2</b> p-values</a></li>
<li class="chapter" data-level="3.3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#f-statistics"><i class="fa fa-check"></i><b>3.3.3</b> F-statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introLinearReg.html"><a href="introLinearReg.html#simple-linear-regression-with-lm-function"><i class="fa fa-check"></i><b>3.4</b> Simple Linear Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="3.5" data-path="introLinearReg.html"><a href="introLinearReg.html#multiple-regression-with-lm-function"><i class="fa fa-check"></i><b>3.5</b> Multiple Regression with <code>lm</code> function</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introClassifier.html"><a href="introClassifier.html"><i class="fa fa-check"></i><b>4</b> Introduction to Classification</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introClassifier.html"><a href="introClassifier.html#visualise-logistic-and-logit-functions"><i class="fa fa-check"></i><b>4.1</b> Visualise logistic and logit functions</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="introClassifier.html"><a href="introClassifier.html#logistic-function"><i class="fa fa-check"></i><b>4.1.1</b> Logistic function</a></li>
<li class="chapter" data-level="4.1.2" data-path="introClassifier.html"><a href="introClassifier.html#logit-function"><i class="fa fa-check"></i><b>4.1.2</b> Logit function</a></li>
<li class="chapter" data-level="4.1.3" data-path="introClassifier.html"><a href="introClassifier.html#visualise-the-distribution"><i class="fa fa-check"></i><b>4.1.3</b> Visualise the distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="introClassifier.html"><a href="introClassifier.html#logistic-regression-on-diabetes"><i class="fa fa-check"></i><b>4.2</b> Logistic regression on Diabetes</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="introClassifier.html"><a href="introClassifier.html#load-pima-indians-diabetes-database"><i class="fa fa-check"></i><b>4.2.1</b> Load Pima Indians Diabetes Database</a></li>
<li class="chapter" data-level="4.2.2" data-path="introClassifier.html"><a href="introClassifier.html#fit-logistic-regression"><i class="fa fa-check"></i><b>4.2.2</b> Fit logistic regression</a></li>
<li class="chapter" data-level="4.2.3" data-path="introClassifier.html"><a href="introClassifier.html#assess-on-test-data"><i class="fa fa-check"></i><b>4.2.3</b> Assess on test data</a></li>
<li class="chapter" data-level="4.2.4" data-path="introClassifier.html"><a href="introClassifier.html#model-selection-and-diagnosis"><i class="fa fa-check"></i><b>4.2.4</b> Model selection and diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="introClassifier.html"><a href="introClassifier.html#cross-validation"><i class="fa fa-check"></i><b>4.3</b> Cross-validation</a></li>
<li class="chapter" data-level="4.4" data-path="introClassifier.html"><a href="introClassifier.html#more-assessment-metrics"><i class="fa fa-check"></i><b>4.4</b> More assessment metrics</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="introClassifier.html"><a href="introClassifier.html#two-types-of-error"><i class="fa fa-check"></i><b>4.4.1</b> Two types of error</a></li>
<li class="chapter" data-level="4.4.2" data-path="introClassifier.html"><a href="introClassifier.html#roc-curve"><i class="fa fa-check"></i><b>4.4.2</b> ROC curve</a></li>
<li class="chapter" data-level="4.4.3" data-path="introClassifier.html"><a href="introClassifier.html#homework"><i class="fa fa-check"></i><b>4.4.3</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Biomedical Data Modules</b></span></li>
<li class="chapter" data-level="5" data-path="image-digital.html"><a href="image-digital.html"><i class="fa fa-check"></i><b>5</b> Medical Image and Digital Health</a></li>
<li class="chapter" data-level="6" data-path="cancer.html"><a href="cancer.html"><i class="fa fa-check"></i><b>6</b> Cancer genomics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="cancer.html"><a href="cancer.html#cancer-case1"><i class="fa fa-check"></i><b>6.1</b> Case study 1: analysis of cBioportal mutation data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="cancer.html"><a href="cancer.html#exploratory-analysis"><i class="fa fa-check"></i><b>6.1.1</b> Exploratory analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="cancer.html"><a href="cancer.html#statistical-analysis"><i class="fa fa-check"></i><b>6.1.2</b> Statistical analysis</a></li>
<li class="chapter" data-level="6.1.3" data-path="cancer.html"><a href="cancer.html#literature-search"><i class="fa fa-check"></i><b>6.1.3</b> Literature search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="epidemiology.html"><a href="epidemiology.html"><i class="fa fa-check"></i><b>7</b> Epidemiology</a></li>
<li class="chapter" data-level="8" data-path="pop-genetics.html"><a href="pop-genetics.html"><i class="fa fa-check"></i><b>8</b> Population Genetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="pop-genetics.html"><a href="pop-genetics.html#case-study-1-heritability-and-human-traits"><i class="fa fa-check"></i><b>8.1</b> Case study 1: Heritability and human traits</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="pop-genetics.html"><a href="pop-genetics.html#part-1"><i class="fa fa-check"></i><b>8.1.1</b> Part 1</a></li>
<li class="chapter" data-level="8.1.2" data-path="pop-genetics.html"><a href="pop-genetics.html#part-2"><i class="fa fa-check"></i><b>8.1.2</b> Part 2</a></li>
<li class="chapter" data-level="8.1.3" data-path="pop-genetics.html"><a href="pop-genetics.html#references"><i class="fa fa-check"></i><b>8.1.3</b> References</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Modules in previous years</b></span></li>
<li class="chapter" data-level="9" data-path="cancer-case2.html"><a href="cancer-case2.html"><i class="fa fa-check"></i><b>9</b> Cancer Epidemiology</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="cancer-case2.html"><a href="cancer-case2.html#scenario"><i class="fa fa-check"></i><b>9.0.1</b> Scenario</a></li>
<li class="chapter" data-level="9.0.2" data-path="cancer-case2.html"><a href="cancer-case2.html#hong-kong-population"><i class="fa fa-check"></i><b>9.0.2</b> Hong Kong population</a></li>
<li class="chapter" data-level="9.0.3" data-path="cancer-case2.html"><a href="cancer-case2.html#cancer-registry-data"><i class="fa fa-check"></i><b>9.0.3</b> Cancer registry data</a></li>
<li class="chapter" data-level="9.0.4" data-path="cancer-case2.html"><a href="cancer-case2.html#existing-cancer-funding-and-publication-data"><i class="fa fa-check"></i><b>9.0.4</b> Existing cancer funding and publication data</a></li>
<li class="chapter" data-level="9.0.5" data-path="cancer-case2.html"><a href="cancer-case2.html#open-discussion"><i class="fa fa-check"></i><b>9.0.5</b> Open discussion</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="genetic-seq.html"><a href="genetic-seq.html"><i class="fa fa-check"></i><b>10</b> Genetic sequence</a>
<ul>
<li class="chapter" data-level="10.1" data-path="genetic-seq.html"><a href="genetic-seq.html#case-study-1-genetic-sequence-analysis"><i class="fa fa-check"></i><b>10.1</b> Case study 1: Genetic sequence analysis</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="genetic-seq.html"><a href="genetic-seq.html#sequence-motif-and-k-mer"><i class="fa fa-check"></i><b>10.1.1</b> Sequence motif and k-mer</a></li>
<li class="chapter" data-level="10.1.2" data-path="genetic-seq.html"><a href="genetic-seq.html#functional-mapping"><i class="fa fa-check"></i><b>10.1.2</b> Functional mapping</a></li>
<li class="chapter" data-level="10.1.3" data-path="genetic-seq.html"><a href="genetic-seq.html#references-1"><i class="fa fa-check"></i><b>10.1.3</b> References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="genomics-regression.html"><a href="genomics-regression.html"><i class="fa fa-check"></i><b>11</b> Genomics and prediction</a>
<ul>
<li class="chapter" data-level="11.1" data-path="genomics-regression.html"><a href="genomics-regression.html#genomics-case1"><i class="fa fa-check"></i><b>11.1</b> Case study 1: splicing fedility prediction</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="genomics-regression.html"><a href="genomics-regression.html#questions-for-discussion-1"><i class="fa fa-check"></i><b>11.1.1</b> Questions for Discussion</a></li>
<li class="chapter" data-level="11.1.2" data-path="genomics-regression.html"><a href="genomics-regression.html#hands-on-with-regression-model"><i class="fa fa-check"></i><b>11.1.2</b> Hands-on with regression model</a></li>
<li class="chapter" data-level="11.1.3" data-path="genomics-regression.html"><a href="genomics-regression.html#open-question"><i class="fa fa-check"></i><b>11.1.3</b> Open question</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i>Appendix A: Install R &amp; RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.1-install-r-4.3.1"><i class="fa fa-check"></i>A.1 Install R (&gt;=4.3.1)</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-windows"><i class="fa fa-check"></i>R on Windows</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-macos"><i class="fa fa-check"></i>R on macOS</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-linux-ubuntu"><i class="fa fa-check"></i>R on Linux (Ubuntu)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.2-install-rstudio"><i class="fa fa-check"></i>A.2 Install RStudio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.3-use-r-inside-rstudio"><i class="fa fa-check"></i>A.3 Use R inside RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-studio"><i class="fa fa-check"></i>R studio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#some-general-knowledge"><i class="fa fa-check"></i>Some general knowledge</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#install-packages"><i class="fa fa-check"></i>Install packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a4.-cloud-computing"><i class="fa fa-check"></i>A4. Cloud computing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-2.html"><a href="references-2.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Data Science - introduction with case studies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genomics-regression" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Genomics and prediction<a href="genomics-regression.html#genomics-regression" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="genomics-case1" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Case study 1: splicing fedility prediction<a href="genomics-regression.html#genomics-case1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Scenario: As a computational biology researcher, you want to find out what sequence-related features function as a regulator for <strong>splicing fidelity</strong>. You conducted a deep RNA-seq experience and quantified the <strong>splicing error frequency (SEF)</strong> for 141 splicing error events in yeast. Based on your prior knowledge and curiosity, you plan to further define a set of sequence-related features and test how predictive they are.</p>
<div id="questions-for-discussion-1" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Questions for Discussion<a href="genomics-regression.html#questions-for-discussion-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Q1.</strong> What sequence-related features you would think of?</p>
<p><strong>Q2.</strong> Given a set of candidate features, what statistical methods can you use to determine if a feature genuinely contributes to splicing fidelity?</p>
<p><strong>Q3.</strong> Given a candidate feature set, what could be the best subset of features for predicting the splicing fidelity and in what sense?</p>
</div>
<div id="hands-on-with-regression-model" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> Hands-on with regression model<a href="genomics-regression.html#hands-on-with-regression-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now, you are given a dataset with 141 splicing error events containing Y for splicing error frequency (SEF) and 14 predictors X. Data set and template notebook are available on Moodle (recommended) and also on this
<a href="https://github.com/StatBiomed/BMDS-book/blob/main/notebooks/module1-genomics/df_SEF_prediction.tsv">GitHub Repo</a>.</p>
<p><strong>The information for columns:</strong></p>
<ul>
<li>RPG: Ribosomal Protein Gene</li>
<li>event: splicing error event type</li>
<li>shift: nucleotide shift between canonical (ref, reference) and erroneous (alt, alternative) splice sites</li>
<li>SEF: splicing error frequency</li>
<li>intronL_ref: intron length for canonical (ref) intron</li>
<li>5ss_bpL_ref: length between 5 splice site to branch point</li>
<li>DeltaG_intron_ref: the energy score in the secondary RNA structure, calculated by <a href="http://rna.tbi.univie.ac.at/cgi-bin/RNAWebSuite/RNAfold.cgi">RNAFold</a>. The lower the more stable.</li>
<li>5ss_motif_ref: the motif score of a sequence. The motif position weight matrix (PWM) is obtained by observed genes. The motif score is calculated as the log likelihood of each sequence given the PWM, followed by resealing to 100 for the maximum value. You can refer to <em>slide 7</em> in lecture topic 1.</li>
<li>intronL_alt: intron length for erroneous (alt) intron</li>
</ul>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="genomics-regression.html#cb200-1" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;df_SEF_prediction.tsv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header=</span><span class="dv">1</span>)</span>
<span id="cb200-2"><a href="genomics-regression.html#cb200-2" tabindex="-1"></a></span>
<span id="cb200-3"><a href="genomics-regression.html#cb200-3" tabindex="-1"></a><span class="fu">head</span>(df)</span>
<span id="cb200-4"><a href="genomics-regression.html#cb200-4" tabindex="-1"></a><span class="co">#&gt;    gene     RPG   event shift         SEF intronL_ref X5ss_bpL_ref</span></span>
<span id="cb200-5"><a href="genomics-regression.html#cb200-5" tabindex="-1"></a><span class="co">#&gt; 1  TFC3 non_RPG down3ss    17 0.479869423          90           77</span></span>
<span id="cb200-6"><a href="genomics-regression.html#cb200-6" tabindex="-1"></a><span class="co">#&gt; 2 SEC17 non_RPG down3ss    47 0.021937843         116           77</span></span>
<span id="cb200-7"><a href="genomics-regression.html#cb200-7" tabindex="-1"></a><span class="co">#&gt; 3  ERD2 non_RPG   up3ss     4 0.059303591          97           63</span></span>
<span id="cb200-8"><a href="genomics-regression.html#cb200-8" tabindex="-1"></a><span class="co">#&gt; 4  LSM2 non_RPG   up5ss    56 0.019908116         128           92</span></span>
<span id="cb200-9"><a href="genomics-regression.html#cb200-9" tabindex="-1"></a><span class="co">#&gt; 5  LSM2 non_RPG   up3ss    47 0.077444760         128           92</span></span>
<span id="cb200-10"><a href="genomics-regression.html#cb200-10" tabindex="-1"></a><span class="co">#&gt; 6  LSM2 non_RPG down3ss    40 0.002844017         128           92</span></span>
<span id="cb200-11"><a href="genomics-regression.html#cb200-11" tabindex="-1"></a><span class="co">#&gt;   DeltaG_intron_ref DeltaG_5ss_bp_ref DeltaG_bp_3ss_ref DeltaG_3ss_ref</span></span>
<span id="cb200-12"><a href="genomics-regression.html#cb200-12" tabindex="-1"></a><span class="co">#&gt; 1          -0.11444          -0.13377               0.0           -6.2</span></span>
<span id="cb200-13"><a href="genomics-regression.html#cb200-13" tabindex="-1"></a><span class="co">#&gt; 2          -0.17155          -0.10779             -10.5          -20.3</span></span>
<span id="cb200-14"><a href="genomics-regression.html#cb200-14" tabindex="-1"></a><span class="co">#&gt; 3          -0.16804          -0.16190              -3.9           -8.7</span></span>
<span id="cb200-15"><a href="genomics-regression.html#cb200-15" tabindex="-1"></a><span class="co">#&gt; 4          -0.13906          -0.12065              -0.9           -7.0</span></span>
<span id="cb200-16"><a href="genomics-regression.html#cb200-16" tabindex="-1"></a><span class="co">#&gt; 5          -0.13906          -0.12065              -0.9           -7.0</span></span>
<span id="cb200-17"><a href="genomics-regression.html#cb200-17" tabindex="-1"></a><span class="co">#&gt; 6          -0.13906          -0.12065              -0.9           -7.0</span></span>
<span id="cb200-18"><a href="genomics-regression.html#cb200-18" tabindex="-1"></a><span class="co">#&gt;   X5ss_motif_ref bp_motif_ref X3ss_motif_ref intronL_alt DeltaG_intron_alt</span></span>
<span id="cb200-19"><a href="genomics-regression.html#cb200-19" tabindex="-1"></a><span class="co">#&gt; 1         0.7213       0.7703         0.0769         107          -0.10841</span></span>
<span id="cb200-20"><a href="genomics-regression.html#cb200-20" tabindex="-1"></a><span class="co">#&gt; 2         0.7695       0.9942         0.1461         163          -0.19202</span></span>
<span id="cb200-21"><a href="genomics-regression.html#cb200-21" tabindex="-1"></a><span class="co">#&gt; 3         0.8495       0.8835         0.1496          93          -0.17527</span></span>
<span id="cb200-22"><a href="genomics-regression.html#cb200-22" tabindex="-1"></a><span class="co">#&gt; 4         0.8305       0.6667         0.3677         184          -0.15217</span></span>
<span id="cb200-23"><a href="genomics-regression.html#cb200-23" tabindex="-1"></a><span class="co">#&gt; 5         0.8305       0.6667         0.3677          81          -0.10864</span></span>
<span id="cb200-24"><a href="genomics-regression.html#cb200-24" tabindex="-1"></a><span class="co">#&gt; 6         0.8305       0.6667         0.3677         168          -0.11667</span></span>
<span id="cb200-25"><a href="genomics-regression.html#cb200-25" tabindex="-1"></a><span class="co">#&gt;   DeltaG_3ss_alt X5ss_motif_alt X3ss_motif_alt</span></span>
<span id="cb200-26"><a href="genomics-regression.html#cb200-26" tabindex="-1"></a><span class="co">#&gt; 1           -6.8          77.00         0.5223</span></span>
<span id="cb200-27"><a href="genomics-regression.html#cb200-27" tabindex="-1"></a><span class="co">#&gt; 2          -13.0          70.99         0.2926</span></span>
<span id="cb200-28"><a href="genomics-regression.html#cb200-28" tabindex="-1"></a><span class="co">#&gt; 3           -8.3          86.10         0.3329</span></span>
<span id="cb200-29"><a href="genomics-regression.html#cb200-29" tabindex="-1"></a><span class="co">#&gt; 4           -7.0          64.20         0.3955</span></span>
<span id="cb200-30"><a href="genomics-regression.html#cb200-30" tabindex="-1"></a><span class="co">#&gt; 5           -8.9          80.76         0.6179</span></span>
<span id="cb200-31"><a href="genomics-regression.html#cb200-31" tabindex="-1"></a><span class="co">#&gt; 6           -4.3          80.76         0.3341</span></span></code></pre></div>
<div id="a-single-feature" class="section level4 hasAnchor" number="11.1.2.1">
<h4><span class="header-section-number">11.1.2.1</span> A single feature<a href="genomics-regression.html#a-single-feature" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q4.</strong> Let’s start with a single feature, <code>bp_motif_ref</code>, is there a linear relationship with the output variable <code>SEF</code> Y? How do you determine it?</p>
<p><strong>Hint</strong>: try scatter plot to visualise the data. See example we mentioned before
<a href="https://statbiomed.github.io/BMDS-book/introR.html#scatter-plot-1" class="uri">https://statbiomed.github.io/BMDS-book/introR.html#scatter-plot-1</a></p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="genomics-regression.html#cb201-1" tabindex="-1"></a><span class="co"># consider visualise it</span></span>
<span id="cb201-2"><a href="genomics-regression.html#cb201-2" tabindex="-1"></a><span class="co"># df$SEF and df$bp_motif_ref</span></span>
<span id="cb201-3"><a href="genomics-regression.html#cb201-3" tabindex="-1"></a></span>
<span id="cb201-4"><a href="genomics-regression.html#cb201-4" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb201-5"><a href="genomics-regression.html#cb201-5" tabindex="-1"></a></span>
<span id="cb201-6"><a href="genomics-regression.html#cb201-6" tabindex="-1"></a><span class="co"># Write your codes here</span></span></code></pre></div>
<p>How about doing a log transform for SEF score? Consider adding a pseudo count, e.g., 0.0001.</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="genomics-regression.html#cb202-1" tabindex="-1"></a>df[<span class="st">&quot;SEF_log&quot;</span>] <span class="ot">=</span> <span class="fu">log10</span>(df<span class="sc">$</span>SEF <span class="sc">+</span> <span class="fl">0.0001</span>)</span>
<span id="cb202-2"><a href="genomics-regression.html#cb202-2" tabindex="-1"></a></span>
<span id="cb202-3"><a href="genomics-regression.html#cb202-3" tabindex="-1"></a><span class="co"># visualize it again</span></span></code></pre></div>
</div>
<div id="all-features-with-lm" class="section level4 hasAnchor" number="11.1.2.2">
<h4><span class="header-section-number">11.1.2.2</span> All features with lm<a href="genomics-regression.html#all-features-with-lm" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q5.</strong> Can you build a linear model to perform the prediction by using all features? How to obtain the assessment statistics?</p>
<p><strong>Hint</strong>: consider using the <code>lm</code> function. See examples we mentioned before
<a href="https://statbiomed.github.io/BMDS-book/introLinearReg.html#simple-linear-regression-with-lm-function" class="uri">https://statbiomed.github.io/BMDS-book/introLinearReg.html#simple-linear-regression-with-lm-function</a></p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="genomics-regression.html#cb203-1" tabindex="-1"></a><span class="co"># For the formula, you can use this one, but you can also write your own:</span></span>
<span id="cb203-2"><a href="genomics-regression.html#cb203-2" tabindex="-1"></a>lr_fm1 <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">&quot;SEF_log ~ 1 + &quot;</span>, <span class="fu">paste</span>(<span class="fu">colnames</span>(df)[<span class="dv">6</span><span class="sc">:</span><span class="dv">19</span>], <span class="at">collapse=</span><span class="st">&quot; + &quot;</span>)))</span>
<span id="cb203-3"><a href="genomics-regression.html#cb203-3" tabindex="-1"></a></span>
<span id="cb203-4"><a href="genomics-regression.html#cb203-4" tabindex="-1"></a><span class="co"># Write your codes for fitting the model here</span></span></code></pre></div>
</div>
<div id="assessment-on-test-data" class="section level4 hasAnchor" number="11.1.2.3">
<h4><span class="header-section-number">11.1.2.3</span> Assessment on test data<a href="genomics-regression.html#assessment-on-test-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q6.</strong> Can you perform the assessment on test data by splitting the data into training and test sets?</p>
<p>We didn’t introduce how perform training &amp; test splitting in linear regression, but you can adapt the scripts that we introduced for logistic regression.</p>
<p><strong>Hint</strong>: you can recall the logistic regress chapter for the training &amp; test spiting: <a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes</a></p>
<p>You can also just use the codes below, but think what is the ratio between training and test sets here:</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="genomics-regression.html#cb204-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb204-2"><a href="genomics-regression.html#cb204-2" tabindex="-1"></a></span>
<span id="cb204-3"><a href="genomics-regression.html#cb204-3" tabindex="-1"></a>idx_train <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(df),  <span class="at">size=</span><span class="fl">0.5</span><span class="sc">*</span><span class="fu">nrow</span>(df), <span class="at">replace =</span> <span class="cn">FALSE</span>)</span>
<span id="cb204-4"><a href="genomics-regression.html#cb204-4" tabindex="-1"></a>df_train <span class="ot">=</span> df[idx_train, ]</span>
<span id="cb204-5"><a href="genomics-regression.html#cb204-5" tabindex="-1"></a>df_test  <span class="ot">=</span> df[<span class="sc">-</span>idx_train, ] <span class="co"># recall the meaning of negative symbol</span></span></code></pre></div>
<p>Now, start training the model on the training set and predict on the test set.</p>
<p><strong>Hint</strong>: follow the codes here: <a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes</a></p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="genomics-regression.html#cb205-1" tabindex="-1"></a><span class="co"># recall the predict function</span></span>
<span id="cb205-2"><a href="genomics-regression.html#cb205-2" tabindex="-1"></a></span>
<span id="cb205-3"><a href="genomics-regression.html#cb205-3" tabindex="-1"></a><span class="co"># lr_train = lm()</span></span>
<span id="cb205-4"><a href="genomics-regression.html#cb205-4" tabindex="-1"></a><span class="co"># res_test = predict()</span></span></code></pre></div>
<p>For calculating R^2, another quick way is using the Pearson’s correlation coefficient R, as it gives equivalent values. We can do it with the following codes with the <code>cor()</code> function.</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="genomics-regression.html#cb206-1" tabindex="-1"></a><span class="co"># Please uncomment the codes below:</span></span>
<span id="cb206-2"><a href="genomics-regression.html#cb206-2" tabindex="-1"></a></span>
<span id="cb206-3"><a href="genomics-regression.html#cb206-3" tabindex="-1"></a><span class="co"># R_sqr = cor(res_test, df_test$SEF_log)**2</span></span>
<span id="cb206-4"><a href="genomics-regression.html#cb206-4" tabindex="-1"></a><span class="co"># R_sqr</span></span>
<span id="cb206-5"><a href="genomics-regression.html#cb206-5" tabindex="-1"></a></span>
<span id="cb206-6"><a href="genomics-regression.html#cb206-6" tabindex="-1"></a><span class="co"># Alternative function</span></span>
<span id="cb206-7"><a href="genomics-regression.html#cb206-7" tabindex="-1"></a><span class="co"># cor.test(res_test, df_test$SEF_log)</span></span></code></pre></div>
</div>
<div id="cross-validation-for-regression" class="section level4 hasAnchor" number="11.1.2.4">
<h4><span class="header-section-number">11.1.2.4</span> Cross-validation for regression<a href="genomics-regression.html#cross-validation-for-regression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q7.</strong> Can you perform a 10-fold cross validation, calculate the R^2 between predicted and observed log(SEF), and plot it with ggplot?</p>
<p><strong>Hint</strong>: We didn’t mention cross-validation for linear regression but again we can adapt the codes from the logistic regression chapter:
<a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#cross-validation" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#cross-validation</a></p>
<p>You can also use the codes below, but try to fill the missing part (with question marks <code>???</code>) yourself.</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="genomics-regression.html#cb207-1" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb207-2"><a href="genomics-regression.html#cb207-2" tabindex="-1"></a></span>
<span id="cb207-3"><a href="genomics-regression.html#cb207-3" tabindex="-1"></a><span class="co"># Define training control</span></span>
<span id="cb207-4"><a href="genomics-regression.html#cb207-4" tabindex="-1"></a><span class="co"># We also want to have savePredictions=TRUE &amp; classProbs=TRUE</span></span>
<span id="cb207-5"><a href="genomics-regression.html#cb207-5" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>) </span>
<span id="cb207-6"><a href="genomics-regression.html#cb207-6" tabindex="-1"></a>my_trControl <span class="ot">&lt;-</span> <span class="fu">trainControl</span>(<span class="at">method =</span> <span class="st">&quot;cv&quot;</span>, <span class="at">number =</span> <span class="dv">10</span>, </span>
<span id="cb207-7"><a href="genomics-regression.html#cb207-7" tabindex="-1"></a>                             <span class="at">savePredictions =</span> <span class="cn">TRUE</span>)</span>
<span id="cb207-8"><a href="genomics-regression.html#cb207-8" tabindex="-1"></a></span>
<span id="cb207-9"><a href="genomics-regression.html#cb207-9" tabindex="-1"></a><span class="co"># Train the model (uncomment the codes below)</span></span>
<span id="cb207-10"><a href="genomics-regression.html#cb207-10" tabindex="-1"></a><span class="co"># cv_model &lt;- train(???, data = ???, </span></span>
<span id="cb207-11"><a href="genomics-regression.html#cb207-11" tabindex="-1"></a><span class="co">#                   method = &quot;lm&quot;,</span></span>
<span id="cb207-12"><a href="genomics-regression.html#cb207-12" tabindex="-1"></a><span class="co">#                   trControl = my_trControl)</span></span>
<span id="cb207-13"><a href="genomics-regression.html#cb207-13" tabindex="-1"></a></span>
<span id="cb207-14"><a href="genomics-regression.html#cb207-14" tabindex="-1"></a><span class="co"># Summarize the results</span></span>
<span id="cb207-15"><a href="genomics-regression.html#cb207-15" tabindex="-1"></a><span class="co"># mean(cv_model$resample$Rsquared)</span></span>
<span id="cb207-16"><a href="genomics-regression.html#cb207-16" tabindex="-1"></a><span class="co"># print(cv_model)</span></span></code></pre></div>
<p>By using <code>print(cv_model)</code>, we can see the <em>Rsquared</em> value, which is an average of all R^2 acoss K folds (each fold has its own R^2 on the test set).</p>
<p>Now, let’s manually calculate the R^2 between observed and predicted Y for all samples in an aggregated manner, despite that the Ys are predicted from different models in different folds.</p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="genomics-regression.html#cb208-1" tabindex="-1"></a><span class="co"># Uncomment the codes below to view the first a few lines of the data frame</span></span>
<span id="cb208-2"><a href="genomics-regression.html#cb208-2" tabindex="-1"></a></span>
<span id="cb208-3"><a href="genomics-regression.html#cb208-3" tabindex="-1"></a><span class="co"># head(cv_model$pred)</span></span></code></pre></div>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="genomics-regression.html#cb209-1" tabindex="-1"></a><span class="co"># Uncomment the codes below to calculate the R^2</span></span>
<span id="cb209-2"><a href="genomics-regression.html#cb209-2" tabindex="-1"></a></span>
<span id="cb209-3"><a href="genomics-regression.html#cb209-3" tabindex="-1"></a><span class="co"># cor(cv_model$pred$pred, cv_model$pred$obs)**2</span></span></code></pre></div>
</div>
<div id="a-subset-of-features" class="section level4 hasAnchor" number="11.1.2.5">
<h4><span class="header-section-number">11.1.2.5</span> A subset of features<a href="genomics-regression.html#a-subset-of-features" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q8.</strong> Can you try a subset of features and see if you can improve the performance?</p>
<p><strong>Hint</strong>: You can consider forward method as mentioned here by checking the <span class="math inline">\(p\)</span> value and only keep the significant ones (think which significance level <span class="math inline">\(\alpha\)</span> you would like to allow:
<a href="https://statbiomed.github.io/BMDS-book/introLinearReg.html#multiple-regression-with-lm-function" class="uri">https://statbiomed.github.io/BMDS-book/introLinearReg.html#multiple-regression-with-lm-function</a></p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="genomics-regression.html#cb210-1" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb210-2"><a href="genomics-regression.html#cb210-2" tabindex="-1"></a></span>
<span id="cb210-3"><a href="genomics-regression.html#cb210-3" tabindex="-1"></a><span class="co"># Define your new formular:</span></span>
<span id="cb210-4"><a href="genomics-regression.html#cb210-4" tabindex="-1"></a><span class="co"># lr_fm2 = as.formula()</span></span>
<span id="cb210-5"><a href="genomics-regression.html#cb210-5" tabindex="-1"></a></span>
<span id="cb210-6"><a href="genomics-regression.html#cb210-6" tabindex="-1"></a></span>
<span id="cb210-7"><a href="genomics-regression.html#cb210-7" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb210-8"><a href="genomics-regression.html#cb210-8" tabindex="-1"></a><span class="co"># Try the 10-fold cross-validation with the new feature set yourself</span></span>
<span id="cb210-9"><a href="genomics-regression.html#cb210-9" tabindex="-1"></a><span class="co"># cv_model2 = ???</span></span></code></pre></div>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="genomics-regression.html#cb211-1" tabindex="-1"></a><span class="co"># Uncomment the codes below to calculate the R^2</span></span>
<span id="cb211-2"><a href="genomics-regression.html#cb211-2" tabindex="-1"></a></span>
<span id="cb211-3"><a href="genomics-regression.html#cb211-3" tabindex="-1"></a><span class="co"># cor(cv_model2$pred$pred, cv_model2$pred$obs)**2</span></span></code></pre></div>
</div>
</div>
<div id="open-question" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> Open question<a href="genomics-regression.html#open-question" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Q9.</strong> Can you plot a scatter plot between log(SEF) vs X3ss_motif_ref, and colored by Ribosomal Protein Genes (RPG)? What did you see?</p>
<p><strong>Hint</strong>: follow the example on the diabetes data:
<a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#load-pima-indians-diabetes-database" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#load-pima-indians-diabetes-database</a></p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="genomics-regression.html#cb212-1" tabindex="-1"></a><span class="co"># Write you codes here with ggplot</span></span></code></pre></div>
<p><strong>Q10.</strong> Is the SEF score mainly driven by the RPG label or is it genuinely related to the 3’ss motif? How to test it?</p>
<p><strong>Hint</strong>: consider adding <code>RPG</code> as a covariate. Is 3’ss motif still significant?</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="genomics-regression.html#cb213-1" tabindex="-1"></a><span class="co"># Define your new model</span></span>
<span id="cb213-2"><a href="genomics-regression.html#cb213-2" tabindex="-1"></a>lr_fm3 <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">&quot;SEF_log ~&quot;</span>, <span class="fu">paste</span>(<span class="fu">colnames</span>(df)[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span><span class="sc">:</span><span class="dv">19</span>)], <span class="at">collapse=</span><span class="st">&quot; + &quot;</span>)))</span>
<span id="cb213-3"><a href="genomics-regression.html#cb213-3" tabindex="-1"></a></span>
<span id="cb213-4"><a href="genomics-regression.html#cb213-4" tabindex="-1"></a><span class="co"># Write you codes below to fit the model</span></span></code></pre></div>
<p><strong>Acknowledgement:</strong> this case study is adapted from the
<a href="https://genome.cshlp.org/content/28/2/203/F4.expansion.html">Figure 4</a> for the following research article:</p>
<ul>
<li>Aslanzadeh et al. <em>Transcription rate strongly affects splicing fidelity and cotranscriptionality in budding yeast</em>. Genome Research, 2018, 28 (2), 203-213</li>
</ul>


</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="genetic-seq.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="install.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/statbiomed/BMDS-book/tree/main/02-data_topics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BMDatSci.pdf", "BMDatSci.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
