<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Personalised Genomic Medicine | Biomedical Data Science - introduction with case studies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.28 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Personalised Genomic Medicine | Biomedical Data Science - introduction with case studies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Personalised Genomic Medicine | Biomedical Data Science - introduction with case studies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="BIOF1001 teaching team" />


<meta name="date" content="2022-10-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introHypoTest.html"/>
<link rel="next" href="image-digital.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">Biomedical Data Science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#introduction-for-readers"><i class="fa fa-check"></i>Introduction for readers</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-hope-you-will-learn-from-this-course-book"><i class="fa fa-check"></i>What we hope you will learn from this course / book</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-recommend-you-do-while-reading-this-book"><i class="fa fa-check"></i>What we recommend you do while reading this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#last-notes"><i class="fa fa-check"></i>Last notes</a></li>
</ul></li>
<li class="part"><span><b>I Data Science Foundations</b></span></li>
<li class="chapter" data-level="1" data-path="introR.html"><a href="introR.html"><i class="fa fa-check"></i><b>1</b> Introduction to R programming</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introR.html"><a href="introR.html#data-types"><i class="fa fa-check"></i><b>1.1</b> Data types</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introR.html"><a href="introR.html#nemeric-or-double"><i class="fa fa-check"></i><b>1.1.1</b> nemeric (or double)</a></li>
<li class="chapter" data-level="1.1.2" data-path="introR.html"><a href="introR.html#integer"><i class="fa fa-check"></i><b>1.1.2</b> integer</a></li>
<li class="chapter" data-level="1.1.3" data-path="introR.html"><a href="introR.html#logical"><i class="fa fa-check"></i><b>1.1.3</b> logical</a></li>
<li class="chapter" data-level="1.1.4" data-path="introR.html"><a href="introR.html#character"><i class="fa fa-check"></i><b>1.1.4</b> character</a></li>
<li class="chapter" data-level="1.1.5" data-path="introR.html"><a href="introR.html#memeory-usage"><i class="fa fa-check"></i><b>1.1.5</b> Memeory usage</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introR.html"><a href="introR.html#data-structures"><i class="fa fa-check"></i><b>1.2</b> Data structures</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introR.html"><a href="introR.html#vector"><i class="fa fa-check"></i><b>1.2.1</b> Vector</a></li>
<li class="chapter" data-level="1.2.2" data-path="introR.html"><a href="introR.html#matrix"><i class="fa fa-check"></i><b>1.2.2</b> Matrix</a></li>
<li class="chapter" data-level="1.2.3" data-path="introR.html"><a href="introR.html#list"><i class="fa fa-check"></i><b>1.2.3</b> List</a></li>
<li class="chapter" data-level="1.2.4" data-path="introR.html"><a href="introR.html#data-frame"><i class="fa fa-check"></i><b>1.2.4</b> Data Frame</a></li>
<li class="chapter" data-level="1.2.5" data-path="introR.html"><a href="introR.html#factor-vs-vector"><i class="fa fa-check"></i><b>1.2.5</b> Factor vs vector</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="introR.html"><a href="introR.html#read-and-write-files-tables"><i class="fa fa-check"></i><b>1.3</b> Read and write files (tables)</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introR.html"><a href="introR.html#read-file"><i class="fa fa-check"></i><b>1.3.1</b> Read file</a></li>
<li class="chapter" data-level="1.3.2" data-path="introR.html"><a href="introR.html#write-file"><i class="fa fa-check"></i><b>1.3.2</b> Write file</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="introR.html"><a href="introR.html#functions-and-packages"><i class="fa fa-check"></i><b>1.4</b> Functions and Packages</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="introR.html"><a href="introR.html#install-packages"><i class="fa fa-check"></i><b>1.4.1</b> Install packages</a></li>
<li class="chapter" data-level="1.4.2" data-path="introR.html"><a href="introR.html#apply-function-repeatly"><i class="fa fa-check"></i><b>1.4.2</b> Apply function repeatly</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="introR.html"><a href="introR.html#plotting"><i class="fa fa-check"></i><b>1.5</b> Plotting</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="introR.html"><a href="introR.html#datasets"><i class="fa fa-check"></i><b>1.5.1</b> datasets</a></li>
<li class="chapter" data-level="1.5.2" data-path="introR.html"><a href="introR.html#basic-plotting"><i class="fa fa-check"></i><b>1.5.2</b> Basic plotting</a></li>
<li class="chapter" data-level="1.5.3" data-path="introR.html"><a href="introR.html#ggplot2"><i class="fa fa-check"></i><b>1.5.3</b> ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="introR.html"><a href="introR.html#scientific-and-statistical-computating"><i class="fa fa-check"></i><b>1.6</b> Scientific and statistical computating</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="introR.html"><a href="introR.html#orders-of-operators"><i class="fa fa-check"></i><b>1.6.1</b> Orders of operators</a></li>
<li class="chapter" data-level="1.6.2" data-path="introR.html"><a href="introR.html#functions-for-statistics"><i class="fa fa-check"></i><b>1.6.2</b> Functions for statistics</a></li>
<li class="chapter" data-level="1.6.3" data-path="introR.html"><a href="introR.html#correlation"><i class="fa fa-check"></i><b>1.6.3</b> Correlation</a></li>
<li class="chapter" data-level="1.6.4" data-path="introR.html"><a href="introR.html#hypothesis-testing-t-test"><i class="fa fa-check"></i><b>1.6.4</b> Hypothesis testing (t test)</a></li>
<li class="chapter" data-level="1.6.5" data-path="introR.html"><a href="introR.html#regression"><i class="fa fa-check"></i><b>1.6.5</b> Regression</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="introR.html"><a href="introR.html#resource-links"><i class="fa fa-check"></i><b>1.7</b> Resource links</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introLinearReg.html"><a href="introLinearReg.html"><i class="fa fa-check"></i><b>2</b> Introduction to Linear Regression</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introLinearReg.html"><a href="introLinearReg.html#linear-regression-using-simulated-data"><i class="fa fa-check"></i><b>2.1</b> Linear Regression Using Simulated Data</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="introLinearReg.html"><a href="introLinearReg.html#simulating-data"><i class="fa fa-check"></i><b>2.1.1</b> Simulating data:</a></li>
<li class="chapter" data-level="2.1.2" data-path="introLinearReg.html"><a href="introLinearReg.html#model-efficacy"><i class="fa fa-check"></i><b>2.1.2</b> Model efficacy</a></li>
<li class="chapter" data-level="2.1.3" data-path="introLinearReg.html"><a href="introLinearReg.html#r-squared"><i class="fa fa-check"></i><b>2.1.3</b> <em>R-Squared</em></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="introLinearReg.html"><a href="introLinearReg.html#least-squares-using-simulated-data"><i class="fa fa-check"></i><b>2.2</b> Least Squares Using Simulated Data</a></li>
<li class="chapter" data-level="2.3" data-path="introLinearReg.html"><a href="introLinearReg.html#diagnostic-check-of-a-fitted-regression-model"><i class="fa fa-check"></i><b>2.3</b> Diagnostic check of a fitted regression model</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#residual-standard-errors-rse"><i class="fa fa-check"></i><b>2.3.1</b> Residual Standard Errors (RSE)</a></li>
<li class="chapter" data-level="2.3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#p-values"><i class="fa fa-check"></i><b>2.3.2</b> p-values</a></li>
<li class="chapter" data-level="2.3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#f-statistics"><i class="fa fa-check"></i><b>2.3.3</b> F-statistics</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="introLinearReg.html"><a href="introLinearReg.html#simple-linear-regression-with-lm-function"><i class="fa fa-check"></i><b>2.4</b> Simple Linear Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="2.5" data-path="introLinearReg.html"><a href="introLinearReg.html#multiple-regression-with-lm-function"><i class="fa fa-check"></i><b>2.5</b> Multiple Regression with <code>lm</code> function</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introClassifier.html"><a href="introClassifier.html"><i class="fa fa-check"></i><b>3</b> Introduction to Classification</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introClassifier.html"><a href="introClassifier.html#visualise-logistic-and-logit-functions"><i class="fa fa-check"></i><b>3.1</b> Visualise logistic and logit functions</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="introClassifier.html"><a href="introClassifier.html#logistic-function"><i class="fa fa-check"></i><b>3.1.1</b> Logistic function</a></li>
<li class="chapter" data-level="3.1.2" data-path="introClassifier.html"><a href="introClassifier.html#logit-function"><i class="fa fa-check"></i><b>3.1.2</b> Logit function</a></li>
<li class="chapter" data-level="3.1.3" data-path="introClassifier.html"><a href="introClassifier.html#visualise-the-distribution"><i class="fa fa-check"></i><b>3.1.3</b> Visualise the distribution</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="introClassifier.html"><a href="introClassifier.html#logistic-regression-on-diabetes"><i class="fa fa-check"></i><b>3.2</b> Logistic regression on Diabetes</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="introClassifier.html"><a href="introClassifier.html#load-pima-indians-diabetes-database"><i class="fa fa-check"></i><b>3.2.1</b> Load Pima Indians Diabetes Database</a></li>
<li class="chapter" data-level="3.2.2" data-path="introClassifier.html"><a href="introClassifier.html#fit-logistic-regression"><i class="fa fa-check"></i><b>3.2.2</b> Fit logistic regression</a></li>
<li class="chapter" data-level="3.2.3" data-path="introClassifier.html"><a href="introClassifier.html#assess-on-test-data"><i class="fa fa-check"></i><b>3.2.3</b> Assess on test data</a></li>
<li class="chapter" data-level="3.2.4" data-path="introClassifier.html"><a href="introClassifier.html#model-selection-and-diagnosis"><i class="fa fa-check"></i><b>3.2.4</b> Model selection and diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introClassifier.html"><a href="introClassifier.html#cross-validation"><i class="fa fa-check"></i><b>3.3</b> Cross-validation</a></li>
<li class="chapter" data-level="3.4" data-path="introClassifier.html"><a href="introClassifier.html#more-assessment-metrics"><i class="fa fa-check"></i><b>3.4</b> More assessment metrics</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="introClassifier.html"><a href="introClassifier.html#two-types-of-error"><i class="fa fa-check"></i><b>3.4.1</b> Two types of error</a></li>
<li class="chapter" data-level="3.4.2" data-path="introClassifier.html"><a href="introClassifier.html#roc-curve"><i class="fa fa-check"></i><b>3.4.2</b> ROC curve</a></li>
<li class="chapter" data-level="3.4.3" data-path="introClassifier.html"><a href="introClassifier.html#homework"><i class="fa fa-check"></i><b>3.4.3</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introHypoTest.html"><a href="introHypoTest.html"><i class="fa fa-check"></i><b>4</b> Introduction to Hypothesis testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-testing-and-p-value"><i class="fa fa-check"></i><b>4.1</b> Hypothesis testing and <em>p</em> value</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-1-probability-of-rolling-a-six"><i class="fa fa-check"></i><b>4.1.1</b> Example 1: probability of rolling a six?</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="introHypoTest.html"><a href="introHypoTest.html#permutation-test"><i class="fa fa-check"></i><b>4.2</b> Permutation test</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-2-difference-in-birth-weight"><i class="fa fa-check"></i><b>4.2.1</b> Example 2: difference in birth weight</a></li>
<li class="chapter" data-level="4.2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-approximated-by-resampling"><i class="fa fa-check"></i><b>4.2.2</b> Null distribution approximated by resampling</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="introHypoTest.html"><a href="introHypoTest.html#t-test"><i class="fa fa-check"></i><b>4.3</b> <em>t</em> test</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introHypoTest.html"><a href="introHypoTest.html#derivation-of-t-distribution"><i class="fa fa-check"></i><b>4.3.1</b> Derivation of t distribution</a></li>
<li class="chapter" data-level="4.3.2" data-path="introHypoTest.html"><a href="introHypoTest.html#direct-use-of-t.test"><i class="fa fa-check"></i><b>4.3.2</b> Direct use of <code>t.test()</code></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introHypoTest.html"><a href="introHypoTest.html#glm-test"><i class="fa fa-check"></i><b>4.4</b> GLM test</a></li>
<li class="chapter" data-level="4.5" data-path="introHypoTest.html"><a href="introHypoTest.html#multiple-testing"><i class="fa fa-check"></i><b>4.5</b> Multiple testing</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-test-statistic"><i class="fa fa-check"></i><b>4.5.1</b> Null distribution (of test statistic)</a></li>
<li class="chapter" data-level="4.5.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-p-value"><i class="fa fa-check"></i><b>4.5.2</b> Null distribution of p value</a></li>
<li class="chapter" data-level="4.5.3" data-path="introHypoTest.html"><a href="introHypoTest.html#minimal-p-values-in-10-tests"><i class="fa fa-check"></i><b>4.5.3</b> Minimal p values in 10 tests</a></li>
<li class="chapter" data-level="4.5.4" data-path="introHypoTest.html"><a href="introHypoTest.html#homework-1"><i class="fa fa-check"></i><b>4.5.4</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Biomedical Data Modules</b></span></li>
<li class="chapter" data-level="5" data-path="genomics.html"><a href="genomics.html"><i class="fa fa-check"></i><b>5</b> Personalised Genomic Medicine</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genomics.html"><a href="genomics.html#genomics-case1"><i class="fa fa-check"></i><b>5.1</b> Case study 1: splicing fedility prediction</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="genomics.html"><a href="genomics.html#questions-for-discussion"><i class="fa fa-check"></i><b>5.1.1</b> Questions for Discussion</a></li>
<li class="chapter" data-level="5.1.2" data-path="genomics.html"><a href="genomics.html#hands-on-with-regression-model"><i class="fa fa-check"></i><b>5.1.2</b> Hands-on with regression model</a></li>
<li class="chapter" data-level="5.1.3" data-path="genomics.html"><a href="genomics.html#open-question"><i class="fa fa-check"></i><b>5.1.3</b> Open question</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="genomics.html"><a href="genomics.html#genomics-case2"><i class="fa fa-check"></i><b>5.2</b> Case study 2: analysis of cBioportal mutation data</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genomics.html"><a href="genomics.html#exploratory-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Exploratory analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="genomics.html"><a href="genomics.html#statistical-analysis"><i class="fa fa-check"></i><b>5.2.2</b> Statistical analysis</a></li>
<li class="chapter" data-level="5.2.3" data-path="genomics.html"><a href="genomics.html#literature-search"><i class="fa fa-check"></i><b>5.2.3</b> Literature search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="image-digital.html"><a href="image-digital.html"><i class="fa fa-check"></i><b>6</b> Medical Image and Digital Health</a></li>
<li class="chapter" data-level="7" data-path="infectious-dis.html"><a href="infectious-dis.html"><i class="fa fa-check"></i><b>7</b> Infectious Disease Informatics</a></li>
<li class="chapter" data-level="8" data-path="pop-genetics.html"><a href="pop-genetics.html"><i class="fa fa-check"></i><b>8</b> Population Genetics and Diseases</a></li>
<li class="chapter" data-level="9" data-path="epidemiology.html"><a href="epidemiology.html"><i class="fa fa-check"></i><b>9</b> Epidemiology of Cancer and Other Diseases</a></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i>Appendix A: Install R &amp; RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.1-install-r"><i class="fa fa-check"></i>A.1 Install R</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-windows"><i class="fa fa-check"></i>R on Windows</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-macos"><i class="fa fa-check"></i>R on macOS</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-linux-ubuntu"><i class="fa fa-check"></i>R on Linux (Ubuntu)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.2-install-rstudio"><i class="fa fa-check"></i>A.2 Install RStudio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.3-use-r-inside-rstudio"><i class="fa fa-check"></i>A.3 Use R inside RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-studio"><i class="fa fa-check"></i>R studio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#some-general-knowledge"><i class="fa fa-check"></i>Some general knowledge</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#install-packages-1"><i class="fa fa-check"></i>Install packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a4.-cloud-computing"><i class="fa fa-check"></i>A4. Cloud computing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Data Science - introduction with case studies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genomics" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Personalised Genomic Medicine<a href="genomics.html#genomics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="genomics-case1" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Case study 1: splicing fedility prediction<a href="genomics.html#genomics-case1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Scenario: As a computational biology researcher, you want to find out what sequence-related features function as a regulator for <strong>splicing fidelity</strong>. You conducted a deep RNA-seq experience and quantified the <strong>splicing error frequency (SEF)</strong> for 141 splicing error events in yeast. Based on your prior knowledge and curiosity, you plan to further define a set of sequence-related features and test how predictive they are.</p>
<div id="questions-for-discussion" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Questions for Discussion<a href="genomics.html#questions-for-discussion" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Q1.</strong> What sequence-related features you would think of?</p>
<p><strong>Q2.</strong> Given a set of candidate features, what statistical methods can you use to determine if a feature genuinely contributes to splicing fidelity?</p>
<p><strong>Q3.</strong> Given a candidate feature set, what could be the best subset of features for predicting the splicing fidelity and in what sense?</p>
</div>
<div id="hands-on-with-regression-model" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Hands-on with regression model<a href="genomics.html#hands-on-with-regression-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now, you are given a dataset with 141 splicing error events containing Y for splicing error frequency (SEF) and 14 predictors X. Data set and template notebook are available on Moodle (recommended) and also on this
<a href="https://github.com/StatBiomed/BMDS-book/blob/main/notebooks/module1-genomics/df_SEF_prediction.tsv">GitHub Repo</a>.</p>
<p><strong>The information for columns:</strong></p>
<ul>
<li>RPG: Ribosomal Protein Gene</li>
<li>event: splicing error event type</li>
<li>shift: nucleotide shift between canonical (ref, reference) and erroneous (alt, alternative) splice sites</li>
<li>SEF: splicing error frequency</li>
<li>intronL_ref: intron length for canonical (ref) intron</li>
<li>5ss_bpL_ref: length between 5 splice site to branch point</li>
<li>DeltaG_intron_ref: the energy score in the secondary RNA structure, calculated by <a href="http://rna.tbi.univie.ac.at/cgi-bin/RNAWebSuite/RNAfold.cgi">RNAFold</a>. The lower the more stable.</li>
<li>5ss_motif_ref: the motif score of a sequence. The motif position weight matrix (PWM) is obtained by observed genes. The motif score is calculated as the log likelihood of each sequence given the PWM, followed by resealing to 100 for the maximum value. You can refer to <em>slide 7</em> in lecture topic 1.</li>
<li>intronL_alt: intron length for erroneous (alt) intron</li>
</ul>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="genomics.html#cb128-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;df_SEF_prediction.tsv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">header=</span><span class="dv">1</span>)</span>
<span id="cb128-2"><a href="genomics.html#cb128-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-3"><a href="genomics.html#cb128-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df)</span>
<span id="cb128-4"><a href="genomics.html#cb128-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    gene     RPG   event shift         SEF intronL_ref X5ss_bpL_ref</span></span>
<span id="cb128-5"><a href="genomics.html#cb128-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1  TFC3 non_RPG down3ss    17 0.479869423          90           77</span></span>
<span id="cb128-6"><a href="genomics.html#cb128-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 SEC17 non_RPG down3ss    47 0.021937843         116           77</span></span>
<span id="cb128-7"><a href="genomics.html#cb128-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3  ERD2 non_RPG   up3ss     4 0.059303591          97           63</span></span>
<span id="cb128-8"><a href="genomics.html#cb128-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4  LSM2 non_RPG   up5ss    56 0.019908116         128           92</span></span>
<span id="cb128-9"><a href="genomics.html#cb128-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  LSM2 non_RPG   up3ss    47 0.077444760         128           92</span></span>
<span id="cb128-10"><a href="genomics.html#cb128-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6  LSM2 non_RPG down3ss    40 0.002844017         128           92</span></span>
<span id="cb128-11"><a href="genomics.html#cb128-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   DeltaG_intron_ref DeltaG_5ss_bp_ref DeltaG_bp_3ss_ref DeltaG_3ss_ref</span></span>
<span id="cb128-12"><a href="genomics.html#cb128-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1          -0.11444          -0.13377               0.0           -6.2</span></span>
<span id="cb128-13"><a href="genomics.html#cb128-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2          -0.17155          -0.10779             -10.5          -20.3</span></span>
<span id="cb128-14"><a href="genomics.html#cb128-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3          -0.16804          -0.16190              -3.9           -8.7</span></span>
<span id="cb128-15"><a href="genomics.html#cb128-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4          -0.13906          -0.12065              -0.9           -7.0</span></span>
<span id="cb128-16"><a href="genomics.html#cb128-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5          -0.13906          -0.12065              -0.9           -7.0</span></span>
<span id="cb128-17"><a href="genomics.html#cb128-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6          -0.13906          -0.12065              -0.9           -7.0</span></span>
<span id="cb128-18"><a href="genomics.html#cb128-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   X5ss_motif_ref bp_motif_ref X3ss_motif_ref intronL_alt DeltaG_intron_alt</span></span>
<span id="cb128-19"><a href="genomics.html#cb128-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1         0.7213       0.7703         0.0769         107          -0.10841</span></span>
<span id="cb128-20"><a href="genomics.html#cb128-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2         0.7695       0.9942         0.1461         163          -0.19202</span></span>
<span id="cb128-21"><a href="genomics.html#cb128-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3         0.8495       0.8835         0.1496          93          -0.17527</span></span>
<span id="cb128-22"><a href="genomics.html#cb128-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4         0.8305       0.6667         0.3677         184          -0.15217</span></span>
<span id="cb128-23"><a href="genomics.html#cb128-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5         0.8305       0.6667         0.3677          81          -0.10864</span></span>
<span id="cb128-24"><a href="genomics.html#cb128-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6         0.8305       0.6667         0.3677         168          -0.11667</span></span>
<span id="cb128-25"><a href="genomics.html#cb128-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   DeltaG_3ss_alt X5ss_motif_alt X3ss_motif_alt</span></span>
<span id="cb128-26"><a href="genomics.html#cb128-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1           -6.8          77.00         0.5223</span></span>
<span id="cb128-27"><a href="genomics.html#cb128-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2          -13.0          70.99         0.2926</span></span>
<span id="cb128-28"><a href="genomics.html#cb128-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3           -8.3          86.10         0.3329</span></span>
<span id="cb128-29"><a href="genomics.html#cb128-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4           -7.0          64.20         0.3955</span></span>
<span id="cb128-30"><a href="genomics.html#cb128-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5           -8.9          80.76         0.6179</span></span>
<span id="cb128-31"><a href="genomics.html#cb128-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6           -4.3          80.76         0.3341</span></span></code></pre></div>
<div id="a-single-feature" class="section level4 hasAnchor" number="5.1.2.1">
<h4><span class="header-section-number">5.1.2.1</span> A single feature<a href="genomics.html#a-single-feature" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q4.</strong> Let’s start with a single feature, <code>bp_motif_ref</code>, is there a linear relationship with the output variable <code>SEF</code> Y? How do you determine it?</p>
<p><strong>Hint</strong>: try scatter plot to visualise the data. See example we mentioned before
<a href="https://statbiomed.github.io/BMDS-book/introR.html#scatter-plot-1" class="uri">https://statbiomed.github.io/BMDS-book/introR.html#scatter-plot-1</a></p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="genomics.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># consider visualise it</span></span>
<span id="cb129-2"><a href="genomics.html#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="co"># df$SEF and df$bp_motif_ref</span></span>
<span id="cb129-3"><a href="genomics.html#cb129-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-4"><a href="genomics.html#cb129-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb129-5"><a href="genomics.html#cb129-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-6"><a href="genomics.html#cb129-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Write your codes here</span></span></code></pre></div>
<p>How about doing a log transform for SEF score? Consider adding a pseudo count, e.g., 0.0001.</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="genomics.html#cb130-1" aria-hidden="true" tabindex="-1"></a>df[<span class="st">&quot;SEF_log&quot;</span>] <span class="ot">=</span> <span class="fu">log10</span>(df<span class="sc">$</span>SEF <span class="sc">+</span> <span class="fl">0.0001</span>)</span>
<span id="cb130-2"><a href="genomics.html#cb130-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-3"><a href="genomics.html#cb130-3" aria-hidden="true" tabindex="-1"></a><span class="co"># visualize it again</span></span></code></pre></div>
</div>
<div id="all-features-with-lm" class="section level4 hasAnchor" number="5.1.2.2">
<h4><span class="header-section-number">5.1.2.2</span> All features with lm<a href="genomics.html#all-features-with-lm" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q5.</strong> Can you build a linear model to perform the prediction by using all features? How to obtain the assessment statistics?</p>
<p><strong>Hint</strong>: consider using the <code>lm</code> function. See examples we mentioned before
<a href="https://statbiomed.github.io/BMDS-book/introLinearReg.html#simple-linear-regression-with-lm-function" class="uri">https://statbiomed.github.io/BMDS-book/introLinearReg.html#simple-linear-regression-with-lm-function</a></p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="genomics.html#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co"># For the formula, you can use this one, but you can also write your own:</span></span>
<span id="cb131-2"><a href="genomics.html#cb131-2" aria-hidden="true" tabindex="-1"></a>lr_fm1 <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">&quot;SEF_log ~ 1 + &quot;</span>, <span class="fu">paste</span>(<span class="fu">colnames</span>(df)[<span class="dv">6</span><span class="sc">:</span><span class="dv">19</span>], <span class="at">collapse=</span><span class="st">&quot; + &quot;</span>)))</span>
<span id="cb131-3"><a href="genomics.html#cb131-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-4"><a href="genomics.html#cb131-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Write your codes for fitting the model here</span></span></code></pre></div>
</div>
<div id="assessment-on-test-data" class="section level4 hasAnchor" number="5.1.2.3">
<h4><span class="header-section-number">5.1.2.3</span> Assessment on test data<a href="genomics.html#assessment-on-test-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q6.</strong> Can you perform the assessment on test data by splitting the data into training and test sets?</p>
<p>We didn’t introduce how perform training &amp; test splitting in linear regression, but you can adapt the scripts that we introduced for logistic regression.</p>
<p><strong>Hint</strong>: you can recall the logistic regress chapter for the training &amp; test spiting: <a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes</a></p>
<p>You can also just use the codes below, but think what is the ratio between training and test sets here:</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="genomics.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb132-2"><a href="genomics.html#cb132-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb132-3"><a href="genomics.html#cb132-3" aria-hidden="true" tabindex="-1"></a>idx_train <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(df),  <span class="at">size=</span><span class="fl">0.5</span><span class="sc">*</span><span class="fu">nrow</span>(df), <span class="at">replace =</span> <span class="cn">FALSE</span>)</span>
<span id="cb132-4"><a href="genomics.html#cb132-4" aria-hidden="true" tabindex="-1"></a>df_train <span class="ot">=</span> df[idx_train, ]</span>
<span id="cb132-5"><a href="genomics.html#cb132-5" aria-hidden="true" tabindex="-1"></a>df_test  <span class="ot">=</span> df[<span class="sc">-</span>idx_train, ] <span class="co"># recall the meaning of negative symbol</span></span></code></pre></div>
<p>Now, start training the model on the training set and predict on the test set.</p>
<p><strong>Hint</strong>: follow the codes here: <a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#logistic-regression-on-diabetes</a></p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="genomics.html#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="co"># recall the predict function</span></span>
<span id="cb133-2"><a href="genomics.html#cb133-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-3"><a href="genomics.html#cb133-3" aria-hidden="true" tabindex="-1"></a><span class="co"># lr_train = lm()</span></span>
<span id="cb133-4"><a href="genomics.html#cb133-4" aria-hidden="true" tabindex="-1"></a><span class="co"># res_test = predict()</span></span></code></pre></div>
<p>For calculating R^2, another quick way is using the Pearson’s correlation coefficient R, as it gives equivalent values. We can do it with the following codes with the <code>cor()</code> function.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="genomics.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Please uncomment the codes below:</span></span>
<span id="cb134-2"><a href="genomics.html#cb134-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-3"><a href="genomics.html#cb134-3" aria-hidden="true" tabindex="-1"></a><span class="co"># R_sqr = cor(res_test, df_test$SEF_log)**2</span></span>
<span id="cb134-4"><a href="genomics.html#cb134-4" aria-hidden="true" tabindex="-1"></a><span class="co"># R_sqr</span></span>
<span id="cb134-5"><a href="genomics.html#cb134-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-6"><a href="genomics.html#cb134-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Alternative function</span></span>
<span id="cb134-7"><a href="genomics.html#cb134-7" aria-hidden="true" tabindex="-1"></a><span class="co"># cor.test(res_test, df_test$SEF_log)</span></span></code></pre></div>
</div>
<div id="cross-validation-for-regression" class="section level4 hasAnchor" number="5.1.2.4">
<h4><span class="header-section-number">5.1.2.4</span> Cross-validation for regression<a href="genomics.html#cross-validation-for-regression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q7.</strong> Can you perform a 10-fold cross validation, calculate the R^2 between predicted and observed log(SEF), and plot it with ggplot?</p>
<p><strong>Hint</strong>: We didn’t mention cross-validation for linear regression but again we can adapt the codes from the logistic regression chapter:
<a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#cross-validation" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#cross-validation</a></p>
<p>You can also use the codes below, but try to fill the missing part (with question marks <code>???</code>) yourself.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="genomics.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb135-2"><a href="genomics.html#cb135-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-3"><a href="genomics.html#cb135-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define training control</span></span>
<span id="cb135-4"><a href="genomics.html#cb135-4" aria-hidden="true" tabindex="-1"></a><span class="co"># We also want to have savePredictions=TRUE &amp; classProbs=TRUE</span></span>
<span id="cb135-5"><a href="genomics.html#cb135-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>) </span>
<span id="cb135-6"><a href="genomics.html#cb135-6" aria-hidden="true" tabindex="-1"></a>my_trControl <span class="ot">&lt;-</span> <span class="fu">trainControl</span>(<span class="at">method =</span> <span class="st">&quot;cv&quot;</span>, <span class="at">number =</span> <span class="dv">10</span>, </span>
<span id="cb135-7"><a href="genomics.html#cb135-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">savePredictions =</span> <span class="cn">TRUE</span>)</span>
<span id="cb135-8"><a href="genomics.html#cb135-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-9"><a href="genomics.html#cb135-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Train the model (uncomment the codes below)</span></span>
<span id="cb135-10"><a href="genomics.html#cb135-10" aria-hidden="true" tabindex="-1"></a><span class="co"># cv_model &lt;- train(???, data = ???, </span></span>
<span id="cb135-11"><a href="genomics.html#cb135-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                   method = &quot;lm&quot;,</span></span>
<span id="cb135-12"><a href="genomics.html#cb135-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                   trControl = my_trControl)</span></span>
<span id="cb135-13"><a href="genomics.html#cb135-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-14"><a href="genomics.html#cb135-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize the results</span></span>
<span id="cb135-15"><a href="genomics.html#cb135-15" aria-hidden="true" tabindex="-1"></a><span class="co"># mean(cv_model$resample$Rsquared)</span></span>
<span id="cb135-16"><a href="genomics.html#cb135-16" aria-hidden="true" tabindex="-1"></a><span class="co"># print(cv_model)</span></span></code></pre></div>
<p>By using <code>print(cv_model)</code>, we can see the <em>Rsquared</em> value, which is an average of all R^2 acoss K folds (each fold has its own R^2 on the test set).</p>
<p>Now, let’s manually calculate the R^2 between observed and predicted Y for all samples in an aggregated manner, despite that the Ys are predicted from different models in different folds.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="genomics.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Uncomment the codes below to view the first a few lines of the data frame</span></span>
<span id="cb136-2"><a href="genomics.html#cb136-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-3"><a href="genomics.html#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="co"># head(cv_model$pred)</span></span></code></pre></div>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="genomics.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Uncomment the codes below to calculate the R^2</span></span>
<span id="cb137-2"><a href="genomics.html#cb137-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb137-3"><a href="genomics.html#cb137-3" aria-hidden="true" tabindex="-1"></a><span class="co"># cor(cv_model$pred$pred, cv_model$pred$obs)**2</span></span></code></pre></div>
</div>
<div id="a-subset-of-features" class="section level4 hasAnchor" number="5.1.2.5">
<h4><span class="header-section-number">5.1.2.5</span> A subset of features<a href="genomics.html#a-subset-of-features" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q8.</strong> Can you try a subset of features and see if you can improve the performance?</p>
<p><strong>Hint</strong>: You can consider forward method as mentioned here by checking the <span class="math inline">\(p\)</span> value and only keep the significant ones (think which significance level <span class="math inline">\(\alpha\)</span> you would like to allow:
<a href="https://statbiomed.github.io/BMDS-book/introLinearReg.html#multiple-regression-with-lm-function" class="uri">https://statbiomed.github.io/BMDS-book/introLinearReg.html#multiple-regression-with-lm-function</a></p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="genomics.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb138-2"><a href="genomics.html#cb138-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-3"><a href="genomics.html#cb138-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define your new formular:</span></span>
<span id="cb138-4"><a href="genomics.html#cb138-4" aria-hidden="true" tabindex="-1"></a><span class="co"># lr_fm2 = as.formula()</span></span>
<span id="cb138-5"><a href="genomics.html#cb138-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-6"><a href="genomics.html#cb138-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-7"><a href="genomics.html#cb138-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb138-8"><a href="genomics.html#cb138-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Try the 10-fold cross-validation with the new feature set yourself</span></span>
<span id="cb138-9"><a href="genomics.html#cb138-9" aria-hidden="true" tabindex="-1"></a><span class="co"># cv_model2 = ???</span></span></code></pre></div>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="genomics.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Uncomment the codes below to calculate the R^2</span></span>
<span id="cb139-2"><a href="genomics.html#cb139-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-3"><a href="genomics.html#cb139-3" aria-hidden="true" tabindex="-1"></a><span class="co"># cor(cv_model2$pred$pred, cv_model2$pred$obs)**2</span></span></code></pre></div>
</div>
</div>
<div id="open-question" class="section level3 hasAnchor" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Open question<a href="genomics.html#open-question" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Q9.</strong> Can you plot a scatter plot between log(SEF) vs X3ss_motif_ref, and colored by Ribosomal Protein Genes (RPG)? What did you see?</p>
<p><strong>Hint</strong>: follow the example on the diabetes data:
<a href="https://statbiomed.github.io/BMDS-book/introClassifier.html#load-pima-indians-diabetes-database" class="uri">https://statbiomed.github.io/BMDS-book/introClassifier.html#load-pima-indians-diabetes-database</a></p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="genomics.html#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Write you codes here with ggplot</span></span></code></pre></div>
<p><strong>Q10.</strong> Is the SEF score mainly driven by the RPG label or is it genuinely related to the 3’ss motif? How to test it?</p>
<p><strong>Hint</strong>: consider adding <code>RPG</code> as a covariate. Is 3’ss motif still significant?</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="genomics.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define your new model</span></span>
<span id="cb141-2"><a href="genomics.html#cb141-2" aria-hidden="true" tabindex="-1"></a>lr_fm3 <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">&quot;SEF_log ~&quot;</span>, <span class="fu">paste</span>(<span class="fu">colnames</span>(df)[<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">6</span><span class="sc">:</span><span class="dv">19</span>)], <span class="at">collapse=</span><span class="st">&quot; + &quot;</span>)))</span>
<span id="cb141-3"><a href="genomics.html#cb141-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-4"><a href="genomics.html#cb141-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Write you codes below to fit the model</span></span></code></pre></div>
<p><strong>Acknowledgement:</strong> this case study is adapted from the
<a href="https://genome.cshlp.org/content/28/2/203/F4.expansion.html">Figure 4</a> for the following research article:</p>
<ul>
<li>Aslanzadeh et al. <em>Transcription rate strongly affects splicing fidelity and cotranscriptionality in budding yeast</em>. Genome Research, 2018, 28 (2), 203-213</li>
</ul>
</div>
</div>
<div id="genomics-case2" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Case study 2: analysis of cBioportal mutation data<a href="genomics.html#genomics-case2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Non-small cell lung cancer is a deadly disease, and we would like to identify
mutations in genes that drive its formation and progression.
George et al. (2015) has sequenced 120 small cell lung cancer samples and this
data is available from <a href="cbioportal">cBioportal</a>.
The original data is in the <a href="https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/">MAF</a>, and we have selected a subset of the
columns, normalized the column names, and made the data available as an RDS
file.</p>
<p>In this exercise, we will explore the mutation data and examine genes that are
frequently mutated in non-small cell lung cancer samples.
We will implement a simple method for identifying candidate tumour suppressors
based on loss-of-function mutations.</p>
<div id="exploratory-analysis" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Exploratory analysis<a href="genomics.html#exploratory-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Q1</strong>. Load the data named <code>mutations_sclc_ucologne_2015.rds</code> into R.
Examine it using <code>head</code>.
In this <code>data.frame</code>, each row is a mutation detected in a particular gene
from a particular sample in the study.
For more information on the columns, see the <a href="https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/">MAF specification</a>.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="genomics.html#cb142-1" aria-hidden="true" tabindex="-1"></a>?readRDS</span>
<span id="cb142-2"><a href="genomics.html#cb142-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-3"><a href="genomics.html#cb142-3" aria-hidden="true" tabindex="-1"></a>?head</span></code></pre></div>
<p><strong>Q2</strong>. Identify the top 10 most frequently mutated genes.</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="genomics.html#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate count frequencies using table</span></span>
<span id="cb143-2"><a href="genomics.html#cb143-2" aria-hidden="true" tabindex="-1"></a>?table</span>
<span id="cb143-3"><a href="genomics.html#cb143-3" aria-hidden="true" tabindex="-1"></a><span class="co"># freqs &lt;- ???</span></span>
<span id="cb143-4"><a href="genomics.html#cb143-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-5"><a href="genomics.html#cb143-5" aria-hidden="true" tabindex="-1"></a><span class="co"># sort the frequency vector in decreasing order</span></span>
<span id="cb143-6"><a href="genomics.html#cb143-6" aria-hidden="true" tabindex="-1"></a>?sort</span>
<span id="cb143-7"><a href="genomics.html#cb143-7" aria-hidden="true" tabindex="-1"></a><span class="co"># freqs &lt;- ???</span></span>
<span id="cb143-8"><a href="genomics.html#cb143-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-9"><a href="genomics.html#cb143-9" aria-hidden="true" tabindex="-1"></a><span class="co"># select the first 10 genes in the freqs</span></span>
<span id="cb143-10"><a href="genomics.html#cb143-10" aria-hidden="true" tabindex="-1"></a><span class="co"># freqs.top &lt;- ???</span></span>
<span id="cb143-11"><a href="genomics.html#cb143-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-12"><a href="genomics.html#cb143-12" aria-hidden="true" tabindex="-1"></a><span class="co"># obtain the gene names</span></span>
<span id="cb143-13"><a href="genomics.html#cb143-13" aria-hidden="true" tabindex="-1"></a><span class="co"># genes &lt;- names(freqs.top);</span></span>
<span id="cb143-14"><a href="genomics.html#cb143-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-15"><a href="genomics.html#cb143-15" aria-hidden="true" tabindex="-1"></a><span class="co"># print(freqs.top)</span></span></code></pre></div>
<p><strong>Q3</strong>. Tabulate the count frequencies of <code>variant_class</code>.
What proportion of variants are <code>Silent</code> or inside an <code>Intron</code>?</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="genomics.html#cb144-1" aria-hidden="true" tabindex="-1"></a>?table</span></code></pre></div>
<p><strong>Q4</strong>. A variant class value <code>frame_shift_del</code> appears to be misspelt due to
case sensitivity. Correct this error.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="genomics.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># x$variant_class[x$variant_class == &quot;frame_shift_del&quot;] &lt;- ???;</span></span></code></pre></div>
<p><strong>Q5</strong>. To help us simplify the variant classes into loss-of-function or neutral,
let us import the <code>mutation_effects.tsv</code> data.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="genomics.html#cb146-1" aria-hidden="true" tabindex="-1"></a>?read.table</span>
<span id="cb146-2"><a href="genomics.html#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Hint: You need to set the `header` and `sep` arguments</span></span></code></pre></div>
<p><strong>Q6</strong>. Define a new column in <code>x</code> that converts the variant classes to effects.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="genomics.html#cb147-1" aria-hidden="true" tabindex="-1"></a>?match</span></code></pre></div>
<p><strong>Q7</strong>. Create a subset of the data for the most frequently mutated genes,
and tabulate the mutation count frequencies of variant classes for each gene.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="genomics.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="co"># x.sub &lt;- x[???, ]</span></span>
<span id="cb148-2"><a href="genomics.html#cb148-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-3"><a href="genomics.html#cb148-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-4"><a href="genomics.html#cb148-4" aria-hidden="true" tabindex="-1"></a><span class="co"># order the factor levels based on `genes`,</span></span>
<span id="cb148-5"><a href="genomics.html#cb148-5" aria-hidden="true" tabindex="-1"></a><span class="co"># which was previously sorted in decreasing order of frequency</span></span>
<span id="cb148-6"><a href="genomics.html#cb148-6" aria-hidden="true" tabindex="-1"></a><span class="co"># x.sub$gene &lt;- factor(x.sub$gene, levels=genes);</span></span>
<span id="cb148-7"><a href="genomics.html#cb148-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-8"><a href="genomics.html#cb148-8" aria-hidden="true" tabindex="-1"></a>?table</span></code></pre></div>
<p><strong>Q8</strong>. Determine the number of mutations in each effect category, across all genes.
Save the result in a variable named <code>overall.counts</code>.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="genomics.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overall.counts &lt;- table(???);</span></span>
<span id="cb149-2"><a href="genomics.html#cb149-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-3"><a href="genomics.html#cb149-3" aria-hidden="true" tabindex="-1"></a><span class="co"># overall.counts</span></span></code></pre></div>
<p><strong>Q9</strong>. Tabulate the mutation count frequencies for each gene and effect.
Save the results in a variable named <code>gene.counts</code>.
Look up the frequently mutated genes in this matrix.</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="genomics.html#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gene.counts &lt;- table(???, ???);</span></span>
<span id="cb150-2"><a href="genomics.html#cb150-2" aria-hidden="true" tabindex="-1"></a><span class="co"># gene.counts[???, ]</span></span></code></pre></div>
</div>
<div id="statistical-analysis" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Statistical analysis<a href="genomics.html#statistical-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we implemented a simple statistical test for assessing whether a
gene is significantly frequently targeted by loss-of-function mutations,
based on the Fisher’s exact test.</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="genomics.html#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Test whether a query gene has significantly more loss-of-function</span></span>
<span id="cb151-2"><a href="genomics.html#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mutations compared with other genes.</span></span>
<span id="cb151-3"><a href="genomics.html#cb151-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Requires global variables `gene.counts` and `overall.counts`</span></span>
<span id="cb151-4"><a href="genomics.html#cb151-4" aria-hidden="true" tabindex="-1"></a>lof_test <span class="ot">&lt;-</span> <span class="cf">function</span>(gene) {</span>
<span id="cb151-5"><a href="genomics.html#cb151-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb151-6"><a href="genomics.html#cb151-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Construct the following contingency table:</span></span>
<span id="cb151-7"><a href="genomics.html#cb151-7" aria-hidden="true" tabindex="-1"></a>    <span class="co">#              neutral    lof</span></span>
<span id="cb151-8"><a href="genomics.html#cb151-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># other genes  a          b</span></span>
<span id="cb151-9"><a href="genomics.html#cb151-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># query gene   c          d</span></span>
<span id="cb151-10"><a href="genomics.html#cb151-10" aria-hidden="true" tabindex="-1"></a>    cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;neutral&quot;</span>, <span class="st">&quot;loss_of_function&quot;</span>);</span>
<span id="cb151-11"><a href="genomics.html#cb151-11" aria-hidden="true" tabindex="-1"></a>    gene.counts.sel <span class="ot">&lt;-</span> gene.counts[gene, cols];</span>
<span id="cb151-12"><a href="genomics.html#cb151-12" aria-hidden="true" tabindex="-1"></a>    ct <span class="ot">&lt;-</span> <span class="fu">matrix</span>(</span>
<span id="cb151-13"><a href="genomics.html#cb151-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">c</span>(</span>
<span id="cb151-14"><a href="genomics.html#cb151-14" aria-hidden="true" tabindex="-1"></a>            overall.counts[cols] <span class="sc">-</span> gene.counts.sel,</span>
<span id="cb151-15"><a href="genomics.html#cb151-15" aria-hidden="true" tabindex="-1"></a>            gene.counts.sel</span>
<span id="cb151-16"><a href="genomics.html#cb151-16" aria-hidden="true" tabindex="-1"></a>        ),</span>
<span id="cb151-17"><a href="genomics.html#cb151-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">byrow=</span><span class="cn">TRUE</span>,</span>
<span id="cb151-18"><a href="genomics.html#cb151-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">ncol=</span><span class="dv">2</span></span>
<span id="cb151-19"><a href="genomics.html#cb151-19" aria-hidden="true" tabindex="-1"></a>    );</span>
<span id="cb151-20"><a href="genomics.html#cb151-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-21"><a href="genomics.html#cb151-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Test whether the lof mutations are greater in frequency compared </span></span>
<span id="cb151-22"><a href="genomics.html#cb151-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># to neutral mutations in the query gene, in comparison to all other genes</span></span>
<span id="cb151-23"><a href="genomics.html#cb151-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">fisher.test</span>(ct, <span class="at">alternative=</span><span class="st">&quot;greater&quot;</span>)</span>
<span id="cb151-24"><a href="genomics.html#cb151-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p><strong>Q10</strong>. Subset the <code>gene.counts</code> table for the <em>TP53</em> gene.
Run <code>lof_test</code> on this gene.</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="genomics.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gene.counts[???, ]</span></span>
<span id="cb152-2"><a href="genomics.html#cb152-2" aria-hidden="true" tabindex="-1"></a><span class="co"># lof_test(???)</span></span></code></pre></div>
<p><strong>Q11</strong>. Now, test to see if <em>TTN</em> and <em>MUC16</em> are significantly frequently
targeted by loss-of-function mutations.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="genomics.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># similar to above</span></span></code></pre></div>
<p><strong>Q12</strong>. Apply the loss-of-function test to all frequently mutated genes.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="genomics.html#cb154-1" aria-hidden="true" tabindex="-1"></a>?lapply</span></code></pre></div>
<p><strong>Q13</strong>. Extract odds ratio and p-values from the test results.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="genomics.html#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="co"># odds.ratios &lt;- vapply(hs, function(h) h$estimate, 0);</span></span>
<span id="cb155-2"><a href="genomics.html#cb155-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ps &lt;- vapply(hs, function(h) ???, 0);</span></span></code></pre></div>
<p><strong>Q14</strong>. Since we tested many genes, we need to adjust for multiple hypothesis
testing so that we can control the false discovery rate.
So, adjust the p-values to obtain q-values.</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="genomics.html#cb156-1" aria-hidden="true" tabindex="-1"></a>?p.adjust</span></code></pre></div>
<p><strong>Q15</strong>. Construct a results <code>data.frame</code> with gene names, odds ratio, p-values,
and q-values that summarize the loss-of-function test results.</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="genomics.html#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="co"># res &lt;- data.frame(</span></span>
<span id="cb157-2"><a href="genomics.html#cb157-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     gene = ???,</span></span>
<span id="cb157-3"><a href="genomics.html#cb157-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     odds_ratio = ???,</span></span>
<span id="cb157-4"><a href="genomics.html#cb157-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     p = ???,</span></span>
<span id="cb157-5"><a href="genomics.html#cb157-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     q = ???</span></span>
<span id="cb157-6"><a href="genomics.html#cb157-6" aria-hidden="true" tabindex="-1"></a><span class="co"># );</span></span>
<span id="cb157-7"><a href="genomics.html#cb157-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-8"><a href="genomics.html#cb157-8" aria-hidden="true" tabindex="-1"></a>?order</span>
<span id="cb157-9"><a href="genomics.html#cb157-9" aria-hidden="true" tabindex="-1"></a><span class="co">#res &lt;- res[order(???), ];</span></span></code></pre></div>
<p><strong>Q16</strong>. Identify the significant genes from the results <code>data.frame</code> at a
false discovery rate of 5% (i.e. q-value threshold of 0.05).</p>
<p><strong>Q17</strong>. Look up the significant genes in the <code>gene.counts</code> matrix.</p>
</div>
<div id="literature-search" class="section level3 hasAnchor" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> Literature search<a href="genomics.html#literature-search" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Q18</strong>. Do the significant genes appear to be involved in cancer?
What general roles do they play?</p>
<p><strong>Q19</strong>. How do these genes contribute specifically to the formation
or progression of small cell lung cancer?</p>
<p><strong>Q20</strong>. What are some possible ways of identifying oncogenes that are activated
by mutations or other genomic alterations?</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introHypoTest.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="image-digital.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/statbiomed/BMDS-book/tree/main/02-data_topics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BMDatSci.pdf", "BMDatSci.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
