<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction to R programming | Biomedical Data Science - introduction with case studies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction to R programming | Biomedical Data Science - introduction with case studies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction to R programming | Biomedical Data Science - introduction with case studies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="BIOF1001 teaching team" />


<meta name="date" content="2023-09-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="preface.html"/>
<link rel="next" href="introLinearReg.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">Biomedical Data Science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#introduction-for-readers"><i class="fa fa-check"></i>Introduction for readers</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-you-will-learn-from-this-coursebook"><i class="fa fa-check"></i>What you will learn from this course/book</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-recommend-you-do-while-reading-this-book"><i class="fa fa-check"></i>What we recommend you do while reading this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#other-reference-books"><i class="fa fa-check"></i>Other reference books</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="part"><span><b>I Data Science Foundations</b></span></li>
<li class="chapter" data-level="1" data-path="introR.html"><a href="introR.html"><i class="fa fa-check"></i><b>1</b> Introduction to R programming</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introR.html"><a href="introR.html#data-types"><i class="fa fa-check"></i><b>1.1</b> Data types</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introR.html"><a href="introR.html#nemeric-or-double"><i class="fa fa-check"></i><b>1.1.1</b> nemeric (or double)</a></li>
<li class="chapter" data-level="1.1.2" data-path="introR.html"><a href="introR.html#integer"><i class="fa fa-check"></i><b>1.1.2</b> integer</a></li>
<li class="chapter" data-level="1.1.3" data-path="introR.html"><a href="introR.html#logical"><i class="fa fa-check"></i><b>1.1.3</b> logical</a></li>
<li class="chapter" data-level="1.1.4" data-path="introR.html"><a href="introR.html#character"><i class="fa fa-check"></i><b>1.1.4</b> character</a></li>
<li class="chapter" data-level="1.1.5" data-path="introR.html"><a href="introR.html#memeory-usage"><i class="fa fa-check"></i><b>1.1.5</b> Memeory usage</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introR.html"><a href="introR.html#data-structures"><i class="fa fa-check"></i><b>1.2</b> Data structures</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introR.html"><a href="introR.html#vector"><i class="fa fa-check"></i><b>1.2.1</b> Vector</a></li>
<li class="chapter" data-level="1.2.2" data-path="introR.html"><a href="introR.html#matrix"><i class="fa fa-check"></i><b>1.2.2</b> Matrix</a></li>
<li class="chapter" data-level="1.2.3" data-path="introR.html"><a href="introR.html#list"><i class="fa fa-check"></i><b>1.2.3</b> List</a></li>
<li class="chapter" data-level="1.2.4" data-path="introR.html"><a href="introR.html#data-frame"><i class="fa fa-check"></i><b>1.2.4</b> Data Frame</a></li>
<li class="chapter" data-level="1.2.5" data-path="introR.html"><a href="introR.html#factor-vs-vector"><i class="fa fa-check"></i><b>1.2.5</b> Factor vs vector</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="introR.html"><a href="introR.html#read-and-write-files-tables"><i class="fa fa-check"></i><b>1.3</b> Read and write files (tables)</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introR.html"><a href="introR.html#read-file"><i class="fa fa-check"></i><b>1.3.1</b> Read file</a></li>
<li class="chapter" data-level="1.3.2" data-path="introR.html"><a href="introR.html#write-file"><i class="fa fa-check"></i><b>1.3.2</b> Write file</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="introR.html"><a href="introR.html#functions-and-packages"><i class="fa fa-check"></i><b>1.4</b> Functions and Packages</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="introR.html"><a href="introR.html#install-packages"><i class="fa fa-check"></i><b>1.4.1</b> Install packages</a></li>
<li class="chapter" data-level="1.4.2" data-path="introR.html"><a href="introR.html#apply-function-repeatly"><i class="fa fa-check"></i><b>1.4.2</b> Apply function repeatly</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="introR.html"><a href="introR.html#plotting"><i class="fa fa-check"></i><b>1.5</b> Plotting</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="introR.html"><a href="introR.html#datasets"><i class="fa fa-check"></i><b>1.5.1</b> datasets</a></li>
<li class="chapter" data-level="1.5.2" data-path="introR.html"><a href="introR.html#basic-plotting"><i class="fa fa-check"></i><b>1.5.2</b> Basic plotting</a></li>
<li class="chapter" data-level="1.5.3" data-path="introR.html"><a href="introR.html#ggplot2"><i class="fa fa-check"></i><b>1.5.3</b> ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="introR.html"><a href="introR.html#scientific-and-statistical-computating"><i class="fa fa-check"></i><b>1.6</b> Scientific and statistical computating</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="introR.html"><a href="introR.html#orders-of-operators"><i class="fa fa-check"></i><b>1.6.1</b> Orders of operators</a></li>
<li class="chapter" data-level="1.6.2" data-path="introR.html"><a href="introR.html#functions-for-statistics"><i class="fa fa-check"></i><b>1.6.2</b> Functions for statistics</a></li>
<li class="chapter" data-level="1.6.3" data-path="introR.html"><a href="introR.html#correlation"><i class="fa fa-check"></i><b>1.6.3</b> Correlation</a></li>
<li class="chapter" data-level="1.6.4" data-path="introR.html"><a href="introR.html#hypothesis-testing-t-test"><i class="fa fa-check"></i><b>1.6.4</b> Hypothesis testing (t test)</a></li>
<li class="chapter" data-level="1.6.5" data-path="introR.html"><a href="introR.html#regression"><i class="fa fa-check"></i><b>1.6.5</b> Regression</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="introR.html"><a href="introR.html#resource-links"><i class="fa fa-check"></i><b>1.7</b> Resource links</a></li>
<li class="chapter" data-level="1.8" data-path="introR.html"><a href="introR.html#exercises"><i class="fa fa-check"></i><b>1.8</b> Exercises</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="introR.html"><a href="introR.html#exercises-p1"><i class="fa fa-check"></i><b>1.8.1</b> Part 1. Basics (~40min)</a></li>
<li class="chapter" data-level="1.8.2" data-path="introR.html"><a href="introR.html#exercises-p2"><i class="fa fa-check"></i><b>1.8.2</b> Part 2. Making plotting (~40min)</a></li>
<li class="chapter" data-level="1.8.3" data-path="introR.html"><a href="introR.html#exercises-p3"><i class="fa fa-check"></i><b>1.8.3</b> Part 3. For loop and repeating processing (~40min)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introLinearReg.html"><a href="introLinearReg.html"><i class="fa fa-check"></i><b>2</b> Introduction to Linear Regression</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introLinearReg.html"><a href="introLinearReg.html#linear-regression-using-simulated-data"><i class="fa fa-check"></i><b>2.1</b> Linear Regression Using Simulated Data</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="introLinearReg.html"><a href="introLinearReg.html#simulating-data"><i class="fa fa-check"></i><b>2.1.1</b> Simulating data:</a></li>
<li class="chapter" data-level="2.1.2" data-path="introLinearReg.html"><a href="introLinearReg.html#model-efficacy"><i class="fa fa-check"></i><b>2.1.2</b> Model efficacy</a></li>
<li class="chapter" data-level="2.1.3" data-path="introLinearReg.html"><a href="introLinearReg.html#r-squared"><i class="fa fa-check"></i><b>2.1.3</b> <em>R-Squared</em></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="introLinearReg.html"><a href="introLinearReg.html#least-squares-using-simulated-data"><i class="fa fa-check"></i><b>2.2</b> Least Squares Using Simulated Data</a></li>
<li class="chapter" data-level="2.3" data-path="introLinearReg.html"><a href="introLinearReg.html#diagnostic-check-of-a-fitted-regression-model"><i class="fa fa-check"></i><b>2.3</b> Diagnostic check of a fitted regression model</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#residual-standard-errors-rse"><i class="fa fa-check"></i><b>2.3.1</b> Residual Standard Errors (RSE)</a></li>
<li class="chapter" data-level="2.3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#p-values"><i class="fa fa-check"></i><b>2.3.2</b> p-values</a></li>
<li class="chapter" data-level="2.3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#f-statistics"><i class="fa fa-check"></i><b>2.3.3</b> F-statistics</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="introLinearReg.html"><a href="introLinearReg.html#simple-linear-regression-with-lm-function"><i class="fa fa-check"></i><b>2.4</b> Simple Linear Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="2.5" data-path="introLinearReg.html"><a href="introLinearReg.html#multiple-regression-with-lm-function"><i class="fa fa-check"></i><b>2.5</b> Multiple Regression with <code>lm</code> function</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introClassifier.html"><a href="introClassifier.html"><i class="fa fa-check"></i><b>3</b> Introduction to Classification</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introClassifier.html"><a href="introClassifier.html#visualise-logistic-and-logit-functions"><i class="fa fa-check"></i><b>3.1</b> Visualise logistic and logit functions</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="introClassifier.html"><a href="introClassifier.html#logistic-function"><i class="fa fa-check"></i><b>3.1.1</b> Logistic function</a></li>
<li class="chapter" data-level="3.1.2" data-path="introClassifier.html"><a href="introClassifier.html#logit-function"><i class="fa fa-check"></i><b>3.1.2</b> Logit function</a></li>
<li class="chapter" data-level="3.1.3" data-path="introClassifier.html"><a href="introClassifier.html#visualise-the-distribution"><i class="fa fa-check"></i><b>3.1.3</b> Visualise the distribution</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="introClassifier.html"><a href="introClassifier.html#logistic-regression-on-diabetes"><i class="fa fa-check"></i><b>3.2</b> Logistic regression on Diabetes</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="introClassifier.html"><a href="introClassifier.html#load-pima-indians-diabetes-database"><i class="fa fa-check"></i><b>3.2.1</b> Load Pima Indians Diabetes Database</a></li>
<li class="chapter" data-level="3.2.2" data-path="introClassifier.html"><a href="introClassifier.html#fit-logistic-regression"><i class="fa fa-check"></i><b>3.2.2</b> Fit logistic regression</a></li>
<li class="chapter" data-level="3.2.3" data-path="introClassifier.html"><a href="introClassifier.html#assess-on-test-data"><i class="fa fa-check"></i><b>3.2.3</b> Assess on test data</a></li>
<li class="chapter" data-level="3.2.4" data-path="introClassifier.html"><a href="introClassifier.html#model-selection-and-diagnosis"><i class="fa fa-check"></i><b>3.2.4</b> Model selection and diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introClassifier.html"><a href="introClassifier.html#cross-validation"><i class="fa fa-check"></i><b>3.3</b> Cross-validation</a></li>
<li class="chapter" data-level="3.4" data-path="introClassifier.html"><a href="introClassifier.html#more-assessment-metrics"><i class="fa fa-check"></i><b>3.4</b> More assessment metrics</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="introClassifier.html"><a href="introClassifier.html#two-types-of-error"><i class="fa fa-check"></i><b>3.4.1</b> Two types of error</a></li>
<li class="chapter" data-level="3.4.2" data-path="introClassifier.html"><a href="introClassifier.html#roc-curve"><i class="fa fa-check"></i><b>3.4.2</b> ROC curve</a></li>
<li class="chapter" data-level="3.4.3" data-path="introClassifier.html"><a href="introClassifier.html#homework"><i class="fa fa-check"></i><b>3.4.3</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introHypoTest.html"><a href="introHypoTest.html"><i class="fa fa-check"></i><b>4</b> Introduction to Hypothesis testing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-testing-and-p-value"><i class="fa fa-check"></i><b>4.1</b> Hypothesis testing and <em>p</em> value</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-1-probability-of-rolling-a-six"><i class="fa fa-check"></i><b>4.1.1</b> Example 1: probability of rolling a six?</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="introHypoTest.html"><a href="introHypoTest.html#permutation-test"><i class="fa fa-check"></i><b>4.2</b> Permutation test</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-2-difference-in-birth-weight"><i class="fa fa-check"></i><b>4.2.1</b> Example 2: difference in birth weight</a></li>
<li class="chapter" data-level="4.2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-approximated-by-resampling"><i class="fa fa-check"></i><b>4.2.2</b> Null distribution approximated by resampling</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="introHypoTest.html"><a href="introHypoTest.html#t-test"><i class="fa fa-check"></i><b>4.3</b> <em>t</em> test</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="introHypoTest.html"><a href="introHypoTest.html#derivation-of-t-distribution"><i class="fa fa-check"></i><b>4.3.1</b> Derivation of t distribution</a></li>
<li class="chapter" data-level="4.3.2" data-path="introHypoTest.html"><a href="introHypoTest.html#direct-use-of-t.test"><i class="fa fa-check"></i><b>4.3.2</b> Direct use of <code>t.test()</code></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="introHypoTest.html"><a href="introHypoTest.html#glm-test"><i class="fa fa-check"></i><b>4.4</b> GLM test</a></li>
<li class="chapter" data-level="4.5" data-path="introHypoTest.html"><a href="introHypoTest.html#multiple-testing"><i class="fa fa-check"></i><b>4.5</b> Multiple testing</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-test-statistic"><i class="fa fa-check"></i><b>4.5.1</b> Null distribution (of test statistic)</a></li>
<li class="chapter" data-level="4.5.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-p-value"><i class="fa fa-check"></i><b>4.5.2</b> Null distribution of p value</a></li>
<li class="chapter" data-level="4.5.3" data-path="introHypoTest.html"><a href="introHypoTest.html#minimal-p-values-in-10-tests"><i class="fa fa-check"></i><b>4.5.3</b> Minimal p values in 10 tests</a></li>
<li class="chapter" data-level="4.5.4" data-path="introHypoTest.html"><a href="introHypoTest.html#homework-1"><i class="fa fa-check"></i><b>4.5.4</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Biomedical Data Modules</b></span></li>
<li class="chapter" data-level="5" data-path="image-digital.html"><a href="image-digital.html"><i class="fa fa-check"></i><b>5</b> Medical Image and Digital Health</a></li>
<li class="chapter" data-level="6" data-path="infectious-dis.html"><a href="infectious-dis.html"><i class="fa fa-check"></i><b>6</b> Infectious Disease Informatics</a></li>
<li class="chapter" data-level="7" data-path="cancer.html"><a href="cancer.html"><i class="fa fa-check"></i><b>7</b> Cancer genomics and epidemiology</a>
<ul>
<li class="chapter" data-level="7.1" data-path="cancer.html"><a href="cancer.html#cancer-case1"><i class="fa fa-check"></i><b>7.1</b> Case study 1: analysis of cBioportal mutation data</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="cancer.html"><a href="cancer.html#exploratory-analysis"><i class="fa fa-check"></i><b>7.1.1</b> Exploratory analysis</a></li>
<li class="chapter" data-level="7.1.2" data-path="cancer.html"><a href="cancer.html#statistical-analysis"><i class="fa fa-check"></i><b>7.1.2</b> Statistical analysis</a></li>
<li class="chapter" data-level="7.1.3" data-path="cancer.html"><a href="cancer.html#literature-search"><i class="fa fa-check"></i><b>7.1.3</b> Literature search</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="cancer.html"><a href="cancer.html#cancer-case2"><i class="fa fa-check"></i><b>7.2</b> Case study 2: Cancer Epidemiology</a></li>
<li class="chapter" data-level="7.3" data-path="cancer.html"><a href="cancer.html#scenario"><i class="fa fa-check"></i><b>7.3</b> 1. Scenario</a></li>
<li class="chapter" data-level="7.4" data-path="cancer.html"><a href="cancer.html#hong-kong-population"><i class="fa fa-check"></i><b>7.4</b> 2. Hong Kong population</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="cancer.html"><a href="cancer.html#download-population-data"><i class="fa fa-check"></i><b>7.4.1</b> Download population data</a></li>
<li class="chapter" data-level="7.4.2" data-path="cancer.html"><a href="cancer.html#format-data-for-plotting"><i class="fa fa-check"></i><b>7.4.2</b> Format data for plotting</a></li>
<li class="chapter" data-level="7.4.3" data-path="cancer.html"><a href="cancer.html#plotting-population-data"><i class="fa fa-check"></i><b>7.4.3</b> Plotting population data</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="cancer.html"><a href="cancer.html#cancer-registry-data"><i class="fa fa-check"></i><b>7.5</b> 3. Cancer registry data</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="cancer.html"><a href="cancer.html#download-cancer-registry-data"><i class="fa fa-check"></i><b>7.5.1</b> Download cancer registry data</a></li>
<li class="chapter" data-level="7.5.2" data-path="cancer.html"><a href="cancer.html#a.-visualise-changes-in-incidence-and-mortality"><i class="fa fa-check"></i><b>7.5.2</b> 3a. Visualise changes in incidence and mortality</a></li>
<li class="chapter" data-level="7.5.3" data-path="cancer.html"><a href="cancer.html#b.-calculate-cancer-risk-and-mortality-rate"><i class="fa fa-check"></i><b>7.5.3</b> 3b. Calculate cancer risk and mortality rate</a></li>
<li class="chapter" data-level="7.5.4" data-path="cancer.html"><a href="cancer.html#c.-mortality-incidence-ratio"><i class="fa fa-check"></i><b>7.5.4</b> 3c. Mortality-incidence ratio</a></li>
<li class="chapter" data-level="7.5.5" data-path="cancer.html"><a href="cancer.html#d.-paired-t-test-on-mortality-incidence-ratio-change"><i class="fa fa-check"></i><b>7.5.5</b> 3d. Paired t-test on mortality-incidence ratio change</a></li>
<li class="chapter" data-level="7.5.6" data-path="cancer.html"><a href="cancer.html#e.-childhood-versus-elderly-cancers"><i class="fa fa-check"></i><b>7.5.6</b> 3e. Childhood versus elderly cancers</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="cancer.html"><a href="cancer.html#existing-cancer-funding-and-publication-data"><i class="fa fa-check"></i><b>7.6</b> 4. Existing cancer funding and publication data</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="cancer.html"><a href="cancer.html#download-hmrf-grants-and-pubmed-data"><i class="fa fa-check"></i><b>7.6.1</b> Download HMRF grants and Pubmed data</a></li>
<li class="chapter" data-level="7.6.2" data-path="cancer.html"><a href="cancer.html#make-word-cloud-for-grants"><i class="fa fa-check"></i><b>7.6.2</b> Make word cloud for grants</a></li>
<li class="chapter" data-level="7.6.3" data-path="cancer.html"><a href="cancer.html#make-compare-grant-funding-with-incidence-and-mortality"><i class="fa fa-check"></i><b>7.6.3</b> Make compare grant funding with incidence and mortality</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="cancer.html"><a href="cancer.html#task-to-be-completed-in-tutotial-2-by-300pm-and-shared-with-other-tutorial-group"><i class="fa fa-check"></i><b>7.7</b> 5. Task (to be completed in Tutotial 2 by 3:00pm and shared with other tutorial group)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="pop-genetics.html"><a href="pop-genetics.html"><i class="fa fa-check"></i><b>8</b> Population Genetics and Diseases</a>
<ul>
<li class="chapter" data-level="8.1" data-path="pop-genetics.html"><a href="pop-genetics.html#case-study-1-heritability-and-human-traits"><i class="fa fa-check"></i><b>8.1</b> Case study 1: Heritability and human traits</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="pop-genetics.html"><a href="pop-genetics.html#part-1"><i class="fa fa-check"></i><b>8.1.1</b> Part 1</a></li>
<li class="chapter" data-level="8.1.2" data-path="pop-genetics.html"><a href="pop-genetics.html#part-2"><i class="fa fa-check"></i><b>8.1.2</b> Part 2</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i>Appendix A: Install R &amp; RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.1-install-r"><i class="fa fa-check"></i>A.1 Install R</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-windows"><i class="fa fa-check"></i>R on Windows</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-macos"><i class="fa fa-check"></i>R on macOS</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-linux-ubuntu"><i class="fa fa-check"></i>R on Linux (Ubuntu)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.2-install-rstudio"><i class="fa fa-check"></i>A.2 Install RStudio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.3-use-r-inside-rstudio"><i class="fa fa-check"></i>A.3 Use R inside RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-studio"><i class="fa fa-check"></i>R studio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#some-general-knowledge"><i class="fa fa-check"></i>Some general knowledge</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#install-packages-1"><i class="fa fa-check"></i>Install packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a4.-cloud-computing"><i class="fa fa-check"></i>A4. Cloud computing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Data Science - introduction with case studies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introR" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction to R programming<a href="introR.html#introR" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This notebook collects the scripts used for teaching in BIOF1001 for
Introduction to R (1 hour teaching). You can get this Rmd file on Moodle or
<a href="https://github.com/StatBiomed/BMDatSci-book/blob/main/notebooks/chapter1-R/01-introR.Rmd">here</a>
(right-click and “save link as” to download).</p>
<p>R is a programming language, particularly popular for its power in statistical
computing, elegant graphics, and also genomic data analysis. It is a free and
open-source software, with active support and development from the community.
Additionally, R is relatively easy to get started for scientific computing.</p>
<blockquote>
<p><strong>Note</strong> To learn and practice R programming, you need to install <code>R</code> and <code>RStudio</code> on
your computer. You can follow the instructions for installation in the
<a href="https://statbiomed.github.io/BMDS-book/install.html">Appendix A</a> chapter.</p>
</blockquote>
<div id="data-types" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Data types<a href="introR.html#data-types" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In R language setting (similar to some other programming languages), there are
a few commonly used data types that are predefined in the built-in environment.
You can use the <code>class()</code> and <code>typeof()</code> functions to check the class and
data type of any variable. In total, R has five data types:</p>
<ol style="list-style-type: decimal">
<li>Numeric</li>
<li>Integers</li>
<li>Complex</li>
<li>Logical</li>
<li>Characters</li>
</ol>
<div id="nemeric-or-double" class="section level3 hasAnchor" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> nemeric (or double)<a href="introR.html#nemeric-or-double" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>numeric</code> is for numeric values, as the most common and the <strong>default</strong>
data type. The <code>numeric</code> datatype saves values in double precision (double
number of bytes in memory), so the type is also <code>double</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="introR.html#cb1-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.0</span>, <span class="fl">2.0</span>, <span class="fl">5.0</span>, <span class="fl">7.0</span>)</span>
<span id="cb1-2"><a href="introR.html#cb1-2" tabindex="-1"></a>x</span>
<span id="cb1-3"><a href="introR.html#cb1-3" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 5 7</span></span>
<span id="cb1-4"><a href="introR.html#cb1-4" tabindex="-1"></a><span class="fu">class</span>(x)</span>
<span id="cb1-5"><a href="introR.html#cb1-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;numeric&quot;</span></span>
<span id="cb1-6"><a href="introR.html#cb1-6" tabindex="-1"></a><span class="fu">typeof</span>(x)</span>
<span id="cb1-7"><a href="introR.html#cb1-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;double&quot;</span></span></code></pre></div>
</div>
<div id="integer" class="section level3 hasAnchor" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> integer<a href="introR.html#integer" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The integer is another data type used for the set of all integers. You can use
the capital ‘L’ notation as a suffix to specify a particular value as the
integer data type. Also, you can convert a value into an integer type using the <code>as.integer()</code> function.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="introR.html#cb2-1" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="dt">L</span>, <span class="dv">2</span><span class="dt">L</span>, <span class="dv">5</span><span class="dt">L</span>, <span class="dv">7</span><span class="dt">L</span>)</span>
<span id="cb2-2"><a href="introR.html#cb2-2" tabindex="-1"></a>y</span>
<span id="cb2-3"><a href="introR.html#cb2-3" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 5 7</span></span>
<span id="cb2-4"><a href="introR.html#cb2-4" tabindex="-1"></a><span class="fu">class</span>(y)</span>
<span id="cb2-5"><a href="introR.html#cb2-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;integer&quot;</span></span>
<span id="cb2-6"><a href="introR.html#cb2-6" tabindex="-1"></a><span class="fu">typeof</span>(y)</span>
<span id="cb2-7"><a href="introR.html#cb2-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;integer&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="introR.html#cb3-1" tabindex="-1"></a><span class="co"># Assign a integer value to y</span></span>
<span id="cb3-2"><a href="introR.html#cb3-2" tabindex="-1"></a>y <span class="ot">=</span> <span class="dv">5</span></span>
<span id="cb3-3"><a href="introR.html#cb3-3" tabindex="-1"></a> </span>
<span id="cb3-4"><a href="introR.html#cb3-4" tabindex="-1"></a><span class="co"># is y an integer?</span></span>
<span id="cb3-5"><a href="introR.html#cb3-5" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">is.integer</span>(y))</span>
<span id="cb3-6"><a href="introR.html#cb3-6" tabindex="-1"></a><span class="co">#&gt; [1] FALSE</span></span></code></pre></div>
</div>
<div id="logical" class="section level3 hasAnchor" number="1.1.3">
<h3><span class="header-section-number">1.1.3</span> logical<a href="introR.html#logical" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In R, the <code>logical</code> data type takes either a value of true or false. A <code>logical</code>
value is often generated when comparing variables.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="introR.html#cb4-1" tabindex="-1"></a>z <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>)</span>
<span id="cb4-2"><a href="introR.html#cb4-2" tabindex="-1"></a>z</span>
<span id="cb4-3"><a href="introR.html#cb4-3" tabindex="-1"></a><span class="co">#&gt; [1]  TRUE  TRUE  TRUE FALSE</span></span>
<span id="cb4-4"><a href="introR.html#cb4-4" tabindex="-1"></a><span class="fu">typeof</span>(z)</span>
<span id="cb4-5"><a href="introR.html#cb4-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;logical&quot;</span></span></code></pre></div>
</div>
<div id="character" class="section level3 hasAnchor" number="1.1.4">
<h3><span class="header-section-number">1.1.4</span> character<a href="introR.html#character" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In R, the <code>character</code> is a data type where you have all the alphabets and
special characters. It stores <code>character</code> values or <code>strings</code>. Strings in R can
contain alphabets, numbers, and symbols. The character type is usually denoted
by wrapping the value inside single or double inverted commas.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="introR.html#cb5-1" tabindex="-1"></a>w <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;aa&quot;</span>, <span class="st">&quot;bb&quot;</span>, <span class="st">&quot;5&quot;</span>, <span class="st">&quot;7&quot;</span>)</span>
<span id="cb5-2"><a href="introR.html#cb5-2" tabindex="-1"></a>w</span>
<span id="cb5-3"><a href="introR.html#cb5-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;aa&quot; &quot;bb&quot; &quot;5&quot;  &quot;7&quot;</span></span>
<span id="cb5-4"><a href="introR.html#cb5-4" tabindex="-1"></a><span class="fu">typeof</span>(w)</span>
<span id="cb5-5"><a href="introR.html#cb5-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;character&quot;</span></span></code></pre></div>
</div>
<div id="memeory-usage" class="section level3 hasAnchor" number="1.1.5">
<h3><span class="header-section-number">1.1.5</span> Memeory usage<a href="introR.html#memeory-usage" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Each data type is explicitly defined, especially the size of the memory.</p>
<p>When initializing a certain data type, there are a few small bytes used to store
basic information. Let’s look at the empty values.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="introR.html#cb6-1" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">numeric</span>())</span>
<span id="cb6-2"><a href="introR.html#cb6-2" tabindex="-1"></a><span class="co">#&gt; 48 bytes</span></span>
<span id="cb6-3"><a href="introR.html#cb6-3" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">integer</span>())</span>
<span id="cb6-4"><a href="introR.html#cb6-4" tabindex="-1"></a><span class="co">#&gt; 48 bytes</span></span>
<span id="cb6-5"><a href="introR.html#cb6-5" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">logical</span>())</span>
<span id="cb6-6"><a href="introR.html#cb6-6" tabindex="-1"></a><span class="co">#&gt; 48 bytes</span></span>
<span id="cb6-7"><a href="introR.html#cb6-7" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">character</span>())</span>
<span id="cb6-8"><a href="introR.html#cb6-8" tabindex="-1"></a><span class="co">#&gt; 48 bytes</span></span></code></pre></div>
<p>To illustrate, let’s use 1000 elements below to show the memory size usage
for each data type.
As we will see, integer and logical use 4 bytes per element, which is only half
of the memory usage by double (numeric) and character of 8 bytes.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="introR.html#cb7-1" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">rep</span>(<span class="dv">1</span>, <span class="dv">1000</span>))</span>
<span id="cb7-2"><a href="introR.html#cb7-2" tabindex="-1"></a><span class="co">#&gt; 8048 bytes</span></span>
<span id="cb7-3"><a href="introR.html#cb7-3" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="dt">L</span>, <span class="dv">1000</span>))</span>
<span id="cb7-4"><a href="introR.html#cb7-4" tabindex="-1"></a><span class="co">#&gt; 4048 bytes</span></span>
<span id="cb7-5"><a href="introR.html#cb7-5" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="dv">1000</span>))</span>
<span id="cb7-6"><a href="introR.html#cb7-6" tabindex="-1"></a><span class="co">#&gt; 4048 bytes</span></span>
<span id="cb7-7"><a href="introR.html#cb7-7" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">rep</span>(<span class="st">&quot;aa&quot;</span>, <span class="dv">1000</span>))</span>
<span id="cb7-8"><a href="introR.html#cb7-8" tabindex="-1"></a><span class="co">#&gt; 8104 bytes</span></span></code></pre></div>
</div>
</div>
<div id="data-structures" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Data structures<a href="introR.html#data-structures" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Data structure is one of the most important features in programming. It involves
how the data is organised and can be accessed and modified. Using an appropriate
data structure may largely improve computing efficiency.</p>
<div id="vector" class="section level3 hasAnchor" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Vector<a href="introR.html#vector" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Vector is a basic data structure in R. It contains elements in the <strong>same</strong> data
type (no matter double, integer, character or others).
You can check the data type by using <code>typeof()</code> function and the length of the
vector by <code>length()</code> function.</p>
<p>Since a vector has elements of the same type, this function will try and coerce elements to the same type, if they are different. Coercion is from lower
to higher types, i.e., from logical to integer to double to a character.</p>
<p>See more introduction <a href="https://www.datamentor.io/r-programming/vector/">here</a>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="introR.html#cb8-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">7</span>)</span>
<span id="cb8-2"><a href="introR.html#cb8-2" tabindex="-1"></a>x</span>
<span id="cb8-3"><a href="introR.html#cb8-3" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 5 7</span></span>
<span id="cb8-4"><a href="introR.html#cb8-4" tabindex="-1"></a><span class="fu">typeof</span>(x)</span>
<span id="cb8-5"><a href="introR.html#cb8-5" tabindex="-1"></a><span class="co">#&gt; [1] &quot;double&quot;</span></span>
<span id="cb8-6"><a href="introR.html#cb8-6" tabindex="-1"></a></span>
<span id="cb8-7"><a href="introR.html#cb8-7" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">3</span>, <span class="dv">5</span>)</span>
<span id="cb8-8"><a href="introR.html#cb8-8" tabindex="-1"></a>x</span>
<span id="cb8-9"><a href="introR.html#cb8-9" tabindex="-1"></a><span class="co">#&gt; [1] 3 3 3 3 3</span></span>
<span id="cb8-10"><a href="introR.html#cb8-10" tabindex="-1"></a><span class="fu">typeof</span>(x)</span>
<span id="cb8-11"><a href="introR.html#cb8-11" tabindex="-1"></a><span class="co">#&gt; [1] &quot;double&quot;</span></span>
<span id="cb8-12"><a href="introR.html#cb8-12" tabindex="-1"></a></span>
<span id="cb8-13"><a href="introR.html#cb8-13" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">12</span> <span class="co"># integer</span></span>
<span id="cb8-14"><a href="introR.html#cb8-14" tabindex="-1"></a>x</span>
<span id="cb8-15"><a href="introR.html#cb8-15" tabindex="-1"></a><span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10 11 12</span></span>
<span id="cb8-16"><a href="introR.html#cb8-16" tabindex="-1"></a><span class="fu">typeof</span>(x)</span>
<span id="cb8-17"><a href="introR.html#cb8-17" tabindex="-1"></a><span class="co">#&gt; [1] &quot;integer&quot;</span></span></code></pre></div>
</div>
<div id="matrix" class="section level3 hasAnchor" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Matrix<a href="introR.html#matrix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Matrix is a two-dimensional data structure. It is in principle built based on
vector but has more convenient built-in functions for computation.
It has rows and columns, both of which can also have names. To check the dimensions,
you can use the <code>dim()</code> function.</p>
<p>See more introduction <a href="https://www.datamentor.io/r-programming/matrix/">here</a>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="introR.html#cb9-1" tabindex="-1"></a>A <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>, <span class="at">nrow=</span><span class="dv">3</span>)</span>
<span id="cb9-2"><a href="introR.html#cb9-2" tabindex="-1"></a>A</span>
<span id="cb9-3"><a href="introR.html#cb9-3" tabindex="-1"></a><span class="co">#&gt;      [,1] [,2] [,3] [,4]</span></span>
<span id="cb9-4"><a href="introR.html#cb9-4" tabindex="-1"></a><span class="co">#&gt; [1,]    1    4    7   10</span></span>
<span id="cb9-5"><a href="introR.html#cb9-5" tabindex="-1"></a><span class="co">#&gt; [2,]    2    5    8   11</span></span>
<span id="cb9-6"><a href="introR.html#cb9-6" tabindex="-1"></a><span class="co">#&gt; [3,]    3    6    9   12</span></span>
<span id="cb9-7"><a href="introR.html#cb9-7" tabindex="-1"></a></span>
<span id="cb9-8"><a href="introR.html#cb9-8" tabindex="-1"></a>B <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>, <span class="at">nrow=</span><span class="dv">3</span>, <span class="at">byrow=</span><span class="cn">TRUE</span>)</span>
<span id="cb9-9"><a href="introR.html#cb9-9" tabindex="-1"></a>B</span>
<span id="cb9-10"><a href="introR.html#cb9-10" tabindex="-1"></a><span class="co">#&gt;      [,1] [,2] [,3] [,4]</span></span>
<span id="cb9-11"><a href="introR.html#cb9-11" tabindex="-1"></a><span class="co">#&gt; [1,]    1    2    3    4</span></span>
<span id="cb9-12"><a href="introR.html#cb9-12" tabindex="-1"></a><span class="co">#&gt; [2,]    5    6    7    8</span></span>
<span id="cb9-13"><a href="introR.html#cb9-13" tabindex="-1"></a><span class="co">#&gt; [3,]    9   10   11   12</span></span>
<span id="cb9-14"><a href="introR.html#cb9-14" tabindex="-1"></a></span>
<span id="cb9-15"><a href="introR.html#cb9-15" tabindex="-1"></a><span class="fu">colnames</span>(A) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;C1&quot;</span>,<span class="st">&quot;C2&quot;</span>,<span class="st">&quot;C3&quot;</span>,<span class="st">&quot;C4&quot;</span>)</span>
<span id="cb9-16"><a href="introR.html#cb9-16" tabindex="-1"></a><span class="fu">rownames</span>(A) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;R1&quot;</span>,<span class="st">&quot;R2&quot;</span>,<span class="st">&quot;R3&quot;</span>)</span>
<span id="cb9-17"><a href="introR.html#cb9-17" tabindex="-1"></a>A</span>
<span id="cb9-18"><a href="introR.html#cb9-18" tabindex="-1"></a><span class="co">#&gt;    C1 C2 C3 C4</span></span>
<span id="cb9-19"><a href="introR.html#cb9-19" tabindex="-1"></a><span class="co">#&gt; R1  1  4  7 10</span></span>
<span id="cb9-20"><a href="introR.html#cb9-20" tabindex="-1"></a><span class="co">#&gt; R2  2  5  8 11</span></span>
<span id="cb9-21"><a href="introR.html#cb9-21" tabindex="-1"></a><span class="co">#&gt; R3  3  6  9 12</span></span></code></pre></div>
<div id="index-vector-and-matrix" class="section level4 hasAnchor" number="1.2.2.1">
<h4><span class="header-section-number">1.2.2.1</span> Index Vector and Matrix<a href="introR.html#index-vector-and-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To index vector, you can use <code>logical</code> or <code>integer</code>, or the <code>element name</code> if it
has.</p>
<p>Note, when using an <code>integer</code> for indexing, the index <em>starts from 1</em> in R, unlike
most programming languages where the index starts from 0.</p>
<p>We can also use negative integers to return all elements except those
specified. But we cannot mix positive and negative integers while indexing and
real numbers if used are truncated to integers.</p>
<p><strong>N.B.</strong>, when using logical as the index, be careful if the index length is
different from the vector.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="introR.html#cb10-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">12</span></span>
<span id="cb10-2"><a href="introR.html#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="introR.html#cb10-3" tabindex="-1"></a>x[<span class="dv">3</span>]</span>
<span id="cb10-4"><a href="introR.html#cb10-4" tabindex="-1"></a><span class="co">#&gt; [1] 3</span></span>
<span id="cb10-5"><a href="introR.html#cb10-5" tabindex="-1"></a></span>
<span id="cb10-6"><a href="introR.html#cb10-6" tabindex="-1"></a>x[<span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb10-7"><a href="introR.html#cb10-7" tabindex="-1"></a><span class="co">#&gt; [1] 2 3 4 5</span></span>
<span id="cb10-8"><a href="introR.html#cb10-8" tabindex="-1"></a></span>
<span id="cb10-9"><a href="introR.html#cb10-9" tabindex="-1"></a>x[<span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">6</span>)]                   <span class="co"># index with integer</span></span>
<span id="cb10-10"><a href="introR.html#cb10-10" tabindex="-1"></a><span class="co">#&gt; [1] 2 5 6</span></span>
<span id="cb10-11"><a href="introR.html#cb10-11" tabindex="-1"></a></span>
<span id="cb10-12"><a href="introR.html#cb10-12" tabindex="-1"></a>x[<span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>)]  <span class="co"># index with logical value</span></span>
<span id="cb10-13"><a href="introR.html#cb10-13" tabindex="-1"></a><span class="co">#&gt; [1]  1  4  5  8  9 12</span></span></code></pre></div>
<p>Now, let’s index a matrix. It is very similar to vector, but it has both rows and columns.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="introR.html#cb11-1" tabindex="-1"></a>A <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>, <span class="at">nrow=</span><span class="dv">3</span>)</span>
<span id="cb11-2"><a href="introR.html#cb11-2" tabindex="-1"></a><span class="fu">colnames</span>(A) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;C1&quot;</span>,<span class="st">&quot;C2&quot;</span>,<span class="st">&quot;C3&quot;</span>,<span class="st">&quot;C4&quot;</span>)</span>
<span id="cb11-3"><a href="introR.html#cb11-3" tabindex="-1"></a><span class="fu">rownames</span>(A) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;R1&quot;</span>,<span class="st">&quot;R2&quot;</span>,<span class="st">&quot;R3&quot;</span>)</span>
<span id="cb11-4"><a href="introR.html#cb11-4" tabindex="-1"></a>A</span>
<span id="cb11-5"><a href="introR.html#cb11-5" tabindex="-1"></a><span class="co">#&gt;    C1 C2 C3 C4</span></span>
<span id="cb11-6"><a href="introR.html#cb11-6" tabindex="-1"></a><span class="co">#&gt; R1  1  4  7 10</span></span>
<span id="cb11-7"><a href="introR.html#cb11-7" tabindex="-1"></a><span class="co">#&gt; R2  2  5  8 11</span></span>
<span id="cb11-8"><a href="introR.html#cb11-8" tabindex="-1"></a><span class="co">#&gt; R3  3  6  9 12</span></span>
<span id="cb11-9"><a href="introR.html#cb11-9" tabindex="-1"></a></span>
<span id="cb11-10"><a href="introR.html#cb11-10" tabindex="-1"></a>A[<span class="dv">1</span>, <span class="dv">2</span>]</span>
<span id="cb11-11"><a href="introR.html#cb11-11" tabindex="-1"></a><span class="co">#&gt; [1] 4</span></span>
<span id="cb11-12"><a href="introR.html#cb11-12" tabindex="-1"></a></span>
<span id="cb11-13"><a href="introR.html#cb11-13" tabindex="-1"></a>A[<span class="dv">1</span>, <span class="st">&quot;C2&quot;</span>]</span>
<span id="cb11-14"><a href="introR.html#cb11-14" tabindex="-1"></a><span class="co">#&gt; [1] 4</span></span>
<span id="cb11-15"><a href="introR.html#cb11-15" tabindex="-1"></a></span>
<span id="cb11-16"><a href="introR.html#cb11-16" tabindex="-1"></a>A[<span class="dv">1</span>, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>)]</span>
<span id="cb11-17"><a href="introR.html#cb11-17" tabindex="-1"></a><span class="co">#&gt; C2 C3 </span></span>
<span id="cb11-18"><a href="introR.html#cb11-18" tabindex="-1"></a><span class="co">#&gt;  4  7</span></span>
<span id="cb11-19"><a href="introR.html#cb11-19" tabindex="-1"></a></span>
<span id="cb11-20"><a href="introR.html#cb11-20" tabindex="-1"></a>A[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>)]</span>
<span id="cb11-21"><a href="introR.html#cb11-21" tabindex="-1"></a><span class="co">#&gt;    C2 C3</span></span>
<span id="cb11-22"><a href="introR.html#cb11-22" tabindex="-1"></a><span class="co">#&gt; R1  4  7</span></span>
<span id="cb11-23"><a href="introR.html#cb11-23" tabindex="-1"></a><span class="co">#&gt; R2  5  8</span></span></code></pre></div>
<p>Single row or column matrix will become a vector, unless <strong>using <code>drop=FALSE</code></strong></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="introR.html#cb12-1" tabindex="-1"></a>A[<span class="dv">1</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb12-2"><a href="introR.html#cb12-2" tabindex="-1"></a><span class="co">#&gt; C2 C3 C4 </span></span>
<span id="cb12-3"><a href="introR.html#cb12-3" tabindex="-1"></a><span class="co">#&gt;  4  7 10</span></span>
<span id="cb12-4"><a href="introR.html#cb12-4" tabindex="-1"></a><span class="fu">dim</span>(A[<span class="dv">1</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>])</span>
<span id="cb12-5"><a href="introR.html#cb12-5" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb12-6"><a href="introR.html#cb12-6" tabindex="-1"></a></span>
<span id="cb12-7"><a href="introR.html#cb12-7" tabindex="-1"></a>A[<span class="dv">1</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb12-8"><a href="introR.html#cb12-8" tabindex="-1"></a><span class="co">#&gt;    C2 C3 C4</span></span>
<span id="cb12-9"><a href="introR.html#cb12-9" tabindex="-1"></a><span class="co">#&gt; R1  4  7 10</span></span>
<span id="cb12-10"><a href="introR.html#cb12-10" tabindex="-1"></a><span class="fu">dim</span>(A[<span class="dv">1</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb12-11"><a href="introR.html#cb12-11" tabindex="-1"></a><span class="co">#&gt; [1] 1 3</span></span></code></pre></div>
</div>
<div id="modify-values" class="section level4 hasAnchor" number="1.2.2.2">
<h4><span class="header-section-number">1.2.2.2</span> Modify values<a href="introR.html#modify-values" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="introR.html#cb13-1" tabindex="-1"></a>A[<span class="dv">1</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="sc">-</span><span class="dv">5</span>, <span class="dv">20</span>)</span>
<span id="cb13-2"><a href="introR.html#cb13-2" tabindex="-1"></a>A</span>
<span id="cb13-3"><a href="introR.html#cb13-3" tabindex="-1"></a><span class="co">#&gt;    C1 C2 C3 C4</span></span>
<span id="cb13-4"><a href="introR.html#cb13-4" tabindex="-1"></a><span class="co">#&gt; R1  1 -3 -5 20</span></span>
<span id="cb13-5"><a href="introR.html#cb13-5" tabindex="-1"></a><span class="co">#&gt; R2  2  5  8 11</span></span>
<span id="cb13-6"><a href="introR.html#cb13-6" tabindex="-1"></a><span class="co">#&gt; R3  3  6  9 12</span></span></code></pre></div>
</div>
</div>
<div id="list" class="section level3 hasAnchor" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> List<a href="introR.html#list" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Different from <code>vector</code> that has all elements in the same data type, the <code>list</code>
data structure can have components of mixed data types.
More broadly, a <code>list</code> can contain a list of any data structure: value, vector,
matrix, etc.</p>
<p>We can use <code>str()</code> function to view the structure of a list (or any object).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="introR.html#cb14-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fl">2.5</span>, <span class="cn">TRUE</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb14-2"><a href="introR.html#cb14-2" tabindex="-1"></a>x</span>
<span id="cb14-3"><a href="introR.html#cb14-3" tabindex="-1"></a><span class="co">#&gt; [[1]]</span></span>
<span id="cb14-4"><a href="introR.html#cb14-4" tabindex="-1"></a><span class="co">#&gt; [1] 2.5</span></span>
<span id="cb14-5"><a href="introR.html#cb14-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-6"><a href="introR.html#cb14-6" tabindex="-1"></a><span class="co">#&gt; [[2]]</span></span>
<span id="cb14-7"><a href="introR.html#cb14-7" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb14-8"><a href="introR.html#cb14-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-9"><a href="introR.html#cb14-9" tabindex="-1"></a><span class="co">#&gt; [[3]]</span></span>
<span id="cb14-10"><a href="introR.html#cb14-10" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 3</span></span>
<span id="cb14-11"><a href="introR.html#cb14-11" tabindex="-1"></a></span>
<span id="cb14-12"><a href="introR.html#cb14-12" tabindex="-1"></a><span class="fu">str</span>(x)</span>
<span id="cb14-13"><a href="introR.html#cb14-13" tabindex="-1"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb14-14"><a href="introR.html#cb14-14" tabindex="-1"></a><span class="co">#&gt;  $ : num 2.5</span></span>
<span id="cb14-15"><a href="introR.html#cb14-15" tabindex="-1"></a><span class="co">#&gt;  $ : logi TRUE</span></span>
<span id="cb14-16"><a href="introR.html#cb14-16" tabindex="-1"></a><span class="co">#&gt;  $ : int [1:3] 1 2 3</span></span></code></pre></div>
<p>We can also have a name for each element:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="introR.html#cb15-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="st">&quot;a&quot;</span> <span class="ot">=</span> <span class="fl">2.5</span>, <span class="st">&quot;b&quot;</span> <span class="ot">=</span> <span class="cn">TRUE</span>, <span class="st">&quot;c&quot;</span> <span class="ot">=</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb15-2"><a href="introR.html#cb15-2" tabindex="-1"></a>x</span>
<span id="cb15-3"><a href="introR.html#cb15-3" tabindex="-1"></a><span class="co">#&gt; $a</span></span>
<span id="cb15-4"><a href="introR.html#cb15-4" tabindex="-1"></a><span class="co">#&gt; [1] 2.5</span></span>
<span id="cb15-5"><a href="introR.html#cb15-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb15-6"><a href="introR.html#cb15-6" tabindex="-1"></a><span class="co">#&gt; $b</span></span>
<span id="cb15-7"><a href="introR.html#cb15-7" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb15-8"><a href="introR.html#cb15-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb15-9"><a href="introR.html#cb15-9" tabindex="-1"></a><span class="co">#&gt; $c</span></span>
<span id="cb15-10"><a href="introR.html#cb15-10" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 3</span></span>
<span id="cb15-11"><a href="introR.html#cb15-11" tabindex="-1"></a><span class="fu">str</span>(x)</span>
<span id="cb15-12"><a href="introR.html#cb15-12" tabindex="-1"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb15-13"><a href="introR.html#cb15-13" tabindex="-1"></a><span class="co">#&gt;  $ a: num 2.5</span></span>
<span id="cb15-14"><a href="introR.html#cb15-14" tabindex="-1"></a><span class="co">#&gt;  $ b: logi TRUE</span></span>
<span id="cb15-15"><a href="introR.html#cb15-15" tabindex="-1"></a><span class="co">#&gt;  $ c: int [1:3] 1 2 3</span></span></code></pre></div>
<div id="indexing-list" class="section level4 hasAnchor" number="1.2.3.1">
<h4><span class="header-section-number">1.2.3.1</span> Indexing list<a href="introR.html#indexing-list" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Different from vector and matrix, for a <code>list</code>, you need to use <strong>double-layer square brackets</strong>,
either by numeric index or name. Alternatively, you can also use <code>$</code> symbol with the name.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="introR.html#cb16-1" tabindex="-1"></a>x[[<span class="dv">3</span>]]</span>
<span id="cb16-2"><a href="introR.html#cb16-2" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 3</span></span>
<span id="cb16-3"><a href="introR.html#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="introR.html#cb16-4" tabindex="-1"></a>x[[<span class="st">&quot;c&quot;</span>]]</span>
<span id="cb16-5"><a href="introR.html#cb16-5" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 3</span></span>
<span id="cb16-6"><a href="introR.html#cb16-6" tabindex="-1"></a></span>
<span id="cb16-7"><a href="introR.html#cb16-7" tabindex="-1"></a>x<span class="sc">$</span>c</span>
<span id="cb16-8"><a href="introR.html#cb16-8" tabindex="-1"></a><span class="co">#&gt; [1] 1 2 3</span></span></code></pre></div>
</div>
</div>
<div id="data-frame" class="section level3 hasAnchor" number="1.2.4">
<h3><span class="header-section-number">1.2.4</span> Data Frame<a href="introR.html#data-frame" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Data frame is widely used for rectangular data, where each column has the same
data type (vector) but different columns can have different data types (like Excel)</p>
<p>As you guess, the data frame is a special type of <code>list</code>: A list of vectors with the same length.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="introR.html#cb17-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&quot;SN&quot;</span> <span class="ot">=</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="st">&quot;Age&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">21</span>,<span class="dv">15</span>), </span>
<span id="cb17-2"><a href="introR.html#cb17-2" tabindex="-1"></a>                 <span class="st">&quot;Name&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;John&quot;</span>,<span class="st">&quot;Dora&quot;</span>))</span>
<span id="cb17-3"><a href="introR.html#cb17-3" tabindex="-1"></a>df</span>
<span id="cb17-4"><a href="introR.html#cb17-4" tabindex="-1"></a><span class="co">#&gt;   SN Age Name</span></span>
<span id="cb17-5"><a href="introR.html#cb17-5" tabindex="-1"></a><span class="co">#&gt; 1  1  21 John</span></span>
<span id="cb17-6"><a href="introR.html#cb17-6" tabindex="-1"></a><span class="co">#&gt; 2  2  15 Dora</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="introR.html#cb18-1" tabindex="-1"></a>df<span class="sc">$</span>Age[<span class="dv">2</span>]</span>
<span id="cb18-2"><a href="introR.html#cb18-2" tabindex="-1"></a><span class="co">#&gt; [1] 15</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="introR.html#cb19-1" tabindex="-1"></a><span class="co"># View(df)</span></span></code></pre></div>
</div>
<div id="factor-vs-vector" class="section level3 hasAnchor" number="1.2.5">
<h3><span class="header-section-number">1.2.5</span> Factor vs vector<a href="introR.html#factor-vs-vector" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For vector, if it only contains pre-defined values (which can have specified
orders), you may consider using factor.</p>
<ul>
<li>Factor is a data structure used for fields that takes only predefined, finite
number of values (categorical data)</li>
<li>The order of predefined values can be specified, instead of alphabetic by default</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="introR.html#cb20-1" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;single&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;single&quot;</span>) <span class="co"># vector</span></span>
<span id="cb20-2"><a href="introR.html#cb20-2" tabindex="-1"></a>x</span>
<span id="cb20-3"><a href="introR.html#cb20-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;single&quot;  &quot;married&quot; &quot;married&quot; &quot;single&quot;</span></span>
<span id="cb20-4"><a href="introR.html#cb20-4" tabindex="-1"></a></span>
<span id="cb20-5"><a href="introR.html#cb20-5" tabindex="-1"></a><span class="fu">class</span>(x)</span>
<span id="cb20-6"><a href="introR.html#cb20-6" tabindex="-1"></a><span class="co">#&gt; [1] &quot;character&quot;</span></span>
<span id="cb20-7"><a href="introR.html#cb20-7" tabindex="-1"></a><span class="fu">typeof</span>(x)</span>
<span id="cb20-8"><a href="introR.html#cb20-8" tabindex="-1"></a><span class="co">#&gt; [1] &quot;character&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="introR.html#cb21-1" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&quot;single&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;single&quot;</span>))</span>
<span id="cb21-2"><a href="introR.html#cb21-2" tabindex="-1"></a>y</span>
<span id="cb21-3"><a href="introR.html#cb21-3" tabindex="-1"></a><span class="co">#&gt; [1] single  married married single </span></span>
<span id="cb21-4"><a href="introR.html#cb21-4" tabindex="-1"></a><span class="co">#&gt; Levels: married single</span></span>
<span id="cb21-5"><a href="introR.html#cb21-5" tabindex="-1"></a></span>
<span id="cb21-6"><a href="introR.html#cb21-6" tabindex="-1"></a><span class="fu">class</span>(y)</span>
<span id="cb21-7"><a href="introR.html#cb21-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;factor&quot;</span></span>
<span id="cb21-8"><a href="introR.html#cb21-8" tabindex="-1"></a><span class="fu">typeof</span>(y)</span>
<span id="cb21-9"><a href="introR.html#cb21-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;integer&quot;</span></span></code></pre></div>
<div id="change-the-order-of-factor-levels" class="section level4 hasAnchor" number="1.2.5.1">
<h4><span class="header-section-number">1.2.5.1</span> Change the order of factor levels<a href="introR.html#change-the-order-of-factor-levels" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="introR.html#cb22-1" tabindex="-1"></a>z <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&quot;single&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;single&quot;</span>) , <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;single&quot;</span>, <span class="st">&quot;married&quot;</span>))</span>
<span id="cb22-2"><a href="introR.html#cb22-2" tabindex="-1"></a>z</span>
<span id="cb22-3"><a href="introR.html#cb22-3" tabindex="-1"></a><span class="co">#&gt; [1] single  married married single </span></span>
<span id="cb22-4"><a href="introR.html#cb22-4" tabindex="-1"></a><span class="co">#&gt; Levels: single married</span></span></code></pre></div>
</div>
<div id="smaller-memory-as-categorical-data-type" class="section level4 hasAnchor" number="1.2.5.2">
<h4><span class="header-section-number">1.2.5.2</span> Smaller memory as categorical data type<a href="introR.html#smaller-memory-as-categorical-data-type" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="introR.html#cb23-1" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;single&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;single&quot;</span>) <span class="co"># vector</span></span>
<span id="cb23-2"><a href="introR.html#cb23-2" tabindex="-1"></a>y <span class="ot">=</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&quot;single&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;married&quot;</span>, <span class="st">&quot;single&quot;</span>))</span>
<span id="cb23-3"><a href="introR.html#cb23-3" tabindex="-1"></a></span>
<span id="cb23-4"><a href="introR.html#cb23-4" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">rep</span>(x, <span class="dv">1000</span>))</span>
<span id="cb23-5"><a href="introR.html#cb23-5" tabindex="-1"></a><span class="co">#&gt; 32160 bytes</span></span>
<span id="cb23-6"><a href="introR.html#cb23-6" tabindex="-1"></a><span class="fu">object.size</span>(<span class="fu">rep</span>(y, <span class="dv">1000</span>))</span>
<span id="cb23-7"><a href="introR.html#cb23-7" tabindex="-1"></a><span class="co">#&gt; 16560 bytes</span></span></code></pre></div>
</div>
</div>
</div>
<div id="read-and-write-files-tables" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> Read and write files (tables)<a href="introR.html#read-and-write-files-tables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Besides operating in the R environment, we also want to read data from files or
write results into files.</p>
<p>For tables, R has convenient built-in function <code>read.table()</code> and <code>write.table()</code>.</p>
<div id="read-file" class="section level3 hasAnchor" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Read file<a href="introR.html#read-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can use <code>read.table()</code> function to read tables, e.g., in comma separated
values (csv) or tab separated values (tsv) formats.
See full manuals: <code>help(read.table)</code> or <code>?read.table</code> or its
<a href="https://www.rdocumentation.org/packages/utils/versions/3.6.2/topics/read.table">online manual</a></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="introR.html#cb24-1" tabindex="-1"></a><span class="fu">help</span>(<span class="st">&quot;read.table&quot;</span>)</span>
<span id="cb24-2"><a href="introR.html#cb24-2" tabindex="-1"></a></span>
<span id="cb24-3"><a href="introR.html#cb24-3" tabindex="-1"></a>?read.table</span></code></pre></div>
<p>Here, let’s read an example file. Data is available on Moodle and on the
<a href="https://github.com/StatBiomed/BMDatSci-book/blob/main/notebooks/chapter1-R">github repository</a></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="introR.html#cb25-1" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;./SRP029880.colData.tsv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb25-2"><a href="introR.html#cb25-2" tabindex="-1"></a>df</span>
<span id="cb25-3"><a href="introR.html#cb25-3" tabindex="-1"></a><span class="co">#&gt;                source_name group</span></span>
<span id="cb25-4"><a href="introR.html#cb25-4" tabindex="-1"></a><span class="co">#&gt; CASE_1 metastasized cancer  CASE</span></span>
<span id="cb25-5"><a href="introR.html#cb25-5" tabindex="-1"></a><span class="co">#&gt; CASE_2 metastasized cancer  CASE</span></span>
<span id="cb25-6"><a href="introR.html#cb25-6" tabindex="-1"></a><span class="co">#&gt; CASE_3 metastasized cancer  CASE</span></span>
<span id="cb25-7"><a href="introR.html#cb25-7" tabindex="-1"></a><span class="co">#&gt; CASE_4 metastasized cancer  CASE</span></span>
<span id="cb25-8"><a href="introR.html#cb25-8" tabindex="-1"></a><span class="co">#&gt; CASE_5 metastasized cancer  CASE</span></span>
<span id="cb25-9"><a href="introR.html#cb25-9" tabindex="-1"></a><span class="co">#&gt; CTRL_1        normal colon  CTRL</span></span>
<span id="cb25-10"><a href="introR.html#cb25-10" tabindex="-1"></a><span class="co">#&gt; CTRL_2        normal colon  CTRL</span></span>
<span id="cb25-11"><a href="introR.html#cb25-11" tabindex="-1"></a><span class="co">#&gt; CTRL_3        normal colon  CTRL</span></span>
<span id="cb25-12"><a href="introR.html#cb25-12" tabindex="-1"></a><span class="co">#&gt; CTRL_4        normal colon  CTRL</span></span>
<span id="cb25-13"><a href="introR.html#cb25-13" tabindex="-1"></a><span class="co">#&gt; CTRL_5        normal colon  CTRL</span></span></code></pre></div>
</div>
<div id="write-file" class="section level3 hasAnchor" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> Write file<a href="introR.html#write-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The above file is loaded as a data frame. Here, let’s add an extra column to
indicate if the sample is frozen or not, then save it into a file.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="introR.html#cb26-1" tabindex="-1"></a>df<span class="sc">$</span>frozen <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>)</span>
<span id="cb26-2"><a href="introR.html#cb26-2" tabindex="-1"></a><span class="fu">write.table</span>(df, <span class="st">&quot;./SRP029880.colData.add_frozen.tsv&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="functions-and-packages" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Functions and Packages<a href="introR.html#functions-and-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As experienced above, we have used function multiple times, e.g.,
<code>read.table</code> and <code>typeof</code>.</p>
<p>As one more example <code>mean()</code> is a function here and it is from the <code>base</code> package</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="introR.html#cb27-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="dv">4</span><span class="sc">:</span><span class="dv">10</span></span>
<span id="cb27-2"><a href="introR.html#cb27-2" tabindex="-1"></a><span class="fu">mean</span>(x)</span>
<span id="cb27-3"><a href="introR.html#cb27-3" tabindex="-1"></a><span class="co">#&gt; [1] 7</span></span>
<span id="cb27-4"><a href="introR.html#cb27-4" tabindex="-1"></a></span>
<span id="cb27-5"><a href="introR.html#cb27-5" tabindex="-1"></a>base<span class="sc">::</span><span class="fu">mean</span>(x)</span>
<span id="cb27-6"><a href="introR.html#cb27-6" tabindex="-1"></a><span class="co">#&gt; [1] 7</span></span></code></pre></div>
<p>Generally speaking, A function is a set of statements organized together to
perform a specific task. Many lines of codes are packed into one function &amp; it’s reusable.</p>
<p>A function can be written in the same R file and loaded, or it can be
distributed as part of a package. For using such functions, we need to install
the corresponding package and load it.</p>
<div id="install-packages" class="section level3 hasAnchor" number="1.4.1">
<h3><span class="header-section-number">1.4.1</span> Install packages<a href="introR.html#install-packages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It depends on where the package is stored. Please refers to the documentation
of the specific package you want to install and use.</p>
<ol style="list-style-type: decimal">
<li>CRAN (the Comprehensive R Archive Network): main platform
<ul>
<li>For example: <code>install.packages("ggplot2")</code></li>
</ul></li>
<li>Bioconductor: primarily for biology related packages
<ul>
<li>For example: <code>BiocManager::install("DESeq2")</code></li>
</ul></li>
</ol>
<p>As an example, we can install the powerful plotting package <code>ggplot2</code> from CRAN.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="introR.html#cb28-1" tabindex="-1"></a><span class="co">#install.packages(&quot;ggplot2&quot;)</span></span></code></pre></div>
</div>
<div id="apply-function-repeatly" class="section level3 hasAnchor" number="1.4.2">
<h3><span class="header-section-number">1.4.2</span> Apply function repeatly<a href="introR.html#apply-function-repeatly" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We may often want to use a certain function for multiple times, e.g., calculate
the sample mean for many genes. There are multiple ways to achieve it, e.g., via a
<code>for loop</code>. Here, we will introduce <code>apply</code> and its variants for this purpose.</p>
<p>See more introductions <a href="https://ademos.people.uic.edu/Chapter4.html">here</a>.</p>
<div id="apply-lapply-sapply-and-vapply" class="section level4 hasAnchor" number="1.4.2.1">
<h4><span class="header-section-number">1.4.2.1</span> apply, lapply, sapply and vapply<a href="introR.html#apply-lapply-sapply-and-vapply" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In short, the <code>apply()</code> function and its variants apply a certain function to
each element of a vector, a matrix, or a list.</p>
<div id="apply-for-matrix" class="section level5 hasAnchor" number="1.4.2.1.1">
<h5><span class="header-section-number">1.4.2.1.1</span> apply for matrix<a href="introR.html#apply-for-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The <code>apply(X, MARGIN, FUN)</code> function works for <code>matrix</code> (or <code>array</code>) for rows or columns. For example, calculating the median of each column:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="introR.html#cb29-1" tabindex="-1"></a>my.matrx <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>, <span class="at">nrow =</span> <span class="dv">5</span>, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb29-2"><a href="introR.html#cb29-2" tabindex="-1"></a>my.matrx</span>
<span id="cb29-3"><a href="introR.html#cb29-3" tabindex="-1"></a><span class="co">#&gt;      [,1] [,2] [,3]</span></span>
<span id="cb29-4"><a href="introR.html#cb29-4" tabindex="-1"></a><span class="co">#&gt; [1,]    1    6   11</span></span>
<span id="cb29-5"><a href="introR.html#cb29-5" tabindex="-1"></a><span class="co">#&gt; [2,]    2    7   12</span></span>
<span id="cb29-6"><a href="introR.html#cb29-6" tabindex="-1"></a><span class="co">#&gt; [3,]    3    8   13</span></span>
<span id="cb29-7"><a href="introR.html#cb29-7" tabindex="-1"></a><span class="co">#&gt; [4,]    4    9   14</span></span>
<span id="cb29-8"><a href="introR.html#cb29-8" tabindex="-1"></a><span class="co">#&gt; [5,]    5   10   15</span></span>
<span id="cb29-9"><a href="introR.html#cb29-9" tabindex="-1"></a></span>
<span id="cb29-10"><a href="introR.html#cb29-10" tabindex="-1"></a><span class="fu">apply</span>(my.matrx, <span class="dv">2</span>, median)</span>
<span id="cb29-11"><a href="introR.html#cb29-11" tabindex="-1"></a><span class="co">#&gt; [1]  3  8 13</span></span></code></pre></div>
</div>
<div id="lapply-sapply-and-vapply-for-list-or-vector" class="section level5 hasAnchor" number="1.4.2.1.2">
<h5><span class="header-section-number">1.4.2.1.2</span> lapply, sapply, and vapply for list or vector<a href="introR.html#lapply-sapply-and-vapply-for-list-or-vector" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The above <code>apply</code> function requires <code>MARGIN</code>, hence won’t work for vector or
list. There are a few variants to support lists or vectors for different purposes.</p>
<p>From the <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/lapply">manual</a>, we can find out the arguments for these three functions:</p>
<ul>
<li><code>lapply(X, FUN, …)</code></li>
<li><code>sapply(X, FUN, …, simplify = TRUE, USE.NAMES = TRUE)</code></li>
<li><code>vapply(X, FUN, FUN.VALUE, …, USE.NAMES = TRUE)</code></li>
</ul>
<p>Let’s look at examples. The <code>lapply</code> works for vector and list and returns a <code>list</code>:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="introR.html#cb30-1" tabindex="-1"></a>A<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>)</span>
<span id="cb30-2"><a href="introR.html#cb30-2" tabindex="-1"></a>B<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>)</span>
<span id="cb30-3"><a href="introR.html#cb30-3" tabindex="-1"></a>C<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span>
<span id="cb30-4"><a href="introR.html#cb30-4" tabindex="-1"></a>my.lst<span class="ot">&lt;-</span><span class="fu">list</span>(A,B,C)</span>
<span id="cb30-5"><a href="introR.html#cb30-5" tabindex="-1"></a></span>
<span id="cb30-6"><a href="introR.html#cb30-6" tabindex="-1"></a><span class="fu">lapply</span>(my.lst, median)</span>
<span id="cb30-7"><a href="introR.html#cb30-7" tabindex="-1"></a><span class="co">#&gt; [[1]]</span></span>
<span id="cb30-8"><a href="introR.html#cb30-8" tabindex="-1"></a><span class="co">#&gt; [1] 5</span></span>
<span id="cb30-9"><a href="introR.html#cb30-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-10"><a href="introR.html#cb30-10" tabindex="-1"></a><span class="co">#&gt; [[2]]</span></span>
<span id="cb30-11"><a href="introR.html#cb30-11" tabindex="-1"></a><span class="co">#&gt; [1] 6.5</span></span>
<span id="cb30-12"><a href="introR.html#cb30-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb30-13"><a href="introR.html#cb30-13" tabindex="-1"></a><span class="co">#&gt; [[3]]</span></span>
<span id="cb30-14"><a href="introR.html#cb30-14" tabindex="-1"></a><span class="co">#&gt; [1] 8</span></span></code></pre></div>
<p>If you want the return as a vector, you can use <code>sapply()</code> for <code>simplified</code> output:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="introR.html#cb31-1" tabindex="-1"></a><span class="fu">sapply</span>(my.lst, median)</span>
<span id="cb31-2"><a href="introR.html#cb31-2" tabindex="-1"></a><span class="co">#&gt; [1] 5.0 6.5 8.0</span></span></code></pre></div>
<p>If you want to check the data type of the return, you can further use the <code>vapply</code> function. An error will be raised if the data type does not match.</p>
<p><strong>Note</strong>, the <code>FUN.VALUE</code> argument takes the value as a template, so the use of
<code>numeric(1)</code> or any numeric value, e.g., <code>3</code> is the same.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="introR.html#cb32-1" tabindex="-1"></a><span class="fu">vapply</span>(my.lst, median, <span class="fu">numeric</span>(<span class="dv">1</span>))</span>
<span id="cb32-2"><a href="introR.html#cb32-2" tabindex="-1"></a><span class="co">#&gt; [1] 5.0 6.5 8.0</span></span>
<span id="cb32-3"><a href="introR.html#cb32-3" tabindex="-1"></a></span>
<span id="cb32-4"><a href="introR.html#cb32-4" tabindex="-1"></a><span class="fu">vapply</span>(my.lst, median, <span class="dv">3</span>)</span>
<span id="cb32-5"><a href="introR.html#cb32-5" tabindex="-1"></a><span class="co">#&gt; [1] 5.0 6.5 8.0</span></span>
<span id="cb32-6"><a href="introR.html#cb32-6" tabindex="-1"></a></span>
<span id="cb32-7"><a href="introR.html#cb32-7" tabindex="-1"></a><span class="co"># try this</span></span>
<span id="cb32-8"><a href="introR.html#cb32-8" tabindex="-1"></a><span class="co"># vapply(my.lst, median, character(1))</span></span></code></pre></div>
</div>
</div>
</div>
</div>
<div id="plotting" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Plotting<a href="introR.html#plotting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="datasets" class="section level3 hasAnchor" number="1.5.1">
<h3><span class="header-section-number">1.5.1</span> datasets<a href="introR.html#datasets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s use a built-in dataset for illustration: <code>iris</code> (4 flower features in 3 plants)</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="introR.html#cb33-1" tabindex="-1"></a><span class="fu">head</span>(iris)</span>
<span id="cb33-2"><a href="introR.html#cb33-2" tabindex="-1"></a><span class="co">#&gt;   Sepal.Length Sepal.Width Petal.Length Petal.Width Species</span></span>
<span id="cb33-3"><a href="introR.html#cb33-3" tabindex="-1"></a><span class="co">#&gt; 1          5.1         3.5          1.4         0.2  setosa</span></span>
<span id="cb33-4"><a href="introR.html#cb33-4" tabindex="-1"></a><span class="co">#&gt; 2          4.9         3.0          1.4         0.2  setosa</span></span>
<span id="cb33-5"><a href="introR.html#cb33-5" tabindex="-1"></a><span class="co">#&gt; 3          4.7         3.2          1.3         0.2  setosa</span></span>
<span id="cb33-6"><a href="introR.html#cb33-6" tabindex="-1"></a><span class="co">#&gt; 4          4.6         3.1          1.5         0.2  setosa</span></span>
<span id="cb33-7"><a href="introR.html#cb33-7" tabindex="-1"></a><span class="co">#&gt; 5          5.0         3.6          1.4         0.2  setosa</span></span>
<span id="cb33-8"><a href="introR.html#cb33-8" tabindex="-1"></a><span class="co">#&gt; 6          5.4         3.9          1.7         0.4  setosa</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="introR.html#cb34-1" tabindex="-1"></a><span class="fu">summary</span>(iris)</span>
<span id="cb34-2"><a href="introR.html#cb34-2" tabindex="-1"></a><span class="co">#&gt;   Sepal.Length    Sepal.Width     Petal.Length    Petal.Width   </span></span>
<span id="cb34-3"><a href="introR.html#cb34-3" tabindex="-1"></a><span class="co">#&gt;  Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100  </span></span>
<span id="cb34-4"><a href="introR.html#cb34-4" tabindex="-1"></a><span class="co">#&gt;  1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300  </span></span>
<span id="cb34-5"><a href="introR.html#cb34-5" tabindex="-1"></a><span class="co">#&gt;  Median :5.800   Median :3.000   Median :4.350   Median :1.300  </span></span>
<span id="cb34-6"><a href="introR.html#cb34-6" tabindex="-1"></a><span class="co">#&gt;  Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199  </span></span>
<span id="cb34-7"><a href="introR.html#cb34-7" tabindex="-1"></a><span class="co">#&gt;  3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800  </span></span>
<span id="cb34-8"><a href="introR.html#cb34-8" tabindex="-1"></a><span class="co">#&gt;  Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500  </span></span>
<span id="cb34-9"><a href="introR.html#cb34-9" tabindex="-1"></a><span class="co">#&gt;        Species  </span></span>
<span id="cb34-10"><a href="introR.html#cb34-10" tabindex="-1"></a><span class="co">#&gt;  setosa    :50  </span></span>
<span id="cb34-11"><a href="introR.html#cb34-11" tabindex="-1"></a><span class="co">#&gt;  versicolor:50  </span></span>
<span id="cb34-12"><a href="introR.html#cb34-12" tabindex="-1"></a><span class="co">#&gt;  virginica :50  </span></span>
<span id="cb34-13"><a href="introR.html#cb34-13" tabindex="-1"></a><span class="co">#&gt;                 </span></span>
<span id="cb34-14"><a href="introR.html#cb34-14" tabindex="-1"></a><span class="co">#&gt;                 </span></span>
<span id="cb34-15"><a href="introR.html#cb34-15" tabindex="-1"></a><span class="co">#&gt; </span></span></code></pre></div>
<p>There are two common ways of plotting:</p>
<ol style="list-style-type: decimal">
<li>the built-in plotting functions</li>
<li>the ggplot format</li>
</ol>
</div>
<div id="basic-plotting" class="section level3 hasAnchor" number="1.5.2">
<h3><span class="header-section-number">1.5.2</span> Basic plotting<a href="introR.html#basic-plotting" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="histogram" class="section level4 hasAnchor" number="1.5.2.1">
<h4><span class="header-section-number">1.5.2.1</span> Histogram<a href="introR.html#histogram" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="introR.html#cb35-1" tabindex="-1"></a><span class="fu">hist</span>(iris<span class="sc">$</span>Sepal.Length)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig1-hist-1.png" width="672" /></p>
</div>
<div id="scatter-plot" class="section level4 hasAnchor" number="1.5.2.2">
<h4><span class="header-section-number">1.5.2.2</span> Scatter plot<a href="introR.html#scatter-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="introR.html#cb36-1" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x=</span>iris<span class="sc">$</span>Sepal.Length, <span class="at">y=</span>iris<span class="sc">$</span>Sepal.Width)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig1-scatter-1.png" width="672" /></p>
</div>
<div id="boxplot" class="section level4 hasAnchor" number="1.5.2.3">
<h4><span class="header-section-number">1.5.2.3</span> boxplot<a href="introR.html#boxplot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="introR.html#cb37-1" tabindex="-1"></a>x1 <span class="ot">&lt;-</span> iris<span class="sc">$</span>Sepal.Length[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;setosa&quot;</span>]</span>
<span id="cb37-2"><a href="introR.html#cb37-2" tabindex="-1"></a>x2 <span class="ot">&lt;-</span> iris<span class="sc">$</span>Sepal.Length[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;versicolor&quot;</span>]</span>
<span id="cb37-3"><a href="introR.html#cb37-3" tabindex="-1"></a>x3 <span class="ot">&lt;-</span> iris<span class="sc">$</span>Sepal.Length[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;virginica&quot;</span>]</span>
<span id="cb37-4"><a href="introR.html#cb37-4" tabindex="-1"></a></span>
<span id="cb37-5"><a href="introR.html#cb37-5" tabindex="-1"></a><span class="fu">boxplot</span>(x1, x2, x3)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig1-box-1.png" width="672" /></p>
</div>
</div>
<div id="ggplot2" class="section level3 hasAnchor" number="1.5.3">
<h3><span class="header-section-number">1.5.3</span> ggplot2<a href="introR.html#ggplot2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>See more instructions: <a href="http://www.sthda.com/english/wiki/ggplot2-essentials" class="uri">http://www.sthda.com/english/wiki/ggplot2-essentials</a></p>
<div id="install-and-load-package" class="section level4 hasAnchor" number="1.5.3.1">
<h4><span class="header-section-number">1.5.3.1</span> Install and load package<a href="introR.html#install-and-load-package" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="introR.html#cb38-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;ggplot2&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb38-2"><a href="introR.html#cb38-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb38-3"><a href="introR.html#cb38-3" tabindex="-1"></a></span>
<span id="cb38-4"><a href="introR.html#cb38-4" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
</div>
<div id="histogram-1" class="section level4 hasAnchor" number="1.5.3.2">
<h4><span class="header-section-number">1.5.3.2</span> Histogram<a href="introR.html#histogram-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="introR.html#cb39-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x=</span>Sepal.Length)) <span class="sc">+</span> </span>
<span id="cb39-2"><a href="introR.html#cb39-2" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">8</span>)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig1-gghist-1.png" width="672" /></p>
</div>
<div id="scatter-plot-1" class="section level4 hasAnchor" number="1.5.3.3">
<h4><span class="header-section-number">1.5.3.3</span> Scatter plot<a href="introR.html#scatter-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="introR.html#cb40-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x=</span>Sepal.Length, <span class="at">y=</span>Sepal.Width)) <span class="sc">+</span> </span>
<span id="cb40-2"><a href="introR.html#cb40-2" tabindex="-1"></a>    <span class="fu">geom_point</span>()</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig1-ggpoint-1.png" width="672" /></p>
</div>
<div id="box-plot" class="section level4 hasAnchor" number="1.5.3.4">
<h4><span class="header-section-number">1.5.3.4</span> Box plot<a href="introR.html#box-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="introR.html#cb41-1" tabindex="-1"></a><span class="fu">ggplot</span>(iris, <span class="fu">aes</span>(<span class="at">x=</span>Species, <span class="at">y=</span>Sepal.Length)) <span class="sc">+</span> </span>
<span id="cb41-2"><a href="introR.html#cb41-2" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>()</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig1-ggbox-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="scientific-and-statistical-computating" class="section level2 hasAnchor" number="1.6">
<h2><span class="header-section-number">1.6</span> Scientific and statistical computating<a href="introR.html#scientific-and-statistical-computating" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="orders-of-operators" class="section level3 hasAnchor" number="1.6.1">
<h3><span class="header-section-number">1.6.1</span> Orders of operators<a href="introR.html#orders-of-operators" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>See lecture slides.</p>
<p>If you are not sure about a certain ordering, use brackets!</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="introR.html#cb42-1" tabindex="-1"></a><span class="dv">5</span> <span class="sc">*</span> <span class="dv">2</span> <span class="sc">&gt;</span> <span class="dv">4</span></span>
<span id="cb42-2"><a href="introR.html#cb42-2" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="introR.html#cb43-1" tabindex="-1"></a><span class="dv">5</span> <span class="sc">*</span> (<span class="dv">2</span> <span class="sc">&gt;</span> <span class="dv">4</span>)</span>
<span id="cb43-2"><a href="introR.html#cb43-2" tabindex="-1"></a><span class="co">#&gt; [1] 0</span></span></code></pre></div>
</div>
<div id="functions-for-statistics" class="section level3 hasAnchor" number="1.6.2">
<h3><span class="header-section-number">1.6.2</span> Functions for statistics<a href="introR.html#functions-for-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>More theory and practice to come in next session</p>
</div>
<div id="correlation" class="section level3 hasAnchor" number="1.6.3">
<h3><span class="header-section-number">1.6.3</span> Correlation<a href="introR.html#correlation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="introR.html#cb44-1" tabindex="-1"></a><span class="fu">cor</span>(iris<span class="sc">$</span>Sepal.Length, iris<span class="sc">$</span>Petal.Length)</span>
<span id="cb44-2"><a href="introR.html#cb44-2" tabindex="-1"></a><span class="co">#&gt; [1] 0.8717538</span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="introR.html#cb45-1" tabindex="-1"></a><span class="fu">cor.test</span>(iris<span class="sc">$</span>Sepal.Length, iris<span class="sc">$</span>Petal.Length)</span>
<span id="cb45-2"><a href="introR.html#cb45-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb45-3"><a href="introR.html#cb45-3" tabindex="-1"></a><span class="co">#&gt;  Pearson&#39;s product-moment correlation</span></span>
<span id="cb45-4"><a href="introR.html#cb45-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb45-5"><a href="introR.html#cb45-5" tabindex="-1"></a><span class="co">#&gt; data:  iris$Sepal.Length and iris$Petal.Length</span></span>
<span id="cb45-6"><a href="introR.html#cb45-6" tabindex="-1"></a><span class="co">#&gt; t = 21.646, df = 148, p-value &lt; 2.2e-16</span></span>
<span id="cb45-7"><a href="introR.html#cb45-7" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true correlation is not equal to 0</span></span>
<span id="cb45-8"><a href="introR.html#cb45-8" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb45-9"><a href="introR.html#cb45-9" tabindex="-1"></a><span class="co">#&gt;  0.8270363 0.9055080</span></span>
<span id="cb45-10"><a href="introR.html#cb45-10" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb45-11"><a href="introR.html#cb45-11" tabindex="-1"></a><span class="co">#&gt;       cor </span></span>
<span id="cb45-12"><a href="introR.html#cb45-12" tabindex="-1"></a><span class="co">#&gt; 0.8717538</span></span></code></pre></div>
</div>
<div id="hypothesis-testing-t-test" class="section level3 hasAnchor" number="1.6.4">
<h3><span class="header-section-number">1.6.4</span> Hypothesis testing (t test)<a href="introR.html#hypothesis-testing-t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="introR.html#cb46-1" tabindex="-1"></a>x1 <span class="ot">&lt;-</span> iris<span class="sc">$</span>Sepal.Length[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;setosa&quot;</span>]</span>
<span id="cb46-2"><a href="introR.html#cb46-2" tabindex="-1"></a>x2 <span class="ot">&lt;-</span> iris<span class="sc">$</span>Sepal.Length[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;versicolor&quot;</span>]</span>
<span id="cb46-3"><a href="introR.html#cb46-3" tabindex="-1"></a>x3 <span class="ot">&lt;-</span> iris<span class="sc">$</span>Sepal.Length[iris<span class="sc">$</span>Species <span class="sc">==</span> <span class="st">&quot;virginica&quot;</span>]</span>
<span id="cb46-4"><a href="introR.html#cb46-4" tabindex="-1"></a></span>
<span id="cb46-5"><a href="introR.html#cb46-5" tabindex="-1"></a><span class="fu">t.test</span>(x2, x3)</span>
<span id="cb46-6"><a href="introR.html#cb46-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb46-7"><a href="introR.html#cb46-7" tabindex="-1"></a><span class="co">#&gt;  Welch Two Sample t-test</span></span>
<span id="cb46-8"><a href="introR.html#cb46-8" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb46-9"><a href="introR.html#cb46-9" tabindex="-1"></a><span class="co">#&gt; data:  x2 and x3</span></span>
<span id="cb46-10"><a href="introR.html#cb46-10" tabindex="-1"></a><span class="co">#&gt; t = -5.6292, df = 94.025, p-value = 1.866e-07</span></span>
<span id="cb46-11"><a href="introR.html#cb46-11" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true difference in means is not equal to 0</span></span>
<span id="cb46-12"><a href="introR.html#cb46-12" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb46-13"><a href="introR.html#cb46-13" tabindex="-1"></a><span class="co">#&gt;  -0.8819731 -0.4220269</span></span>
<span id="cb46-14"><a href="introR.html#cb46-14" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb46-15"><a href="introR.html#cb46-15" tabindex="-1"></a><span class="co">#&gt; mean of x mean of y </span></span>
<span id="cb46-16"><a href="introR.html#cb46-16" tabindex="-1"></a><span class="co">#&gt;     5.936     6.588</span></span></code></pre></div>
</div>
<div id="regression" class="section level3 hasAnchor" number="1.6.5">
<h3><span class="header-section-number">1.6.5</span> Regression<a href="introR.html#regression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="introR.html#cb47-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lm</span>(Sepal.Length <span class="sc">~</span> Sepal.Width <span class="sc">+</span> Petal.Length <span class="sc">+</span> Petal.Width, <span class="at">data=</span>iris)</span>
<span id="cb47-2"><a href="introR.html#cb47-2" tabindex="-1"></a><span class="fu">summary</span>(fit) <span class="co"># show results</span></span>
<span id="cb47-3"><a href="introR.html#cb47-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb47-4"><a href="introR.html#cb47-4" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb47-5"><a href="introR.html#cb47-5" tabindex="-1"></a><span class="co">#&gt; lm(formula = Sepal.Length ~ Sepal.Width + Petal.Length + Petal.Width, </span></span>
<span id="cb47-6"><a href="introR.html#cb47-6" tabindex="-1"></a><span class="co">#&gt;     data = iris)</span></span>
<span id="cb47-7"><a href="introR.html#cb47-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb47-8"><a href="introR.html#cb47-8" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb47-9"><a href="introR.html#cb47-9" tabindex="-1"></a><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span id="cb47-10"><a href="introR.html#cb47-10" tabindex="-1"></a><span class="co">#&gt; -0.82816 -0.21989  0.01875  0.19709  0.84570 </span></span>
<span id="cb47-11"><a href="introR.html#cb47-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb47-12"><a href="introR.html#cb47-12" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb47-13"><a href="introR.html#cb47-13" tabindex="-1"></a><span class="co">#&gt;              Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb47-14"><a href="introR.html#cb47-14" tabindex="-1"></a><span class="co">#&gt; (Intercept)   1.85600    0.25078   7.401 9.85e-12 ***</span></span>
<span id="cb47-15"><a href="introR.html#cb47-15" tabindex="-1"></a><span class="co">#&gt; Sepal.Width   0.65084    0.06665   9.765  &lt; 2e-16 ***</span></span>
<span id="cb47-16"><a href="introR.html#cb47-16" tabindex="-1"></a><span class="co">#&gt; Petal.Length  0.70913    0.05672  12.502  &lt; 2e-16 ***</span></span>
<span id="cb47-17"><a href="introR.html#cb47-17" tabindex="-1"></a><span class="co">#&gt; Petal.Width  -0.55648    0.12755  -4.363 2.41e-05 ***</span></span>
<span id="cb47-18"><a href="introR.html#cb47-18" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb47-19"><a href="introR.html#cb47-19" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb47-20"><a href="introR.html#cb47-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb47-21"><a href="introR.html#cb47-21" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 0.3145 on 146 degrees of freedom</span></span>
<span id="cb47-22"><a href="introR.html#cb47-22" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.8586, Adjusted R-squared:  0.8557 </span></span>
<span id="cb47-23"><a href="introR.html#cb47-23" tabindex="-1"></a><span class="co">#&gt; F-statistic: 295.5 on 3 and 146 DF,  p-value: &lt; 2.2e-16</span></span></code></pre></div>
<p>This means the fitted regression is:</p>
<p><code>Sepal.Length ~ 1.856 + 0.65*Sepal.Width + 0.709*Petal.Length - 0.556*Petal.Width</code></p>
<p>We can check how good the regression is by plotting it out</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="introR.html#cb48-1" tabindex="-1"></a>y_pred <span class="ot">&lt;-</span> fit<span class="sc">$</span>coefficients[<span class="dv">1</span>] <span class="sc">+</span> </span>
<span id="cb48-2"><a href="introR.html#cb48-2" tabindex="-1"></a>  fit<span class="sc">$</span>coefficients[<span class="dv">2</span>] <span class="sc">*</span> iris<span class="sc">$</span>Sepal.Width <span class="sc">+</span></span>
<span id="cb48-3"><a href="introR.html#cb48-3" tabindex="-1"></a>  fit<span class="sc">$</span>coefficients[<span class="dv">3</span>] <span class="sc">*</span> iris<span class="sc">$</span>Petal.Length <span class="sc">+</span></span>
<span id="cb48-4"><a href="introR.html#cb48-4" tabindex="-1"></a>  fit<span class="sc">$</span>coefficients[<span class="dv">4</span>] <span class="sc">*</span> iris<span class="sc">$</span>Petal.Width</span>
<span id="cb48-5"><a href="introR.html#cb48-5" tabindex="-1"></a></span>
<span id="cb48-6"><a href="introR.html#cb48-6" tabindex="-1"></a><span class="fu">cor</span>(iris<span class="sc">$</span>Sepal.Length, y_pred)</span>
<span id="cb48-7"><a href="introR.html#cb48-7" tabindex="-1"></a><span class="co">#&gt; [1] 0.926613</span></span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="introR.html#cb49-1" tabindex="-1"></a><span class="fu">plot</span>(iris<span class="sc">$</span>Sepal.Length, y_pred)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig1-reg-1.png" width="672" /></p>
</div>
</div>
<div id="resource-links" class="section level2 hasAnchor" number="1.7">
<h2><span class="header-section-number">1.7</span> Resource links<a href="introR.html#resource-links" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This notebook adapts contents from these resources:</p>
<ul>
<li><a href="https://www.datamentor.io/r-programming/" class="uri">https://www.datamentor.io/r-programming/</a></li>
<li><a href="https://www.geeksforgeeks.org/r-data-types/" class="uri">https://www.geeksforgeeks.org/r-data-types/</a></li>
<li><a href="https://data-flair.training/blogs/r-data-types/" class="uri">https://data-flair.training/blogs/r-data-types/</a></li>
<li><a href="http://www.sthda.com/english/wiki/ggplot2-essentials" class="uri">http://www.sthda.com/english/wiki/ggplot2-essentials</a></li>
</ul>
</div>
<div id="exercises" class="section level2 hasAnchor" number="1.8">
<h2><span class="header-section-number">1.8</span> Exercises<a href="introR.html#exercises" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This list of exercises will serve as a 2-hour demonstration in the BIOF1001 course.
For other learners, you can go through it as homework. The expected time is
around two hours if you have read the R materials carefully together with
trying them with your own R environment.</p>
<blockquote>
<p><strong>Note</strong>, before you get started, please make sure that you are familiar with
the panels in RStudio. You may watch this
<a href="https://www.youtube.com/watch?v=FIrsOBy5k58">YouTube Video</a> or check this
<a href="https://rstudio.github.io/cheatsheets/rstudio-ide.pdf">RStudio cheatsheet</a>.</p>
</blockquote>
<div id="exercises-p1" class="section level3 hasAnchor" number="1.8.1">
<h3><span class="header-section-number">1.8.1</span> Part 1. Basics (~40min)<a href="introR.html#exercises-p1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Q1: We have 25 students in BIOF1001, to store the final marks (0 to 100 with
a precision of 0.1), what data type will we use?</li>
</ul>
<!-- > Numeric (or float or double) -->
<ul>
<li>Q2: For the grades (<code>A+</code> to <code>F</code>), what data type and data structure can be
used to keep the memory minimal?</li>
</ul>
<!-- > `character` and `factor` -->
<ul>
<li>Q3: Make a matrix with name <code>my_matrix</code>, shape of 5 rows and 2 columns and values
from 3 to 12. The first row is 3 and 4.
<em>Hint</em>: for making a vector from 3 to 12, you may use <code>seq()</code> or <code>:</code>.</li>
</ul>
<!-- > my_matrix = matrix(seq(3, 12), nrow=5, byrow=TRUE) -->
<!-- > my_matrix = matrix(3:12, nrow=5, byrow=TRUE) -->
<ul>
<li>Q4: Based on <code>Q3</code>, add the row names to <code>Day1</code> to <code>Day5</code> and column names to
<code>Lunch</code> and <code>Dinner</code>.</li>
</ul>
<!-- > rownames(my_matrix) = c("Day1", "Day2", "Day3", "Day4", "Day5") -->
<!-- > colnames(my_matrix) = c("Lunch", "Dinner") -->
<!-- > rownames(my_matrix) = paste0("Day", seq(1,5)) -->
<ul>
<li>Q5: Based on <code>Q4</code>, get a matrix with shape of 3x1 and values of 6, 8, 10 from
the matrix <code>my_matrix</code>.</li>
</ul>
<!-- > Any of the following codes: -->
<!-- > my_matrix[2:4, 2, drop=FALSE] -->
<!-- > my_matrix[c(2, 3, 4), 2, drop=FALSE] -->
<!-- > my_matrix[c("Day2", "Day3", "Day4"), 2, drop=FALSE] -->
<ul>
<li>Q6: What will you get for <code>my_matrix[c(TRUE, FALSE, FALSE, TRUE), ]</code>?
<em>Hint</em>: think of recycling if the index length is different from the query
dimension (Over-flexibility comes with a price of wrong use).</li>
</ul>
<!-- > it will give row length of 3, for row 1, 4 and 5. -->
<ul>
<li>Q7: If you have two vectors the BIOF1001 <code>marks</code> and <code>grades</code> and one
<code>character</code> for teaching performance <code>"good"</code>, and you want to store them
into one variable, which data structure will you use?</li>
</ul>
<!-- > `list` data structure -->
<ul>
<li><p>Q8: Now, on your <code>Desktop folder</code>, make a subfolder with name <code>R_exercises</code>
and download this file of differentailly expressed genes results
<a href="https://github.com/StatBiomed/BMDS-book/blob/main/notebooks/chapter1-R/Diff_Expression_results.tsv">Diff_Expression_results.tsv</a>
to the folder. Check your current work directory by
<code>getwd()</code> function and change the work directory to the folder you just created.
<em>Hint</em>: you may use <code>setwd()</code> to change the work directory or use the <code>Session</code>
button of RStudio.</p></li>
<li><p>Q9: Related to Q8, use the <code>read.table()</code> function to load the file into a
data frame with the variable name <code>df_DEG</code>.
<em>Hint</em>: You may consider using the full path or just the file name if it’s in
the same work directory. Please keep <code>header=1</code> for the argument. Think how to
find help page for a certain function.</p></li>
</ul>
<!-- > df_DEG = read.table("Diff_Expression_results.tsv", sep="\t", header=1) -->
<ul>
<li>Q10: Can you calculate the <code>mean</code> and <code>standard deviation</code> of the <code>log2FoldChange</code>?
If the mean &gt;0.1 and the standard deviation &lt; 3, set the <code>trend</code> variable as
“increase”, otherwise “not_increase”. What will happen if you add this <code>trend</code>
variable to the data frame, and why?
<em>Hint</em>: use <code>mean()</code> and <code>st()</code> functions for calculating mean and
standard deviation.</li>
</ul>
<!-- > trend = "not_increase" -->
<!-- > if (mean(df_DEG$log2FoldChange) > 0.1 && sd(df_DEG$log2FoldChange) < 3) trend = "increase" -->
<!-- > df_DEG[["trend"]] = trend -->
<!-- > data frame is also a list -->
</div>
<div id="exercises-p2" class="section level3 hasAnchor" number="1.8.2">
<h3><span class="header-section-number">1.8.2</span> Part 2. Making plotting (~40min)<a href="introR.html#exercises-p2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Q11: Keep use the <code>df_DEG</code> from part 1 Q9. Now, make a histogram of the
<code>log2FoldChange</code> with both basic plotting function <code>hist()</code> and <code>ggplot2</code>.</p></li>
<li><p>Q12: Make a plot with x-axis of <code>log10(baseMean)</code> and y-axis of <code>log2FoldChange</code>.
Please try both the basic <code>plot()</code> function and <code>ggplot2</code>.</p></li>
<li><p>Q13: Now, manipulate the dataframe by adding two columns:</p>
<ul>
<li>Add a column <code>log2FC_clip</code> for clipping log2FoldChange to <code>[-5, +5]</code></li>
<li>Add a column <code>is_DE</code> for <code>padj &lt; 0.05</code></li>
</ul></li>
<li><p>Q14: Try the <code>summary()</code> function with the above <code>df_DEG</code> data frame, and also
<code>table()</code> function for the <code>is_DE</code> column.</p></li>
<li><p>Q15: Based on ggplot2, add the color by the newly added column <code>is_DE</code>.</p></li>
</ul>
<!-- ggplot(df, aes(x=log10(baseMean),           y=log2FoldChange_clip)) +  -->
<!--   geom_point(aes(color=is_DE)) +  -->
<!--   scale_color_manual(values = c("red", "grey")) -->
<ul>
<li><p>Q16: Set the colors to “red” and “grey”, and make it in the order of TRUE and
FALSE.
<em>Hint</em>: use <code>factor</code> and set the <code>levels</code> parameter.</p></li>
<li><p>Q17: Save the generated figure into “My_DEG_results.pdf”. Use the
<a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave()</a> function. Please
set <code>width = 5, height = 4</code>.</p></li>
</ul>
<p>You are expected a figure like this</p>
<div class="float">
<img src="https://github.com/StatBiomed/BMDS-book/raw/main/notebooks/chapter1-R/DEG_results.png" alt="DEG figure" />
<div class="figcaption">DEG figure</div>
</div>
<p>If you want to change labels on x-axis or y-axis and font size, etc., you can
simply Google and find examples.</p>
</div>
<div id="exercises-p3" class="section level3 hasAnchor" number="1.8.3">
<h3><span class="header-section-number">1.8.3</span> Part 3. For loop and repeating processing (~40min)<a href="introR.html#exercises-p3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Q18: Load the following table from this file on GitHub and <em>View</em> it in RStudio.</p>
<p>It contains expressoon of 619 transcription factors from 7 Nasopharyngeal
carcinoma (NPC) samples and 3 nasopharyngeal lymphatic hyperplasia (NLH) samples.
<a href="https://github.com/StatBiomed/NegabinGLM/blob/main/data/NPC_NLH-Tcell-donorX.tsv" class="uri">https://github.com/StatBiomed/NegabinGLM/blob/main/data/NPC_NLH-Tcell-donorX.tsv</a></p>
<p><code>df_NPC = read.table("https://github.com/StatBiomed/NegabinGLM/raw/main/data/NPC_NLH-Tcell-donorX.tsv", sep="\t", header=1)</code></p></li>
<li><p>Q19: Extract the column 5 (HES5) to 623 (LEK4) and make it into a matrix <code>TF_mat</code>
<em>Hint</em>: how to index a data frame like index a matrix.</p></li>
</ul>
<!-- > TF_mat = as.matrix(df_NPC[, colnames(df_NPC)[5:623]]) -->
<!-- > TF_mat = as.matrix(df_NPC[, 5:623]) -->
<ul>
<li><p>Q20: perform normalization. Divide the <code>TF_mat</code> by the <code>df_NPC$total_counts</code>
and multiply by 1000000, and assign it to a new matrix named <code>TF_mat_norm</code>.</p>
<p>You may further consider transformation by log1p(), i.e., log(TF_mat_norm + 1).</p></li>
<li><p>Q21: calculate the log fold change on the first gene TP73 in <code>TF_mat_norm</code>
between NPC (row 1 to 7) and NLH (row 8 to 10) and perform t-test return the
<code>p</code> value and log fold change.</p></li>
<li><p>Q22: perform t-test on the <em>all</em> gene in <code>TF_mat_norm</code> between
NPC (row 1 to 7) and NLH (row 8 to 10).
<em>Hint</em>: think of for loop or <code>apply()</code> function.</p></li>
</ul>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="preface.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introLinearReg.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/statbiomed/BMDS-book/tree/main/01-introduction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BMDatSci.pdf", "BMDatSci.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
