<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Introduction to Hypothesis test | Biomedical Data Science - introduction with case studies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Introduction to Hypothesis test | Biomedical Data Science - introduction with case studies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Introduction to Hypothesis test | Biomedical Data Science - introduction with case studies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="BIOF1001 teaching team" />


<meta name="date" content="2025-10-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introR.html"/>
<link rel="next" href="introLinearReg.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">Biomedical Data Science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#introduction-for-readers"><i class="fa fa-check"></i>Introduction for readers</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-you-will-learn-from-this-coursebook"><i class="fa fa-check"></i>What you will learn from this course/book</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-recommend-you-do-while-reading-this-book"><i class="fa fa-check"></i>What we recommend you do while reading this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#other-reference-books"><i class="fa fa-check"></i>Other reference books</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="part"><span><b>I Data Science Foundations</b></span></li>
<li class="chapter" data-level="1" data-path="introR.html"><a href="introR.html"><i class="fa fa-check"></i><b>1</b> Introduction to R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introR.html"><a href="introR.html#introR-w1"><i class="fa fa-check"></i><b>1.1</b> Intro to R programming (Session 1)</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introR.html"><a href="introR.html#r-as-a-basic-calculator"><i class="fa fa-check"></i><b>1.1.1</b> R as a basic calculator</a></li>
<li class="chapter" data-level="1.1.2" data-path="introR.html"><a href="introR.html#variables"><i class="fa fa-check"></i><b>1.1.2</b> Variables</a></li>
<li class="chapter" data-level="1.1.3" data-path="introR.html"><a href="introR.html#data-types"><i class="fa fa-check"></i><b>1.1.3</b> Data types</a></li>
<li class="chapter" data-level="1.1.4" data-path="introR.html"><a href="introR.html#functions"><i class="fa fa-check"></i><b>1.1.4</b> Functions</a></li>
<li class="chapter" data-level="1.1.5" data-path="introR.html"><a href="introR.html#r-scripts"><i class="fa fa-check"></i><b>1.1.5</b> R scripts</a></li>
<li class="chapter" data-level="1.1.6" data-path="introR.html"><a href="introR.html#resource-links"><i class="fa fa-check"></i><b>1.1.6</b> Resource links</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introR.html"><a href="introR.html#introR-w2"><i class="fa fa-check"></i><b>1.2</b> Intro to R programming (Session 2)</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introR.html"><a href="introR.html#data-structures"><i class="fa fa-check"></i><b>1.2.1</b> Data structures</a></li>
<li class="chapter" data-level="1.2.2" data-path="introR.html"><a href="introR.html#vector"><i class="fa fa-check"></i><b>1.2.2</b> Vector</a></li>
<li class="chapter" data-level="1.2.3" data-path="introR.html"><a href="introR.html#matrix"><i class="fa fa-check"></i><b>1.2.3</b> Matrix</a></li>
<li class="chapter" data-level="1.2.4" data-path="introR.html"><a href="introR.html#list"><i class="fa fa-check"></i><b>1.2.4</b> List</a></li>
<li class="chapter" data-level="1.2.5" data-path="introR.html"><a href="introR.html#data-frame"><i class="fa fa-check"></i><b>1.2.5</b> Data Frame</a></li>
<li class="chapter" data-level="1.2.6" data-path="introR.html"><a href="introR.html#readingwriting-data-fromto-r"><i class="fa fa-check"></i><b>1.2.6</b> Reading/writing data from/to R</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="introR.html"><a href="introR.html#introR-w3"><i class="fa fa-check"></i><b>1.3</b> Intro to R programming (Session 4)</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introR.html"><a href="introR.html#operators"><i class="fa fa-check"></i><b>1.3.1</b> Operators</a></li>
<li class="chapter" data-level="1.3.2" data-path="introR.html"><a href="introR.html#ifelse"><i class="fa fa-check"></i><b>1.3.2</b> If/else</a></li>
<li class="chapter" data-level="1.3.3" data-path="introR.html"><a href="introR.html#loop"><i class="fa fa-check"></i><b>1.3.3</b> Loop</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introHypoTest.html"><a href="introHypoTest.html"><i class="fa fa-check"></i><b>2</b> Introduction to Hypothesis test</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-test-and-p-value"><i class="fa fa-check"></i><b>2.1</b> Hypothesis test and <em>p</em>-value</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="introHypoTest.html"><a href="introHypoTest.html#extreme-events-and-random-chance"><i class="fa fa-check"></i><b>2.1.1</b> Extreme events and random chance</a></li>
<li class="chapter" data-level="2.1.2" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-test-for-statistical-decision"><i class="fa fa-check"></i><b>2.1.2</b> Hypothesis test for statistical decision</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#basic-hypothesis-test-methods"><i class="fa fa-check"></i><b>2.2</b> Basic hypothesis test methods</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#permutation-test"><i class="fa fa-check"></i><b>2.2.1</b> Permutation test</a></li>
<li class="chapter" data-level="2.2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#t-test-and-regression-based-test"><i class="fa fa-check"></i><b>2.2.2</b> <em>t</em>-test and regression-based test</a></li>
<li class="chapter" data-level="2.2.3" data-path="introHypoTest.html"><a href="introHypoTest.html#fishers-exact-test"><i class="fa fa-check"></i><b>2.2.3</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introHypoTest.html"><a href="introHypoTest.html#multiple-testing-and-errors-to-be-updated"><i class="fa fa-check"></i><b>2.3</b> Multiple testing and errors (To be updated)</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-test-statistic"><i class="fa fa-check"></i><b>2.3.1</b> Null distribution (of test statistic)</a></li>
<li class="chapter" data-level="2.3.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-p-value"><i class="fa fa-check"></i><b>2.3.2</b> Null distribution of p value</a></li>
<li class="chapter" data-level="2.3.3" data-path="introHypoTest.html"><a href="introHypoTest.html#minimal-p-values-in-10-tests"><i class="fa fa-check"></i><b>2.3.3</b> Minimal p values in 10 tests</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="introHypoTest.html"><a href="introHypoTest.html#power-analysis"><i class="fa fa-check"></i><b>2.4</b> Explore power and sample size (optional)</a></li>
<li class="chapter" data-level="2.5" data-path="introHypoTest.html"><a href="introHypoTest.html#summary"><i class="fa fa-check"></i><b>2.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introLinearReg.html"><a href="introLinearReg.html"><i class="fa fa-check"></i><b>3</b> Introduction to Linear Regression</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#linear-regression-using-simulated-data"><i class="fa fa-check"></i><b>3.1</b> Linear Regression Using Simulated Data</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="introLinearReg.html"><a href="introLinearReg.html#simulating-data"><i class="fa fa-check"></i><b>3.1.1</b> Simulating data:</a></li>
<li class="chapter" data-level="3.1.2" data-path="introLinearReg.html"><a href="introLinearReg.html#model-efficacy"><i class="fa fa-check"></i><b>3.1.2</b> Model efficacy</a></li>
<li class="chapter" data-level="3.1.3" data-path="introLinearReg.html"><a href="introLinearReg.html#r-squared"><i class="fa fa-check"></i><b>3.1.3</b> <em>R-Squared</em></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#least-squares-using-simulated-data"><i class="fa fa-check"></i><b>3.2</b> Least Squares Using Simulated Data</a></li>
<li class="chapter" data-level="3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#diagnostic-check-of-a-fitted-regression-model"><i class="fa fa-check"></i><b>3.3</b> Diagnostic check of a fitted regression model</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#residual-standard-errors-rse"><i class="fa fa-check"></i><b>3.3.1</b> Residual Standard Errors (RSE)</a></li>
<li class="chapter" data-level="3.3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#p-values"><i class="fa fa-check"></i><b>3.3.2</b> p-values</a></li>
<li class="chapter" data-level="3.3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#f-statistics"><i class="fa fa-check"></i><b>3.3.3</b> F-statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introLinearReg.html"><a href="introLinearReg.html#simple-linear-regression-with-lm-function"><i class="fa fa-check"></i><b>3.4</b> Simple Linear Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="3.5" data-path="introLinearReg.html"><a href="introLinearReg.html#multiple-regression-with-lm-function"><i class="fa fa-check"></i><b>3.5</b> Multiple Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="3.6" data-path="introLinearReg.html"><a href="introLinearReg.html#exercise-with-the-real-estate-data"><i class="fa fa-check"></i><b>3.6</b> Exercise with the real estate data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introClassifier.html"><a href="introClassifier.html"><i class="fa fa-check"></i><b>4</b> Introduction to Classification</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introClassifier.html"><a href="introClassifier.html#visualise-logistic-and-logit-functions"><i class="fa fa-check"></i><b>4.1</b> Visualise logistic and logit functions</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="introClassifier.html"><a href="introClassifier.html#logistic-function"><i class="fa fa-check"></i><b>4.1.1</b> Logistic function</a></li>
<li class="chapter" data-level="4.1.2" data-path="introClassifier.html"><a href="introClassifier.html#logit-function"><i class="fa fa-check"></i><b>4.1.2</b> Logit function</a></li>
<li class="chapter" data-level="4.1.3" data-path="introClassifier.html"><a href="introClassifier.html#visualise-the-distribution"><i class="fa fa-check"></i><b>4.1.3</b> Visualise the distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="introClassifier.html"><a href="introClassifier.html#logistic-regression-on-diabetes"><i class="fa fa-check"></i><b>4.2</b> Logistic regression on Diabetes</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="introClassifier.html"><a href="introClassifier.html#load-pima-indians-diabetes-database"><i class="fa fa-check"></i><b>4.2.1</b> Load Pima Indians Diabetes Database</a></li>
<li class="chapter" data-level="4.2.2" data-path="introClassifier.html"><a href="introClassifier.html#fit-logistic-regression"><i class="fa fa-check"></i><b>4.2.2</b> Fit logistic regression</a></li>
<li class="chapter" data-level="4.2.3" data-path="introClassifier.html"><a href="introClassifier.html#assess-on-test-data"><i class="fa fa-check"></i><b>4.2.3</b> Assess on test data</a></li>
<li class="chapter" data-level="4.2.4" data-path="introClassifier.html"><a href="introClassifier.html#model-selection-and-diagnosis"><i class="fa fa-check"></i><b>4.2.4</b> Model selection and diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="introClassifier.html"><a href="introClassifier.html#cross-validation"><i class="fa fa-check"></i><b>4.3</b> Cross-validation</a></li>
<li class="chapter" data-level="4.4" data-path="introClassifier.html"><a href="introClassifier.html#more-assessment-metrics"><i class="fa fa-check"></i><b>4.4</b> More assessment metrics</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="introClassifier.html"><a href="introClassifier.html#two-types-of-error"><i class="fa fa-check"></i><b>4.4.1</b> Two types of error</a></li>
<li class="chapter" data-level="4.4.2" data-path="introClassifier.html"><a href="introClassifier.html#roc-curve"><i class="fa fa-check"></i><b>4.4.2</b> ROC curve</a></li>
<li class="chapter" data-level="4.4.3" data-path="introClassifier.html"><a href="introClassifier.html#homework"><i class="fa fa-check"></i><b>4.4.3</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Biomedical Data Modules</b></span></li>
<li class="chapter" data-level="5" data-path="image-digital.html"><a href="image-digital.html"><i class="fa fa-check"></i><b>5</b> Medical Image and Digital Health</a></li>
<li class="chapter" data-level="6" data-path="cancer.html"><a href="cancer.html"><i class="fa fa-check"></i><b>6</b> Cancer genomics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="cancer.html"><a href="cancer.html#cancer-case1"><i class="fa fa-check"></i><b>6.1</b> Case study 1: analysis of cBioportal mutation data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="cancer.html"><a href="cancer.html#exploratory-analysis"><i class="fa fa-check"></i><b>6.1.1</b> Exploratory analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="cancer.html"><a href="cancer.html#statistical-analysis"><i class="fa fa-check"></i><b>6.1.2</b> Statistical analysis</a></li>
<li class="chapter" data-level="6.1.3" data-path="cancer.html"><a href="cancer.html#literature-search"><i class="fa fa-check"></i><b>6.1.3</b> Literature search</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="epidemiology.html"><a href="epidemiology.html"><i class="fa fa-check"></i><b>7</b> Epidemiology</a></li>
<li class="chapter" data-level="8" data-path="pop-genetics.html"><a href="pop-genetics.html"><i class="fa fa-check"></i><b>8</b> Population Genetics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="pop-genetics.html"><a href="pop-genetics.html#case-study-1-heritability-and-human-traits"><i class="fa fa-check"></i><b>8.1</b> Case study 1: Heritability and human traits</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="pop-genetics.html"><a href="pop-genetics.html#part-1"><i class="fa fa-check"></i><b>8.1.1</b> Part 1</a></li>
<li class="chapter" data-level="8.1.2" data-path="pop-genetics.html"><a href="pop-genetics.html#part-2"><i class="fa fa-check"></i><b>8.1.2</b> Part 2</a></li>
<li class="chapter" data-level="8.1.3" data-path="pop-genetics.html"><a href="pop-genetics.html#references"><i class="fa fa-check"></i><b>8.1.3</b> References</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Modules in previous years</b></span></li>
<li class="chapter" data-level="9" data-path="cancer-case2.html"><a href="cancer-case2.html"><i class="fa fa-check"></i><b>9</b> Cancer Epidemiology</a>
<ul>
<li class="chapter" data-level="9.0.1" data-path="cancer-case2.html"><a href="cancer-case2.html#scenario"><i class="fa fa-check"></i><b>9.0.1</b> Scenario</a></li>
<li class="chapter" data-level="9.0.2" data-path="cancer-case2.html"><a href="cancer-case2.html#hong-kong-population"><i class="fa fa-check"></i><b>9.0.2</b> Hong Kong population</a></li>
<li class="chapter" data-level="9.0.3" data-path="cancer-case2.html"><a href="cancer-case2.html#cancer-registry-data"><i class="fa fa-check"></i><b>9.0.3</b> Cancer registry data</a></li>
<li class="chapter" data-level="9.0.4" data-path="cancer-case2.html"><a href="cancer-case2.html#existing-cancer-funding-and-publication-data"><i class="fa fa-check"></i><b>9.0.4</b> Existing cancer funding and publication data</a></li>
<li class="chapter" data-level="9.0.5" data-path="cancer-case2.html"><a href="cancer-case2.html#open-discussion"><i class="fa fa-check"></i><b>9.0.5</b> Open discussion</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="genetic-seq.html"><a href="genetic-seq.html"><i class="fa fa-check"></i><b>10</b> Genetic sequence</a>
<ul>
<li class="chapter" data-level="10.1" data-path="genetic-seq.html"><a href="genetic-seq.html#case-study-1-genetic-sequence-analysis"><i class="fa fa-check"></i><b>10.1</b> Case study 1: Genetic sequence analysis</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="genetic-seq.html"><a href="genetic-seq.html#sequence-motif-and-k-mer"><i class="fa fa-check"></i><b>10.1.1</b> Sequence motif and k-mer</a></li>
<li class="chapter" data-level="10.1.2" data-path="genetic-seq.html"><a href="genetic-seq.html#functional-mapping"><i class="fa fa-check"></i><b>10.1.2</b> Functional mapping</a></li>
<li class="chapter" data-level="10.1.3" data-path="genetic-seq.html"><a href="genetic-seq.html#references-1"><i class="fa fa-check"></i><b>10.1.3</b> References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="genomics-regression.html"><a href="genomics-regression.html"><i class="fa fa-check"></i><b>11</b> Genomics and prediction</a>
<ul>
<li class="chapter" data-level="11.1" data-path="genomics-regression.html"><a href="genomics-regression.html#genomics-case1"><i class="fa fa-check"></i><b>11.1</b> Case study 1: splicing fedility prediction</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="genomics-regression.html"><a href="genomics-regression.html#questions-for-discussion-1"><i class="fa fa-check"></i><b>11.1.1</b> Questions for Discussion</a></li>
<li class="chapter" data-level="11.1.2" data-path="genomics-regression.html"><a href="genomics-regression.html#hands-on-with-regression-model"><i class="fa fa-check"></i><b>11.1.2</b> Hands-on with regression model</a></li>
<li class="chapter" data-level="11.1.3" data-path="genomics-regression.html"><a href="genomics-regression.html#open-question"><i class="fa fa-check"></i><b>11.1.3</b> Open question</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i>Appendix A: Install R &amp; RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.1-install-r-4.3.1"><i class="fa fa-check"></i>A.1 Install R (&gt;=4.3.1)</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-windows"><i class="fa fa-check"></i>R on Windows</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-macos"><i class="fa fa-check"></i>R on macOS</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-linux-ubuntu"><i class="fa fa-check"></i>R on Linux (Ubuntu)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.2-install-rstudio"><i class="fa fa-check"></i>A.2 Install RStudio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.3-use-r-inside-rstudio"><i class="fa fa-check"></i>A.3 Use R inside RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-studio"><i class="fa fa-check"></i>R studio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#some-general-knowledge"><i class="fa fa-check"></i>Some general knowledge</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#install-packages"><i class="fa fa-check"></i>Install packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a4.-cloud-computing"><i class="fa fa-check"></i>A4. Cloud computing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-2.html"><a href="references-2.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Data Science - introduction with case studies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introHypoTest" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Introduction to Hypothesis test<a href="introHypoTest.html#introHypoTest" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="hypothesis-test-and-p-value" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Hypothesis test and <em>p</em>-value<a href="introHypoTest.html#hypothesis-test-and-p-value" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Hypothesis test is a method for statistical inference, specifically to
determine whether the collected data provide sufficient evidence to reject a
particular hypothesis.</p>
<div id="extreme-events-and-random-chance" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Extreme events and random chance<a href="introHypoTest.html#extreme-events-and-random-chance" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p><strong>Example 1:</strong> In 1930s, Sir Ronald Fisher did a famous experiment, “The Lady Tasting Tea”, which aimed to determine whether Lady Bristol could reliably distinguish whether milk or tea was added first when making English milk tea. In the experiment, eight cups—four with milk added first and four with tea added first—were shuffled before being presented to the lady. She correctly identified the order in all eight cups. This raises the question: does the lady genuinely have the ability to distinguish whether milk or tea was added first? In hypothesis testing, we can consider two hypotheses:<br />
* <span class="math inline">\(H_0\)</span>: The lady has no ability to distinguish between milk-first and tea-first; her correct identifications are due to chance.<br />
* <span class="math inline">\(H_A\)</span>: The lady has some ability to distinguish; her correct identifications are better than what would be expected by chance.</p>
</blockquote>
<p><strong>Table 1.</strong> Contingency table for the lady tasting tea experiment.</p>
<table>
<colgroup>
<col width="15%" />
<col width="26%" />
<col width="28%" />
<col width="29%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>Predicted tea first</th>
<th>Predicted milk first</th>
<th><strong>Total</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Tea first</td>
<td>a = 4</td>
<td>b = 0</td>
<td>a + b = 4</td>
</tr>
<tr class="even">
<td>Milk first</td>
<td>c = 0</td>
<td>d = 4</td>
<td>c + d = 4</td>
</tr>
<tr class="odd">
<td><strong>Total</strong></td>
<td>a +c = 4</td>
<td>b + d = 4</td>
<td>n = a + b + c + d = 8</td>
</tr>
</tbody>
</table>
<p><br /></p>
<p>Let’s consider a different perspective. How likely is it for this to happen purely by chance?</p>
<p>Using our mathematical skills (or <code>choose()</code> function in R), we find that there are 70 possible combinations. Out of these, only one combination results in an outcome as extreme as the lady’s identification. Therefore, if the results were purely random, the probability of observing a completely correct identification is 1 out of 70, or approximately 0.014.</p>
<p><strong>In another word, the chance we are fooled by randomness is 1/70 = 0.014.</strong></p>
<details>
<summary>
<strong>How to calculate the number of combinations with R and math:</strong>
</summary>
<p><strong>R</strong></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="introHypoTest.html#cb56-1" tabindex="-1"></a><span class="fu">choose</span>(<span class="dv">8</span>, <span class="dv">4</span>)</span>
<span id="cb56-2"><a href="introHypoTest.html#cb56-2" tabindex="-1"></a><span class="co">#&gt; [1] 70</span></span></code></pre></div>
<p><strong>Math</strong>
Consider permutations where the first choice is 8, the second is 7, and so on. However, we must also account for the permutations within the selected cups—that is, the arrangements among the chosen 4 cups. Therefore, the number of combinations can be calculated as follows.</p>
<span class="math display">\[\frac{8 \times 7 \times 6 \times 5} {4 \times 3 \times 2 \times 1} = 70\]</span>
</details>
<p><br /></p>
<blockquote>
<p><strong>Example 2:</strong> Your arch-nemesis, Blofeld, always seems to win at Ludo, leading you to suspect that he might be using a loaded die. You have observed the following outcomes from 100 rolls of his die:
<br />
Based on these observations, you want to evaluate two hypotheses:<br />
* <span class="math inline">\(H_0\)</span>: probability of rolling a 6 is p = 1/6<br />
* <span class="math inline">\(H_A\)</span>: probability of rolling a 6 is p &gt; 1/6</p>
</blockquote>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="introHypoTest.html#cb57-1" tabindex="-1"></a>data <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">3</span>, <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">2</span>,</span>
<span id="cb57-2"><a href="introHypoTest.html#cb57-2" tabindex="-1"></a>       <span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">3</span>,</span>
<span id="cb57-3"><a href="introHypoTest.html#cb57-3" tabindex="-1"></a>       <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">5</span>,</span>
<span id="cb57-4"><a href="introHypoTest.html#cb57-4" tabindex="-1"></a>       <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">3</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">6</span>,</span>
<span id="cb57-5"><a href="introHypoTest.html#cb57-5" tabindex="-1"></a>       <span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">4</span>, <span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">5</span>)</span>
<span id="cb57-6"><a href="introHypoTest.html#cb57-6" tabindex="-1"></a></span>
<span id="cb57-7"><a href="introHypoTest.html#cb57-7" tabindex="-1"></a>stat_k <span class="ot">=</span> <span class="fu">sum</span>(data <span class="sc">==</span> <span class="dv">6</span>)</span>
<span id="cb57-8"><a href="introHypoTest.html#cb57-8" tabindex="-1"></a>stat_k</span>
<span id="cb57-9"><a href="introHypoTest.html#cb57-9" tabindex="-1"></a><span class="co">#&gt; [1] 43</span></span></code></pre></div>
<p>After counting, you found that the number 6 appears 43 times out of 100 rolls. You then ask: what is the probability of observing this many or more sixes if the die is fair? Assuming the die is unbiased, you can calculate the probability of getting 6 for <span class="math inline">\(K\)</span> times out of 100 trials, via a <a href="https://en.wikipedia.org/wiki/Binomial_distribution">binomial distribution</a>.</p>
<p><span class="math display">\[\binom{100}{k} p^k (1-p)^{(100-k)}; p=\frac{1}{6}\]</span>
You can also calculate in R with this function <code>dbinom(k, size=100, prob=1/6)</code>.</p>
<p>Now, let’s visualize the probability of seeing any times of 6 via R scripts.</p>
<details>
<summary>
<strong>Visualizing with R:</strong>
</summary>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="introHypoTest.html#cb58-1" tabindex="-1"></a><span class="co"># plot the probability mass function of null distribution</span></span>
<span id="cb58-2"><a href="introHypoTest.html#cb58-2" tabindex="-1"></a></span>
<span id="cb58-3"><a href="introHypoTest.html#cb58-3" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">101</span>)</span>
<span id="cb58-4"><a href="introHypoTest.html#cb58-4" tabindex="-1"></a>pmf <span class="ot">=</span> <span class="fu">dbinom</span>(x, <span class="at">size=</span><span class="dv">100</span>, <span class="at">prob=</span><span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>)</span>
<span id="cb58-5"><a href="introHypoTest.html#cb58-5" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">x=</span>x, <span class="at">pmf=</span>pmf, <span class="at">extreme=</span>(x <span class="sc">&gt;=</span> stat_k))</span>
<span id="cb58-6"><a href="introHypoTest.html#cb58-6" tabindex="-1"></a></span>
<span id="cb58-7"><a href="introHypoTest.html#cb58-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb58-8"><a href="introHypoTest.html#cb58-8" tabindex="-1"></a></span>
<span id="cb58-9"><a href="introHypoTest.html#cb58-9" tabindex="-1"></a>fig4_dice <span class="ot">=</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>x)) <span class="sc">+</span> </span>
<span id="cb58-10"><a href="introHypoTest.html#cb58-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y=</span>pmf, <span class="at">color=</span>extreme)) <span class="sc">+</span></span>
<span id="cb58-11"><a href="introHypoTest.html#cb58-11" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;red&quot;</span>)) <span class="sc">+</span></span>
<span id="cb58-12"><a href="introHypoTest.html#cb58-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&#39;Number of sixes&#39;</span>) <span class="sc">+</span> </span>
<span id="cb58-13"><a href="introHypoTest.html#cb58-13" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&#39;Probability Mass Function&#39;</span>) <span class="sc">+</span></span>
<span id="cb58-14"><a href="introHypoTest.html#cb58-14" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&#39;Distribution of n_six under the null hypothesis&#39;</span>)</span></code></pre></div>
</details>
<p><br /></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="introHypoTest.html#cb59-1" tabindex="-1"></a>fig4_dice</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-dice-1.png" width="60%" /></p>
<p>Here, if we sum up the probability of the events at least as extreme as we observed, namely seeing 43 or more sixes, we can have a probability (or <code>p-value</code>; see definition below), using math or R function <code>1 - pbinom(42, 100, 1/6)</code>.</p>
<p><span class="math display">\[
p(k \geq 43 | p=1/6) = \sum_{k=34}^{n=100} \binom{100}{k} p^k (1-p)^{(100-k)} = 5.4\times 10^{-10}
\]</span></p>
<details>
<summary>
<strong>Calculating with R:</strong>
</summary>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="introHypoTest.html#cb60-1" tabindex="-1"></a>?pbinom</span>
<span id="cb60-2"><a href="introHypoTest.html#cb60-2" tabindex="-1"></a><span class="dv">1</span> <span class="sc">-</span> <span class="fu">pbinom</span>(<span class="dv">42</span>, <span class="dv">100</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>)</span>
<span id="cb60-3"><a href="introHypoTest.html#cb60-3" tabindex="-1"></a><span class="co">#&gt; [1] 5.439087e-10</span></span></code></pre></div>
</details>
<p><br /></p>
<p>Therefore, you may think the chance to see such extreme results is very low (5.4e-10)!</p>
<p><strong>Random chance to blame:</strong>
The purpose of hypothesis test is to help us learn whether random chance might be responsible for observations. Think:</p>
<ul>
<li><p>In Example 1, can the lady genuinely distinguish whether the milk or tea was poured first? To what extent should we attribute her correct identification to random chance?</p></li>
<li><p>In Example 2, how much should we consider the occurrence of 43 or more sixes as a result by random?</p></li>
</ul>
<p><strong>N.B.</strong>, random chance is random but not always in a uniform or normal distribution. The distribution sometimes can be quite complex.</p>
</div>
<div id="hypothesis-test-for-statistical-decision" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Hypothesis test for statistical decision<a href="introHypoTest.html#hypothesis-test-for-statistical-decision" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Is there a formal way for us to make such a statistical decision?
<strong>Yes, hypothesis testing is a formal way for decision making</strong>. Let us see how it is defined.</p>
<blockquote>
<p><strong>Definition of hypothesis test:</strong> A hypothesis test is a formal procedure in statistical inference used to assess whether the observed data provide sufficient evidence to reject a specific null hypothesis. The process typically involves calculating a test statistic, which is a scalar value derived from the data. This observed test statistic is then compared to its distribution under the null hypothesis, or alternatively, a p-value is computed from the test statistic. Based on this comparison or p-value, a decision is made to either reject or fail to reject the null hypothesis.</p>
</blockquote>
<p><strong>Null and alternative hypotheses:</strong></p>
<ul>
<li><p>Null hypothesis (<span class="math inline">\(H_0\)</span>): The hypothesis that random chance is to blame.</p></li>
<li><p>Alternative hypothesis (<span class="math inline">\(H_1\)</span> or <span class="math inline">\(H_a\)</span>): Counterpart to the null; namely the hypothesis you want to prove.</p></li>
</ul>
<blockquote>
<p><strong>Example 3</strong>, Consider an A/B test examining COVID-19 recovery times, comparing patients who receive drug D (Group A) to those who do not (Group B). The question is: can drug D actually reduce recovery time? Or, alternatively, is the observed difference between the two groups simply due to random chance?
<br />
* <span class="math inline">\(H_0\)</span>: The drug has no effect on Covid-19 recovery time, i.e., <span class="math inline">\(\mu_A = \mu_B\)</span><br />
* <span class="math inline">\(H_1\)</span>: The drug does have an effect on recovery time, i.e., <span class="math inline">\(\mu_A \neq \mu_B\)</span></p>
</blockquote>
<p><strong>One reason why we need hypothesis test</strong></p>
<ul>
<li><p>We have no idea of the testing statistic in the alternative hypothesis (the “right”), but we can know how it looks when the null hypothesis is correct.</p></li>
<li><p>If we know both, we may consider classification - think the difference.</p></li>
</ul>
<p>Therefore, the main goal of hypothesis testing is: <em>Given our null and alternative hypotheses, we aim to determine whether the observed data provides enough evidence to reject the null hypothesis</em>. This involves three key steps::</p>
<ol style="list-style-type: decimal">
<li><p>Define the test statistic. First, select an appropriate <strong>test statistic</strong>, such as the difference in means between groups A and B, the difference in medians, or the variance of the group means.</p></li>
<li><p>Approximate the null distribution. Next, under the null hypothesis, determine the distribution of the test statistic—either through resampling methods or analytical calculations—referred to as the <strong>null distribution</strong>.</p></li>
<li><p>Calculte the p-value. Last, from the the null distribution we can calculate the probabilityof seeing the test statistic at least as extreme as the observed value, termed as <strong>p-value</strong>.</p></li>
</ol>
<p><strong>Definition of p-value:</strong> Finally, compute the probability of observing a test statistic as extreme or more extreme than the one obtained, assuming the null hypothesis is true. This probability is known as the <strong>p-value</strong>.</p>
<p>The p-value represents the probability of observing data as extreme or more extreme than what was actually observed, assuming the null hypothesis is true. In other words, it measures how surprising the observed data is under the null hypothesis.</p>
<p>Therefore, the p-value can help guide our statistical decision:</p>
<ul>
<li><p>If the p-value is very small, it indicates that such an extreme result would be unlikely if the null hypothesis were true, suggesting that the null hypothesis may be false.</p></li>
<li><p>If the p-value is large, it implies that the observed data is consistent with the null hypothesis, providing <em>not enough evidence to reject it</em>.</p></li>
</ul>
<p><strong>Definition of significance level:</strong> significance level, denoted by <span class="math inline">\(\alpha\)</span>, is the probability of rejecting the null hypothesis when it is actually true. In other words, it represents the threshold for determining when a result is considered statistically significant.</p>
<p>In practical terms, statistical significance measures whether an observed result is more extreme than what would typically be expected due to chance alone.</p>
<ul>
<li><p>The significance level is <span class="math inline">\(\alpha\)</span> (Alpha) predefined cutoff for “unusualness.” If the p-value falls below , the observed result is deemed statistically significant, suggesting that it is unlikely to have occurred under the null hypothesis.</p></li>
<li><p>We reject reject <span class="math inline">\(H_0\)</span> when <span class="math inline">\(p&lt;\alpha\)</span>. Since is the probability of making a Type I error (incorrectly rejecting a true null hypothesis), choosing a smaller <span class="math inline">\(\alpha\)</span> makes the test more conservative.</p></li>
<li><p>Example: if <span class="math inline">\(\alpha = 0.05, p = 0.004\)</span>, then since <span class="math inline">\(p &lt; \alpha\)</span>, we reject <span class="math inline">\(H_0\)</span>.</p></li>
</ul>
<p><strong>Key steps in performing a hypothesis test</strong></p>
<ol style="list-style-type: decimal">
<li>Propose a research question;</li>
<li>Formulate the null hypothesis H0and alternative hypothesis <span class="math inline">\(H_1\)</span>;</li>
<li>Choose an appropriate statistical test, including defining the <strong>test statistic</strong> and its <strong>null distribution</strong>;</li>
<li>Choose an appropriate significance level, <span class="math inline">\(\alpha\)</span>;</li>
<li>Calculate the test statistic based on the observed data;</li>
<li>Compute the p-value corresponding to the test statistic;</li>
<li>Make a decision: reject <span class="math inline">\(H_0\)</span> if <span class="math inline">\(p &lt; \alpha\)</span>, otherwise, there is not enough evidence to reject <span class="math inline">\(H_0\)</span>.</li>
</ol>
</div>
</div>
<div id="basic-hypothesis-test-methods" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Basic hypothesis test methods<a href="introHypoTest.html#basic-hypothesis-test-methods" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A crucial step in conducting a hypothesis test is selecting an appropriate method, which involves defining the test statistic and approximating the null distribution. Here, we will introduce several commonly used methods as an overview.</p>
<div id="permutation-test" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Permutation test<a href="introHypoTest.html#permutation-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The permutation test is a resampling method that involves sampling with replacement to approximate the null distribution. In this context, we will use an A/B test as an example, where the goal is to determine whether the means differ between groups A and B. Let’s begin with the following example.</p>
<blockquote>
<p><strong>Example 4</strong>: The difference in birth weights between babies born to nonsmoking and heavy-smoking mothers was analyzed. Birth weights (in kilograms) were measured for a sample of mothers divided into two groups: nonsmokers (n=13) and heavy smokers (n=15).</p>
</blockquote>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="introHypoTest.html#cb61-1" tabindex="-1"></a>data_heavysmoking <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">3.18</span>, <span class="fl">2.84</span>, <span class="fl">2.90</span>, <span class="fl">3.27</span>, <span class="fl">3.85</span>, <span class="fl">3.52</span>, <span class="fl">3.23</span>, <span class="fl">2.76</span>, </span>
<span id="cb61-2"><a href="introHypoTest.html#cb61-2" tabindex="-1"></a>                      <span class="fl">3.60</span>, <span class="fl">3.75</span>, <span class="fl">3.59</span>, <span class="fl">3.63</span>, <span class="fl">2.38</span>, <span class="fl">2.34</span>, <span class="fl">2.44</span>) </span>
<span id="cb61-3"><a href="introHypoTest.html#cb61-3" tabindex="-1"></a>data_nonsmoking   <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">3.99</span>, <span class="fl">3.79</span>, <span class="fl">3.60</span>, <span class="fl">3.73</span>, <span class="fl">3.21</span>, <span class="fl">3.60</span>, <span class="fl">4.08</span>, <span class="fl">3.61</span>, </span>
<span id="cb61-4"><a href="introHypoTest.html#cb61-4" tabindex="-1"></a>                      <span class="fl">3.83</span>, <span class="fl">3.31</span>, <span class="fl">4.13</span>, <span class="fl">3.26</span>, <span class="fl">3.54</span>)</span>
<span id="cb61-5"><a href="introHypoTest.html#cb61-5" tabindex="-1"></a><span class="fu">c</span>(<span class="fu">mean</span>(data_nonsmoking), <span class="fu">mean</span>(data_heavysmoking))</span>
<span id="cb61-6"><a href="introHypoTest.html#cb61-6" tabindex="-1"></a><span class="co">#&gt; [1] 3.667692 3.152000</span></span></code></pre></div>
<p>Now, let’s apply the seven steps outlined in the previous section to perform a hypothesis test.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="introHypoTest.html#cb62-1" tabindex="-1"></a><span class="do">## 1) Research question: </span></span>
<span id="cb62-2"><a href="introHypoTest.html#cb62-2" tabindex="-1"></a><span class="co"># We want to know whether there is a significant difference in mean birth weight</span></span>
<span id="cb62-3"><a href="introHypoTest.html#cb62-3" tabindex="-1"></a><span class="co"># between the two groups.</span></span>
<span id="cb62-4"><a href="introHypoTest.html#cb62-4" tabindex="-1"></a></span>
<span id="cb62-5"><a href="introHypoTest.html#cb62-5" tabindex="-1"></a><span class="do">## 2) Write down the hypotheses</span></span>
<span id="cb62-6"><a href="introHypoTest.html#cb62-6" tabindex="-1"></a><span class="co"># H0: there is no difference in mean birth weight between groups: d == 0</span></span>
<span id="cb62-7"><a href="introHypoTest.html#cb62-7" tabindex="-1"></a><span class="co"># H1: there is a difference, d != 0</span></span>
<span id="cb62-8"><a href="introHypoTest.html#cb62-8" tabindex="-1"></a></span>
<span id="cb62-9"><a href="introHypoTest.html#cb62-9" tabindex="-1"></a><span class="do">## 3) Testing method</span></span>
<span id="cb62-10"><a href="introHypoTest.html#cb62-10" tabindex="-1"></a><span class="co"># Method: Permutation test</span></span>
<span id="cb62-11"><a href="introHypoTest.html#cb62-11" tabindex="-1"></a><span class="co"># Test statistic: difference of group means</span></span>
<span id="cb62-12"><a href="introHypoTest.html#cb62-12" tabindex="-1"></a><span class="co"># Approximate null distribution: resampling method</span></span>
<span id="cb62-13"><a href="introHypoTest.html#cb62-13" tabindex="-1"></a></span>
<span id="cb62-14"><a href="introHypoTest.html#cb62-14" tabindex="-1"></a><span class="do">## 4) Choose a significance level</span></span>
<span id="cb62-15"><a href="introHypoTest.html#cb62-15" tabindex="-1"></a><span class="co"># alpha = 0.05</span></span>
<span id="cb62-16"><a href="introHypoTest.html#cb62-16" tabindex="-1"></a></span>
<span id="cb62-17"><a href="introHypoTest.html#cb62-17" tabindex="-1"></a><span class="do">## 5) Calculate test statistic</span></span>
<span id="cb62-18"><a href="introHypoTest.html#cb62-18" tabindex="-1"></a><span class="co"># \mu = \mu_A - \mu_B = -0.51</span></span>
<span id="cb62-19"><a href="introHypoTest.html#cb62-19" tabindex="-1"></a></span>
<span id="cb62-20"><a href="introHypoTest.html#cb62-20" tabindex="-1"></a><span class="do">## 6) Compute the p-value</span></span>
<span id="cb62-21"><a href="introHypoTest.html#cb62-21" tabindex="-1"></a><span class="co"># To be determined below</span></span>
<span id="cb62-22"><a href="introHypoTest.html#cb62-22" tabindex="-1"></a></span>
<span id="cb62-23"><a href="introHypoTest.html#cb62-23" tabindex="-1"></a><span class="do">## 7) Make a decision</span></span>
<span id="cb62-24"><a href="introHypoTest.html#cb62-24" tabindex="-1"></a><span class="co"># To be determined below, depending on the p-value</span></span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="introHypoTest.html#cb63-1" tabindex="-1"></a><span class="do">## 5) Calculate test statistic: difference of group mean</span></span>
<span id="cb63-2"><a href="introHypoTest.html#cb63-2" tabindex="-1"></a>stat_mu <span class="ot">=</span> <span class="fu">mean</span>(data_heavysmoking) <span class="sc">-</span> <span class="fu">mean</span>(data_nonsmoking)</span>
<span id="cb63-3"><a href="introHypoTest.html#cb63-3" tabindex="-1"></a>stat_mu</span>
<span id="cb63-4"><a href="introHypoTest.html#cb63-4" tabindex="-1"></a><span class="co">#&gt; [1] -0.5156923</span></span></code></pre></div>
<div id="resampling-method-approximates-null-distribution" class="section level4 hasAnchor" number="2.2.1.1">
<h4><span class="header-section-number">2.2.1.1</span> Resampling method approximates null distribution<a href="introHypoTest.html#resampling-method-approximates-null-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here, we will introduce a sampling-based method to approximate the null distribution. To facilitate reuse, we will implement it as an
<a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/function">R function</a>.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="introHypoTest.html#cb64-1" tabindex="-1"></a><span class="co">#&#39; Simple function to generate permutation distribution</span></span>
<span id="cb64-2"><a href="introHypoTest.html#cb64-2" tabindex="-1"></a><span class="co">#&#39; @param x1 A vector with length n1, the observed values in group 1</span></span>
<span id="cb64-3"><a href="introHypoTest.html#cb64-3" tabindex="-1"></a><span class="co">#&#39; @param x2 A vector with length n2, the observed values in group 2</span></span>
<span id="cb64-4"><a href="introHypoTest.html#cb64-4" tabindex="-1"></a><span class="co">#&#39; @param n_permute A scalar, the number of permutation to perform</span></span>
<span id="cb64-5"><a href="introHypoTest.html#cb64-5" tabindex="-1"></a><span class="co">#&#39; @return a vector of n_permute mean differences between two permuted groups</span></span>
<span id="cb64-6"><a href="introHypoTest.html#cb64-6" tabindex="-1"></a>get_permutation_null <span class="ot">&lt;-</span> <span class="cf">function</span>(x1, x2, <span class="at">n_permute=</span><span class="dv">1000</span>) {</span>
<span id="cb64-7"><a href="introHypoTest.html#cb64-7" tabindex="-1"></a>  n1 <span class="ot">=</span> <span class="fu">length</span>(x1)</span>
<span id="cb64-8"><a href="introHypoTest.html#cb64-8" tabindex="-1"></a>  n2 <span class="ot">=</span> <span class="fu">length</span>(x2)</span>
<span id="cb64-9"><a href="introHypoTest.html#cb64-9" tabindex="-1"></a>  </span>
<span id="cb64-10"><a href="introHypoTest.html#cb64-10" tabindex="-1"></a>  <span class="co"># pool data sets</span></span>
<span id="cb64-11"><a href="introHypoTest.html#cb64-11" tabindex="-1"></a>  x_pool <span class="ot">=</span> <span class="fu">c</span>(x1, x2)</span>
<span id="cb64-12"><a href="introHypoTest.html#cb64-12" tabindex="-1"></a>  </span>
<span id="cb64-13"><a href="introHypoTest.html#cb64-13" tabindex="-1"></a>  null_distr <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">0</span>, n_permute)</span>
<span id="cb64-14"><a href="introHypoTest.html#cb64-14" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq</span>(n_permute)) {</span>
<span id="cb64-15"><a href="introHypoTest.html#cb64-15" tabindex="-1"></a>    <span class="co"># split</span></span>
<span id="cb64-16"><a href="introHypoTest.html#cb64-16" tabindex="-1"></a>    idx <span class="ot">=</span> <span class="fu">sample</span>(n1 <span class="sc">+</span> n2, <span class="at">size=</span>n1)</span>
<span id="cb64-17"><a href="introHypoTest.html#cb64-17" tabindex="-1"></a>    x1_perm <span class="ot">=</span> x_pool[idx]</span>
<span id="cb64-18"><a href="introHypoTest.html#cb64-18" tabindex="-1"></a>    x2_perm <span class="ot">=</span> x_pool[<span class="sc">-</span>idx]</span>
<span id="cb64-19"><a href="introHypoTest.html#cb64-19" tabindex="-1"></a>    </span>
<span id="cb64-20"><a href="introHypoTest.html#cb64-20" tabindex="-1"></a>    <span class="co"># calculate test statistic</span></span>
<span id="cb64-21"><a href="introHypoTest.html#cb64-21" tabindex="-1"></a>    null_distr[i] <span class="ot">=</span> <span class="fu">mean</span>(x1_perm) <span class="sc">-</span> <span class="fu">mean</span>(x2_perm)</span>
<span id="cb64-22"><a href="introHypoTest.html#cb64-22" tabindex="-1"></a>  }</span>
<span id="cb64-23"><a href="introHypoTest.html#cb64-23" tabindex="-1"></a>  </span>
<span id="cb64-24"><a href="introHypoTest.html#cb64-24" tabindex="-1"></a>  <span class="fu">return</span>(null_distr)</span>
<span id="cb64-25"><a href="introHypoTest.html#cb64-25" tabindex="-1"></a>}</span></code></pre></div>
<p>To visualize the null distribution and the test statistic, let’s run the <code>get_permutation_null()</code> function.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="introHypoTest.html#cb65-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb65-2"><a href="introHypoTest.html#cb65-2" tabindex="-1"></a>perm_null <span class="ot">=</span> <span class="fu">get_permutation_null</span>(data_heavysmoking, data_nonsmoking)</span>
<span id="cb65-3"><a href="introHypoTest.html#cb65-3" tabindex="-1"></a><span class="fu">head</span>(perm_null, <span class="dv">5</span>)</span>
<span id="cb65-4"><a href="introHypoTest.html#cb65-4" tabindex="-1"></a><span class="co">#&gt; [1]  0.004102564 -0.087794872  0.096000000 -0.229948718  0.216615385</span></span></code></pre></div>
<p>We can visualize the null distribution by plotting its histogram obtained through resampling. Additionally, we can add lines indicating the observed statistic value (e.g., mu) and any other extreme values. Depending on the hypothesis test, we may consider one tail or both tails as representing extreme values.</p>
<details>
<summary>
<strong>Click for R scripts for histogram and a vertical line - try it yourself first</strong>
</summary>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="introHypoTest.html#cb66-1" tabindex="-1"></a>df_perm <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">perm_null =</span> perm_null)</span>
<span id="cb66-2"><a href="introHypoTest.html#cb66-2" tabindex="-1"></a></span>
<span id="cb66-3"><a href="introHypoTest.html#cb66-3" tabindex="-1"></a>fig4_perm_twoside <span class="ot">=</span> <span class="fu">ggplot</span>(df_perm, <span class="fu">aes</span>(<span class="at">x=</span>perm_null)) <span class="sc">+</span> </span>
<span id="cb66-4"><a href="introHypoTest.html#cb66-4" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins=</span><span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb66-5"><a href="introHypoTest.html#cb66-5" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span>stat_mu, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;tomato&quot;</span>) <span class="sc">+</span></span>
<span id="cb66-6"><a href="introHypoTest.html#cb66-6" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="sc">-</span>stat_mu, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;tomato&quot;</span>) <span class="sc">+</span></span>
<span id="cb66-7"><a href="introHypoTest.html#cb66-7" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&#39;Difference of group mean&#39;</span>) <span class="sc">+</span></span>
<span id="cb66-8"><a href="introHypoTest.html#cb66-8" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&#39;Resampling frequency&#39;</span>) <span class="sc">+</span></span>
<span id="cb66-9"><a href="introHypoTest.html#cb66-9" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&#39;Distribution of mu under the null hypothesis&#39;</span>)</span></code></pre></div>
</details>
<p><br /></p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="introHypoTest.html#cb67-1" tabindex="-1"></a>fig4_perm_twoside</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-perm-twoside-1.png" width="60%" /></p>
<p>Now, we can finish the above steps 6 and 7 by computing the p-value=0.003 and make the decision that we reject the null hypothesis.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="introHypoTest.html#cb68-1" tabindex="-1"></a>p_value <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">abs</span>(perm_null) <span class="sc">&gt;=</span> <span class="fu">abs</span>(stat_mu))</span>
<span id="cb68-2"><a href="introHypoTest.html#cb68-2" tabindex="-1"></a>p_value</span>
<span id="cb68-3"><a href="introHypoTest.html#cb68-3" tabindex="-1"></a><span class="co">#&gt; [1] 0.003</span></span></code></pre></div>
</div>
<div id="two-tailed-or-one-tailed-alternative-hypothesis" class="section level4 hasAnchor" number="2.2.1.2">
<h4><span class="header-section-number">2.2.1.2</span> Two-tailed or one-tailed alternative hypothesis<a href="introHypoTest.html#two-tailed-or-one-tailed-alternative-hypothesis" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In the above case, we consider both <span class="math inline">\(\mu&lt;-0.51\)</span> and <span class="math inline">\(\mu&gt;0.51\)</span> as extreme events relative to the observed value. However, in some situations, we may be interested in extreme events only in one direction— for example, <span class="math inline">\(\mu&lt;-0.51\)</span>. In such cases, the p-value is calculated based on a one-tailed test.</p>
<p>There are more specific definitions for one-tailed and two-tailed p-values, and below is how we can compute the one-tailed p-value:</p>
<p><strong>Definition of one-tailed and two-tailed p-values</strong></p>
<ul>
<li>One-tailed: Corresponds to the probability of observing a test statistic as extreme as or more extreme than the observed value in a specific direction, based on the alternative hypothesis (e.g., <span class="math inline">\(H_a: \mu &lt; 0\)</span>).</li>
<li>Two-tailed: Corresponds to the probability of observing a test statistic as extreme as or more extreme than the observed value in either direction, without specifying a direction in the alternative hypothesis (e.g., <span class="math inline">\(H_a: \mu \neq 0\)</span>).</li>
</ul>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="introHypoTest.html#cb69-1" tabindex="-1"></a><span class="do">## Two tailed p value</span></span>
<span id="cb69-2"><a href="introHypoTest.html#cb69-2" tabindex="-1"></a>p_two_tailed <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">abs</span>(perm_null) <span class="sc">&gt;=</span> <span class="fu">abs</span>(stat_mu))</span>
<span id="cb69-3"><a href="introHypoTest.html#cb69-3" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Two tailed p value:&quot;</span>, <span class="fu">round</span>(p_two_tailed, <span class="dv">5</span>)))</span>
<span id="cb69-4"><a href="introHypoTest.html#cb69-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Two tailed p value: 0.003&quot;</span></span>
<span id="cb69-5"><a href="introHypoTest.html#cb69-5" tabindex="-1"></a></span>
<span id="cb69-6"><a href="introHypoTest.html#cb69-6" tabindex="-1"></a><span class="do">## Left-tailed p value</span></span>
<span id="cb69-7"><a href="introHypoTest.html#cb69-7" tabindex="-1"></a>p_one_tailed <span class="ot">=</span> <span class="fu">mean</span>(perm_null <span class="sc">&lt;</span> stat_mu)</span>
<span id="cb69-8"><a href="introHypoTest.html#cb69-8" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;One (left) tailed p value:&quot;</span>, <span class="fu">round</span>(p_one_tailed, <span class="dv">5</span>)))</span>
<span id="cb69-9"><a href="introHypoTest.html#cb69-9" tabindex="-1"></a><span class="co">#&gt; [1] &quot;One (left) tailed p value: 0.002&quot;</span></span></code></pre></div>
</div>
</div>
<div id="t-test-and-regression-based-test" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> <em>t</em>-test and regression-based test<a href="introHypoTest.html#t-test-and-regression-based-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In the above example, we observe how resampling methods enable us to approximate the null distribution of the difference between group means. Conversely, there are also analytical methods that can directly calculate the null distribution, often based on certain assumptions.</p>
<ul>
<li>Resampling methods, like permutation test, are <em>one-size-fits-all</em> methods and become increasingly popular with advances in computational power;</li>
<li>Analytic methods, or formula-based approaches, are generally faster and precise when the underlying assumptions are reasonably met and not heavily violated.</li>
</ul>
<p>In this section, we will introduce the t-test, which uses the t-distribution to approximate the null distribution. The t-distribution was first described in a paper published in <a href="https://www.jstor.org/stable/2331554"><em>Biometrika</em> in 1908</a> by an author under the pseudonym “Student,” which is why it is also known as Student’s t-test <a href="https://en.wikipedia.org/wiki/Student%27s_t-distribution">Student’s t-distribution</a>.</p>
<p>We will not delve into the mathematical details of the t-distribution here. However, to summarize, it is a distribution that resembles and extends the Gaussian (normal) distribution by incorporating an additional parameter called the degree of freedom (df). Typically, the degree of freedom represents the number of samples minus the number of parameters estimated, meaning how many variables can independently change.</p>
<p>When the degree of freedom is low (e.g., df=3), the t-distribution has longer tails than the normal distribution. As the degree of freedom increases, the t-distribution gradually approaches the Gaussian distribution, with minimal differences observed when df≥30.</p>
<p>In R, there are multiple implementations of t distribution. The most common one is
<a href="https://stat.ethz.ch/R-manual/R-devel/library/stats/html/TDist.html">?stats::TDist</a>,
for a standardized the t distribution (i.e., mean: 0, std: 1).
For non-standardized one with supporting location and scale parameters, we may consider using
<a href="https://cran.r-project.org/web/packages/extraDistr/refman/extraDistr.html#LocationScaleT">?extraDistr::LocationScaleT</a>.</p>
<details>
<summary>
<strong>Click for R scripts to plot t-distributions</strong>
</summary>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="introHypoTest.html#cb70-1" tabindex="-1"></a><span class="do">## Comparison between standard t and normal distributions.</span></span>
<span id="cb70-2"><a href="introHypoTest.html#cb70-2" tabindex="-1"></a><span class="co"># for non-standardized variable y = (x - loc) / scale, its pdf is dt(y, df) / scale</span></span>
<span id="cb70-3"><a href="introHypoTest.html#cb70-3" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">4</span>, <span class="fl">0.01</span>)</span>
<span id="cb70-4"><a href="introHypoTest.html#cb70-4" tabindex="-1"></a>p1 <span class="ot">=</span> <span class="fu">dt</span>(x, <span class="at">df=</span><span class="dv">1</span>)</span>
<span id="cb70-5"><a href="introHypoTest.html#cb70-5" tabindex="-1"></a>p3 <span class="ot">=</span> <span class="fu">dt</span>(x, <span class="at">df=</span><span class="dv">3</span>)</span>
<span id="cb70-6"><a href="introHypoTest.html#cb70-6" tabindex="-1"></a>p10 <span class="ot">=</span> <span class="fu">dt</span>(x, <span class="at">df=</span><span class="dv">10</span>)</span>
<span id="cb70-7"><a href="introHypoTest.html#cb70-7" tabindex="-1"></a>p30 <span class="ot">=</span> <span class="fu">dt</span>(x, <span class="at">df=</span><span class="dv">30</span>)</span>
<span id="cb70-8"><a href="introHypoTest.html#cb70-8" tabindex="-1"></a>p_z <span class="ot">=</span> <span class="fu">dnorm</span>(x)</span>
<span id="cb70-9"><a href="introHypoTest.html#cb70-9" tabindex="-1"></a>models <span class="ot">=</span> <span class="fu">factor</span>(</span>
<span id="cb70-10"><a href="introHypoTest.html#cb70-10" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;t(df=1)&quot;</span>, <span class="st">&quot;t(df=3)&quot;</span>, <span class="st">&quot;t(df=10)&quot;</span>, <span class="st">&quot;t(df=30)&quot;</span>, <span class="st">&quot;normal&quot;</span>), <span class="at">each =</span> <span class="fu">length</span>(x)),</span>
<span id="cb70-11"><a href="introHypoTest.html#cb70-11" tabindex="-1"></a>  <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;t(df=1)&quot;</span>, <span class="st">&quot;t(df=3)&quot;</span>, <span class="st">&quot;t(df=10)&quot;</span>, <span class="st">&quot;t(df=30)&quot;</span>, <span class="st">&quot;normal&quot;</span>)</span>
<span id="cb70-12"><a href="introHypoTest.html#cb70-12" tabindex="-1"></a>)</span>
<span id="cb70-13"><a href="introHypoTest.html#cb70-13" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">x=</span><span class="fu">rep</span>(x, <span class="dv">5</span>), <span class="at">PDF=</span><span class="fu">c</span>(p1, p3, p10, p30, p_z), <span class="at">models =</span> models)</span>
<span id="cb70-14"><a href="introHypoTest.html#cb70-14" tabindex="-1"></a>fig_tdist <span class="ot">=</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> PDF, <span class="at">color =</span> models)) <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="dv">1</span>) </span></code></pre></div>
</details>
<p><br /></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="introHypoTest.html#cb71-1" tabindex="-1"></a>fig_tdist</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-t-vs-normal-1.png" width="60%" /></p>
<div id="null-distribution-approximated-by-t-distribution" class="section level4 hasAnchor" number="2.2.2.1">
<h4><span class="header-section-number">2.2.2.1</span> Null distribution approximated by <span class="math inline">\(t\)</span> distribution<a href="introHypoTest.html#null-distribution-approximated-by-t-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s revisit Example 4, where our goal is to approximate the null distribution of the difference between group means, this time using the t-distribution.</p>
<p>Before proceeding, it is important to acknowledge certain assumptions necessary for applying the t-distribution for this approximation:</p>
<ol style="list-style-type: decimal">
<li>The two groups (non-smokers and heavy-smokers) are measured independently of each other.</li>
<li>Both groups are drawn from normal distributions.</li>
<li>The two groups are assumed to have the same unknown variance.</li>
</ol>
<p><strong>N.B.,</strong> Assumption 2) can be relaxed if the sample sizes are sufficiently large, for example, greater than 30. This is because we are comparing the means of the two groups instead of the samples themselves, and with larger samples, the normality assumption becomes less critical, as guaranteed by the <em>Central limit theorem</em>.</p>
<p><strong>Law of large number and Central limit theorem</strong></p>
<p>In simple terms, the <a href="https://en.wikipedia.org/wiki/Law_of_large_numbers">Law of large number</a> states that the sample mean will <strong>converge</strong> to the true population mean as the sample size increases. Additionally, the <a href="https://en.wikipedia.org/wiki/Central_limit_theorem">Central limit theorem</a>
indicates that the sample mean will have a <strong>distribution</strong> asymptotically to a normal distribution, centered at the population mean <span class="math inline">\(\mu\)</span>, with a variance of <span class="math inline">\(\sigma^2 / n\)</span>, where <span class="math inline">\(\sigma^2\)</span> is the population variance and <span class="math inline">\(n\)</span> is the sample size.</p>
<p>However, when the sample size is small—typically less than 30—Assumption 2) becomes more critical. Moreover, in such low-sample-size cases, using the t-distribution to approximate the distribution of the sample mean is more accurate than using a normal distribution.</p>
<blockquote>
<p><strong>Thinking:</strong> Why do we talk about the distribution of the sample mean? You might ask this since, in practice, we only have a single sample set <span class="math inline">\(S^{(1)} = \{X_1, X_2, ..., X_n\}\)</span>, which yields one sample mean.
When we refer to the distribution of the sample mean, we are actually considering many hypothetical sample sets of the same size—say, <span class="math inline">\(S^{(1)}, S^{(2)}, ...\)</span>—all drawn from the same population. The distribution of these sample means describes the variability we would observe across many such samples.</p>
</blockquote>
</div>
<div id="one-group-t-test" class="section level4 hasAnchor" number="2.2.2.2">
<h4><span class="header-section-number">2.2.2.2</span> One-group t-test<a href="introHypoTest.html#one-group-t-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If we only have one group <span class="math inline">\(S = \{X_1, X_2, ..., X_n\}\)</span>— for example, the nonsmoking group — and we want to test whether the population mean is equal to a specific value <span class="math inline">\(\mu_0\)</span> (for example 3.2kg), we can directly apply the <em>Central Limit Theorem</em> to approximate the distribution of the sample mean.
In a hypothesis testing framework, we set up the null and alternative hypotheses as follows:</p>
<ul>
<li><span class="math inline">\(H_0\)</span>: <span class="math inline">\(\mu - \mu_0 = 0\)</span> (the population mean equals to <span class="math inline">\(\mu_0\)</span>);</li>
<li><span class="math inline">\(H_a\)</span>: <span class="math inline">\(\mu - \mu_0 \neq 0\)</span> (the population mean does not equal to <span class="math inline">\(\mu_0\)</span>)</li>
</ul>
<p>Our test statistic is <span class="math inline">\(\bar{X} - \mu_0\)</span>, with an observed value—for example, 0.468. If the null hypothesis is true, we can approximate the null distribution with a t-distribution: <span class="math inline">\(\mu - \mu_0 \sim t(0, \hat\sigma^2, \text{df}=n-1)\)</span>, where we only need to estimate the <span class="math inline">\(\hat{\sigma}\)</span>. When the sample size <span class="math inline">\(n\)</span> is large, the Central Limit Theorem justifies using this approximation.</p>
<p><span class="math display">\[\hat{\sigma}^2 \approx \frac{\mathtt{Var}(X)}{n} = \frac{\frac{1}{n-1} \sum_{i=1}^n{(X_i-\bar{X})}^2}{n}\]</span></p>
<details>
<summary>
<strong>Click for R scripts to plot one-group null distribution via t distribution</strong>
</summary>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="introHypoTest.html#cb72-1" tabindex="-1"></a><span class="do">## Install extraDistr pkg for using location-scale version of the t-distribution</span></span>
<span id="cb72-2"><a href="introHypoTest.html#cb72-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;extraDistr&quot;</span>)) {</span>
<span id="cb72-3"><a href="introHypoTest.html#cb72-3" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;extraDistr&quot;</span>)</span>
<span id="cb72-4"><a href="introHypoTest.html#cb72-4" tabindex="-1"></a>}</span>
<span id="cb72-5"><a href="introHypoTest.html#cb72-5" tabindex="-1"></a></span>
<span id="cb72-6"><a href="introHypoTest.html#cb72-6" tabindex="-1"></a>data_nonsmoking   <span class="ot">=</span> <span class="fu">c</span>(<span class="fl">3.99</span>, <span class="fl">3.79</span>, <span class="fl">3.60</span>, <span class="fl">3.73</span>, <span class="fl">3.21</span>, <span class="fl">3.60</span>, <span class="fl">4.08</span>, <span class="fl">3.61</span>, </span>
<span id="cb72-7"><a href="introHypoTest.html#cb72-7" tabindex="-1"></a>                      <span class="fl">3.83</span>, <span class="fl">3.31</span>, <span class="fl">4.13</span>, <span class="fl">3.26</span>, <span class="fl">3.54</span>)</span>
<span id="cb72-8"><a href="introHypoTest.html#cb72-8" tabindex="-1"></a>sigma_hat <span class="ot">=</span> <span class="fu">sqrt</span>(<span class="fu">var</span>(data_nonsmoking) <span class="sc">/</span> <span class="fu">length</span>(data_nonsmoking))</span>
<span id="cb72-9"><a href="introHypoTest.html#cb72-9" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span>, <span class="fl">0.01</span>)</span>
<span id="cb72-10"><a href="introHypoTest.html#cb72-10" tabindex="-1"></a>p <span class="ot">=</span> extraDistr<span class="sc">::</span><span class="fu">dlst</span>(x, <span class="at">df =</span> <span class="fu">length</span>(data_nonsmoking) <span class="sc">-</span> <span class="dv">1</span>, <span class="at">sigma =</span> sigma_hat)</span>
<span id="cb72-11"><a href="introHypoTest.html#cb72-11" tabindex="-1"></a><span class="co">#p = dt(x / sigma_hat, df = length(data_nonsmoking) - 1) / sigma_hat</span></span>
<span id="cb72-12"><a href="introHypoTest.html#cb72-12" tabindex="-1"></a>events <span class="ot">=</span> <span class="fu">rep</span>(<span class="st">&quot;non-extreme&quot;</span>, <span class="fu">length</span>(x))</span>
<span id="cb72-13"><a href="introHypoTest.html#cb72-13" tabindex="-1"></a>events[x <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fl">0.468</span>] <span class="ot">=</span> <span class="st">&quot;extreme-left&quot;</span></span>
<span id="cb72-14"><a href="introHypoTest.html#cb72-14" tabindex="-1"></a>events[x <span class="sc">&gt;=</span> <span class="fl">0.468</span>] <span class="ot">=</span> <span class="st">&quot;extreme-right&quot;</span></span>
<span id="cb72-15"><a href="introHypoTest.html#cb72-15" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">x=</span>x, <span class="at">PDF=</span>p, <span class="at">events=</span>events)</span>
<span id="cb72-16"><a href="introHypoTest.html#cb72-16" tabindex="-1"></a>fig_1group <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x=</span>x, <span class="at">PDF=</span>p), <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> PDF, <span class="at">color=</span>events)) <span class="sc">+</span> </span>
<span id="cb72-17"><a href="introHypoTest.html#cb72-17" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;black&quot;</span>))</span></code></pre></div>
</details>
<p><br /></p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="introHypoTest.html#cb73-1" tabindex="-1"></a>cdf_left <span class="ot">=</span> extraDistr<span class="sc">::</span><span class="fu">plst</span>(<span class="sc">-</span><span class="fl">0.468</span>, <span class="at">df =</span> <span class="fu">length</span>(data_nonsmoking) <span class="sc">-</span> <span class="dv">1</span>, <span class="at">sigma =</span> sigma_hat)</span>
<span id="cb73-2"><a href="introHypoTest.html#cb73-2" tabindex="-1"></a>cdf_right <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> extraDistr<span class="sc">::</span><span class="fu">plst</span>(<span class="fl">0.468</span>, <span class="at">df =</span> <span class="fu">length</span>(data_nonsmoking) <span class="sc">-</span> <span class="dv">1</span>, <span class="at">sigma =</span> sigma_hat)</span>
<span id="cb73-3"><a href="introHypoTest.html#cb73-3" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(cdf_left, cdf_right))</span>
<span id="cb73-4"><a href="introHypoTest.html#cb73-4" tabindex="-1"></a><span class="co">#&gt; [1] 5.250456e-05 5.250456e-05</span></span>
<span id="cb73-5"><a href="introHypoTest.html#cb73-5" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;p-value:&quot;</span>, cdf_left <span class="sc">+</span> cdf_right))</span>
<span id="cb73-6"><a href="introHypoTest.html#cb73-6" tabindex="-1"></a><span class="co">#&gt; [1] &quot;p-value: 0.000105009117330513&quot;</span></span>
<span id="cb73-7"><a href="introHypoTest.html#cb73-7" tabindex="-1"></a>fig_1group</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-t-onegroup-1.png" width="60%" /></p>
</div>
<div id="two-group-t-test" class="section level4 hasAnchor" number="2.2.2.3">
<h4><span class="header-section-number">2.2.2.3</span> Two-group t-test<a href="introHypoTest.html#two-group-t-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Aside from comparing a single sample mean to a specific value, a more common scenario involves comparing the means between two groups—for example, a treatment group (such as disease status or intervention) versus a control group. This approach is often referred to as an A/B test, which assesses whether the sample means of groups A and B are statistically different.</p>
<p>Let’s revisit Example 4, where we compare birth weights between two groups: heavy smokers and non-smokers. The observed difference in mean birth weight is -0.516 kg. To analyze this, we can set up the following hypotheses, similar to the permutation test:</p>
<ul>
<li>H0: there is no difference in mean birth weight between groups: <span class="math inline">\(d = 0\)</span></li>
<li>H1: there is a difference, <span class="math inline">\(d \neq 0\)</span></li>
</ul>
<p>This time, instead of using resampling, we will now use the t-distribution to approximate the distribution of the difference in sample means, <span class="math inline">\(d\)</span>, under the null hypothesis. Assuming the null hypothesis is true, we can model the distribution of the observed difference using a t-distribution as follows.</p>
<p><span class="math display">\[
d \sim t(\mu=0, \hat\sigma^2, \text{df}=n_1 + n_2 - 2)
\]</span>
Under the null hypothesis, the t-distribution has a mean of <span class="math inline">\(\mu=0\)</span>, with degrees of freedom df=<span class="math inline">\(n_1+n_2-2\)</span>, since the means of the two samples are considered fixed. Similar to the one-sample test, the only unknown parameter to estimate is
<span class="math inline">\(\hat\sigma\)</span>. Assuming <strong>Assumption 3</strong>—<em>that the two groups have the same unknown variance</em>—we can approximate this parameter using the following calculation:</p>
<p><span class="math display">\[\hat\sigma^2 \approx {s_p^2 \cdot (\frac{1}{n_{1}} + \frac{1}{n_{2}}})\]</span>,</p>
<p>where <span class="math inline">\(s_p^2\)</span> is the pooled variance and is given by <span class="math inline">\(s_1^2\)</span> and <span class="math inline">\(s_2^2\)</span>, sample variance (i.e., the unbiased estimator of the population variance) in groups 1 and 2, as follows,</p>
<p><span class="math display">\[
s_p^2 = \frac{(n_1 - 1)s_1^2 + (n_2 - 1)s_2^2}{n_1 + n_2 - 2}; \quad
s_1^2 = \frac{\sum {(X_1^{(1)} - \bar{X}^{(1)})^2}}{n_1 - 1} ; \quad
s_2^2 = \frac{\sum {(X_1^{(2)} - \bar{X}^{(2)})^2}}{n_2 - 1}
\]</span>
Note, if the variances between the groups differ, the calculation would need to be adjusted, using the <a href="https://en.wikipedia.org/wiki/Welch%27s_t-test">Welch’s t-test</a></p>
<details>
<summary>
<strong>Click for R scripts to calculate t distribution for two-group comparison</strong>
</summary>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="introHypoTest.html#cb74-1" tabindex="-1"></a><span class="do">## Install extraDistr pkg for using location-scale version of the t-distribution</span></span>
<span id="cb74-2"><a href="introHypoTest.html#cb74-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;extraDistr&quot;</span>)) {</span>
<span id="cb74-3"><a href="introHypoTest.html#cb74-3" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;extraDistr&quot;</span>)</span>
<span id="cb74-4"><a href="introHypoTest.html#cb74-4" tabindex="-1"></a>}</span>
<span id="cb74-5"><a href="introHypoTest.html#cb74-5" tabindex="-1"></a></span>
<span id="cb74-6"><a href="introHypoTest.html#cb74-6" tabindex="-1"></a>d_obs <span class="ot">=</span> <span class="fu">mean</span>(data_heavysmoking) <span class="sc">-</span> <span class="fu">mean</span>(data_nonsmoking)</span>
<span id="cb74-7"><a href="introHypoTest.html#cb74-7" tabindex="-1"></a></span>
<span id="cb74-8"><a href="introHypoTest.html#cb74-8" tabindex="-1"></a>n_ns <span class="ot">=</span> <span class="fu">length</span>(data_nonsmoking)</span>
<span id="cb74-9"><a href="introHypoTest.html#cb74-9" tabindex="-1"></a>n_hs <span class="ot">=</span> <span class="fu">length</span>(data_heavysmoking)</span>
<span id="cb74-10"><a href="introHypoTest.html#cb74-10" tabindex="-1"></a></span>
<span id="cb74-11"><a href="introHypoTest.html#cb74-11" tabindex="-1"></a>s_ns <span class="ot">=</span> <span class="fu">sd</span>(data_nonsmoking) <span class="co"># degree of freedom: n-1</span></span>
<span id="cb74-12"><a href="introHypoTest.html#cb74-12" tabindex="-1"></a>s_hs <span class="ot">=</span> <span class="fu">sd</span>(data_heavysmoking)</span>
<span id="cb74-13"><a href="introHypoTest.html#cb74-13" tabindex="-1"></a></span>
<span id="cb74-14"><a href="introHypoTest.html#cb74-14" tabindex="-1"></a><span class="co"># the pooled standard deviation</span></span>
<span id="cb74-15"><a href="introHypoTest.html#cb74-15" tabindex="-1"></a>sp <span class="ot">=</span> <span class="fu">sqrt</span>(((n_ns <span class="sc">-</span> <span class="dv">1</span>)<span class="sc">*</span>s_ns<span class="sc">**</span><span class="dv">2</span> <span class="sc">+</span> (n_hs <span class="sc">-</span> <span class="dv">1</span>)<span class="sc">*</span>s_hs<span class="sc">**</span><span class="dv">2</span>) <span class="sc">/</span> (n_ns <span class="sc">+</span> n_hs <span class="sc">-</span> <span class="dv">2</span>))</span>
<span id="cb74-16"><a href="introHypoTest.html#cb74-16" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;Pooled standard deviation:&quot;</span>, sp))</span>
<span id="cb74-17"><a href="introHypoTest.html#cb74-17" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Pooled standard deviation:0.428057812829366&quot;</span></span>
<span id="cb74-18"><a href="introHypoTest.html#cb74-18" tabindex="-1"></a></span>
<span id="cb74-19"><a href="introHypoTest.html#cb74-19" tabindex="-1"></a>d_sigma_hat <span class="ot">=</span> sp <span class="sc">*</span> <span class="fu">sqrt</span>(<span class="dv">1</span><span class="sc">/</span>n_ns <span class="sc">+</span> <span class="dv">1</span><span class="sc">/</span>n_hs)</span>
<span id="cb74-20"><a href="introHypoTest.html#cb74-20" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Estimated standard error of mean difference:&quot;</span>, d_sigma_hat))</span>
<span id="cb74-21"><a href="introHypoTest.html#cb74-21" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Estimated standard error of mean difference: 0.162204962956089&quot;</span></span>
<span id="cb74-22"><a href="introHypoTest.html#cb74-22" tabindex="-1"></a></span>
<span id="cb74-23"><a href="introHypoTest.html#cb74-23" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span>, <span class="fl">0.01</span>)</span>
<span id="cb74-24"><a href="introHypoTest.html#cb74-24" tabindex="-1"></a>p <span class="ot">=</span> extraDistr<span class="sc">::</span><span class="fu">dlst</span>(x, <span class="at">df =</span> n_ns <span class="sc">+</span> n_hs <span class="sc">-</span> <span class="dv">2</span>, <span class="at">sigma =</span> d_sigma_hat)</span>
<span id="cb74-25"><a href="introHypoTest.html#cb74-25" tabindex="-1"></a>events <span class="ot">=</span> <span class="fu">rep</span>(<span class="st">&quot;non-extreme&quot;</span>, <span class="fu">length</span>(x))</span>
<span id="cb74-26"><a href="introHypoTest.html#cb74-26" tabindex="-1"></a>events[x <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">abs</span>(d_obs)] <span class="ot">=</span> <span class="st">&quot;extreme-left&quot;</span></span>
<span id="cb74-27"><a href="introHypoTest.html#cb74-27" tabindex="-1"></a>events[x <span class="sc">&gt;=</span> <span class="fu">abs</span>(d_obs)] <span class="ot">=</span> <span class="st">&quot;extreme-right&quot;</span></span>
<span id="cb74-28"><a href="introHypoTest.html#cb74-28" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">x=</span>x, <span class="at">PDF=</span>p, <span class="at">events=</span>events)</span>
<span id="cb74-29"><a href="introHypoTest.html#cb74-29" tabindex="-1"></a>fig_2group <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x=</span>x, <span class="at">PDF=</span>p), <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> PDF, <span class="at">color=</span>events)) <span class="sc">+</span> </span>
<span id="cb74-30"><a href="introHypoTest.html#cb74-30" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;black&quot;</span>))</span></code></pre></div>
</details>
<p><br /></p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="introHypoTest.html#cb75-1" tabindex="-1"></a>cdf_left <span class="ot">=</span> extraDistr<span class="sc">::</span><span class="fu">plst</span>(<span class="sc">-</span><span class="fu">abs</span>(d_obs), <span class="at">df =</span> n_ns <span class="sc">+</span> n_hs <span class="sc">-</span> <span class="dv">2</span>, <span class="at">sigma =</span> d_sigma_hat)</span>
<span id="cb75-2"><a href="introHypoTest.html#cb75-2" tabindex="-1"></a>cdf_right <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> extraDistr<span class="sc">::</span><span class="fu">plst</span>(<span class="fu">abs</span>(d_obs), <span class="at">df =</span> n_ns <span class="sc">+</span> n_hs <span class="sc">-</span> <span class="dv">2</span>, <span class="at">sigma =</span> d_sigma_hat)</span>
<span id="cb75-3"><a href="introHypoTest.html#cb75-3" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(cdf_left, cdf_right))</span>
<span id="cb75-4"><a href="introHypoTest.html#cb75-4" tabindex="-1"></a><span class="co">#&gt; [1] 0.00189629 0.00189629</span></span>
<span id="cb75-5"><a href="introHypoTest.html#cb75-5" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;p-value:&quot;</span>, cdf_left <span class="sc">+</span> cdf_right))</span>
<span id="cb75-6"><a href="introHypoTest.html#cb75-6" tabindex="-1"></a><span class="co">#&gt; [1] &quot;p-value: 0.00379258038486997&quot;</span></span>
<span id="cb75-7"><a href="introHypoTest.html#cb75-7" tabindex="-1"></a>fig_2group</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-t-twogroups-1.png" width="60%" /></p>
<p><strong>Transform to a standardized <span class="math inline">\(t\)</span>-distribution</strong></p>
<p>In addition to using the non-standardized t-distribution with its location and scale parameters, it is common to transform the t-statistic into a standardized form. Specifically, if we compute the t-statistic as <span class="math inline">\(t_{\text{statistic}} = d / \hat{\sigma}\)</span>, it follows a standardized t-distribution with the the same degree of freedom. This allows us to report both the t-statistic and the corresponding p-value for hypothesis testing.</p>
<p><span class="math display">\[ t_{\text{statistic}} = \frac{\bar{X}_{1} - \bar{X}_{2}}{s_p \cdot \sqrt{\frac{1}{n_{1}} + \frac{1}{n_{2}}}} \]</span></p>
<p>We can visualize the null distribution and the extreme values in a standardized form as follows:</p>
<details>
<summary>
<strong>Click for R scripts to calculate t distribution for two-group comparison</strong>
</summary>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="introHypoTest.html#cb76-1" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span>, <span class="fl">0.01</span>) <span class="sc">/</span> d_sigma_hat</span>
<span id="cb76-2"><a href="introHypoTest.html#cb76-2" tabindex="-1"></a>p <span class="ot">=</span> stats<span class="sc">::</span><span class="fu">dt</span>(x, <span class="at">df =</span> n_ns <span class="sc">+</span> n_hs <span class="sc">-</span> <span class="dv">2</span>)</span>
<span id="cb76-3"><a href="introHypoTest.html#cb76-3" tabindex="-1"></a>events <span class="ot">=</span> <span class="fu">rep</span>(<span class="st">&quot;non-extreme&quot;</span>, <span class="fu">length</span>(x))</span>
<span id="cb76-4"><a href="introHypoTest.html#cb76-4" tabindex="-1"></a>events[x <span class="sc">&lt;=</span> <span class="sc">-</span><span class="fu">abs</span>(d_obs <span class="sc">/</span> d_sigma_hat)] <span class="ot">=</span> <span class="st">&quot;extreme-left&quot;</span></span>
<span id="cb76-5"><a href="introHypoTest.html#cb76-5" tabindex="-1"></a>events[x <span class="sc">&gt;=</span> <span class="fu">abs</span>(d_obs <span class="sc">/</span> d_sigma_hat)] <span class="ot">=</span> <span class="st">&quot;extreme-right&quot;</span></span>
<span id="cb76-6"><a href="introHypoTest.html#cb76-6" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">x=</span>x, <span class="at">PDF=</span>p, <span class="at">events=</span>events)</span>
<span id="cb76-7"><a href="introHypoTest.html#cb76-7" tabindex="-1"></a>fig_2group_st <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x=</span>x, <span class="at">PDF=</span>p), <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> PDF, <span class="at">color=</span>events)) <span class="sc">+</span> </span>
<span id="cb76-8"><a href="introHypoTest.html#cb76-8" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;black&quot;</span>))</span></code></pre></div>
</details>
<p><br /></p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="introHypoTest.html#cb77-1" tabindex="-1"></a>cdf_left <span class="ot">=</span> stats<span class="sc">::</span><span class="fu">pt</span>(<span class="sc">-</span><span class="fu">abs</span>(d_obs <span class="sc">/</span> d_sigma_hat), <span class="at">df =</span> n_ns <span class="sc">+</span> n_hs <span class="sc">-</span> <span class="dv">2</span>)</span>
<span id="cb77-2"><a href="introHypoTest.html#cb77-2" tabindex="-1"></a>cdf_right <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> stats<span class="sc">::</span><span class="fu">pt</span>(<span class="fu">abs</span>(d_obs <span class="sc">/</span> d_sigma_hat), <span class="at">df =</span> n_ns <span class="sc">+</span> n_hs <span class="sc">-</span> <span class="dv">2</span>)</span>
<span id="cb77-3"><a href="introHypoTest.html#cb77-3" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(cdf_left, cdf_right))</span>
<span id="cb77-4"><a href="introHypoTest.html#cb77-4" tabindex="-1"></a><span class="co">#&gt; [1] 0.00189629 0.00189629</span></span>
<span id="cb77-5"><a href="introHypoTest.html#cb77-5" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;p-value:&quot;</span>, cdf_left <span class="sc">+</span> cdf_right))</span>
<span id="cb77-6"><a href="introHypoTest.html#cb77-6" tabindex="-1"></a><span class="co">#&gt; [1] &quot;p-value: 0.00379258038486997&quot;</span></span>
<span id="cb77-7"><a href="introHypoTest.html#cb77-7" tabindex="-1"></a>fig_2group_st</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-t-twogroups_st-1.png" width="60%" /></p>
</div>
<div id="direct-use-of-t.test" class="section level4 hasAnchor" number="2.2.2.4">
<h4><span class="header-section-number">2.2.2.4</span> Direct use of <code>t.test()</code><a href="introHypoTest.html#direct-use-of-t.test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In the course and most of your future analyses, you can directly use the built-in
<code>t.test()</code> function. If you assume that the variances in the two groups are the same, you can set the <code>var.equal = TRUE</code>; Note, the default value is <code>FALSE</code>.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="introHypoTest.html#cb78-1" tabindex="-1"></a><span class="co"># Note, we assumed the variance in both groups are the same, </span></span>
<span id="cb78-2"><a href="introHypoTest.html#cb78-2" tabindex="-1"></a><span class="co"># we so need to set var.equal = TRUE</span></span>
<span id="cb78-3"><a href="introHypoTest.html#cb78-3" tabindex="-1"></a></span>
<span id="cb78-4"><a href="introHypoTest.html#cb78-4" tabindex="-1"></a>t_res <span class="ot">=</span> <span class="fu">t.test</span>(data_nonsmoking, data_heavysmoking, <span class="at">var.equal =</span> <span class="cn">TRUE</span>)</span>
<span id="cb78-5"><a href="introHypoTest.html#cb78-5" tabindex="-1"></a></span>
<span id="cb78-6"><a href="introHypoTest.html#cb78-6" tabindex="-1"></a>t_res<span class="sc">$</span>p.value</span>
<span id="cb78-7"><a href="introHypoTest.html#cb78-7" tabindex="-1"></a><span class="co">#&gt; [1] 0.00379258</span></span>
<span id="cb78-8"><a href="introHypoTest.html#cb78-8" tabindex="-1"></a></span>
<span id="cb78-9"><a href="introHypoTest.html#cb78-9" tabindex="-1"></a>t_res</span>
<span id="cb78-10"><a href="introHypoTest.html#cb78-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb78-11"><a href="introHypoTest.html#cb78-11" tabindex="-1"></a><span class="co">#&gt;  Two Sample t-test</span></span>
<span id="cb78-12"><a href="introHypoTest.html#cb78-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb78-13"><a href="introHypoTest.html#cb78-13" tabindex="-1"></a><span class="co">#&gt; data:  data_nonsmoking and data_heavysmoking</span></span>
<span id="cb78-14"><a href="introHypoTest.html#cb78-14" tabindex="-1"></a><span class="co">#&gt; t = 3.1793, df = 26, p-value = 0.003793</span></span>
<span id="cb78-15"><a href="introHypoTest.html#cb78-15" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true difference in means is not equal to 0</span></span>
<span id="cb78-16"><a href="introHypoTest.html#cb78-16" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb78-17"><a href="introHypoTest.html#cb78-17" tabindex="-1"></a><span class="co">#&gt;  0.1822752 0.8491094</span></span>
<span id="cb78-18"><a href="introHypoTest.html#cb78-18" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb78-19"><a href="introHypoTest.html#cb78-19" tabindex="-1"></a><span class="co">#&gt; mean of x mean of y </span></span>
<span id="cb78-20"><a href="introHypoTest.html#cb78-20" tabindex="-1"></a><span class="co">#&gt;  3.667692  3.152000</span></span></code></pre></div>
</div>
<div id="regression-based-test" class="section level4 hasAnchor" number="2.2.2.5">
<h4><span class="header-section-number">2.2.2.5</span> Regression-based test<a href="introHypoTest.html#regression-based-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We can also perform a t-test (Welch’s t-test) in a Generalized Linear Model (GLM) setting to test if a coefficient is zero or not.
Here, we use the <code>marketing</code> dataset as an example, and we can load the data and visualize it as follows using R scripts.</p>
<details>
<summary>
<strong>Click for R scripts to load the data</strong>
</summary>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="introHypoTest.html#cb79-1" tabindex="-1"></a><span class="co"># Install datarium library if you haven&#39;t</span></span>
<span id="cb79-2"><a href="introHypoTest.html#cb79-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;datarium&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb79-3"><a href="introHypoTest.html#cb79-3" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;datarium&quot;</span>)</span>
<span id="cb79-4"><a href="introHypoTest.html#cb79-4" tabindex="-1"></a>}</span>
<span id="cb79-5"><a href="introHypoTest.html#cb79-5" tabindex="-1"></a></span>
<span id="cb79-6"><a href="introHypoTest.html#cb79-6" tabindex="-1"></a><span class="fu">library</span>(datarium)</span>
<span id="cb79-7"><a href="introHypoTest.html#cb79-7" tabindex="-1"></a></span>
<span id="cb79-8"><a href="introHypoTest.html#cb79-8" tabindex="-1"></a><span class="co"># Load data: then we will have a data.frame with name marketing</span></span>
<span id="cb79-9"><a href="introHypoTest.html#cb79-9" tabindex="-1"></a><span class="fu">data</span>(marketing)</span>
<span id="cb79-10"><a href="introHypoTest.html#cb79-10" tabindex="-1"></a><span class="fu">head</span>(marketing)</span>
<span id="cb79-11"><a href="introHypoTest.html#cb79-11" tabindex="-1"></a><span class="co">#&gt;   youtube facebook newspaper sales</span></span>
<span id="cb79-12"><a href="introHypoTest.html#cb79-12" tabindex="-1"></a><span class="co">#&gt; 1  276.12    45.36     83.04 26.52</span></span>
<span id="cb79-13"><a href="introHypoTest.html#cb79-13" tabindex="-1"></a><span class="co">#&gt; 2   53.40    47.16     54.12 12.48</span></span>
<span id="cb79-14"><a href="introHypoTest.html#cb79-14" tabindex="-1"></a><span class="co">#&gt; 3   20.64    55.08     83.16 11.16</span></span>
<span id="cb79-15"><a href="introHypoTest.html#cb79-15" tabindex="-1"></a><span class="co">#&gt; 4  181.80    49.56     70.20 22.20</span></span>
<span id="cb79-16"><a href="introHypoTest.html#cb79-16" tabindex="-1"></a><span class="co">#&gt; 5  216.96    12.96     70.08 15.48</span></span>
<span id="cb79-17"><a href="introHypoTest.html#cb79-17" tabindex="-1"></a><span class="co">#&gt; 6   10.44    58.68     90.00  8.64</span></span></code></pre></div>
</details>
<p><br /></p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="introHypoTest.html#cb80-1" tabindex="-1"></a><span class="fu">ggplot</span>(marketing, <span class="fu">aes</span>(<span class="at">x=</span>newspaper, <span class="at">y=</span>sales)) <span class="sc">+</span> </span>
<span id="cb80-2"><a href="introHypoTest.html#cb80-2" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm, <span class="at">formula =</span> <span class="st">&#39;y ~ x&#39;</span>)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-glm-scatter-1.png" width="60%" /></p>
<p>We can further use the <code>lm</code> function to fit the regression model, which will estimate the coefficients for each covariate as well as the intercept. In addition to the estimates, the function provides the standard errors of each coefficient, allowing us to derive the null distribution of each coefficient in the form of a t-distribution. By transforming the observed statistic—dividing the estimated coefficient by its standard error (similar to the two-group t-test described earlier)—we obtain the t-statistic and its corresponding p-value. All of these results are accessible through the output of the <code>lm</code> function, which returns a list containing each value.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="introHypoTest.html#cb81-1" tabindex="-1"></a><span class="co"># Fit linear regression</span></span>
<span id="cb81-2"><a href="introHypoTest.html#cb81-2" tabindex="-1"></a>res.lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(sales <span class="sc">~</span> newspaper, <span class="at">data =</span> marketing)</span>
<span id="cb81-3"><a href="introHypoTest.html#cb81-3" tabindex="-1"></a></span>
<span id="cb81-4"><a href="introHypoTest.html#cb81-4" tabindex="-1"></a><span class="co"># We can check the test via the summary() function</span></span>
<span id="cb81-5"><a href="introHypoTest.html#cb81-5" tabindex="-1"></a><span class="fu">summary</span>(res.lm)</span>
<span id="cb81-6"><a href="introHypoTest.html#cb81-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-7"><a href="introHypoTest.html#cb81-7" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb81-8"><a href="introHypoTest.html#cb81-8" tabindex="-1"></a><span class="co">#&gt; lm(formula = sales ~ newspaper, data = marketing)</span></span>
<span id="cb81-9"><a href="introHypoTest.html#cb81-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-10"><a href="introHypoTest.html#cb81-10" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb81-11"><a href="introHypoTest.html#cb81-11" tabindex="-1"></a><span class="co">#&gt;     Min      1Q  Median      3Q     Max </span></span>
<span id="cb81-12"><a href="introHypoTest.html#cb81-12" tabindex="-1"></a><span class="co">#&gt; -13.473  -4.065  -1.007   4.207  15.330 </span></span>
<span id="cb81-13"><a href="introHypoTest.html#cb81-13" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-14"><a href="introHypoTest.html#cb81-14" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb81-15"><a href="introHypoTest.html#cb81-15" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb81-16"><a href="introHypoTest.html#cb81-16" tabindex="-1"></a><span class="co">#&gt; (Intercept) 14.82169    0.74570   19.88  &lt; 2e-16 ***</span></span>
<span id="cb81-17"><a href="introHypoTest.html#cb81-17" tabindex="-1"></a><span class="co">#&gt; newspaper    0.05469    0.01658    3.30  0.00115 ** </span></span>
<span id="cb81-18"><a href="introHypoTest.html#cb81-18" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb81-19"><a href="introHypoTest.html#cb81-19" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb81-20"><a href="introHypoTest.html#cb81-20" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-21"><a href="introHypoTest.html#cb81-21" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 6.111 on 198 degrees of freedom</span></span>
<span id="cb81-22"><a href="introHypoTest.html#cb81-22" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.05212,    Adjusted R-squared:  0.04733 </span></span>
<span id="cb81-23"><a href="introHypoTest.html#cb81-23" tabindex="-1"></a><span class="co">#&gt; F-statistic: 10.89 on 1 and 198 DF,  p-value: 0.001148</span></span></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="introHypoTest.html#cb82-1" tabindex="-1"></a>glm_t_val <span class="ot">=</span> <span class="fu">summary</span>(res.lm)<span class="sc">$</span>coefficients[<span class="st">&quot;newspaper&quot;</span>, <span class="st">&quot;t value&quot;</span>]</span>
<span id="cb82-2"><a href="introHypoTest.html#cb82-2" tabindex="-1"></a></span>
<span id="cb82-3"><a href="introHypoTest.html#cb82-3" tabindex="-1"></a>xx <span class="ot">=</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">5</span>, <span class="fl">0.01</span>)</span>
<span id="cb82-4"><a href="introHypoTest.html#cb82-4" tabindex="-1"></a>yy <span class="ot">=</span> <span class="fu">dt</span>(xx, <span class="dv">198</span>)</span>
<span id="cb82-5"><a href="introHypoTest.html#cb82-5" tabindex="-1"></a>df_ttest <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x=</span>xx, <span class="at">PDF=</span>yy)</span>
<span id="cb82-6"><a href="introHypoTest.html#cb82-6" tabindex="-1"></a></span>
<span id="cb82-7"><a href="introHypoTest.html#cb82-7" tabindex="-1"></a><span class="fu">ggplot</span>(df_ttest, <span class="fu">aes</span>(<span class="at">x=</span>x, <span class="at">y=</span>PDF)) <span class="sc">+</span> </span>
<span id="cb82-8"><a href="introHypoTest.html#cb82-8" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span> </span>
<span id="cb82-9"><a href="introHypoTest.html#cb82-9" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> glm_t_val, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;tomato&quot;</span>) <span class="sc">+</span></span>
<span id="cb82-10"><a href="introHypoTest.html#cb82-10" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="sc">-</span>glm_t_val, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&#39;tomato&#39;</span>)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-glm-1.png" width="60%" /></p>
</div>
</div>
<div id="fishers-exact-test" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Fisher’s exact test<a href="introHypoTest.html#fishers-exact-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Fisher’s exact test is a widely used hypothesis testing method for comparing categorical variables between groups. In genomics, an example of its application is gene set enrichment analysis within a specific annotated pathway. This involves testing whether a gene set of interest contains a significantly higher proportion of genes from the pathway compared to non-interest genes.</p>
<div id="binomial-test" class="section level4 hasAnchor" number="2.2.3.1">
<h4><span class="header-section-number">2.2.3.1</span> Binomial test<a href="introHypoTest.html#binomial-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>A similar, often simpler, approach is the binomial test. Let’s revisit Example 2, where we suspect that a die is biased toward rolling a six. We can formulate the hypotheses as follows:</p>
<ul>
<li><span class="math inline">\(H_0\)</span> (Null hypothesis): The die is fair, with a probability of rolling a six of <span class="math inline">\(p=1/6\)</span>.</li>
<li><span class="math inline">\(H_a\)</span> (Alternative hypothesis): The die is unfair, with a probability of rolling a six greater 1/6, name <span class="math inline">\(p&gt;1/6\)</span>.</li>
</ul>
<p>In this scenario, the exact null distribution follows a binomial distribution with success probability <span class="math inline">\(p=1/6\)</span>, meaning we do not need to approximate the null distribution. This is also why binomial test and Fisher’s test are exact tests.</p>
<p>Therefore, in this example with observing 43 sixes in 100 rolls, the events at least as extreme the observed number are 43, 44, …, 100. By using the definition of <a href="https://en.wikipedia.org/wiki/Binomial_distribution">binomial distribution</a>, we can calculate the p-value as follows,</p>
<p><span class="math display">\[\text{p_value} = \sum_{k=\{43...,100\}}{\binom{100}{k} p^{k} (1-p)^{100-k}}; \quad p=1/6\]</span></p>
<p>Alternatively, in R, we can calculate the p-value manually with <code>choose()</code> for number of combinations or directly using <code>binom.test()</code> for the hypothesis test, as follows.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="introHypoTest.html#cb83-1" tabindex="-1"></a><span class="co"># manual calculation with extreme values</span></span>
<span id="cb83-2"><a href="introHypoTest.html#cb83-2" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">seq</span>(<span class="dv">43</span>, <span class="dv">100</span>)</span>
<span id="cb83-3"><a href="introHypoTest.html#cb83-3" tabindex="-1"></a>p_val <span class="ot">=</span> <span class="fu">sum</span>(<span class="fu">choose</span>(<span class="dv">100</span>, x) <span class="sc">*</span> (<span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>)<span class="sc">^</span>(x) <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> <span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>)<span class="sc">^</span>(<span class="dv">100</span><span class="sc">-</span>x))</span>
<span id="cb83-4"><a href="introHypoTest.html#cb83-4" tabindex="-1"></a>p_val</span>
<span id="cb83-5"><a href="introHypoTest.html#cb83-5" tabindex="-1"></a><span class="co">#&gt; [1] 5.439087e-10</span></span>
<span id="cb83-6"><a href="introHypoTest.html#cb83-6" tabindex="-1"></a></span>
<span id="cb83-7"><a href="introHypoTest.html#cb83-7" tabindex="-1"></a><span class="co"># using built-in function</span></span>
<span id="cb83-8"><a href="introHypoTest.html#cb83-8" tabindex="-1"></a><span class="fu">binom.test</span>(<span class="dv">43</span>, <span class="dv">100</span>, <span class="at">p=</span><span class="dv">1</span><span class="sc">/</span><span class="dv">6</span>, <span class="at">alternative =</span> <span class="st">&quot;greater&quot;</span>)</span>
<span id="cb83-9"><a href="introHypoTest.html#cb83-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb83-10"><a href="introHypoTest.html#cb83-10" tabindex="-1"></a><span class="co">#&gt;  Exact binomial test</span></span>
<span id="cb83-11"><a href="introHypoTest.html#cb83-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb83-12"><a href="introHypoTest.html#cb83-12" tabindex="-1"></a><span class="co">#&gt; data:  43 and 100</span></span>
<span id="cb83-13"><a href="introHypoTest.html#cb83-13" tabindex="-1"></a><span class="co">#&gt; number of successes = 43, number of trials = 100, p-value = 5.439e-10</span></span>
<span id="cb83-14"><a href="introHypoTest.html#cb83-14" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true probability of success is greater than 0.1666667</span></span>
<span id="cb83-15"><a href="introHypoTest.html#cb83-15" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb83-16"><a href="introHypoTest.html#cb83-16" tabindex="-1"></a><span class="co">#&gt;  0.3460073 1.0000000</span></span>
<span id="cb83-17"><a href="introHypoTest.html#cb83-17" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb83-18"><a href="introHypoTest.html#cb83-18" tabindex="-1"></a><span class="co">#&gt; probability of success </span></span>
<span id="cb83-19"><a href="introHypoTest.html#cb83-19" tabindex="-1"></a><span class="co">#&gt;                   0.43</span></span></code></pre></div>
</div>
<div id="hypergeometric-distribution" class="section level4 hasAnchor" number="2.2.3.2">
<h4><span class="header-section-number">2.2.3.2</span> Hypergeometric distribution<a href="introHypoTest.html#hypergeometric-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In a certain sense, you may feel that the binomial test is somewhat similar to a one-sample t-test, as it assesses whether the success rate (or proportion) equals a <strong>specified value</strong>, using the number of successes as the test statistic. (Note, binomial test focuses on categorical variables and the t-test on continuous variables).</p>
<p>Similarly, a more common scenario involves comparing success rates (or hit rates) between two conditions—such as treated versus untreated groups. Let’s consider Example 5:</p>
<blockquote>
<p><strong>Example 5</strong>
Is handedness associated with sex? To investigate this, we set up the hypotheses:<br />
* <span class="math inline">\(H_0\)</span> (null hypothesis): The proportion of right-handed individuals is the same for males and females, i.e., <span class="math inline">\(p_{male} = p_{female}\)</span>.<br />
* <span class="math inline">\(H_a\)</span> (alternative hypothesis): The proportion of right-handed individuals differs between males and females, i.e., <span class="math inline">\(p_{male} \neq p_{female}\)</span>.<br /></p>
</blockquote>
<p><strong>Table 2.</strong> Contingency table for handedness between male and female.</p>
<table>
<thead>
<tr class="header">
<th></th>
<th>Right-handed</th>
<th>Left-handed</th>
<th><strong>Total</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Male</td>
<td>a = 43</td>
<td>b = 9</td>
<td>a + b = 52</td>
</tr>
<tr class="even">
<td>Female</td>
<td>c = 44</td>
<td>d = 4</td>
<td>c + d = 48</td>
</tr>
<tr class="odd">
<td><strong>Total</strong></td>
<td>a +c = 87</td>
<td>b + d = 13</td>
<td>n = 100</td>
</tr>
</tbody>
</table>
<p><br /></p>
<p>Before diving into hypothesis testing, let’s briefly review the hypergeometric distribution. In this context, we can arbitrarily choose any of the variables a, b, c, or d as the test statistic; without loss of generality, we choose <span class="math inline">\(a\)</span>. To avoid confusion between variables and observed values, we’ll use <span class="math inline">\(x\)</span> to denote the test statistic—in this case, the number of right-handed individuals among males (corresponding to <span class="math inline">\(a\)</span>).</p>
<p>The key requirement for the hypergeometric distribution is that the totals a+b (number of males), c+d (number of females), and a+c (number of right-handed individuals) are fixed. For illustration, suppose we randomly assign the 87 right-handed individuals into 52 males and 48 females. (Think about how this can be simulated.)</p>
<p>Notice that if <span class="math inline">\(x\)</span> (the number of right-handed males) changes, the values of b, c, and d must also change accordingly, given that the totals a+b, c+d, and a+c are fixed. To ensure all counts are non-negative, <span class="math inline">\(x\)</span> is bounded within a specific range:</p>
<p><span class="math display">\[\text{max}(0, (a+c)-(c+d)) &lt;= x &lt;= \text{min}(a+b, a+c)\]</span>
In our example, this translates to 39&lt;=x&lt;=52. Under the null hypothesis, the most probable values of <span class="math inline">\(x\)</span> are around 45 and 46. For a two-tailed test, the extreme values are those from 39 to 43 and symmetrically from 48 to 52.</p>
<p>If you were thinking of how to make the simulation and obtain the null distribution, you may realize that the probability mass function is the number of combinations to obtain a out of a+c, multiplying the number of combinations to obtain c out of c+d, dividing the combinations to obtain a+c out of a+b+c+d. We can write it as follows,
<span class="math display">\[p(x, m=a+b, n=c+d, k=a+c) = \frac{\binom{a+b}{x} * \binom{c+d}{c}}{ \binom{a+c}{a+b+c+d}}
\]</span></p>
<p>This distribution is called the <a href="https://en.wikipedia.org/wiki/Hypergeometric_distribution">hypergeomatric distribution</a>. In the terminology used above (for <span class="math inline">\(m, n, k\)</span>), it corresponds to the R function <a href="https://stat.ethz.ch/R-manual/R-devel/library/stats/html/Hypergeometric.html">stats::Hypergeometric</a>, which allows us to compute the probability density function (PDF) for specific values.</p>
<p>With the <strong>extreme events</strong> and <strong>null distribution</strong> defined, we can now calculate the two-sided p-value with manual calculation, using built-in <code>dhyper()</code> function, or directly use <code>fisher.test()</code> for the test.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="introHypoTest.html#cb84-1" tabindex="-1"></a><span class="do">## extreme events</span></span>
<span id="cb84-2"><a href="introHypoTest.html#cb84-2" tabindex="-1"></a>x_extreme <span class="ot">=</span> <span class="fu">c</span>(<span class="dv">39</span><span class="sc">:</span><span class="dv">43</span>, <span class="dv">48</span><span class="sc">:</span><span class="dv">52</span>)</span>
<span id="cb84-3"><a href="introHypoTest.html#cb84-3" tabindex="-1"></a></span>
<span id="cb84-4"><a href="introHypoTest.html#cb84-4" tabindex="-1"></a><span class="do">## manual calculation</span></span>
<span id="cb84-5"><a href="introHypoTest.html#cb84-5" tabindex="-1"></a>p_manual <span class="ot">=</span> <span class="fu">sum</span>(<span class="fu">choose</span>(<span class="dv">52</span>, x_extreme) <span class="sc">*</span> <span class="fu">choose</span>(<span class="dv">48</span>, <span class="dv">87</span> <span class="sc">-</span> x_extreme) <span class="sc">/</span> <span class="fu">choose</span>(<span class="dv">100</span>, <span class="dv">87</span>))</span>
<span id="cb84-6"><a href="introHypoTest.html#cb84-6" tabindex="-1"></a>p_manual</span>
<span id="cb84-7"><a href="introHypoTest.html#cb84-7" tabindex="-1"></a><span class="co">#&gt; [1] 0.239157</span></span>
<span id="cb84-8"><a href="introHypoTest.html#cb84-8" tabindex="-1"></a></span>
<span id="cb84-9"><a href="introHypoTest.html#cb84-9" tabindex="-1"></a><span class="do">## hypergeometric distribution</span></span>
<span id="cb84-10"><a href="introHypoTest.html#cb84-10" tabindex="-1"></a>p_hyper <span class="ot">=</span> <span class="fu">sum</span>(<span class="fu">dhyper</span>(x_extreme, <span class="at">m =</span> <span class="dv">52</span>, <span class="at">n =</span> <span class="dv">48</span>, <span class="at">k =</span> <span class="dv">87</span>))</span>
<span id="cb84-11"><a href="introHypoTest.html#cb84-11" tabindex="-1"></a>p_hyper</span>
<span id="cb84-12"><a href="introHypoTest.html#cb84-12" tabindex="-1"></a><span class="co">#&gt; [1] 0.239157</span></span>
<span id="cb84-13"><a href="introHypoTest.html#cb84-13" tabindex="-1"></a></span>
<span id="cb84-14"><a href="introHypoTest.html#cb84-14" tabindex="-1"></a><span class="do">## fisher&#39;s exact test</span></span>
<span id="cb84-15"><a href="introHypoTest.html#cb84-15" tabindex="-1"></a><span class="fu">fisher.test</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">43</span>, <span class="dv">44</span>, <span class="dv">9</span>, <span class="dv">4</span>), <span class="at">nrow =</span> <span class="dv">2</span>), <span class="at">alternative =</span> <span class="st">&quot;two.sided&quot;</span>)</span>
<span id="cb84-16"><a href="introHypoTest.html#cb84-16" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb84-17"><a href="introHypoTest.html#cb84-17" tabindex="-1"></a><span class="co">#&gt;  Fisher&#39;s Exact Test for Count Data</span></span>
<span id="cb84-18"><a href="introHypoTest.html#cb84-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb84-19"><a href="introHypoTest.html#cb84-19" tabindex="-1"></a><span class="co">#&gt; data:  matrix(c(43, 44, 9, 4), nrow = 2)</span></span>
<span id="cb84-20"><a href="introHypoTest.html#cb84-20" tabindex="-1"></a><span class="co">#&gt; p-value = 0.2392</span></span>
<span id="cb84-21"><a href="introHypoTest.html#cb84-21" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true odds ratio is not equal to 1</span></span>
<span id="cb84-22"><a href="introHypoTest.html#cb84-22" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb84-23"><a href="introHypoTest.html#cb84-23" tabindex="-1"></a><span class="co">#&gt;  0.09150811 1.71527769</span></span>
<span id="cb84-24"><a href="introHypoTest.html#cb84-24" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb84-25"><a href="introHypoTest.html#cb84-25" tabindex="-1"></a><span class="co">#&gt; odds ratio </span></span>
<span id="cb84-26"><a href="introHypoTest.html#cb84-26" tabindex="-1"></a><span class="co">#&gt;  0.4378606</span></span></code></pre></div>
<p><strong>Exercise</strong>:
try it yourself by calculating the one-tailed p-value for the alternative hypothesis that the right-handed proportion is lower in male than female, namely p_male &lt; p_female.</p>
</div>
</div>
</div>
<div id="multiple-testing-and-errors-to-be-updated" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Multiple testing and errors (To be updated)<a href="introHypoTest.html#multiple-testing-and-errors-to-be-updated" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Hypothetical null distribution.
Feel feel to try any null distribution, examples below</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="introHypoTest.html#cb85-1" tabindex="-1"></a><span class="do">## Example null distributions</span></span>
<span id="cb85-2"><a href="introHypoTest.html#cb85-2" tabindex="-1"></a><span class="co"># t, normal or anything. we use chi-squared distribution as an example</span></span>
<span id="cb85-3"><a href="introHypoTest.html#cb85-3" tabindex="-1"></a></span>
<span id="cb85-4"><a href="introHypoTest.html#cb85-4" tabindex="-1"></a>x_random <span class="ot">=</span> <span class="fu">rchisq</span>(<span class="at">n=</span><span class="dv">1000</span>, <span class="at">df=</span><span class="dv">3</span>)</span>
<span id="cb85-5"><a href="introHypoTest.html#cb85-5" tabindex="-1"></a>any_null_dist <span class="ot">=</span> <span class="fu">dchisq</span>(x_random, <span class="at">df=</span><span class="dv">3</span>)</span>
<span id="cb85-6"><a href="introHypoTest.html#cb85-6" tabindex="-1"></a></span>
<span id="cb85-7"><a href="introHypoTest.html#cb85-7" tabindex="-1"></a>pvals_null <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fu">pchisq</span>(x_random, <span class="at">df=</span><span class="dv">3</span>)</span></code></pre></div>
<div id="null-distribution-of-test-statistic" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Null distribution (of test statistic)<a href="introHypoTest.html#null-distribution-of-test-statistic" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="introHypoTest.html#cb86-1" tabindex="-1"></a><span class="co"># Null distribution of test statistic</span></span>
<span id="cb86-2"><a href="introHypoTest.html#cb86-2" tabindex="-1"></a><span class="fu">hist</span>(x_random)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-stat-null-1.png" width="60%" /></p>
</div>
<div id="null-distribution-of-p-value" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Null distribution of p value<a href="introHypoTest.html#null-distribution-of-p-value" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="introHypoTest.html#cb87-1" tabindex="-1"></a><span class="co"># Null distribution of test statistic</span></span>
<span id="cb87-2"><a href="introHypoTest.html#cb87-2" tabindex="-1"></a><span class="fu">hist</span>(pvals_null)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-pval-null-1.png" width="60%" /></p>
</div>
<div id="minimal-p-values-in-10-tests" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Minimal p values in 10 tests<a href="introHypoTest.html#minimal-p-values-in-10-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="introHypoTest.html#cb88-1" tabindex="-1"></a><span class="co"># We use matrix to group 100 trials into a column</span></span>
<span id="cb88-2"><a href="introHypoTest.html#cb88-2" tabindex="-1"></a><span class="co"># We then use apply() to calculate min value for each column</span></span>
<span id="cb88-3"><a href="introHypoTest.html#cb88-3" tabindex="-1"></a></span>
<span id="cb88-4"><a href="introHypoTest.html#cb88-4" tabindex="-1"></a>pval_null_mtx <span class="ot">=</span> <span class="fu">matrix</span>(pvals_null, <span class="at">nrow=</span><span class="dv">10</span>)</span>
<span id="cb88-5"><a href="introHypoTest.html#cb88-5" tabindex="-1"></a></span>
<span id="cb88-6"><a href="introHypoTest.html#cb88-6" tabindex="-1"></a>p_min_in10 <span class="ot">=</span> <span class="fu">apply</span>(pval_null_mtx, <span class="at">MARGIN=</span><span class="dv">2</span>, <span class="at">FUN=</span>min)</span></code></pre></div>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="introHypoTest.html#cb89-1" tabindex="-1"></a><span class="fu">hist</span>(p_min_in10)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/fig4-pmin-null-1.png" width="60%" /></p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="introHypoTest.html#cb90-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;Proportion of tests with min(p) &lt; 0.05:&#39;</span>, <span class="fu">mean</span>(p_min_in10 <span class="sc">&lt;</span> <span class="fl">0.05</span>)))</span>
<span id="cb90-2"><a href="introHypoTest.html#cb90-2" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Proportion of tests with min(p) &lt; 0.05: 0.43&quot;</span></span></code></pre></div>
</div>
</div>
<div id="power-analysis" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Explore power and sample size (optional)<a href="introHypoTest.html#power-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>Make a simulation of score: group A and B</li>
<li>B follows normal(mean=0, std=1); A follows normal(mean=0.1, std=1)</li>
<li>Generate 100 samples for each group, and do a t test, is difference significant? Please use set.seed(0) beforehand.</li>
<li>Try 3) again but general 3,00 samples this time, later 1,000 samples. What
do you find? Think the relation between power and sample size.</li>
</ol>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="introHypoTest.html#cb91-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">0</span>)</span>
<span id="cb91-2"><a href="introHypoTest.html#cb91-2" tabindex="-1"></a></span>
<span id="cb91-3"><a href="introHypoTest.html#cb91-3" tabindex="-1"></a>n_sample <span class="ot">=</span> <span class="dv">100</span> <span class="co"># change this value to 1000 and 10000</span></span>
<span id="cb91-4"><a href="introHypoTest.html#cb91-4" tabindex="-1"></a>xB <span class="ot">=</span> <span class="fu">rnorm</span>(n_sample)</span>
<span id="cb91-5"><a href="introHypoTest.html#cb91-5" tabindex="-1"></a>xA <span class="ot">=</span> <span class="fu">rnorm</span>(n_sample, <span class="at">mean=</span><span class="fl">0.1</span>)</span>
<span id="cb91-6"><a href="introHypoTest.html#cb91-6" tabindex="-1"></a><span class="fu">t.test</span>(xA, xB, <span class="at">var.equal =</span> <span class="cn">TRUE</span>)</span>
<span id="cb91-7"><a href="introHypoTest.html#cb91-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb91-8"><a href="introHypoTest.html#cb91-8" tabindex="-1"></a><span class="co">#&gt;  Two Sample t-test</span></span>
<span id="cb91-9"><a href="introHypoTest.html#cb91-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb91-10"><a href="introHypoTest.html#cb91-10" tabindex="-1"></a><span class="co">#&gt; data:  xA and xB</span></span>
<span id="cb91-11"><a href="introHypoTest.html#cb91-11" tabindex="-1"></a><span class="co">#&gt; t = 0.24294, df = 198, p-value = 0.8083</span></span>
<span id="cb91-12"><a href="introHypoTest.html#cb91-12" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true difference in means is not equal to 0</span></span>
<span id="cb91-13"><a href="introHypoTest.html#cb91-13" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb91-14"><a href="introHypoTest.html#cb91-14" tabindex="-1"></a><span class="co">#&gt;  -0.2261882  0.2897482</span></span>
<span id="cb91-15"><a href="introHypoTest.html#cb91-15" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb91-16"><a href="introHypoTest.html#cb91-16" tabindex="-1"></a><span class="co">#&gt;  mean of x  mean of y </span></span>
<span id="cb91-17"><a href="introHypoTest.html#cb91-17" tabindex="-1"></a><span class="co">#&gt; 0.05444844 0.02266845</span></span></code></pre></div>
</div>
<div id="summary" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Summary<a href="introHypoTest.html#summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A statistical hypothesis test is a method of statistical inference used to decide whether the data provide sufficient evidence to reject a particular hypothesis.</p>
<p>If you haven’t heard much about hypothesis testing or <code>p value</code>, you will hear
them again and again throughout of your whole journey of scientific study and
research. This might be your first time study it, but very likely will not be
the last time study it (via other courses or self-study). Its importance will
not be over rated, as misuse or misunderstanding can easily lead to errors in
scientific conclusions.</p>
<p>Before we started with examples, I will concisely summarize the key elements
here, so you can come back after read through this chapter (or anytime you need
to review).</p>
<blockquote>
<p>When to perform hypothesis test
- unsure if the observed results are caused by randomness (randomness are
everywhere but understanding or even admitting them are not always easy)</p>
</blockquote>
<blockquote>
<p>What are the key elements in the hypothesis test
- choose a proper testing statistics
- approximate the null distribution of the test statistic</p>
</blockquote>
<blockquote>
<p>What is the major challenge?
- understand the randomness of certain variables in the system / process
- estimate the testing statistic under the null hypothesis
- find a more effective test statistic (or the null distribution estimateion)</p>
</blockquote>
<p>That’s why you need to study more statistical course e.g., STAT2601 for the
mathematical of common variable distribution (or its more application-focused
alternative BIOF2013) and STAT2602 for advanced skills in deriving analytic or
asymptotic null distribution.</p>
<p>Namely, to determine how surprising my observation is, we can calculating a p-value!</p>
<p>There are many circumstances where we simply want to check whether an observation looks like it is compatible with the null hypothesis, <span class="math inline">\(H_{0}\)</span>.</p>
<p>Having decided on a significance level <span class="math inline">\(\alpha\)</span> and whether the situation warrants a one-tailed or a two-tailed test, we can use the cdf of the null distribution to calculate a p-value for the observation.</p>
<p><strong>Acknowledgement</strong>: examples are from Dr John Pinney <a href="https://github.com/johnpinney/sampling_and_hypothesis_testing/blob/master/python_version/hypothesis_testing_python.ipynb">link here</a></p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introR.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introLinearReg.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/statbiomed/BMDS-book/tree/main/01-introduction.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BMDatSci.pdf", "BMDatSci.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
