<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Population Genetics and Diseases | Biomedical Data Science - introduction with case studies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Population Genetics and Diseases | Biomedical Data Science - introduction with case studies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Population Genetics and Diseases | Biomedical Data Science - introduction with case studies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="BIOF1001 teaching team" />


<meta name="date" content="2023-11-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cancer.html"/>
<link rel="next" href="install.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">Biomedical Data Science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#introduction-for-readers"><i class="fa fa-check"></i>Introduction for readers</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-you-will-learn-from-this-coursebook"><i class="fa fa-check"></i>What you will learn from this course/book</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-recommend-you-do-while-reading-this-book"><i class="fa fa-check"></i>What we recommend you do while reading this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#other-reference-books"><i class="fa fa-check"></i>Other reference books</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="part"><span><b>I Data Science Foundations</b></span></li>
<li class="chapter" data-level="1" data-path="introR.html"><a href="introR.html"><i class="fa fa-check"></i><b>1</b> Introduction to R programming</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introR.html"><a href="introR.html#data-types"><i class="fa fa-check"></i><b>1.1</b> Data types</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introR.html"><a href="introR.html#nemeric-or-double"><i class="fa fa-check"></i><b>1.1.1</b> nemeric (or double)</a></li>
<li class="chapter" data-level="1.1.2" data-path="introR.html"><a href="introR.html#integer"><i class="fa fa-check"></i><b>1.1.2</b> integer</a></li>
<li class="chapter" data-level="1.1.3" data-path="introR.html"><a href="introR.html#logical"><i class="fa fa-check"></i><b>1.1.3</b> logical</a></li>
<li class="chapter" data-level="1.1.4" data-path="introR.html"><a href="introR.html#character"><i class="fa fa-check"></i><b>1.1.4</b> character</a></li>
<li class="chapter" data-level="1.1.5" data-path="introR.html"><a href="introR.html#memeory-usage"><i class="fa fa-check"></i><b>1.1.5</b> Memeory usage</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introR.html"><a href="introR.html#data-structures"><i class="fa fa-check"></i><b>1.2</b> Data structures</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introR.html"><a href="introR.html#vector"><i class="fa fa-check"></i><b>1.2.1</b> Vector</a></li>
<li class="chapter" data-level="1.2.2" data-path="introR.html"><a href="introR.html#matrix"><i class="fa fa-check"></i><b>1.2.2</b> Matrix</a></li>
<li class="chapter" data-level="1.2.3" data-path="introR.html"><a href="introR.html#list"><i class="fa fa-check"></i><b>1.2.3</b> List</a></li>
<li class="chapter" data-level="1.2.4" data-path="introR.html"><a href="introR.html#data-frame"><i class="fa fa-check"></i><b>1.2.4</b> Data Frame</a></li>
<li class="chapter" data-level="1.2.5" data-path="introR.html"><a href="introR.html#factor-vs-vector"><i class="fa fa-check"></i><b>1.2.5</b> Factor vs vector</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="introR.html"><a href="introR.html#read-and-write-files-tables"><i class="fa fa-check"></i><b>1.3</b> Read and write files (tables)</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introR.html"><a href="introR.html#read-file"><i class="fa fa-check"></i><b>1.3.1</b> Read file</a></li>
<li class="chapter" data-level="1.3.2" data-path="introR.html"><a href="introR.html#write-file"><i class="fa fa-check"></i><b>1.3.2</b> Write file</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="introR.html"><a href="introR.html#functions-and-packages"><i class="fa fa-check"></i><b>1.4</b> Functions and Packages</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="introR.html"><a href="introR.html#install-packages"><i class="fa fa-check"></i><b>1.4.1</b> Install packages</a></li>
<li class="chapter" data-level="1.4.2" data-path="introR.html"><a href="introR.html#apply-function-repeatly"><i class="fa fa-check"></i><b>1.4.2</b> Apply function repeatly</a></li>
<li class="chapter" data-level="1.4.3" data-path="introR.html"><a href="introR.html#pattern-match"><i class="fa fa-check"></i><b>1.4.3</b> Pattern match</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="introR.html"><a href="introR.html#flow-control"><i class="fa fa-check"></i><b>1.5</b> Flow Control</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="introR.html"><a href="introR.html#logical-operator"><i class="fa fa-check"></i><b>1.5.1</b> Logical operator</a></li>
<li class="chapter" data-level="1.5.2" data-path="introR.html"><a href="introR.html#if-else-statements"><i class="fa fa-check"></i><b>1.5.2</b> if-else statements</a></li>
<li class="chapter" data-level="1.5.3" data-path="introR.html"><a href="introR.html#for-loop"><i class="fa fa-check"></i><b>1.5.3</b> for-loop</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="introR.html"><a href="introR.html#plotting"><i class="fa fa-check"></i><b>1.6</b> Plotting</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="introR.html"><a href="introR.html#datasets"><i class="fa fa-check"></i><b>1.6.1</b> datasets</a></li>
<li class="chapter" data-level="1.6.2" data-path="introR.html"><a href="introR.html#basic-plotting"><i class="fa fa-check"></i><b>1.6.2</b> Basic plotting</a></li>
<li class="chapter" data-level="1.6.3" data-path="introR.html"><a href="introR.html#ggplot2"><i class="fa fa-check"></i><b>1.6.3</b> ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="introR.html"><a href="introR.html#scientific-computating"><i class="fa fa-check"></i><b>1.7</b> Scientific computating</a>
<ul>
<li class="chapter" data-level="1.7.1" data-path="introR.html"><a href="introR.html#orders-of-operators"><i class="fa fa-check"></i><b>1.7.1</b> Orders of operators</a></li>
<li class="chapter" data-level="1.7.2" data-path="introR.html"><a href="introR.html#functions-for-statistics"><i class="fa fa-check"></i><b>1.7.2</b> Functions for statistics</a></li>
<li class="chapter" data-level="1.7.3" data-path="introR.html"><a href="introR.html#correlation"><i class="fa fa-check"></i><b>1.7.3</b> Correlation</a></li>
<li class="chapter" data-level="1.7.4" data-path="introR.html"><a href="introR.html#hypothesis-testing-t-test"><i class="fa fa-check"></i><b>1.7.4</b> Hypothesis testing (t test)</a></li>
<li class="chapter" data-level="1.7.5" data-path="introR.html"><a href="introR.html#regression"><i class="fa fa-check"></i><b>1.7.5</b> Regression</a></li>
<li class="chapter" data-level="1.7.6" data-path="introR.html"><a href="introR.html#resource-links"><i class="fa fa-check"></i><b>1.7.6</b> Resource links</a></li>
<li class="chapter" data-level="1.7.7" data-path="introR.html"><a href="introR.html#coding-styling"><i class="fa fa-check"></i><b>1.7.7</b> Coding styling</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="introR.html"><a href="introR.html#exercises"><i class="fa fa-check"></i><b>1.8</b> Exercises</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="introR.html"><a href="introR.html#exercises-p1"><i class="fa fa-check"></i><b>1.8.1</b> Part 1. Basics (~40min)</a></li>
<li class="chapter" data-level="1.8.2" data-path="introR.html"><a href="introR.html#exercises-p2"><i class="fa fa-check"></i><b>1.8.2</b> Part 2. Making plotting (~40min)</a></li>
<li class="chapter" data-level="1.8.3" data-path="introR.html"><a href="introR.html#exercises-p3"><i class="fa fa-check"></i><b>1.8.3</b> Part 3. For loop and repeating processing (~40min)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introHypoTest.html"><a href="introHypoTest.html"><i class="fa fa-check"></i><b>2</b> Introduction to Hypothesis testing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-testing-and-p-value"><i class="fa fa-check"></i><b>2.1</b> Hypothesis testing and <em>p</em> value</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-1-probability-of-rolling-a-six"><i class="fa fa-check"></i><b>2.1.1</b> Example 1: probability of rolling a six?</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#permutation-test"><i class="fa fa-check"></i><b>2.2</b> Permutation test</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-2-difference-in-birth-weight"><i class="fa fa-check"></i><b>2.2.1</b> Example 2: difference in birth weight</a></li>
<li class="chapter" data-level="2.2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-approximated-by-resampling"><i class="fa fa-check"></i><b>2.2.2</b> Null distribution approximated by resampling</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introHypoTest.html"><a href="introHypoTest.html#t-test"><i class="fa fa-check"></i><b>2.3</b> <em>t</em> test</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introHypoTest.html"><a href="introHypoTest.html#derivation-of-t-distribution"><i class="fa fa-check"></i><b>2.3.1</b> Derivation of t distribution</a></li>
<li class="chapter" data-level="2.3.2" data-path="introHypoTest.html"><a href="introHypoTest.html#direct-use-of-t.test"><i class="fa fa-check"></i><b>2.3.2</b> Direct use of <code>t.test()</code></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="introHypoTest.html"><a href="introHypoTest.html#glm-test"><i class="fa fa-check"></i><b>2.4</b> GLM test</a></li>
<li class="chapter" data-level="2.5" data-path="introHypoTest.html"><a href="introHypoTest.html#multiple-testing"><i class="fa fa-check"></i><b>2.5</b> Multiple testing</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-test-statistic"><i class="fa fa-check"></i><b>2.5.1</b> Null distribution (of test statistic)</a></li>
<li class="chapter" data-level="2.5.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-p-value"><i class="fa fa-check"></i><b>2.5.2</b> Null distribution of p value</a></li>
<li class="chapter" data-level="2.5.3" data-path="introHypoTest.html"><a href="introHypoTest.html#minimal-p-values-in-10-tests"><i class="fa fa-check"></i><b>2.5.3</b> Minimal p values in 10 tests</a></li>
<li class="chapter" data-level="2.5.4" data-path="introHypoTest.html"><a href="introHypoTest.html#homework"><i class="fa fa-check"></i><b>2.5.4</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introLinearReg.html"><a href="introLinearReg.html"><i class="fa fa-check"></i><b>3</b> Introduction to Linear Regression</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#linear-regression-using-simulated-data"><i class="fa fa-check"></i><b>3.1</b> Linear Regression Using Simulated Data</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="introLinearReg.html"><a href="introLinearReg.html#simulating-data"><i class="fa fa-check"></i><b>3.1.1</b> Simulating data:</a></li>
<li class="chapter" data-level="3.1.2" data-path="introLinearReg.html"><a href="introLinearReg.html#model-efficacy"><i class="fa fa-check"></i><b>3.1.2</b> Model efficacy</a></li>
<li class="chapter" data-level="3.1.3" data-path="introLinearReg.html"><a href="introLinearReg.html#r-squared"><i class="fa fa-check"></i><b>3.1.3</b> <em>R-Squared</em></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#least-squares-using-simulated-data"><i class="fa fa-check"></i><b>3.2</b> Least Squares Using Simulated Data</a></li>
<li class="chapter" data-level="3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#diagnostic-check-of-a-fitted-regression-model"><i class="fa fa-check"></i><b>3.3</b> Diagnostic check of a fitted regression model</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#residual-standard-errors-rse"><i class="fa fa-check"></i><b>3.3.1</b> Residual Standard Errors (RSE)</a></li>
<li class="chapter" data-level="3.3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#p-values"><i class="fa fa-check"></i><b>3.3.2</b> p-values</a></li>
<li class="chapter" data-level="3.3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#f-statistics"><i class="fa fa-check"></i><b>3.3.3</b> F-statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introLinearReg.html"><a href="introLinearReg.html#simple-linear-regression-with-lm-function"><i class="fa fa-check"></i><b>3.4</b> Simple Linear Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="3.5" data-path="introLinearReg.html"><a href="introLinearReg.html#multiple-regression-with-lm-function"><i class="fa fa-check"></i><b>3.5</b> Multiple Regression with <code>lm</code> function</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introClassifier.html"><a href="introClassifier.html"><i class="fa fa-check"></i><b>4</b> Introduction to Classification</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introClassifier.html"><a href="introClassifier.html#visualise-logistic-and-logit-functions"><i class="fa fa-check"></i><b>4.1</b> Visualise logistic and logit functions</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="introClassifier.html"><a href="introClassifier.html#logistic-function"><i class="fa fa-check"></i><b>4.1.1</b> Logistic function</a></li>
<li class="chapter" data-level="4.1.2" data-path="introClassifier.html"><a href="introClassifier.html#logit-function"><i class="fa fa-check"></i><b>4.1.2</b> Logit function</a></li>
<li class="chapter" data-level="4.1.3" data-path="introClassifier.html"><a href="introClassifier.html#visualise-the-distribution"><i class="fa fa-check"></i><b>4.1.3</b> Visualise the distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="introClassifier.html"><a href="introClassifier.html#logistic-regression-on-diabetes"><i class="fa fa-check"></i><b>4.2</b> Logistic regression on Diabetes</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="introClassifier.html"><a href="introClassifier.html#load-pima-indians-diabetes-database"><i class="fa fa-check"></i><b>4.2.1</b> Load Pima Indians Diabetes Database</a></li>
<li class="chapter" data-level="4.2.2" data-path="introClassifier.html"><a href="introClassifier.html#fit-logistic-regression"><i class="fa fa-check"></i><b>4.2.2</b> Fit logistic regression</a></li>
<li class="chapter" data-level="4.2.3" data-path="introClassifier.html"><a href="introClassifier.html#assess-on-test-data"><i class="fa fa-check"></i><b>4.2.3</b> Assess on test data</a></li>
<li class="chapter" data-level="4.2.4" data-path="introClassifier.html"><a href="introClassifier.html#model-selection-and-diagnosis"><i class="fa fa-check"></i><b>4.2.4</b> Model selection and diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="introClassifier.html"><a href="introClassifier.html#cross-validation"><i class="fa fa-check"></i><b>4.3</b> Cross-validation</a></li>
<li class="chapter" data-level="4.4" data-path="introClassifier.html"><a href="introClassifier.html#more-assessment-metrics"><i class="fa fa-check"></i><b>4.4</b> More assessment metrics</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="introClassifier.html"><a href="introClassifier.html#two-types-of-error"><i class="fa fa-check"></i><b>4.4.1</b> Two types of error</a></li>
<li class="chapter" data-level="4.4.2" data-path="introClassifier.html"><a href="introClassifier.html#roc-curve"><i class="fa fa-check"></i><b>4.4.2</b> ROC curve</a></li>
<li class="chapter" data-level="4.4.3" data-path="introClassifier.html"><a href="introClassifier.html#homework-1"><i class="fa fa-check"></i><b>4.4.3</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Biomedical Data Modules</b></span></li>
<li class="chapter" data-level="5" data-path="image-digital.html"><a href="image-digital.html"><i class="fa fa-check"></i><b>5</b> Medical Image and Digital Health</a></li>
<li class="chapter" data-level="6" data-path="infectious-dis.html"><a href="infectious-dis.html"><i class="fa fa-check"></i><b>6</b> Infectious Disease Informatics</a></li>
<li class="chapter" data-level="7" data-path="cancer.html"><a href="cancer.html"><i class="fa fa-check"></i><b>7</b> Cancer genomics and epidemiology</a>
<ul>
<li class="chapter" data-level="7.1" data-path="cancer.html"><a href="cancer.html#cancer-case1"><i class="fa fa-check"></i><b>7.1</b> Case study 1: analysis of cBioportal mutation data</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="cancer.html"><a href="cancer.html#exploratory-analysis"><i class="fa fa-check"></i><b>7.1.1</b> Exploratory analysis</a></li>
<li class="chapter" data-level="7.1.2" data-path="cancer.html"><a href="cancer.html#statistical-analysis"><i class="fa fa-check"></i><b>7.1.2</b> Statistical analysis</a></li>
<li class="chapter" data-level="7.1.3" data-path="cancer.html"><a href="cancer.html#literature-search"><i class="fa fa-check"></i><b>7.1.3</b> Literature search</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="cancer.html"><a href="cancer.html#cancer-case2"><i class="fa fa-check"></i><b>7.2</b> Case study 2: Cancer Epidemiology</a></li>
<li class="chapter" data-level="7.3" data-path="cancer.html"><a href="cancer.html#scenario"><i class="fa fa-check"></i><b>7.3</b> 1. Scenario</a></li>
<li class="chapter" data-level="7.4" data-path="cancer.html"><a href="cancer.html#hong-kong-population"><i class="fa fa-check"></i><b>7.4</b> 2. Hong Kong population</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="cancer.html"><a href="cancer.html#download-population-data"><i class="fa fa-check"></i><b>7.4.1</b> Download population data</a></li>
<li class="chapter" data-level="7.4.2" data-path="cancer.html"><a href="cancer.html#format-data-for-plotting"><i class="fa fa-check"></i><b>7.4.2</b> Format data for plotting</a></li>
<li class="chapter" data-level="7.4.3" data-path="cancer.html"><a href="cancer.html#plotting-population-data"><i class="fa fa-check"></i><b>7.4.3</b> Plotting population data</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="cancer.html"><a href="cancer.html#cancer-registry-data"><i class="fa fa-check"></i><b>7.5</b> 3. Cancer registry data</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="cancer.html"><a href="cancer.html#download-cancer-registry-data"><i class="fa fa-check"></i><b>7.5.1</b> Download cancer registry data</a></li>
<li class="chapter" data-level="7.5.2" data-path="cancer.html"><a href="cancer.html#a.-visualise-changes-in-incidence-and-mortality"><i class="fa fa-check"></i><b>7.5.2</b> 3a. Visualise changes in incidence and mortality</a></li>
<li class="chapter" data-level="7.5.3" data-path="cancer.html"><a href="cancer.html#b.-calculate-cancer-risk-and-mortality-rate"><i class="fa fa-check"></i><b>7.5.3</b> 3b. Calculate cancer risk and mortality rate</a></li>
<li class="chapter" data-level="7.5.4" data-path="cancer.html"><a href="cancer.html#c.-mortality-incidence-ratio"><i class="fa fa-check"></i><b>7.5.4</b> 3c. Mortality-incidence ratio</a></li>
<li class="chapter" data-level="7.5.5" data-path="cancer.html"><a href="cancer.html#d.-paired-t-test-on-mortality-incidence-ratio-change"><i class="fa fa-check"></i><b>7.5.5</b> 3d. Paired t-test on mortality-incidence ratio change</a></li>
<li class="chapter" data-level="7.5.6" data-path="cancer.html"><a href="cancer.html#e.-childhood-versus-elderly-cancers"><i class="fa fa-check"></i><b>7.5.6</b> 3e. Childhood versus elderly cancers</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="cancer.html"><a href="cancer.html#existing-cancer-funding-and-publication-data"><i class="fa fa-check"></i><b>7.6</b> 4. Existing cancer funding and publication data</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="cancer.html"><a href="cancer.html#download-hmrf-grants-and-pubmed-data"><i class="fa fa-check"></i><b>7.6.1</b> Download HMRF grants and Pubmed data</a></li>
<li class="chapter" data-level="7.6.2" data-path="cancer.html"><a href="cancer.html#make-word-cloud-for-grants"><i class="fa fa-check"></i><b>7.6.2</b> Make word cloud for grants</a></li>
<li class="chapter" data-level="7.6.3" data-path="cancer.html"><a href="cancer.html#make-compare-grant-funding-with-incidence-and-mortality"><i class="fa fa-check"></i><b>7.6.3</b> Make compare grant funding with incidence and mortality</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="cancer.html"><a href="cancer.html#open-disucssion"><i class="fa fa-check"></i><b>7.7</b> 5. Open disucssion</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="pop-genetics.html"><a href="pop-genetics.html"><i class="fa fa-check"></i><b>8</b> Population Genetics and Diseases</a>
<ul>
<li class="chapter" data-level="8.1" data-path="pop-genetics.html"><a href="pop-genetics.html#case-study-1-heritability-and-human-traits"><i class="fa fa-check"></i><b>8.1</b> Case study 1: Heritability and human traits</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="pop-genetics.html"><a href="pop-genetics.html#part-1"><i class="fa fa-check"></i><b>8.1.1</b> Part 1</a></li>
<li class="chapter" data-level="8.1.2" data-path="pop-genetics.html"><a href="pop-genetics.html#part-2"><i class="fa fa-check"></i><b>8.1.2</b> Part 2</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i>Appendix A: Install R &amp; RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.1-install-r"><i class="fa fa-check"></i>A.1 Install R</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-windows"><i class="fa fa-check"></i>R on Windows</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-macos"><i class="fa fa-check"></i>R on macOS</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-linux-ubuntu"><i class="fa fa-check"></i>R on Linux (Ubuntu)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.2-install-rstudio"><i class="fa fa-check"></i>A.2 Install RStudio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.3-use-r-inside-rstudio"><i class="fa fa-check"></i>A.3 Use R inside RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-studio"><i class="fa fa-check"></i>R studio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#some-general-knowledge"><i class="fa fa-check"></i>Some general knowledge</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#install-packages-1"><i class="fa fa-check"></i>Install packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a4.-cloud-computing"><i class="fa fa-check"></i>A4. Cloud computing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Data Science - introduction with case studies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pop-genetics" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Population Genetics and Diseases<a href="pop-genetics.html#pop-genetics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="case-study-1-heritability-and-human-traits" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Case study 1: Heritability and human traits<a href="pop-genetics.html#case-study-1-heritability-and-human-traits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="part-1" class="section level3 hasAnchor" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> Part 1<a href="pop-genetics.html#part-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Scenario:</strong> You are a researcher working on a twin study on cardiovascular traits to assess the genetic and environmental contribution relevant to metabolism and cardiovascular disease risk. You have recruited a cohort of volunteer adult twins of the same ancestry. The volunteers have undergone a series of baseline clinical evaluations and performed genotyping on a panel of single nucleotide polymorphisms that may be associated with the traits.</p>
<div id="questions-for-discussion" class="section level4 hasAnchor" number="8.1.1.1">
<h4><span class="header-section-number">8.1.1.1</span> Questions for Discussion<a href="pop-genetics.html#questions-for-discussion" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<strong>Q1.</strong> Besides the clinical measurements, what data do you need to collect from the subjects?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Sex</li>
<li>Age</li>
<li>Other confounding factors, e.g. BMI, blood pressure, smoking status, etc.</li>
</ul>
</details>
<strong>Q2.</strong> How is genotype data represented for statistical genetic analysis?<br />

<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Allele: 0/1, 1/2, A/C, etc</li>
<li>Genotype: 0 0, 0 1, 1 0, 1 1</li>
<li>Genotype probabilities: P(0/0)=0, P(0/1)=1, P(1/1)=0</li>
<li>Genotype dosage: 0/1/2, 0.678 (continuous from 0-1 or 0-2)</li>
</ul>
</details>
<strong>Q3.</strong> How can you test for association between genotypes and phenotypes (binary and quantitative)?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Allelic chi-square test</li>
<li>Fisher’s exact test</li>
<li>Linear/Logistic regression</li>
<li>Linear mixed model</li>
</ul>
</details>
</div>
<div id="hands-on-exercise-association-test" class="section level4 hasAnchor" number="8.1.1.2">
<h4><span class="header-section-number">8.1.1.2</span> Hands-on exercise : Association test<a href="pop-genetics.html#hands-on-exercise-association-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now, you are given a dataset of age- and sex-matched twin cohort with two cardiovascular phenotypes and 5 quantitative trait loci (QTL). Data set and template notebook are available on Moodle (recommended) and also on this
<a href="https://github.com/StatBiomed/BMDS-book/blob/main/notebooks/module4-genetics/dataTwin2023.dat">GitHub Repo</a>.</p>
<p><strong>The information for columns:</strong></p>
<ul>
<li>zygosity: 1 for monozygotic (MZ) and 2 for dizygotic (DZ) twin</li>
<li><code>T1QTL_A[1-5]</code> and <code>T2QTL_A[1-5]</code>: 5 quantitative loci (A1-A5) in additive coding for Twin 1 (T1) and Twin 2 (T2) respectively</li>
<li>The same 5 QTL (D1-D5) in dominance coding for T1 and T2</li>
<li>Phenotype scores of T1 and T2 for the two quantitative cardiovascular traits</li>
</ul>
<p><strong>Download the data <code>dataTwin.dat</code> to your working directory.</strong> Start the RStudio program and set the working directory.</p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="pop-genetics.html#cb168-1" tabindex="-1"></a>dataTwin <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;dataTwin2023.dat&quot;</span>,<span class="at">h=</span>T)</span></code></pre></div>
<p><strong>Exploratory analysis</strong></p>
<ul>
<li>A1-5: The QTLs are biallelic with two alleles A and a. The genotypes aa, Aa, and AA are coded additively as 0 (aa), 1 (Aa) and 2 (AA).</li>
<li>D1-5: The genotypes aa, Aa, and AA are coded as 0 (aa), 1 (Aa) and 0 (AA).</li>
</ul>
<p><strong>Q1.</strong> How many MZ and DZ volunteers are there?</p>
<p><strong>Q2.</strong> How are the genotypes represented?</p>
<p><strong>Q3.</strong> Are the QTL independent of each other?</p>
<p><strong>Q4.</strong> Are there outliers in phenotypes?</p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="pop-genetics.html#cb169-1" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>zygosity)        <span class="co"># Q1: shows number of MZ and DZ twin pairs</span></span>
<span id="cb169-2"><a href="pop-genetics.html#cb169-2" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>T1QTL_A1)        <span class="co"># Q2: shows the distribution of QTL_A1</span></span>
<span id="cb169-3"><a href="pop-genetics.html#cb169-3" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>T1QTL_D1)        <span class="co"># Q2: shows the distribution of QTL_D1</span></span>
<span id="cb169-4"><a href="pop-genetics.html#cb169-4" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>T1QTL_A1, dataTwin<span class="sc">$</span>T1QTL_D1)     <span class="co"># Q2: shows the distribution of QTL_A1 in relation to QTL_D1</span></span>
<span id="cb169-5"><a href="pop-genetics.html#cb169-5" tabindex="-1"></a><span class="fu">cor</span>(dataTwin[,<span class="dv">2</span><span class="sc">:</span><span class="dv">11</span>])                            <span class="co"># Q3: shows the correlation between QTL_As</span></span>
<span id="cb169-6"><a href="pop-genetics.html#cb169-6" tabindex="-1"></a><span class="fu">cor</span>(dataTwin[,<span class="dv">2</span><span class="sc">:</span><span class="dv">11</span>])<span class="sc">&gt;</span><span class="fl">0.2</span></span>
<span id="cb169-7"><a href="pop-genetics.html#cb169-7" tabindex="-1"></a><span class="fu">apply</span>(dataTwin[<span class="dv">22</span><span class="sc">:</span><span class="dv">25</span>],<span class="dv">2</span>,<span class="cf">function</span>(x){ <span class="fu">any</span>(x <span class="sc">&lt;</span> (<span class="fu">mean</span>(x) <span class="sc">-</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span>(x))) })  <span class="co"># Q4: any outlier &gt;4 SD from the mean for the two quantitative phenotypes</span></span></code></pre></div>
<p><strong>Association test</strong></p>
<p>Test for association between QTL and pheno1 for T1</p>
<ul>
<li>Regress <code>pheno1_T1</code> on <code>T1QTL_A1</code> to estimate the proportion of variance explained (R2).</li>
<li>Model: pheno1_T1 = b0 + b1* T1QTL_A1 + e</li>
<li>Calculate the conditional mean of phenotype (i.e. phenotypic mean conditional genotype)</li>
</ul>
<p>If the relationship between the QTL and the phenotype is perfectly linear, the regression line should pass through the conditional means (c_means), and the differences between the conditional means should
be about equal.</p>
<p><strong>Q5.</strong> What are the values of b0, b1? Is QTL1 significant associated with the phenotype at alpha&lt;0.01 (multiple testing of 5 loci)?</p>
<p><strong>Q6.</strong> What is the proportion of phenotypic variance explained?</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="pop-genetics.html#cb170-1" tabindex="-1"></a>linA1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1<span class="sc">~</span>T1QTL_A1, <span class="at">data=</span>dataTwin)</span>
<span id="cb170-2"><a href="pop-genetics.html#cb170-2" tabindex="-1"></a><span class="fu">summary</span>(linA1)</span>
<span id="cb170-3"><a href="pop-genetics.html#cb170-3" tabindex="-1"></a><span class="fu">summary</span>(linA1)<span class="sc">$</span>r.squared            <span class="co"># proportion of explained variance by additive component</span></span>
<span id="cb170-4"><a href="pop-genetics.html#cb170-4" tabindex="-1"></a></span>
<span id="cb170-5"><a href="pop-genetics.html#cb170-5" tabindex="-1"></a>c_means <span class="ot">&lt;-</span> <span class="fu">by</span>(dataTwin<span class="sc">$</span>pheno1_T1,dataTwin<span class="sc">$</span>T1QTL_A1,mean)</span>
<span id="cb170-6"><a href="pop-genetics.html#cb170-6" tabindex="-1"></a><span class="fu">plot</span>(dataTwin<span class="sc">$</span>pheno1_T1 <span class="sc">~</span> dataTwin<span class="sc">$</span>T1QTL_A1, <span class="at">col=</span><span class="st">&#39;grey&#39;</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>))</span>
<span id="cb170-7"><a href="pop-genetics.html#cb170-7" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>), c_means, <span class="at">type=</span><span class="st">&quot;p&quot;</span>, <span class="at">col=</span><span class="dv">6</span>, <span class="at">lwd=</span><span class="dv">8</span>)</span>
<span id="cb170-8"><a href="pop-genetics.html#cb170-8" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">sort</span>(dataTwin<span class="sc">$</span>T1QTL_A1),<span class="fu">sort</span>(linA1<span class="sc">$</span>fitted.values), <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">col=</span><span class="st">&quot;dark green&quot;</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span></code></pre></div>
<p>To test this “linearity”, we can use the dominance coding of the QTL and add the dominance term to the regression model.</p>
<ul>
<li>Model: pheno1_T1 = b0 + b1* T1QTL_A1 + b2* T1QTL_D1 + e</li>
<li>Repeat for T2.</li>
</ul>
<p><strong>Q7.</strong> Why can’t we analyse T1 and T2 together?</p>
<p><strong>Q8.</strong> Is there a dominance effect?</p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="pop-genetics.html#cb171-1" tabindex="-1"></a>linAD1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A1 <span class="sc">+</span> T1QTL_D1, <span class="at">data=</span>dataTwin)</span>
<span id="cb171-2"><a href="pop-genetics.html#cb171-2" tabindex="-1"></a><span class="fu">summary</span>(linA1)  <span class="co"># results lm(phenoT1~T1QTL_A1)</span></span>
<span id="cb171-3"><a href="pop-genetics.html#cb171-3" tabindex="-1"></a><span class="fu">summary</span>(linAD1) <span class="co"># results lm(phenoT1~T1QTL_A1+T1QTL_D1)</span></span>
<span id="cb171-4"><a href="pop-genetics.html#cb171-4" tabindex="-1"></a></span>
<span id="cb171-5"><a href="pop-genetics.html#cb171-5" tabindex="-1"></a><span class="fu">plot</span>(dataTwin<span class="sc">$</span>pheno1_T1 <span class="sc">~</span> dataTwin<span class="sc">$</span>T1QTL_A1, <span class="at">col=</span><span class="st">&#39;grey&#39;</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>))</span>
<span id="cb171-6"><a href="pop-genetics.html#cb171-6" tabindex="-1"></a><span class="fu">abline</span>(linA1, <span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb171-7"><a href="pop-genetics.html#cb171-7" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>), c_means, <span class="at">type=</span><span class="st">&#39;p&#39;</span>, <span class="at">col=</span><span class="dv">6</span>, <span class="at">lwd=</span><span class="dv">8</span>)</span>
<span id="cb171-8"><a href="pop-genetics.html#cb171-8" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">sort</span>(dataTwin<span class="sc">$</span>T1QTL_A1),<span class="fu">sort</span>(linA1<span class="sc">$</span>fitted.values), <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">col=</span><span class="st">&quot;dark green&quot;</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb171-9"><a href="pop-genetics.html#cb171-9" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">sort</span>(dataTwin<span class="sc">$</span>T1QTL_A1),<span class="fu">sort</span>(linAD1<span class="sc">$</span>fitted.values), <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span></code></pre></div>
<p><strong>Q9.</strong> Repeat for the other 4 QTL and determine which QTL shows strongest association with the phenotype T1</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="pop-genetics.html#cb172-1" tabindex="-1"></a>allQTL_A_T1 <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span></span>
<span id="cb172-2"><a href="pop-genetics.html#cb172-2" tabindex="-1"></a>cpheno1_T1 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(dataTwin)<span class="sc">==</span><span class="st">&quot;pheno1_T1&quot;</span>)</span>
<span id="cb172-3"><a href="pop-genetics.html#cb172-3" tabindex="-1"></a><span class="do">## Additive</span></span>
<span id="cb172-4"><a href="pop-genetics.html#cb172-4" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">lapply</span>(allQTL_A_T1,<span class="cf">function</span>(x){ fstat<span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> ., <span class="at">data=</span>dataTwin[,<span class="fu">c</span>(x,cpheno1_T1)]))<span class="sc">$</span>fstatistic;  <span class="fu">pf</span>(fstat[<span class="dv">1</span>],fstat[<span class="dv">2</span>],fstat[<span class="dv">3</span>],<span class="at">lower.tail =</span> F) }))</span>
<span id="cb172-5"><a href="pop-genetics.html#cb172-5" tabindex="-1"></a><span class="do">## Dominance</span></span>
<span id="cb172-6"><a href="pop-genetics.html#cb172-6" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">lapply</span>(allQTL_A_T1,<span class="cf">function</span>(x){ fstat<span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> ., <span class="at">data=</span>dataTwin[,<span class="fu">c</span>(x,x<span class="sc">+</span><span class="dv">10</span>,cpheno1_T1)]))<span class="sc">$</span>fstatistic;  <span class="fu">pf</span>(fstat[<span class="dv">1</span>],fstat[<span class="dv">2</span>],fstat[<span class="dv">3</span>],<span class="at">lower.tail =</span> F) }))</span>
<span id="cb172-7"><a href="pop-genetics.html#cb172-7" tabindex="-1"></a></span>
<span id="cb172-8"><a href="pop-genetics.html#cb172-8" tabindex="-1"></a><span class="co">#Q9: QTL3 shows the strongest association with P=7.771588e-25</span></span>
<span id="cb172-9"><a href="pop-genetics.html#cb172-9" tabindex="-1"></a>linAD3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A3 <span class="sc">+</span> T1QTL_D3, <span class="at">data=</span>dataTwin)</span>
<span id="cb172-10"><a href="pop-genetics.html#cb172-10" tabindex="-1"></a><span class="fu">summary</span>(linAD3) <span class="co"># results lm(phenoT1~T1QTL_A1+T1QTL_D1)</span></span></code></pre></div>
<p>If the subjects with top 5% of the phenotype score are considered as cases, perform case-control association test for most significant SNP (from Q9) and interpret the result.</p>
<p><strong>Q10.</strong> What are the odds ratio, p-value, and 95% confidence interval?</p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="pop-genetics.html#cb173-1" tabindex="-1"></a>quant05 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(<span class="fu">c</span>(dataTwin<span class="sc">$</span>pheno1_T1,dataTwin<span class="sc">$</span>pheno1_T2),<span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.05</span>))</span>
<span id="cb173-2"><a href="pop-genetics.html#cb173-2" tabindex="-1"></a>dataTwin<span class="sc">$</span>CaseT1 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dataTwin<span class="sc">$</span>pheno1_T1<span class="sc">&gt;</span>quant05[<span class="dv">20</span>]) </span>
<span id="cb173-3"><a href="pop-genetics.html#cb173-3" tabindex="-1"></a>dataTwin<span class="sc">$</span>CaseT2 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dataTwin<span class="sc">$</span>pheno1_T2<span class="sc">&gt;</span>quant05[<span class="dv">20</span>])</span>
<span id="cb173-4"><a href="pop-genetics.html#cb173-4" tabindex="-1"></a>logisticAD1 <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">glm</span>(CaseT1 <span class="sc">~</span> T1QTL_A3 <span class="sc">+</span> T1QTL_D3, <span class="at">data=</span>dataTwin, <span class="at">family=</span><span class="st">&quot;binomial&quot;</span>))</span>
<span id="cb173-5"><a href="pop-genetics.html#cb173-5" tabindex="-1"></a><span class="fu">exp</span>(logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>])                                     <span class="co"># odds ratio</span></span>
<span id="cb173-6"><a href="pop-genetics.html#cb173-6" tabindex="-1"></a><span class="fu">exp</span>(logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>]<span class="sc">-</span><span class="fl">1.96</span><span class="sc">*</span>logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>])  <span class="co"># lower 95% confidence interval</span></span>
<span id="cb173-7"><a href="pop-genetics.html#cb173-7" tabindex="-1"></a><span class="fu">exp</span>(logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>]<span class="sc">+</span><span class="fl">1.96</span><span class="sc">*</span>logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>])  <span class="co"># upper 95% confidence interval</span></span></code></pre></div>
</div>
</div>
<div id="part-2" class="section level3 hasAnchor" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> Part 2<a href="pop-genetics.html#part-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Scenario:</strong> You are asked to estimate the additive genetic variance, dominance genetic variance and/or shared environmental variance using regression-based method and a classical twin design.</p>
<p><span class="math display">\[\begin{align*}

\text{For ADE model :  }~ &amp; \sigma^{2}_{P} = \sigma^{2}_{A} + \sigma^{2}_{D} + \sigma^{2}_{E}\\

\text{For ACE model :  }~ &amp; \sigma^{2}_{P} = \sigma^{2}_{A} + \sigma^{2}_{C} + \sigma^{2}_{E}, \quad \text{where} \\

\sigma^{2}_{P} &amp; \text{ is the phenotypic variance}, \\

\sigma^{2}_{A} &amp; \text{ is additive genetic variance}, \\

\sigma^{2}_{D} &amp; \text{ is dominance genetic variance}, \\

\sigma^{2}_{C} &amp; \text{ is shared environmental variance, and} \\

\sigma^{2}_{E} &amp; \text{ is unshared environmental variance.}
\end{align*}\]</span></p>
<p>For ADE model,</p>
<p><span class="math display">\[\begin{align*}

  cov(MZ) = cor(MZ) &amp; = rMZ = \sigma^{2}_{A} + \sigma^{2}_{D} \\
  cov(DZ) = cor(DZ) &amp; = rDZ = 0.5 * \sigma^{2}_{A} + 0.25 * \sigma^{2}_{D} \quad \text{ , where} \\

\end{align*}\]</span>
the coefficients 1/2 and 1/4 are based on quantitative genetic theory (Mather &amp; Jinks, 1971).</p>
<p>By solving the unknowns, the Falconer’s equations for the ADE model:</p>
<p><span class="math display">\[\begin{align*}

\sigma^{2}_{A} &amp; = 4*rDZ - rMZ \\

\sigma^{2}_{D} &amp; = 2*rMZ - 4*rDZ \\
\sigma^{2}_{E} &amp; = 1 - \sigma^{2}_{A} - \sigma^{2}_{D} \\

\end{align*}\]</span></p>
<p>For ACE model,</p>
<p><span class="math display">\[\begin{align*}

  cov(MZ) = cor(MZ) &amp; = rMZ= \sigma^{2}_{A} + \sigma^{2}_{C} \\
  cov(DZ) = cor(DZ) &amp; = rDZ = 0.5 * \sigma^{2}_{A} + \sigma^{2}_{C} \quad \text{ , where} \\

\end{align*}\]</span></p>
<p>By solving the unknowns, the Falconer’s equations for the ACE model:</p>
<p><span class="math display">\[\begin{align*}

\sigma^{2}_{A} &amp; = 2*(rMZ - rDZ) \\
\sigma^{2}_{C} &amp; = 2*rDZ - rMZ \\
\sigma^{2}_{E} &amp; = 1 - \sigma^{2}_{A} - \sigma^{2}_{C} = 1 - rMZ

\end{align*}\]</span></p>
<div id="questions-for-discussions" class="section level4 hasAnchor" number="8.1.2.1">
<h4><span class="header-section-number">8.1.2.1</span> Questions for discussions :<a href="pop-genetics.html#questions-for-discussions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q1.</strong> What is missing heritability of common traits in the era of genome-wide association analysis (GWAS)?</p>
<p><strong>Q2.</strong> What are the potential sources of missing heritability?</p>
<ul>
<li>Suggested reading:
<ul>
<li>Manolio TA, Collins FS, Cox NJ, et al. Finding the missing heritability of complex diseases. Nature. 2009 ;461(7265):747-753. <a href="doi:10.1038/nature08494" class="uri">doi:10.1038/nature08494</a></li>
</ul></li>
</ul>
</div>
<div id="hands-on-exercise-variance-explained-using-regression-based-method" class="section level4 hasAnchor" number="8.1.2.2">
<h4><span class="header-section-number">8.1.2.2</span> Hands-on exercise : variance explained using regression-based method<a href="pop-genetics.html#hands-on-exercise-variance-explained-using-regression-based-method" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q1.</strong> What is the variance of the phenotype?</p>
<p><strong>Q2.</strong> Compute the explained variance attributable to the additive genetic component of the QTL with strongest association in Part 1.</p>
<p><strong>Q3.</strong> Compute the explained variance attributable to the dominance genetic component of the QTL with strongest association in Part 1.</p>
<p>R2 from the regression represents the proportion of phenotypic variance explained; thus the raw explained variance component is R2 times the variance of the phenotype (var_pheno).</p>
<details>
<summary>
<strong>Example for T1 on QTL1</strong>
</summary>
<ul>
<li>The proportion of explained variance are 0.02732 (additive) and 0.03658 (total: additive + dominance).</li>
<li>As the predictors are uncorrelated, the proportion of explained variance by dominance = 0.03658 - 0.02732 = 0.00926</li>
<li>Given the phenotypic variance of 15.102, then
<ul>
<li>Total genetic: 0.03658*15.102 = 0.552</li>
<li>Additive genetic: 0.02732*15.102 = 0.412</li>
<li>Dominance genetic: 0.00926*15.102 = 0.140</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="pop-genetics.html#cb174-1" tabindex="-1"></a>var_pheno <span class="ot">&lt;-</span> <span class="fu">var</span>(dataTwin<span class="sc">$</span>pheno1_T1)  <span class="co"># the variance of the phenotype</span></span>
<span id="cb174-2"><a href="pop-genetics.html#cb174-2" tabindex="-1"></a>var_pheno</span>
<span id="cb174-3"><a href="pop-genetics.html#cb174-3" tabindex="-1"></a><span class="fu">summary</span>(linAD1)<span class="sc">$</span>r.squared           <span class="co"># proportion of explained variance by total genetic component</span></span>
<span id="cb174-4"><a href="pop-genetics.html#cb174-4" tabindex="-1"></a><span class="fu">summary</span>(linA1)<span class="sc">$</span>r.squared            <span class="co"># proportion of explained variance by additive component</span></span>
<span id="cb174-5"><a href="pop-genetics.html#cb174-5" tabindex="-1"></a><span class="fu">summary</span>(linAD1)<span class="sc">$</span>r.squared<span class="sc">*</span>var_pheno <span class="co"># (raw) variance component of total genetic component</span></span>
<span id="cb174-6"><a href="pop-genetics.html#cb174-6" tabindex="-1"></a><span class="fu">summary</span>(linA1)<span class="sc">$</span>r.squared<span class="sc">*</span>var_pheno  <span class="co"># (raw) variance component of additive genetic component</span></span>
<span id="cb174-7"><a href="pop-genetics.html#cb174-7" tabindex="-1"></a>(<span class="fu">summary</span>(linAD1)<span class="sc">$</span>r.squared<span class="sc">-</span><span class="fu">summary</span>(linA1)<span class="sc">$</span>r.squared)<span class="sc">*</span>var_pheno  <span class="co"># (raw) variance component of dominance genetic component</span></span></code></pre></div>
</details>
<p><strong>Q4.</strong> Estimate the variance explained by all the QTL using linear regression.</p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="pop-genetics.html#cb175-1" tabindex="-1"></a><span class="co"># compute for all 5 QTL</span></span>
<span id="cb175-2"><a href="pop-genetics.html#cb175-2" tabindex="-1"></a>linAD5<span class="ot">=</span>(<span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A1 <span class="sc">+</span> T1QTL_A2 <span class="sc">+</span> T1QTL_A3 <span class="sc">+</span> T1QTL_A4 <span class="sc">+</span> T1QTL_A5 <span class="sc">+</span></span>
<span id="cb175-3"><a href="pop-genetics.html#cb175-3" tabindex="-1"></a>                       T1QTL_D1 <span class="sc">+</span> T1QTL_D2 <span class="sc">+</span> T1QTL_D3 <span class="sc">+</span> T1QTL_D4 <span class="sc">+</span> T1QTL_D5,</span>
<span id="cb175-4"><a href="pop-genetics.html#cb175-4" tabindex="-1"></a>            <span class="at">data=</span>dataTwin))</span>
<span id="cb175-5"><a href="pop-genetics.html#cb175-5" tabindex="-1"></a></span>
<span id="cb175-6"><a href="pop-genetics.html#cb175-6" tabindex="-1"></a><span class="fu">summary</span>(linAD5)<span class="sc">$</span>r.squared              <span class="co"># proportion of explained variance by total genetic component</span></span>
<span id="cb175-7"><a href="pop-genetics.html#cb175-7" tabindex="-1"></a><span class="fu">summary</span>(linAD5)<span class="sc">$</span>r.squared<span class="sc">*</span>var_pheno    <span class="co"># (raw) variance component of total genetic component</span></span></code></pre></div>
</div>
<div id="hands-on-exercise-variance-explained-using-a-classical-twin-design." class="section level4 hasAnchor" number="8.1.2.3">
<h4><span class="header-section-number">8.1.2.3</span> Hands-on exercise : variance explained using a classical twin design.<a href="pop-genetics.html#hands-on-exercise-variance-explained-using-a-classical-twin-design." class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Based on our regression results, we have estimates of the total genetic variance as well as the A and D components for phenotype 1. In practice, it is impossible to know all the variants associated with any polygenic trait.</p>
<p>Given <code>rMZ &gt; 2*rDZ</code>, we can use Falconer’s formula based on ADE model to estimate the A (additive genetic) and D (dominance) variance with the classical twin design for phenotype 1 without genotypes.</p>
<p><strong>Q5.</strong> Compute rMZ and rDZ.</p>
<p><strong>Q6.</strong> Estimate the proportion of additive and dominance genetic variances using the Falconer’s equations for the ADE model.</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="pop-genetics.html#cb176-1" tabindex="-1"></a>dataMZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">1</span>, <span class="fu">c</span>(<span class="st">&#39;pheno1_T1&#39;</span>, <span class="st">&#39;pheno1_T2&#39;</span>)] <span class="co"># MZ data frame</span></span>
<span id="cb176-2"><a href="pop-genetics.html#cb176-2" tabindex="-1"></a>dataDZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&#39;pheno1_T1&#39;</span>, <span class="st">&#39;pheno1_T2&#39;</span>)] <span class="co"># DZ data frame</span></span>
<span id="cb176-3"><a href="pop-genetics.html#cb176-3" tabindex="-1"></a></span>
<span id="cb176-4"><a href="pop-genetics.html#cb176-4" tabindex="-1"></a>rMZ<span class="ot">=</span><span class="fu">cor</span>(dataMZ)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the MZ correlation matrix</span></span>
<span id="cb176-5"><a href="pop-genetics.html#cb176-5" tabindex="-1"></a>rDZ<span class="ot">=</span><span class="fu">cor</span>(dataDZ)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the DZ correlation matrix</span></span>
<span id="cb176-6"><a href="pop-genetics.html#cb176-6" tabindex="-1"></a></span>
<span id="cb176-7"><a href="pop-genetics.html#cb176-7" tabindex="-1"></a>sA2 <span class="ot">=</span> <span class="dv">4</span><span class="sc">*</span>rDZ <span class="sc">-</span> rMZ</span>
<span id="cb176-8"><a href="pop-genetics.html#cb176-8" tabindex="-1"></a>sD2 <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span>rMZ <span class="sc">-</span> <span class="dv">4</span><span class="sc">*</span>rDZ</span>
<span id="cb176-9"><a href="pop-genetics.html#cb176-9" tabindex="-1"></a>sE2 <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> sA2 <span class="sc">-</span> sD2</span>
<span id="cb176-10"><a href="pop-genetics.html#cb176-10" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(sA2, sD2, sE2))</span></code></pre></div>
<p>Similarly, for phenotype 2, we can estimate the proportion of additive and/or dominance genetic variances as well as shared environmental variance using the Falconer’s formula.</p>
<p><strong>Q7.</strong> Which model (ACE or ADE) should be considered for phenotype 2?</p>
<p><strong>Q8.</strong> Estimate the proportion of A, C/D and E variance components for phenotype 2.</p>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="pop-genetics.html#cb177-1" tabindex="-1"></a>dataMZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">1</span>, <span class="fu">c</span>(<span class="st">&#39;pheno2_T1&#39;</span>, <span class="st">&#39;pheno2_T2&#39;</span>)] <span class="co"># MZ data frame</span></span>
<span id="cb177-2"><a href="pop-genetics.html#cb177-2" tabindex="-1"></a>dataDZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&#39;pheno2_T1&#39;</span>, <span class="st">&#39;pheno2_T2&#39;</span>)] <span class="co"># DZ data frame</span></span>
<span id="cb177-3"><a href="pop-genetics.html#cb177-3" tabindex="-1"></a></span>
<span id="cb177-4"><a href="pop-genetics.html#cb177-4" tabindex="-1"></a>rMZ<span class="ot">=</span><span class="fu">cor</span>(dataMZ)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the MZ correlation matrix</span></span>
<span id="cb177-5"><a href="pop-genetics.html#cb177-5" tabindex="-1"></a>rDZ<span class="ot">=</span><span class="fu">cor</span>(dataDZ)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the DZ correlation matrix</span></span>
<span id="cb177-6"><a href="pop-genetics.html#cb177-6" tabindex="-1"></a></span>
<span id="cb177-7"><a href="pop-genetics.html#cb177-7" tabindex="-1"></a>sA2 <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span>(rMZ <span class="sc">-</span> rDZ)</span>
<span id="cb177-8"><a href="pop-genetics.html#cb177-8" tabindex="-1"></a>sC2 <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span>rDZ <span class="sc">-</span> rMZ</span>
<span id="cb177-9"><a href="pop-genetics.html#cb177-9" tabindex="-1"></a>sE2 <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> rMZ</span>
<span id="cb177-10"><a href="pop-genetics.html#cb177-10" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(sA2, sC2, sE2))</span></code></pre></div>


</div>
</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="cancer.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="install.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/statbiomed/BMDS-book/tree/main/02-data_topics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BMDatSci.pdf", "BMDatSci.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
