<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Population Genetics and Diseases | Biomedical Data Science - introduction with case studies</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Population Genetics and Diseases | Biomedical Data Science - introduction with case studies" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Population Genetics and Diseases | Biomedical Data Science - introduction with case studies" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="BIOF1001 teaching team" />


<meta name="date" content="2024-10-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="cancer.html"/>
<link rel="next" href="install.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">Biomedical Data Science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>Preface</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#introduction-for-readers"><i class="fa fa-check"></i>Introduction for readers</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-you-will-learn-from-this-coursebook"><i class="fa fa-check"></i>What you will learn from this course/book</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#what-we-recommend-you-do-while-reading-this-book"><i class="fa fa-check"></i>What we recommend you do while reading this book</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#other-reference-books"><i class="fa fa-check"></i>Other reference books</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a></li>
</ul></li>
<li class="part"><span><b>I Data Science Foundations</b></span></li>
<li class="chapter" data-level="1" data-path="introR.html"><a href="introR.html"><i class="fa fa-check"></i><b>1</b> Introduction to R programming</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introR.html"><a href="introR.html#data-types"><i class="fa fa-check"></i><b>1.1</b> Data types</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="introR.html"><a href="introR.html#nemeric-or-double"><i class="fa fa-check"></i><b>1.1.1</b> nemeric (or double)</a></li>
<li class="chapter" data-level="1.1.2" data-path="introR.html"><a href="introR.html#integer"><i class="fa fa-check"></i><b>1.1.2</b> integer</a></li>
<li class="chapter" data-level="1.1.3" data-path="introR.html"><a href="introR.html#logical"><i class="fa fa-check"></i><b>1.1.3</b> logical</a></li>
<li class="chapter" data-level="1.1.4" data-path="introR.html"><a href="introR.html#character"><i class="fa fa-check"></i><b>1.1.4</b> character</a></li>
<li class="chapter" data-level="1.1.5" data-path="introR.html"><a href="introR.html#memeory-usage"><i class="fa fa-check"></i><b>1.1.5</b> Memeory usage</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="introR.html"><a href="introR.html#data-structures"><i class="fa fa-check"></i><b>1.2</b> Data structures</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="introR.html"><a href="introR.html#vector"><i class="fa fa-check"></i><b>1.2.1</b> Vector</a></li>
<li class="chapter" data-level="1.2.2" data-path="introR.html"><a href="introR.html#matrix"><i class="fa fa-check"></i><b>1.2.2</b> Matrix</a></li>
<li class="chapter" data-level="1.2.3" data-path="introR.html"><a href="introR.html#list"><i class="fa fa-check"></i><b>1.2.3</b> List</a></li>
<li class="chapter" data-level="1.2.4" data-path="introR.html"><a href="introR.html#data-frame"><i class="fa fa-check"></i><b>1.2.4</b> Data Frame</a></li>
<li class="chapter" data-level="1.2.5" data-path="introR.html"><a href="introR.html#factor-vs-vector"><i class="fa fa-check"></i><b>1.2.5</b> Factor vs vector</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="introR.html"><a href="introR.html#read-and-write-files-tables"><i class="fa fa-check"></i><b>1.3</b> Read and write files (tables)</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introR.html"><a href="introR.html#read-file"><i class="fa fa-check"></i><b>1.3.1</b> Read file</a></li>
<li class="chapter" data-level="1.3.2" data-path="introR.html"><a href="introR.html#write-file"><i class="fa fa-check"></i><b>1.3.2</b> Write file</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="introR.html"><a href="introR.html#functions-and-packages"><i class="fa fa-check"></i><b>1.4</b> Functions and Packages</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="introR.html"><a href="introR.html#install-packages"><i class="fa fa-check"></i><b>1.4.1</b> Install packages</a></li>
<li class="chapter" data-level="1.4.2" data-path="introR.html"><a href="introR.html#apply-function-repeatly"><i class="fa fa-check"></i><b>1.4.2</b> Apply function repeatly</a></li>
<li class="chapter" data-level="1.4.3" data-path="introR.html"><a href="introR.html#pattern-match"><i class="fa fa-check"></i><b>1.4.3</b> Pattern match</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="introR.html"><a href="introR.html#flow-control"><i class="fa fa-check"></i><b>1.5</b> Flow Control</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="introR.html"><a href="introR.html#logical-operator"><i class="fa fa-check"></i><b>1.5.1</b> Logical operator</a></li>
<li class="chapter" data-level="1.5.2" data-path="introR.html"><a href="introR.html#if-else-statements"><i class="fa fa-check"></i><b>1.5.2</b> if-else statements</a></li>
<li class="chapter" data-level="1.5.3" data-path="introR.html"><a href="introR.html#for-loop"><i class="fa fa-check"></i><b>1.5.3</b> for-loop</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="introR.html"><a href="introR.html#plotting"><i class="fa fa-check"></i><b>1.6</b> Plotting</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="introR.html"><a href="introR.html#datasets"><i class="fa fa-check"></i><b>1.6.1</b> datasets</a></li>
<li class="chapter" data-level="1.6.2" data-path="introR.html"><a href="introR.html#basic-plotting"><i class="fa fa-check"></i><b>1.6.2</b> Basic plotting</a></li>
<li class="chapter" data-level="1.6.3" data-path="introR.html"><a href="introR.html#ggplot2"><i class="fa fa-check"></i><b>1.6.3</b> ggplot2</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="introR.html"><a href="introR.html#scientific-computating"><i class="fa fa-check"></i><b>1.7</b> Scientific computating</a>
<ul>
<li class="chapter" data-level="1.7.1" data-path="introR.html"><a href="introR.html#orders-of-operators"><i class="fa fa-check"></i><b>1.7.1</b> Orders of operators</a></li>
<li class="chapter" data-level="1.7.2" data-path="introR.html"><a href="introR.html#functions-for-statistics"><i class="fa fa-check"></i><b>1.7.2</b> Functions for statistics</a></li>
<li class="chapter" data-level="1.7.3" data-path="introR.html"><a href="introR.html#correlation"><i class="fa fa-check"></i><b>1.7.3</b> Correlation</a></li>
<li class="chapter" data-level="1.7.4" data-path="introR.html"><a href="introR.html#hypothesis-testing-t-test"><i class="fa fa-check"></i><b>1.7.4</b> Hypothesis testing (t test)</a></li>
<li class="chapter" data-level="1.7.5" data-path="introR.html"><a href="introR.html#regression"><i class="fa fa-check"></i><b>1.7.5</b> Regression</a></li>
<li class="chapter" data-level="1.7.6" data-path="introR.html"><a href="introR.html#resource-links"><i class="fa fa-check"></i><b>1.7.6</b> Resource links</a></li>
<li class="chapter" data-level="1.7.7" data-path="introR.html"><a href="introR.html#coding-styling"><i class="fa fa-check"></i><b>1.7.7</b> Coding styling</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="introR.html"><a href="introR.html#exercises"><i class="fa fa-check"></i><b>1.8</b> Exercises</a>
<ul>
<li class="chapter" data-level="1.8.1" data-path="introR.html"><a href="introR.html#exercises-p1"><i class="fa fa-check"></i><b>1.8.1</b> Part 1. Basics (~40min)</a></li>
<li class="chapter" data-level="1.8.2" data-path="introR.html"><a href="introR.html#exercises-p2"><i class="fa fa-check"></i><b>1.8.2</b> Part 2. Making plotting (~40min)</a></li>
<li class="chapter" data-level="1.8.3" data-path="introR.html"><a href="introR.html#exercises-p3"><i class="fa fa-check"></i><b>1.8.3</b> Part 3. For loop and repeating processing (~40min)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introHypoTest.html"><a href="introHypoTest.html"><i class="fa fa-check"></i><b>2</b> Introduction to Hypothesis testing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#hypothesis-testing-and-p-value"><i class="fa fa-check"></i><b>2.1</b> Hypothesis testing and <em>p</em> value</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-1-probability-of-rolling-a-six"><i class="fa fa-check"></i><b>2.1.1</b> Example 1: probability of rolling a six?</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#permutation-test"><i class="fa fa-check"></i><b>2.2</b> Permutation test</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introHypoTest.html"><a href="introHypoTest.html#example-2-difference-in-birth-weight"><i class="fa fa-check"></i><b>2.2.1</b> Example 2: difference in birth weight</a></li>
<li class="chapter" data-level="2.2.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-approximated-by-resampling"><i class="fa fa-check"></i><b>2.2.2</b> Null distribution approximated by resampling</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introHypoTest.html"><a href="introHypoTest.html#t-test"><i class="fa fa-check"></i><b>2.3</b> <em>t</em> test</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="introHypoTest.html"><a href="introHypoTest.html#derivation-of-t-distribution"><i class="fa fa-check"></i><b>2.3.1</b> Derivation of t distribution</a></li>
<li class="chapter" data-level="2.3.2" data-path="introHypoTest.html"><a href="introHypoTest.html#direct-use-of-t.test"><i class="fa fa-check"></i><b>2.3.2</b> Direct use of <code>t.test()</code></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="introHypoTest.html"><a href="introHypoTest.html#regression-based-test"><i class="fa fa-check"></i><b>2.4</b> regression-based test</a></li>
<li class="chapter" data-level="2.5" data-path="introHypoTest.html"><a href="introHypoTest.html#multiple-testing"><i class="fa fa-check"></i><b>2.5</b> Multiple testing</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-test-statistic"><i class="fa fa-check"></i><b>2.5.1</b> Null distribution (of test statistic)</a></li>
<li class="chapter" data-level="2.5.2" data-path="introHypoTest.html"><a href="introHypoTest.html#null-distribution-of-p-value"><i class="fa fa-check"></i><b>2.5.2</b> Null distribution of p value</a></li>
<li class="chapter" data-level="2.5.3" data-path="introHypoTest.html"><a href="introHypoTest.html#minimal-p-values-in-10-tests"><i class="fa fa-check"></i><b>2.5.3</b> Minimal p values in 10 tests</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="introHypoTest.html"><a href="introHypoTest.html#explore-power-and-sample-size-optional-power"><i class="fa fa-check"></i><b>2.6</b> Explore power and sample size (optional) { power }</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introLinearReg.html"><a href="introLinearReg.html"><i class="fa fa-check"></i><b>3</b> Introduction to Linear Regression</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#linear-regression-using-simulated-data"><i class="fa fa-check"></i><b>3.1</b> Linear Regression Using Simulated Data</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="introLinearReg.html"><a href="introLinearReg.html#simulating-data"><i class="fa fa-check"></i><b>3.1.1</b> Simulating data:</a></li>
<li class="chapter" data-level="3.1.2" data-path="introLinearReg.html"><a href="introLinearReg.html#model-efficacy"><i class="fa fa-check"></i><b>3.1.2</b> Model efficacy</a></li>
<li class="chapter" data-level="3.1.3" data-path="introLinearReg.html"><a href="introLinearReg.html#r-squared"><i class="fa fa-check"></i><b>3.1.3</b> <em>R-Squared</em></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#least-squares-using-simulated-data"><i class="fa fa-check"></i><b>3.2</b> Least Squares Using Simulated Data</a></li>
<li class="chapter" data-level="3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#diagnostic-check-of-a-fitted-regression-model"><i class="fa fa-check"></i><b>3.3</b> Diagnostic check of a fitted regression model</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="introLinearReg.html"><a href="introLinearReg.html#residual-standard-errors-rse"><i class="fa fa-check"></i><b>3.3.1</b> Residual Standard Errors (RSE)</a></li>
<li class="chapter" data-level="3.3.2" data-path="introLinearReg.html"><a href="introLinearReg.html#p-values"><i class="fa fa-check"></i><b>3.3.2</b> p-values</a></li>
<li class="chapter" data-level="3.3.3" data-path="introLinearReg.html"><a href="introLinearReg.html#f-statistics"><i class="fa fa-check"></i><b>3.3.3</b> F-statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introLinearReg.html"><a href="introLinearReg.html#simple-linear-regression-with-lm-function"><i class="fa fa-check"></i><b>3.4</b> Simple Linear Regression with <code>lm</code> function</a></li>
<li class="chapter" data-level="3.5" data-path="introLinearReg.html"><a href="introLinearReg.html#multiple-regression-with-lm-function"><i class="fa fa-check"></i><b>3.5</b> Multiple Regression with <code>lm</code> function</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="introClassifier.html"><a href="introClassifier.html"><i class="fa fa-check"></i><b>4</b> Introduction to Classification</a>
<ul>
<li class="chapter" data-level="4.1" data-path="introClassifier.html"><a href="introClassifier.html#visualise-logistic-and-logit-functions"><i class="fa fa-check"></i><b>4.1</b> Visualise logistic and logit functions</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="introClassifier.html"><a href="introClassifier.html#logistic-function"><i class="fa fa-check"></i><b>4.1.1</b> Logistic function</a></li>
<li class="chapter" data-level="4.1.2" data-path="introClassifier.html"><a href="introClassifier.html#logit-function"><i class="fa fa-check"></i><b>4.1.2</b> Logit function</a></li>
<li class="chapter" data-level="4.1.3" data-path="introClassifier.html"><a href="introClassifier.html#visualise-the-distribution"><i class="fa fa-check"></i><b>4.1.3</b> Visualise the distribution</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="introClassifier.html"><a href="introClassifier.html#logistic-regression-on-diabetes"><i class="fa fa-check"></i><b>4.2</b> Logistic regression on Diabetes</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="introClassifier.html"><a href="introClassifier.html#load-pima-indians-diabetes-database"><i class="fa fa-check"></i><b>4.2.1</b> Load Pima Indians Diabetes Database</a></li>
<li class="chapter" data-level="4.2.2" data-path="introClassifier.html"><a href="introClassifier.html#fit-logistic-regression"><i class="fa fa-check"></i><b>4.2.2</b> Fit logistic regression</a></li>
<li class="chapter" data-level="4.2.3" data-path="introClassifier.html"><a href="introClassifier.html#assess-on-test-data"><i class="fa fa-check"></i><b>4.2.3</b> Assess on test data</a></li>
<li class="chapter" data-level="4.2.4" data-path="introClassifier.html"><a href="introClassifier.html#model-selection-and-diagnosis"><i class="fa fa-check"></i><b>4.2.4</b> Model selection and diagnosis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="introClassifier.html"><a href="introClassifier.html#cross-validation"><i class="fa fa-check"></i><b>4.3</b> Cross-validation</a></li>
<li class="chapter" data-level="4.4" data-path="introClassifier.html"><a href="introClassifier.html#more-assessment-metrics"><i class="fa fa-check"></i><b>4.4</b> More assessment metrics</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="introClassifier.html"><a href="introClassifier.html#two-types-of-error"><i class="fa fa-check"></i><b>4.4.1</b> Two types of error</a></li>
<li class="chapter" data-level="4.4.2" data-path="introClassifier.html"><a href="introClassifier.html#roc-curve"><i class="fa fa-check"></i><b>4.4.2</b> ROC curve</a></li>
<li class="chapter" data-level="4.4.3" data-path="introClassifier.html"><a href="introClassifier.html#homework"><i class="fa fa-check"></i><b>4.4.3</b> Homework</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Biomedical Data Modules</b></span></li>
<li class="chapter" data-level="5" data-path="image-digital.html"><a href="image-digital.html"><i class="fa fa-check"></i><b>5</b> Medical Image and Digital Health</a></li>
<li class="chapter" data-level="6" data-path="cancer.html"><a href="cancer.html"><i class="fa fa-check"></i><b>6</b> Cancer genomics and epidemiology</a>
<ul>
<li class="chapter" data-level="6.1" data-path="cancer.html"><a href="cancer.html#cancer-case1"><i class="fa fa-check"></i><b>6.1</b> Case study 1: analysis of cBioportal mutation data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="cancer.html"><a href="cancer.html#exploratory-analysis"><i class="fa fa-check"></i><b>6.1.1</b> Exploratory analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="cancer.html"><a href="cancer.html#statistical-analysis"><i class="fa fa-check"></i><b>6.1.2</b> Statistical analysis</a></li>
<li class="chapter" data-level="6.1.3" data-path="cancer.html"><a href="cancer.html#literature-search"><i class="fa fa-check"></i><b>6.1.3</b> Literature search</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="cancer.html"><a href="cancer.html#cancer-case2"><i class="fa fa-check"></i><b>6.2</b> Case study 2: Cancer Epidemiology</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="cancer.html"><a href="cancer.html#scenario"><i class="fa fa-check"></i><b>6.2.1</b> Scenario</a></li>
<li class="chapter" data-level="6.2.2" data-path="cancer.html"><a href="cancer.html#hong-kong-population"><i class="fa fa-check"></i><b>6.2.2</b> Hong Kong population</a></li>
<li class="chapter" data-level="6.2.3" data-path="cancer.html"><a href="cancer.html#cancer-registry-data"><i class="fa fa-check"></i><b>6.2.3</b> Cancer registry data</a></li>
<li class="chapter" data-level="6.2.4" data-path="cancer.html"><a href="cancer.html#existing-cancer-funding-and-publication-data"><i class="fa fa-check"></i><b>6.2.4</b> Existing cancer funding and publication data</a></li>
<li class="chapter" data-level="6.2.5" data-path="cancer.html"><a href="cancer.html#open-discussion"><i class="fa fa-check"></i><b>6.2.5</b> Open discussion</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="pop-genetics.html"><a href="pop-genetics.html"><i class="fa fa-check"></i><b>7</b> Population Genetics and Diseases</a>
<ul>
<li class="chapter" data-level="7.1" data-path="pop-genetics.html"><a href="pop-genetics.html#case-study-1-heritability-and-human-traits"><i class="fa fa-check"></i><b>7.1</b> Case study 1: Heritability and human traits</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="pop-genetics.html"><a href="pop-genetics.html#part-1"><i class="fa fa-check"></i><b>7.1.1</b> Part 1</a></li>
<li class="chapter" data-level="7.1.2" data-path="pop-genetics.html"><a href="pop-genetics.html#part-2"><i class="fa fa-check"></i><b>7.1.2</b> Part 2</a></li>
<li class="chapter" data-level="7.1.3" data-path="pop-genetics.html"><a href="pop-genetics.html#references"><i class="fa fa-check"></i><b>7.1.3</b> References</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>III Appendix</b></span></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i>Appendix A: Install R &amp; RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.1-install-r-4.3.1"><i class="fa fa-check"></i>A.1 Install R (&gt;=4.3.1)</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-windows"><i class="fa fa-check"></i>R on Windows</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-macos"><i class="fa fa-check"></i>R on macOS</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-on-linux-ubuntu"><i class="fa fa-check"></i>R on Linux (Ubuntu)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.2-install-rstudio"><i class="fa fa-check"></i>A.2 Install RStudio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a.3-use-r-inside-rstudio"><i class="fa fa-check"></i>A.3 Use R inside RStudio</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#r-studio"><i class="fa fa-check"></i>R studio</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#set-working-directory"><i class="fa fa-check"></i>Set working directory</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#some-general-knowledge"><i class="fa fa-check"></i>Some general knowledge</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#install-packages-1"><i class="fa fa-check"></i>Install packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#a4.-cloud-computing"><i class="fa fa-check"></i>A4. Cloud computing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Biomedical Data Science - introduction with case studies</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pop-genetics" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Population Genetics and Diseases<a href="pop-genetics.html#pop-genetics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="case-study-1-heritability-and-human-traits" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Case study 1: Heritability and human traits<a href="pop-genetics.html#case-study-1-heritability-and-human-traits" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="part-1" class="section level3 hasAnchor" number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> Part 1<a href="pop-genetics.html#part-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Scenario:</strong> You are a researcher working on a twin study on cardiovascular traits to assess the genetic and environmental contribution relevant to metabolism and cardiovascular disease risk. You have recruited a cohort of volunteer adult twins of the same ancestry. The volunteers have undergone a series of baseline clinical evaluations and performed genotyping on a panel of single nucleotide polymorphisms that may be associated with the traits.</p>
<div id="questions-for-discussion" class="section level4 hasAnchor" number="7.1.1.1">
<h4><span class="header-section-number">7.1.1.1</span> Questions for Discussion<a href="pop-genetics.html#questions-for-discussion" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<strong>Q1.</strong> Besides the clinical measurements, what data do you need to collect from the subjects?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Sex</li>
<li>Age</li>
<li>Other confounding factors, e.g. BMI, blood pressure, smoking status, etc.</li>
</ul>
</details>
<strong>Q2.</strong> How is genotype data represented for statistical genetic analysis?<br />

<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Allele: 0/1, 1/2, A/C, etc</li>
<li>Genotype: 0 0, 0 1, 1 0, 1 1</li>
<li>Genotype probabilities: P(0/0)=0, P(0/1)=1, P(1/1)=0</li>
<li>Genotype dosage: 0/1/2, 0.678 (continuous from 0-1 or 0-2)</li>
</ul>
</details>
<strong>Q3.</strong> How can you test for association between genotypes and phenotypes (binary and quantitative)?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Allelic chi-square test</li>
<li>Fisher’s exact test</li>
<li>Linear/Logistic regression</li>
<li>Linear mixed model</li>
</ul>
</details>
</div>
<div id="hands-on-exercise-association-test" class="section level4 hasAnchor" number="7.1.1.2">
<h4><span class="header-section-number">7.1.1.2</span> Hands-on exercise : Association test<a href="pop-genetics.html#hands-on-exercise-association-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now, you are given a dataset of age- and sex-matched twin cohort with two cardiovascular phenotypes and 5 quantitative trait loci (QTL). Data set and template notebook are available on Moodle (recommended) and also on this
<a href="https://github.com/StatBiomed/BMDS-book/blob/main/notebooks/module4-genetics/dataTwin2023.dat">GitHub Repo</a>.</p>
<p><strong>The information for columns:</strong></p>
<ul>
<li>zygosity: 1 for monozygotic (MZ) and 2 for dizygotic (DZ) twin</li>
<li><code>T1QTL_A[1-5]</code> and <code>T2QTL_A[1-5]</code>: 5 quantitative loci (A1-A5) in additive coding for Twin 1 (T1) and Twin 2 (T2) respectively</li>
<li>The same 5 QTL (D1-D5) in dominance coding for T1 and T2</li>
<li>Phenotype scores of T1 and T2 for the two quantitative cardiovascular traits</li>
</ul>
<p><strong>Download the data <code>dataTwin.dat</code> to your working directory.</strong> Start the RStudio program and set the working directory.</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="pop-genetics.html#cb170-1" tabindex="-1"></a>dataTwin <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;dataTwin2023.dat&quot;</span>,<span class="at">h=</span>T)</span></code></pre></div>
<p><strong>Exploratory analysis</strong></p>
<ul>
<li>A1-5: The QTLs are biallelic with two alleles A and a. The genotypes aa, Aa, and AA are coded additively as 0 (aa), 1 (Aa) and 2 (AA).</li>
<li>D1-5: The genotypes aa, Aa, and AA are coded as 0 (aa), 1 (Aa) and 0 (AA).</li>
</ul>
<strong>Q1.</strong> How many MZ and DZ volunteers are there?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>1000 MZ and 1000 DZ</li>
</ul>
</details>
<strong>Q2.</strong> How are the genotypes represented?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Dosage/Count of non-reference allele : 0, 1, and 2</li>
</ul>
</details>
<strong>Q3.</strong> Are the QTL independent of each other?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Yes. The pairwise correlations are low (&lt;0.2).</li>
</ul>
</details>
<strong>Q4.</strong> Are there outliers in phenotypes?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Yes. T2 individual 1303 has phenotype score (-4.21) being 4 SD below the mean.</li>
</ul>
</details>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="pop-genetics.html#cb171-1" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>zygosity)        <span class="co"># Q1: shows number of MZ and DZ twin pairs</span></span>
<span id="cb171-2"><a href="pop-genetics.html#cb171-2" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb171-3"><a href="pop-genetics.html#cb171-3" tabindex="-1"></a><span class="co">#&gt;    1    2 </span></span>
<span id="cb171-4"><a href="pop-genetics.html#cb171-4" tabindex="-1"></a><span class="co">#&gt; 1000 1000</span></span>
<span id="cb171-5"><a href="pop-genetics.html#cb171-5" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>T1QTL_A1)        <span class="co"># Q2: shows the distribution of QTL_A1</span></span>
<span id="cb171-6"><a href="pop-genetics.html#cb171-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb171-7"><a href="pop-genetics.html#cb171-7" tabindex="-1"></a><span class="co">#&gt;    0    1    2 </span></span>
<span id="cb171-8"><a href="pop-genetics.html#cb171-8" tabindex="-1"></a><span class="co">#&gt;  474 1021  505</span></span>
<span id="cb171-9"><a href="pop-genetics.html#cb171-9" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>T1QTL_D1)        <span class="co"># Q2: shows the distribution of QTL_D1</span></span>
<span id="cb171-10"><a href="pop-genetics.html#cb171-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb171-11"><a href="pop-genetics.html#cb171-11" tabindex="-1"></a><span class="co">#&gt;    0    1 </span></span>
<span id="cb171-12"><a href="pop-genetics.html#cb171-12" tabindex="-1"></a><span class="co">#&gt;  979 1021</span></span>
<span id="cb171-13"><a href="pop-genetics.html#cb171-13" tabindex="-1"></a><span class="fu">table</span>(dataTwin<span class="sc">$</span>T1QTL_A1, dataTwin<span class="sc">$</span>T1QTL_D1)     <span class="co"># Q2: shows the distribution of QTL_A1 in relation to QTL_D1</span></span>
<span id="cb171-14"><a href="pop-genetics.html#cb171-14" tabindex="-1"></a><span class="co">#&gt;    </span></span>
<span id="cb171-15"><a href="pop-genetics.html#cb171-15" tabindex="-1"></a><span class="co">#&gt;        0    1</span></span>
<span id="cb171-16"><a href="pop-genetics.html#cb171-16" tabindex="-1"></a><span class="co">#&gt;   0  474    0</span></span>
<span id="cb171-17"><a href="pop-genetics.html#cb171-17" tabindex="-1"></a><span class="co">#&gt;   1    0 1021</span></span>
<span id="cb171-18"><a href="pop-genetics.html#cb171-18" tabindex="-1"></a><span class="co">#&gt;   2  505    0</span></span>
<span id="cb171-19"><a href="pop-genetics.html#cb171-19" tabindex="-1"></a><span class="fu">cor</span>(dataTwin[,<span class="dv">2</span><span class="sc">:</span><span class="dv">11</span>])                            <span class="co"># Q3: shows the correlation between QTL_As</span></span>
<span id="cb171-20"><a href="pop-genetics.html#cb171-20" tabindex="-1"></a><span class="co">#&gt;             T1QTL_A1     T1QTL_A2     T1QTL_A3    T1QTL_A4     T1QTL_A5</span></span>
<span id="cb171-21"><a href="pop-genetics.html#cb171-21" tabindex="-1"></a><span class="co">#&gt; T1QTL_A1  1.00000000 -0.005470340  0.021705688  0.01940408  0.016278190</span></span>
<span id="cb171-22"><a href="pop-genetics.html#cb171-22" tabindex="-1"></a><span class="co">#&gt; T1QTL_A2 -0.00547034  1.000000000  0.017344822 -0.01421677 -0.008678746</span></span>
<span id="cb171-23"><a href="pop-genetics.html#cb171-23" tabindex="-1"></a><span class="co">#&gt; T1QTL_A3  0.02170569  0.017344822  1.000000000  0.01335711 -0.036751338</span></span>
<span id="cb171-24"><a href="pop-genetics.html#cb171-24" tabindex="-1"></a><span class="co">#&gt; T1QTL_A4  0.01940408 -0.014216767  0.013357109  1.00000000  0.074899996</span></span>
<span id="cb171-25"><a href="pop-genetics.html#cb171-25" tabindex="-1"></a><span class="co">#&gt; T1QTL_A5  0.01627819 -0.008678746 -0.036751338  0.07490000  1.000000000</span></span>
<span id="cb171-26"><a href="pop-genetics.html#cb171-26" tabindex="-1"></a><span class="co">#&gt; T2QTL_A1  0.53243815  0.004201635 -0.013909013  0.03252724  0.020081970</span></span>
<span id="cb171-27"><a href="pop-genetics.html#cb171-27" tabindex="-1"></a><span class="co">#&gt; T2QTL_A2 -0.04561174  0.464131160 -0.005044127  0.01324172 -0.003012277</span></span>
<span id="cb171-28"><a href="pop-genetics.html#cb171-28" tabindex="-1"></a><span class="co">#&gt; T2QTL_A3  0.03316574 -0.003552831  0.521253656  0.02045423  0.009081830</span></span>
<span id="cb171-29"><a href="pop-genetics.html#cb171-29" tabindex="-1"></a><span class="co">#&gt; T2QTL_A4  0.03271254 -0.033419904  0.020422583  0.48641289  0.019247531</span></span>
<span id="cb171-30"><a href="pop-genetics.html#cb171-30" tabindex="-1"></a><span class="co">#&gt; T2QTL_A5 -0.01285323  0.030413269 -0.045121964  0.08288145  0.457962222</span></span>
<span id="cb171-31"><a href="pop-genetics.html#cb171-31" tabindex="-1"></a><span class="co">#&gt;              T2QTL_A1     T2QTL_A2     T2QTL_A3     T2QTL_A4    T2QTL_A5</span></span>
<span id="cb171-32"><a href="pop-genetics.html#cb171-32" tabindex="-1"></a><span class="co">#&gt; T1QTL_A1  0.532438150 -0.045611740  0.033165736  0.032712539 -0.01285323</span></span>
<span id="cb171-33"><a href="pop-genetics.html#cb171-33" tabindex="-1"></a><span class="co">#&gt; T1QTL_A2  0.004201635  0.464131160 -0.003552831 -0.033419904  0.03041327</span></span>
<span id="cb171-34"><a href="pop-genetics.html#cb171-34" tabindex="-1"></a><span class="co">#&gt; T1QTL_A3 -0.013909013 -0.005044127  0.521253656  0.020422583 -0.04512196</span></span>
<span id="cb171-35"><a href="pop-genetics.html#cb171-35" tabindex="-1"></a><span class="co">#&gt; T1QTL_A4  0.032527239  0.013241725  0.020454234  0.486412895  0.08288145</span></span>
<span id="cb171-36"><a href="pop-genetics.html#cb171-36" tabindex="-1"></a><span class="co">#&gt; T1QTL_A5  0.020081970 -0.003012277  0.009081830  0.019247531  0.45796222</span></span>
<span id="cb171-37"><a href="pop-genetics.html#cb171-37" tabindex="-1"></a><span class="co">#&gt; T2QTL_A1  1.000000000  0.006179257 -0.013129314  0.048294183 -0.01325839</span></span>
<span id="cb171-38"><a href="pop-genetics.html#cb171-38" tabindex="-1"></a><span class="co">#&gt; T2QTL_A2  0.006179257  1.000000000 -0.020860987  0.002164782 -0.01131418</span></span>
<span id="cb171-39"><a href="pop-genetics.html#cb171-39" tabindex="-1"></a><span class="co">#&gt; T2QTL_A3 -0.013129314 -0.020860987  1.000000000 -0.010583797 -0.02101270</span></span>
<span id="cb171-40"><a href="pop-genetics.html#cb171-40" tabindex="-1"></a><span class="co">#&gt; T2QTL_A4  0.048294183  0.002164782 -0.010583797  1.000000000  0.04350925</span></span>
<span id="cb171-41"><a href="pop-genetics.html#cb171-41" tabindex="-1"></a><span class="co">#&gt; T2QTL_A5 -0.013258394 -0.011314179 -0.021012699  0.043509251  1.00000000</span></span>
<span id="cb171-42"><a href="pop-genetics.html#cb171-42" tabindex="-1"></a><span class="fu">cor</span>(dataTwin[,<span class="dv">2</span><span class="sc">:</span><span class="dv">11</span>])<span class="sc">&gt;</span><span class="fl">0.2</span></span>
<span id="cb171-43"><a href="pop-genetics.html#cb171-43" tabindex="-1"></a><span class="co">#&gt;          T1QTL_A1 T1QTL_A2 T1QTL_A3 T1QTL_A4 T1QTL_A5 T2QTL_A1 T2QTL_A2</span></span>
<span id="cb171-44"><a href="pop-genetics.html#cb171-44" tabindex="-1"></a><span class="co">#&gt; T1QTL_A1     TRUE    FALSE    FALSE    FALSE    FALSE     TRUE    FALSE</span></span>
<span id="cb171-45"><a href="pop-genetics.html#cb171-45" tabindex="-1"></a><span class="co">#&gt; T1QTL_A2    FALSE     TRUE    FALSE    FALSE    FALSE    FALSE     TRUE</span></span>
<span id="cb171-46"><a href="pop-genetics.html#cb171-46" tabindex="-1"></a><span class="co">#&gt; T1QTL_A3    FALSE    FALSE     TRUE    FALSE    FALSE    FALSE    FALSE</span></span>
<span id="cb171-47"><a href="pop-genetics.html#cb171-47" tabindex="-1"></a><span class="co">#&gt; T1QTL_A4    FALSE    FALSE    FALSE     TRUE    FALSE    FALSE    FALSE</span></span>
<span id="cb171-48"><a href="pop-genetics.html#cb171-48" tabindex="-1"></a><span class="co">#&gt; T1QTL_A5    FALSE    FALSE    FALSE    FALSE     TRUE    FALSE    FALSE</span></span>
<span id="cb171-49"><a href="pop-genetics.html#cb171-49" tabindex="-1"></a><span class="co">#&gt; T2QTL_A1     TRUE    FALSE    FALSE    FALSE    FALSE     TRUE    FALSE</span></span>
<span id="cb171-50"><a href="pop-genetics.html#cb171-50" tabindex="-1"></a><span class="co">#&gt; T2QTL_A2    FALSE     TRUE    FALSE    FALSE    FALSE    FALSE     TRUE</span></span>
<span id="cb171-51"><a href="pop-genetics.html#cb171-51" tabindex="-1"></a><span class="co">#&gt; T2QTL_A3    FALSE    FALSE     TRUE    FALSE    FALSE    FALSE    FALSE</span></span>
<span id="cb171-52"><a href="pop-genetics.html#cb171-52" tabindex="-1"></a><span class="co">#&gt; T2QTL_A4    FALSE    FALSE    FALSE     TRUE    FALSE    FALSE    FALSE</span></span>
<span id="cb171-53"><a href="pop-genetics.html#cb171-53" tabindex="-1"></a><span class="co">#&gt; T2QTL_A5    FALSE    FALSE    FALSE    FALSE     TRUE    FALSE    FALSE</span></span>
<span id="cb171-54"><a href="pop-genetics.html#cb171-54" tabindex="-1"></a><span class="co">#&gt;          T2QTL_A3 T2QTL_A4 T2QTL_A5</span></span>
<span id="cb171-55"><a href="pop-genetics.html#cb171-55" tabindex="-1"></a><span class="co">#&gt; T1QTL_A1    FALSE    FALSE    FALSE</span></span>
<span id="cb171-56"><a href="pop-genetics.html#cb171-56" tabindex="-1"></a><span class="co">#&gt; T1QTL_A2    FALSE    FALSE    FALSE</span></span>
<span id="cb171-57"><a href="pop-genetics.html#cb171-57" tabindex="-1"></a><span class="co">#&gt; T1QTL_A3     TRUE    FALSE    FALSE</span></span>
<span id="cb171-58"><a href="pop-genetics.html#cb171-58" tabindex="-1"></a><span class="co">#&gt; T1QTL_A4    FALSE     TRUE    FALSE</span></span>
<span id="cb171-59"><a href="pop-genetics.html#cb171-59" tabindex="-1"></a><span class="co">#&gt; T1QTL_A5    FALSE    FALSE     TRUE</span></span>
<span id="cb171-60"><a href="pop-genetics.html#cb171-60" tabindex="-1"></a><span class="co">#&gt; T2QTL_A1    FALSE    FALSE    FALSE</span></span>
<span id="cb171-61"><a href="pop-genetics.html#cb171-61" tabindex="-1"></a><span class="co">#&gt; T2QTL_A2    FALSE    FALSE    FALSE</span></span>
<span id="cb171-62"><a href="pop-genetics.html#cb171-62" tabindex="-1"></a><span class="co">#&gt; T2QTL_A3     TRUE    FALSE    FALSE</span></span>
<span id="cb171-63"><a href="pop-genetics.html#cb171-63" tabindex="-1"></a><span class="co">#&gt; T2QTL_A4    FALSE     TRUE    FALSE</span></span>
<span id="cb171-64"><a href="pop-genetics.html#cb171-64" tabindex="-1"></a><span class="co">#&gt; T2QTL_A5    FALSE    FALSE     TRUE</span></span>
<span id="cb171-65"><a href="pop-genetics.html#cb171-65" tabindex="-1"></a><span class="fu">apply</span>(dataTwin[<span class="dv">22</span><span class="sc">:</span><span class="dv">25</span>],<span class="dv">2</span>,<span class="cf">function</span>(x){ <span class="fu">any</span>(x <span class="sc">&lt;</span> (<span class="fu">mean</span>(x) <span class="sc">-</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span>(x))) })  <span class="co"># Q4: any outlier &lt; 4 SD from the mean for the two quantitative phenotypes</span></span>
<span id="cb171-66"><a href="pop-genetics.html#cb171-66" tabindex="-1"></a><span class="co">#&gt; pheno1_T1 pheno1_T2 pheno2_T1 pheno2_T2 </span></span>
<span id="cb171-67"><a href="pop-genetics.html#cb171-67" tabindex="-1"></a><span class="co">#&gt;     FALSE     FALSE     FALSE      TRUE</span></span>
<span id="cb171-68"><a href="pop-genetics.html#cb171-68" tabindex="-1"></a><span class="fu">apply</span>(dataTwin[<span class="dv">22</span><span class="sc">:</span><span class="dv">25</span>],<span class="dv">2</span>,<span class="cf">function</span>(x){ <span class="fu">any</span>(x <span class="sc">&gt;</span> (<span class="fu">mean</span>(x) <span class="sc">+</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span>(x))) })  <span class="co"># Q4: any outlier &gt; 4 SD from the mean for the two quantitative phenotypes</span></span>
<span id="cb171-69"><a href="pop-genetics.html#cb171-69" tabindex="-1"></a><span class="co">#&gt; pheno1_T1 pheno1_T2 pheno2_T1 pheno2_T2 </span></span>
<span id="cb171-70"><a href="pop-genetics.html#cb171-70" tabindex="-1"></a><span class="co">#&gt;     FALSE     FALSE     FALSE     FALSE</span></span>
<span id="cb171-71"><a href="pop-genetics.html#cb171-71" tabindex="-1"></a><span class="co"># remove the phenotype score of the outlier (T2) for the phenotype 2 (pheno2_T2)</span></span>
<span id="cb171-72"><a href="pop-genetics.html#cb171-72" tabindex="-1"></a>outlier<span class="ot">&lt;-</span> <span class="fu">which</span>(dataTwin<span class="sc">$</span>pheno2_T2 <span class="sc">&lt;</span> (<span class="fu">mean</span>(dataTwin<span class="sc">$</span>pheno2_T2) <span class="sc">-</span> <span class="dv">4</span><span class="sc">*</span><span class="fu">sd</span>(dataTwin<span class="sc">$</span>pheno2_T2) )) </span>
<span id="cb171-73"><a href="pop-genetics.html#cb171-73" tabindex="-1"></a>outlier</span>
<span id="cb171-74"><a href="pop-genetics.html#cb171-74" tabindex="-1"></a><span class="co">#&gt; [1] 1303</span></span>
<span id="cb171-75"><a href="pop-genetics.html#cb171-75" tabindex="-1"></a>dataTwin<span class="sc">$</span>pheno2_T2[outlier]</span>
<span id="cb171-76"><a href="pop-genetics.html#cb171-76" tabindex="-1"></a><span class="co">#&gt; [1] -4.21</span></span>
<span id="cb171-77"><a href="pop-genetics.html#cb171-77" tabindex="-1"></a>dataTwin<span class="sc">$</span>pheno2_T2[outlier] <span class="ot">&lt;-</span> <span class="cn">NA</span></span></code></pre></div>
<p><strong>Association test</strong></p>
<p>Test for association between QTL and pheno1 for T1</p>
<ul>
<li>Regress <code>pheno1_T1</code> on <code>T1QTL_A1</code> to estimate the proportion of variance explained (R2).</li>
<li>Model: pheno1_T1 = b0 + b1* T1QTL_A1 + e</li>
<li>Calculate the conditional mean of phenotype (i.e. phenotypic mean conditional genotype)</li>
</ul>
<p>If the relationship between the QTL and the phenotype is perfectly linear, the regression line should pass through the conditional means (c_means), and the differences between the conditional means should be about equal.</p>
<strong>Q5.</strong> What are the values of b0, b1? Is QTL1 significant associated with the phenotype at alpha&lt;0.01 (multiple testing of 5 loci)?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>b0 = 4.1464</li>
<li>b1 = 0.9180</li>
<li>QTL1 is significantly associated with the phenotype with <span class="math inline">\(P = 1.02\times 10^{-13}\)</span></li>
</ul>
</details>
<strong>Q6.</strong> What is the proportion of phenotypic variance explained?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Proportion of phenotypic variance explained = 0.027</li>
</ul>
</details>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="pop-genetics.html#cb172-1" tabindex="-1"></a>linA1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1<span class="sc">~</span>T1QTL_A1, <span class="at">data=</span>dataTwin)</span>
<span id="cb172-2"><a href="pop-genetics.html#cb172-2" tabindex="-1"></a><span class="fu">summary</span>(linA1)</span>
<span id="cb172-3"><a href="pop-genetics.html#cb172-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb172-4"><a href="pop-genetics.html#cb172-4" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb172-5"><a href="pop-genetics.html#cb172-5" tabindex="-1"></a><span class="co">#&gt; lm(formula = pheno1_T1 ~ T1QTL_A1, data = dataTwin)</span></span>
<span id="cb172-6"><a href="pop-genetics.html#cb172-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb172-7"><a href="pop-genetics.html#cb172-7" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb172-8"><a href="pop-genetics.html#cb172-8" tabindex="-1"></a><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span id="cb172-9"><a href="pop-genetics.html#cb172-9" tabindex="-1"></a><span class="co">#&gt; -15.1225  -2.4435   0.1105   2.7775  12.2555 </span></span>
<span id="cb172-10"><a href="pop-genetics.html#cb172-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb172-11"><a href="pop-genetics.html#cb172-11" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb172-12"><a href="pop-genetics.html#cb172-12" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb172-13"><a href="pop-genetics.html#cb172-13" tabindex="-1"></a><span class="co">#&gt; (Intercept)   4.1464     0.1511  27.438  &lt; 2e-16 ***</span></span>
<span id="cb172-14"><a href="pop-genetics.html#cb172-14" tabindex="-1"></a><span class="co">#&gt; T1QTL_A1      0.9180     0.1226   7.491 1.02e-13 ***</span></span>
<span id="cb172-15"><a href="pop-genetics.html#cb172-15" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb172-16"><a href="pop-genetics.html#cb172-16" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb172-17"><a href="pop-genetics.html#cb172-17" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb172-18"><a href="pop-genetics.html#cb172-18" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 3.834 on 1998 degrees of freedom</span></span>
<span id="cb172-19"><a href="pop-genetics.html#cb172-19" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.02732,    Adjusted R-squared:  0.02683 </span></span>
<span id="cb172-20"><a href="pop-genetics.html#cb172-20" tabindex="-1"></a><span class="co">#&gt; F-statistic: 56.11 on 1 and 1998 DF,  p-value: 1.022e-13</span></span>
<span id="cb172-21"><a href="pop-genetics.html#cb172-21" tabindex="-1"></a><span class="fu">summary</span>(linA1)<span class="sc">$</span>r.squared  <span class="co"># proportion of explained variance by additive component</span></span>
<span id="cb172-22"><a href="pop-genetics.html#cb172-22" tabindex="-1"></a><span class="co">#&gt; [1] 0.02731675</span></span>
<span id="cb172-23"><a href="pop-genetics.html#cb172-23" tabindex="-1"></a></span>
<span id="cb172-24"><a href="pop-genetics.html#cb172-24" tabindex="-1"></a>c_means <span class="ot">&lt;-</span> <span class="fu">by</span>(dataTwin<span class="sc">$</span>pheno1_T1,dataTwin<span class="sc">$</span>T1QTL_A1,mean)</span>
<span id="cb172-25"><a href="pop-genetics.html#cb172-25" tabindex="-1"></a><span class="fu">plot</span>(dataTwin<span class="sc">$</span>pheno1_T1 <span class="sc">~</span> dataTwin<span class="sc">$</span>T1QTL_A1, <span class="at">col=</span><span class="st">&#39;grey&#39;</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>))</span>
<span id="cb172-26"><a href="pop-genetics.html#cb172-26" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>), c_means, <span class="at">type=</span><span class="st">&quot;p&quot;</span>, <span class="at">col=</span><span class="dv">6</span>, <span class="at">lwd=</span><span class="dv">8</span>)</span>
<span id="cb172-27"><a href="pop-genetics.html#cb172-27" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">sort</span>(dataTwin<span class="sc">$</span>T1QTL_A1),<span class="fu">sort</span>(linA1<span class="sc">$</span>fitted.values), <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">col=</span><span class="st">&quot;dark green&quot;</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/lmAR-1.png" width="672" /></p>
<p>To test for the non-linearity, we can use the dominance coding of the QTL and add the dominance term to the regression model.</p>
<ul>
<li>Model: pheno1_T1 = b0 + b1* T1QTL_A1 + b2* T1QTL_D1 + e</li>
<li>Repeat for T2.</li>
</ul>
<strong>Q7.</strong> Why can’t we analyse T1 and T2 together?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>As T1 and T2 are biologically related as MZ or DZ twins, the genotypes of the QTLs are not independent. Treating the genotypes of T1 and T2 as independent observations will introduce bias.</li>
</ul>
</details>
<strong>Q8.</strong> Is there a dominance effect?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Yes. The model with dominance provides a better goodness of fit (lower p-value)</li>
</ul>
</details>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="pop-genetics.html#cb173-1" tabindex="-1"></a>linAD1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A1 <span class="sc">+</span> T1QTL_D1, <span class="at">data=</span>dataTwin)</span>
<span id="cb173-2"><a href="pop-genetics.html#cb173-2" tabindex="-1"></a><span class="fu">summary</span>(linAD1) <span class="co"># results lm(phenoT1~T1QTL_A1+T1QTL_D1)</span></span>
<span id="cb173-3"><a href="pop-genetics.html#cb173-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb173-4"><a href="pop-genetics.html#cb173-4" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb173-5"><a href="pop-genetics.html#cb173-5" tabindex="-1"></a><span class="co">#&gt; lm(formula = pheno1_T1 ~ T1QTL_A1 + T1QTL_D1, data = dataTwin)</span></span>
<span id="cb173-6"><a href="pop-genetics.html#cb173-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb173-7"><a href="pop-genetics.html#cb173-7" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb173-8"><a href="pop-genetics.html#cb173-8" tabindex="-1"></a><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span id="cb173-9"><a href="pop-genetics.html#cb173-9" tabindex="-1"></a><span class="co">#&gt; -14.7524  -2.5131   0.0586   2.8215  11.8894 </span></span>
<span id="cb173-10"><a href="pop-genetics.html#cb173-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb173-11"><a href="pop-genetics.html#cb173-11" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb173-12"><a href="pop-genetics.html#cb173-12" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb173-13"><a href="pop-genetics.html#cb173-13" tabindex="-1"></a><span class="co">#&gt; (Intercept)   3.7522     0.1753  21.405  &lt; 2e-16 ***</span></span>
<span id="cb173-14"><a href="pop-genetics.html#cb173-14" tabindex="-1"></a><span class="co">#&gt; T1QTL_A1      0.9301     0.1220   7.622 3.83e-14 ***</span></span>
<span id="cb173-15"><a href="pop-genetics.html#cb173-15" tabindex="-1"></a><span class="co">#&gt; T1QTL_D1      0.7483     0.1708   4.382 1.24e-05 ***</span></span>
<span id="cb173-16"><a href="pop-genetics.html#cb173-16" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb173-17"><a href="pop-genetics.html#cb173-17" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb173-18"><a href="pop-genetics.html#cb173-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb173-19"><a href="pop-genetics.html#cb173-19" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 3.816 on 1997 degrees of freedom</span></span>
<span id="cb173-20"><a href="pop-genetics.html#cb173-20" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.03658,    Adjusted R-squared:  0.03562 </span></span>
<span id="cb173-21"><a href="pop-genetics.html#cb173-21" tabindex="-1"></a><span class="co">#&gt; F-statistic: 37.91 on 2 and 1997 DF,  p-value: &lt; 2.2e-16</span></span>
<span id="cb173-22"><a href="pop-genetics.html#cb173-22" tabindex="-1"></a></span>
<span id="cb173-23"><a href="pop-genetics.html#cb173-23" tabindex="-1"></a><span class="fu">plot</span>(dataTwin<span class="sc">$</span>pheno1_T1 <span class="sc">~</span> dataTwin<span class="sc">$</span>T1QTL_A1, <span class="at">col=</span><span class="st">&#39;grey&#39;</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>))</span>
<span id="cb173-24"><a href="pop-genetics.html#cb173-24" tabindex="-1"></a><span class="fu">abline</span>(linA1, <span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb173-25"><a href="pop-genetics.html#cb173-25" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">2</span>), c_means, <span class="at">type=</span><span class="st">&#39;p&#39;</span>, <span class="at">col=</span><span class="dv">6</span>, <span class="at">lwd=</span><span class="dv">8</span>)</span>
<span id="cb173-26"><a href="pop-genetics.html#cb173-26" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">sort</span>(dataTwin<span class="sc">$</span>T1QTL_A1),<span class="fu">sort</span>(linA1<span class="sc">$</span>fitted.values), <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">col=</span><span class="st">&quot;dark green&quot;</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb173-27"><a href="pop-genetics.html#cb173-27" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">sort</span>(dataTwin<span class="sc">$</span>T1QTL_A1),<span class="fu">sort</span>(linAD1<span class="sc">$</span>fitted.values), <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>, <span class="at">lwd=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="BMDatSci_files/figure-html/lmADR-1.png" width="672" /></p>
<strong>Q9.</strong> Repeat for the other 4 QTL and determine which QTL shows strongest association with the phenotype T1
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>QTL3 with <span class="math inline">\(P = 7.77 \times 10^{-25}\)</span> for model with dominance</li>
</ul>
</details>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="pop-genetics.html#cb174-1" tabindex="-1"></a>allQTL_A_T1 <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span></span>
<span id="cb174-2"><a href="pop-genetics.html#cb174-2" tabindex="-1"></a>cpheno1_T1 <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(dataTwin)<span class="sc">==</span><span class="st">&quot;pheno1_T1&quot;</span>)</span>
<span id="cb174-3"><a href="pop-genetics.html#cb174-3" tabindex="-1"></a><span class="do">## Additive</span></span>
<span id="cb174-4"><a href="pop-genetics.html#cb174-4" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">lapply</span>(allQTL_A_T1,<span class="cf">function</span>(x){ fstat<span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> ., <span class="at">data=</span>dataTwin[,<span class="fu">c</span>(x,cpheno1_T1)]))<span class="sc">$</span>fstatistic;  <span class="fu">pf</span>(fstat[<span class="dv">1</span>],fstat[<span class="dv">2</span>],fstat[<span class="dv">3</span>],<span class="at">lower.tail =</span> F) }))</span>
<span id="cb174-5"><a href="pop-genetics.html#cb174-5" tabindex="-1"></a><span class="co">#&gt;      [,1]        </span></span>
<span id="cb174-6"><a href="pop-genetics.html#cb174-6" tabindex="-1"></a><span class="co">#&gt; [1,] 1.021942e-13</span></span>
<span id="cb174-7"><a href="pop-genetics.html#cb174-7" tabindex="-1"></a><span class="co">#&gt; [2,] 8.329416e-15</span></span>
<span id="cb174-8"><a href="pop-genetics.html#cb174-8" tabindex="-1"></a><span class="co">#&gt; [3,] 1.007527e-13</span></span>
<span id="cb174-9"><a href="pop-genetics.html#cb174-9" tabindex="-1"></a><span class="co">#&gt; [4,] 4.523758e-18</span></span>
<span id="cb174-10"><a href="pop-genetics.html#cb174-10" tabindex="-1"></a><span class="co">#&gt; [5,] 8.207842e-13</span></span>
<span id="cb174-11"><a href="pop-genetics.html#cb174-11" tabindex="-1"></a><span class="do">## Dominance</span></span>
<span id="cb174-12"><a href="pop-genetics.html#cb174-12" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">lapply</span>(allQTL_A_T1,<span class="cf">function</span>(x){ fstat<span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> ., <span class="at">data=</span>dataTwin[,<span class="fu">c</span>(x,x<span class="sc">+</span><span class="dv">10</span>,cpheno1_T1)]))<span class="sc">$</span>fstatistic;  <span class="fu">pf</span>(fstat[<span class="dv">1</span>],fstat[<span class="dv">2</span>],fstat[<span class="dv">3</span>],<span class="at">lower.tail =</span> F) }))</span>
<span id="cb174-13"><a href="pop-genetics.html#cb174-13" tabindex="-1"></a><span class="co">#&gt;      [,1]        </span></span>
<span id="cb174-14"><a href="pop-genetics.html#cb174-14" tabindex="-1"></a><span class="co">#&gt; [1,] 6.907834e-17</span></span>
<span id="cb174-15"><a href="pop-genetics.html#cb174-15" tabindex="-1"></a><span class="co">#&gt; [2,] 2.166957e-22</span></span>
<span id="cb174-16"><a href="pop-genetics.html#cb174-16" tabindex="-1"></a><span class="co">#&gt; [3,] 7.771588e-25</span></span>
<span id="cb174-17"><a href="pop-genetics.html#cb174-17" tabindex="-1"></a><span class="co">#&gt; [4,] 8.124437e-25</span></span>
<span id="cb174-18"><a href="pop-genetics.html#cb174-18" tabindex="-1"></a><span class="co">#&gt; [5,] 4.312127e-21</span></span>
<span id="cb174-19"><a href="pop-genetics.html#cb174-19" tabindex="-1"></a></span>
<span id="cb174-20"><a href="pop-genetics.html#cb174-20" tabindex="-1"></a><span class="co">#Q9: QTL3 shows the strongest association with P=7.771588e-25</span></span>
<span id="cb174-21"><a href="pop-genetics.html#cb174-21" tabindex="-1"></a>linAD3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A3 <span class="sc">+</span> T1QTL_D3, <span class="at">data=</span>dataTwin)</span>
<span id="cb174-22"><a href="pop-genetics.html#cb174-22" tabindex="-1"></a><span class="fu">summary</span>(linAD3) <span class="co"># results lm(phenoT1~T1QTL_A1+T1QTL_D1)</span></span>
<span id="cb174-23"><a href="pop-genetics.html#cb174-23" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb174-24"><a href="pop-genetics.html#cb174-24" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb174-25"><a href="pop-genetics.html#cb174-25" tabindex="-1"></a><span class="co">#&gt; lm(formula = pheno1_T1 ~ T1QTL_A3 + T1QTL_D3, data = dataTwin)</span></span>
<span id="cb174-26"><a href="pop-genetics.html#cb174-26" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb174-27"><a href="pop-genetics.html#cb174-27" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb174-28"><a href="pop-genetics.html#cb174-28" tabindex="-1"></a><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span id="cb174-29"><a href="pop-genetics.html#cb174-29" tabindex="-1"></a><span class="co">#&gt; -14.4988  -2.5701   0.1843   2.6991  11.5974 </span></span>
<span id="cb174-30"><a href="pop-genetics.html#cb174-30" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb174-31"><a href="pop-genetics.html#cb174-31" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb174-32"><a href="pop-genetics.html#cb174-32" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb174-33"><a href="pop-genetics.html#cb174-33" tabindex="-1"></a><span class="co">#&gt; (Intercept)   3.5437     0.1684  21.038  &lt; 2e-16 ***</span></span>
<span id="cb174-34"><a href="pop-genetics.html#cb174-34" tabindex="-1"></a><span class="co">#&gt; T1QTL_A3      0.9076     0.1181   7.683 2.43e-14 ***</span></span>
<span id="cb174-35"><a href="pop-genetics.html#cb174-35" tabindex="-1"></a><span class="co">#&gt; T1QTL_D3      1.2714     0.1692   7.515 8.55e-14 ***</span></span>
<span id="cb174-36"><a href="pop-genetics.html#cb174-36" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb174-37"><a href="pop-genetics.html#cb174-37" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb174-38"><a href="pop-genetics.html#cb174-38" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb174-39"><a href="pop-genetics.html#cb174-39" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 3.782 on 1997 degrees of freedom</span></span>
<span id="cb174-40"><a href="pop-genetics.html#cb174-40" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.05408,    Adjusted R-squared:  0.05313 </span></span>
<span id="cb174-41"><a href="pop-genetics.html#cb174-41" tabindex="-1"></a><span class="co">#&gt; F-statistic: 57.09 on 2 and 1997 DF,  p-value: &lt; 2.2e-16</span></span></code></pre></div>
<p>If the subjects with top 5% of the phenotype score are considered as cases, perform case-control association test for most significant SNP (from Q9) and interpret the result.</p>
<strong>Q10.</strong> What are the odds ratio, p-value, and 95% confidence interval (CI)?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Odds ratio is 3.40 and the 95% CI is (1.65, 7.03)</li>
</ul>
</details>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="pop-genetics.html#cb175-1" tabindex="-1"></a>quant05 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(<span class="fu">c</span>(dataTwin<span class="sc">$</span>pheno1_T1,dataTwin<span class="sc">$</span>pheno1_T2),<span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="fl">0.05</span>))</span>
<span id="cb175-2"><a href="pop-genetics.html#cb175-2" tabindex="-1"></a>dataTwin<span class="sc">$</span>CaseT1 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dataTwin<span class="sc">$</span>pheno1_T1<span class="sc">&gt;</span>quant05[<span class="dv">20</span>]) </span>
<span id="cb175-3"><a href="pop-genetics.html#cb175-3" tabindex="-1"></a>dataTwin<span class="sc">$</span>CaseT2 <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dataTwin<span class="sc">$</span>pheno1_T2<span class="sc">&gt;</span>quant05[<span class="dv">20</span>])</span>
<span id="cb175-4"><a href="pop-genetics.html#cb175-4" tabindex="-1"></a>logisticAD1 <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">glm</span>(CaseT1 <span class="sc">~</span> T1QTL_A3 <span class="sc">+</span> T1QTL_D3, <span class="at">data=</span>dataTwin, <span class="at">family=</span><span class="st">&quot;binomial&quot;</span>))</span>
<span id="cb175-5"><a href="pop-genetics.html#cb175-5" tabindex="-1"></a><span class="fu">exp</span>(logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>])                                     <span class="co"># odds ratio</span></span>
<span id="cb175-6"><a href="pop-genetics.html#cb175-6" tabindex="-1"></a><span class="co">#&gt; [1] 3.404757</span></span>
<span id="cb175-7"><a href="pop-genetics.html#cb175-7" tabindex="-1"></a><span class="fu">exp</span>(logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>]<span class="sc">-</span><span class="fl">1.96</span><span class="sc">*</span>logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>])  <span class="co"># lower 95% confidence interval</span></span>
<span id="cb175-8"><a href="pop-genetics.html#cb175-8" tabindex="-1"></a><span class="co">#&gt; [1] 1.648816</span></span>
<span id="cb175-9"><a href="pop-genetics.html#cb175-9" tabindex="-1"></a><span class="fu">exp</span>(logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">1</span>]<span class="sc">+</span><span class="fl">1.96</span><span class="sc">*</span>logisticAD1<span class="sc">$</span>coefficients[<span class="dv">2</span>,<span class="dv">2</span>])  <span class="co"># upper 95% confidence interval</span></span>
<span id="cb175-10"><a href="pop-genetics.html#cb175-10" tabindex="-1"></a><span class="co">#&gt; [1] 7.030723</span></span></code></pre></div>
</div>
</div>
<div id="part-2" class="section level3 hasAnchor" number="7.1.2">
<h3><span class="header-section-number">7.1.2</span> Part 2<a href="pop-genetics.html#part-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Scenario:</strong> You are asked to estimate the additive genetic variance, dominance genetic variance and/or shared environmental variance using regression-based method and a classical twin design.</p>
<p><span class="math display">\[\begin{align*}

\text{For ADE model :  }~ &amp; \sigma^{2}_{P} = \sigma^{2}_{A} + \sigma^{2}_{D} + \sigma^{2}_{E}\\

\text{For ACE model :  }~ &amp; \sigma^{2}_{P} = \sigma^{2}_{A} + \sigma^{2}_{C} + \sigma^{2}_{E}, \quad \text{where} \\

\sigma^{2}_{P} &amp; \text{ is the phenotypic variance}, \\

\sigma^{2}_{A} &amp; \text{ is additive genetic variance}, \\

\sigma^{2}_{D} &amp; \text{ is dominance genetic variance}, \\

\sigma^{2}_{C} &amp; \text{ is shared environmental variance, and} \\

\sigma^{2}_{E} &amp; \text{ is unshared environmental variance.}
\end{align*}\]</span></p>
<p>For ADE model,</p>
<p><span class="math display">\[\begin{align*}

  cov(MZ) = cor(MZ) &amp; = rMZ = \sigma^{2}_{A} + \sigma^{2}_{D} \\
  cov(DZ) = cor(DZ) &amp; = rDZ = 0.5 * \sigma^{2}_{A} + 0.25 * \sigma^{2}_{D} \quad \text{ , where} \\

\end{align*}\]</span>
the coefficients 1/2 and 1/4 are based on quantitative genetic theory (Mather &amp; Jinks, 1971).</p>
<p>By solving the unknowns, the Falconer’s equations for the ADE model:</p>
<p><span class="math display">\[\begin{align*}

\sigma^{2}_{A} &amp; = 4*rDZ - rMZ \\

\sigma^{2}_{D} &amp; = 2*rMZ - 4*rDZ \\
\sigma^{2}_{E} &amp; = 1 - \sigma^{2}_{A} - \sigma^{2}_{D} \\

\end{align*}\]</span></p>
<p>For ACE model,</p>
<p><span class="math display">\[\begin{align*}

  cov(MZ) = cor(MZ) &amp; = rMZ= \sigma^{2}_{A} + \sigma^{2}_{C} \\
  cov(DZ) = cor(DZ) &amp; = rDZ = 0.5 * \sigma^{2}_{A} + \sigma^{2}_{C} \quad \text{ , where} \\

\end{align*}\]</span></p>
<p>By solving the unknowns, the Falconer’s equations for the ACE model:</p>
<p><span class="math display">\[\begin{align*}

\sigma^{2}_{A} &amp; = 2*(rMZ - rDZ) \\
\sigma^{2}_{C} &amp; = 2*rDZ - rMZ \\
\sigma^{2}_{E} &amp; = 1 - \sigma^{2}_{A} - \sigma^{2}_{C} = 1 - rMZ

\end{align*}\]</span></p>
<div id="questions-for-discussions" class="section level4 hasAnchor" number="7.1.2.1">
<h4><span class="header-section-number">7.1.2.1</span> Questions for discussions :<a href="pop-genetics.html#questions-for-discussions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<strong>Q1.</strong> What is missing heritability of common traits in the era of genome-wide association analysis (GWAS)?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Missing heritability refers to the discrepancy between twin/family-estimated heritability and the amount of variance explained by disease/trait-associated loci identified in GWAS</li>
</ul>
</details>
<strong>Q2.</strong> What are the potential sources of missing heritability?
<details>
<summary>
<strong>Answers:</strong>
</summary>
<ul>
<li>Large number of variants with small effect not reaching GWAS significance due to inadequate power of GWAS</li>
<li>Poor detection of rarer disease/trait-associated variants by genotyping arrays</li>
<li>Structural variants poorly captured by genotyping arrays</li>
<li>Low power to detect gene–gene interactions</li>
<li>Underestimation of shared environment among relatives in twin/family-based studies</li>
</ul>
</details>
<ul>
<li>Suggested reading:
<ul>
<li>Manolio TA, Collins FS, Cox NJ, et al. Nature. 2009 ;461(7265):747-753. <a href="doi:10.1038/nature08494" class="uri">doi:10.1038/nature08494</a></li>
</ul></li>
</ul>
</div>
<div id="hands-on-exercise-variance-explained-using-regression-based-method" class="section level4 hasAnchor" number="7.1.2.2">
<h4><span class="header-section-number">7.1.2.2</span> Hands-on exercise : variance explained using regression-based method<a href="pop-genetics.html#hands-on-exercise-variance-explained-using-regression-based-method" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Q1.</strong> What is the variance of the phenotype?</p>
<p><strong>Q2.</strong> Compute the explained variance attributable to the additive genetic component of the QTL with strongest association in Part 1.</p>
<p><strong>Q3.</strong> Compute the explained variance attributable to the dominance genetic component of the QTL with strongest association in Part 1.</p>
<p>R2 from the regression represents the proportion of phenotypic variance explained; thus the raw explained variance component is R2 times the variance of the phenotype (var_pheno).</p>
<details>
<summary>
<strong>Answers</strong>
</summary>
<ul>
<li>The proportion of explained variance are 0.0273 (additive) and 0.0541 (total: additive + dominance).</li>
<li>As the predictors are uncorrelated, the proportion of explained variance by dominance = 0.0541 - 0.0273 = 0.0267</li>
<li>Given the phenotypic variance of 15.102, then
<ul>
<li>Total genetic: 0.0541*15.102 = 0.8168</li>
<li>Additive genetic: 0.0273*15.102 = 0.4128</li>
<li>Dominance genetic: 0.0267*15.102 = 0.4040</li>
</ul></li>
</ul>
</details>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="pop-genetics.html#cb176-1" tabindex="-1"></a>var_pheno <span class="ot">&lt;-</span> <span class="fu">var</span>(dataTwin<span class="sc">$</span>pheno1_T1)  <span class="co"># the variance of the phenotype</span></span>
<span id="cb176-2"><a href="pop-genetics.html#cb176-2" tabindex="-1"></a>var_pheno</span>
<span id="cb176-3"><a href="pop-genetics.html#cb176-3" tabindex="-1"></a><span class="co">#&gt; [1] 15.10257</span></span>
<span id="cb176-4"><a href="pop-genetics.html#cb176-4" tabindex="-1"></a></span>
<span id="cb176-5"><a href="pop-genetics.html#cb176-5" tabindex="-1"></a>linAD3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A3 <span class="sc">+</span> T1QTL_D3, <span class="at">data=</span>dataTwin)</span>
<span id="cb176-6"><a href="pop-genetics.html#cb176-6" tabindex="-1"></a>linA3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A3, <span class="at">data=</span>dataTwin)</span>
<span id="cb176-7"><a href="pop-genetics.html#cb176-7" tabindex="-1"></a></span>
<span id="cb176-8"><a href="pop-genetics.html#cb176-8" tabindex="-1"></a><span class="fu">summary</span>(linAD3)<span class="sc">$</span>r.squared           <span class="co"># proportion of explained variance by total genetic component</span></span>
<span id="cb176-9"><a href="pop-genetics.html#cb176-9" tabindex="-1"></a><span class="co">#&gt; [1] 0.05408025</span></span>
<span id="cb176-10"><a href="pop-genetics.html#cb176-10" tabindex="-1"></a><span class="fu">summary</span>(linA3)<span class="sc">$</span>r.squared            <span class="co"># proportion of explained variance by additive component</span></span>
<span id="cb176-11"><a href="pop-genetics.html#cb176-11" tabindex="-1"></a><span class="co">#&gt; [1] 0.02733034</span></span>
<span id="cb176-12"><a href="pop-genetics.html#cb176-12" tabindex="-1"></a><span class="fu">summary</span>(linAD3)<span class="sc">$</span>r.squared<span class="sc">*</span>var_pheno <span class="co"># (raw) variance component of total genetic component</span></span>
<span id="cb176-13"><a href="pop-genetics.html#cb176-13" tabindex="-1"></a><span class="co">#&gt; [1] 0.8167509</span></span>
<span id="cb176-14"><a href="pop-genetics.html#cb176-14" tabindex="-1"></a><span class="fu">summary</span>(linA3)<span class="sc">$</span>r.squared<span class="sc">*</span>var_pheno  <span class="co"># (raw) variance component of additive genetic component</span></span>
<span id="cb176-15"><a href="pop-genetics.html#cb176-15" tabindex="-1"></a><span class="co">#&gt; [1] 0.4127585</span></span>
<span id="cb176-16"><a href="pop-genetics.html#cb176-16" tabindex="-1"></a>(<span class="fu">summary</span>(linAD3)<span class="sc">$</span>r.squared<span class="sc">-</span><span class="fu">summary</span>(linA3)<span class="sc">$</span>r.squared)<span class="sc">*</span>var_pheno  <span class="co"># (raw) variance component of dominance genetic component</span></span>
<span id="cb176-17"><a href="pop-genetics.html#cb176-17" tabindex="-1"></a><span class="co">#&gt; [1] 0.4039924</span></span></code></pre></div>
</details>
<p><strong>Q4.</strong> Estimate the variance explained by all the QTL using linear regression.</p>
<details>
<summary>
<strong>Answers</strong>
</summary>
<ul>
<li>Proportion of variance explained by all 5 QTLs with dominance = 0.23 and the total variance explained = 3.52.</li>
</ul>
</details>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="pop-genetics.html#cb177-1" tabindex="-1"></a><span class="co"># compute for all 5 QTL</span></span>
<span id="cb177-2"><a href="pop-genetics.html#cb177-2" tabindex="-1"></a>linAD5<span class="ot">=</span>(<span class="fu">lm</span>(pheno1_T1 <span class="sc">~</span> T1QTL_A1 <span class="sc">+</span> T1QTL_A2 <span class="sc">+</span> T1QTL_A3 <span class="sc">+</span> T1QTL_A4 <span class="sc">+</span> T1QTL_A5 <span class="sc">+</span></span>
<span id="cb177-3"><a href="pop-genetics.html#cb177-3" tabindex="-1"></a>                       T1QTL_D1 <span class="sc">+</span> T1QTL_D2 <span class="sc">+</span> T1QTL_D3 <span class="sc">+</span> T1QTL_D4 <span class="sc">+</span> T1QTL_D5,</span>
<span id="cb177-4"><a href="pop-genetics.html#cb177-4" tabindex="-1"></a>            <span class="at">data=</span>dataTwin))</span>
<span id="cb177-5"><a href="pop-genetics.html#cb177-5" tabindex="-1"></a></span>
<span id="cb177-6"><a href="pop-genetics.html#cb177-6" tabindex="-1"></a><span class="fu">summary</span>(linAD5)<span class="sc">$</span>r.squared              <span class="co"># proportion of explained variance by total genetic component</span></span>
<span id="cb177-7"><a href="pop-genetics.html#cb177-7" tabindex="-1"></a><span class="co">#&gt; [1] 0.2330307</span></span>
<span id="cb177-8"><a href="pop-genetics.html#cb177-8" tabindex="-1"></a><span class="fu">summary</span>(linAD5)<span class="sc">$</span>r.squared<span class="sc">*</span>var_pheno    <span class="co"># (raw) variance component of total genetic component</span></span>
<span id="cb177-9"><a href="pop-genetics.html#cb177-9" tabindex="-1"></a><span class="co">#&gt; [1] 3.519363</span></span></code></pre></div>
</div>
<div id="hands-on-exercise-variance-explained-using-a-classical-twin-design." class="section level4 hasAnchor" number="7.1.2.3">
<h4><span class="header-section-number">7.1.2.3</span> Hands-on exercise : variance explained using a classical twin design.<a href="pop-genetics.html#hands-on-exercise-variance-explained-using-a-classical-twin-design." class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Based on our regression results, we have estimates of the total genetic variance as well as the A and D components for phenotype 1. In practice, it is impossible to know all the variants associated with any polygenic trait.</p>
<p>Given <code>rMZ &gt; 2*rDZ</code>, we can use Falconer’s formula based on ADE model to estimate the A (additive genetic) and D (dominance) variance with the classical twin design for phenotype 1 without genotypes.</p>
<strong>Q5.</strong> Compute rMZ and rDZ.
<details>
<summary>
<strong>Answers</strong>
</summary>
<ul>
<li>rMZ = 0.5434</li>
<li>rDZ = 0.1904</li>
</ul>
</details>
<strong>Q6.</strong> Estimate the proportion of additive and dominance genetic variances using the Falconer’s equations for the ADE model.
<details>
<summary>
<strong>Answers</strong>
</summary>
<ul>
<li><span class="math inline">\(\sigma^{2}_{A} = 0.2181\)</span></li>
<li><span class="math inline">\(\sigma^{2}_{D} = 0.3253\)</span></li>
<li><span class="math inline">\(\sigma^{2}_{E} = 0.4566\)</span></li>
</ul>
</details>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="pop-genetics.html#cb178-1" tabindex="-1"></a>dataMZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">1</span>, <span class="fu">c</span>(<span class="st">&#39;pheno1_T1&#39;</span>, <span class="st">&#39;pheno1_T2&#39;</span>)] <span class="co"># MZ data frame</span></span>
<span id="cb178-2"><a href="pop-genetics.html#cb178-2" tabindex="-1"></a>dataDZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&#39;pheno1_T1&#39;</span>, <span class="st">&#39;pheno1_T2&#39;</span>)] <span class="co"># DZ data frame</span></span>
<span id="cb178-3"><a href="pop-genetics.html#cb178-3" tabindex="-1"></a></span>
<span id="cb178-4"><a href="pop-genetics.html#cb178-4" tabindex="-1"></a>rMZ<span class="ot">=</span><span class="fu">cor</span>(dataMZ)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the MZ correlation matrix</span></span>
<span id="cb178-5"><a href="pop-genetics.html#cb178-5" tabindex="-1"></a>rDZ<span class="ot">=</span><span class="fu">cor</span>(dataDZ)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the DZ correlation matrix</span></span>
<span id="cb178-6"><a href="pop-genetics.html#cb178-6" tabindex="-1"></a>rMZ</span>
<span id="cb178-7"><a href="pop-genetics.html#cb178-7" tabindex="-1"></a>rDZ</span>
<span id="cb178-8"><a href="pop-genetics.html#cb178-8" tabindex="-1"></a></span>
<span id="cb178-9"><a href="pop-genetics.html#cb178-9" tabindex="-1"></a>sA2 <span class="ot">=</span> <span class="dv">4</span><span class="sc">*</span>rDZ <span class="sc">-</span> rMZ</span>
<span id="cb178-10"><a href="pop-genetics.html#cb178-10" tabindex="-1"></a>sD2 <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span>rMZ <span class="sc">-</span> <span class="dv">4</span><span class="sc">*</span>rDZ</span>
<span id="cb178-11"><a href="pop-genetics.html#cb178-11" tabindex="-1"></a>sE2 <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> sA2 <span class="sc">-</span> sD2</span>
<span id="cb178-12"><a href="pop-genetics.html#cb178-12" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(sA2, sD2, sE2))</span></code></pre></div>
<p>Similarly, for phenotype 2, we can estimate the proportion of additive and/or dominance genetic variances as well as shared environmental variance using the Falconer’s formula.</p>
<strong>Q7.</strong> Which model (ACE or ADE) should be considered for phenotype 2?
<details>
<summary>
<strong>Answers</strong>
</summary>
<ul>
<li>ACE as rMZ &lt; 2*rDZ</li>
</ul>
</details>
<strong>Q8.</strong> Estimate the proportion of A, C/D and E variance components for phenotype 2.
<details>
<summary>
<strong>Answers</strong>
</summary>
<ul>
<li><span class="math inline">\(\sigma^{2}_{A} = 0.3526\)</span></li>
<li><span class="math inline">\(\sigma^{2}_{C} = 0.1610\)</span></li>
<li><span class="math inline">\(\sigma^{2}_{E} = 0.4864\)</span></li>
</ul>
</details>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="pop-genetics.html#cb179-1" tabindex="-1"></a>dataMZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">1</span>, <span class="fu">c</span>(<span class="st">&#39;pheno2_T1&#39;</span>, <span class="st">&#39;pheno2_T2&#39;</span>)] <span class="co"># MZ data frame</span></span>
<span id="cb179-2"><a href="pop-genetics.html#cb179-2" tabindex="-1"></a>dataDZ <span class="ot">=</span> dataTwin[dataTwin<span class="sc">$</span>zygosity<span class="sc">==</span><span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&#39;pheno2_T1&#39;</span>, <span class="st">&#39;pheno2_T2&#39;</span>)] <span class="co"># DZ data frame</span></span>
<span id="cb179-3"><a href="pop-genetics.html#cb179-3" tabindex="-1"></a></span>
<span id="cb179-4"><a href="pop-genetics.html#cb179-4" tabindex="-1"></a>rMZ<span class="ot">=</span><span class="fu">cor</span>(dataMZ, <span class="at">use=</span><span class="st">&quot;complete.obs&quot;</span>)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the MZ correlation matrix</span></span>
<span id="cb179-5"><a href="pop-genetics.html#cb179-5" tabindex="-1"></a>rDZ<span class="ot">=</span><span class="fu">cor</span>(dataDZ, <span class="at">use=</span><span class="st">&quot;complete.obs&quot;</span>)[<span class="dv">2</span>,<span class="dv">1</span>] <span class="co"># element 2,1 in the DZ correlation matrix</span></span>
<span id="cb179-6"><a href="pop-genetics.html#cb179-6" tabindex="-1"></a>rMZ</span>
<span id="cb179-7"><a href="pop-genetics.html#cb179-7" tabindex="-1"></a>rDZ</span>
<span id="cb179-8"><a href="pop-genetics.html#cb179-8" tabindex="-1"></a></span>
<span id="cb179-9"><a href="pop-genetics.html#cb179-9" tabindex="-1"></a>sA2 <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span>(rMZ <span class="sc">-</span> rDZ)</span>
<span id="cb179-10"><a href="pop-genetics.html#cb179-10" tabindex="-1"></a>sC2 <span class="ot">=</span> <span class="dv">2</span><span class="sc">*</span>rDZ <span class="sc">-</span> rMZ</span>
<span id="cb179-11"><a href="pop-genetics.html#cb179-11" tabindex="-1"></a>sE2 <span class="ot">=</span> <span class="dv">1</span> <span class="sc">-</span> rMZ</span>
<span id="cb179-12"><a href="pop-genetics.html#cb179-12" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">c</span>(sA2, sC2, sE2))</span></code></pre></div>
</div>
</div>
<div id="references" class="section level3 hasAnchor" number="7.1.3">
<h3><span class="header-section-number">7.1.3</span> References<a href="pop-genetics.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Evans DM, Gillespie NA, Martin NG. Biometrical genetics. Biol Psychol. 2002 Oct;61(1-2):33-51. doi: 10.1016/s0301-0511(02)00051-0. PMID: 12385668. [Review article]</p></li>
<li><p>Falconer, D.S. and Mackay, T.F.C. (1996) Introduction to Quantitative Genetics. 4th Edition, Addison Wesley Longman, Harlow. [Most classical; a lot of online version]</p></li>
<li><p>Neale, B., Ferreira, M., Medland, S., &amp; Posthuma, D. (Eds.). (2007). Statistical Genetics: Gene Mapping Through Linkage and Association (1st ed.). Taylor &amp; Francis. <a href="https://doi.org/10.1201/9780203967201" class="uri">https://doi.org/10.1201/9780203967201</a> [chapter on biometrical genetics; can be borrowed from HKU lib]</p></li>
<li><p><a href="https://ibg.colorado.edu/cdrom2020/dolan/biometricalGenetics/biom_gen_2020.pdf" class="uri">https://ibg.colorado.edu/cdrom2020/dolan/biometricalGenetics/biom_gen_2020.pdf</a> [Course material of the Boulder IBG workshop co-organized by top statistical geneticists]</p></li>
</ol>


</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="cancer.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="install.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/statbiomed/BMDS-book/tree/main/02-data_topics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BMDatSci.pdf", "BMDatSci.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
